// gm_api/src/preference/data.ts
import { getLocale } from './lng'

interface Enum_List_Item {
  text: string
  value: any
}

interface Enum_Map {
  [key: string]: string
}
export const list_Status_Code: Enum_List_Item[] = [
  { value: 20190001, text: getLocale('Status_Code__INVENTORY_SETTINGS_DUPLICATE_SETTINGS') },
  { value: 20190002, text: getLocale('Status_Code__DUPLICATE_PRINTING_TEMPLATE_NAME') },
  { value: 20190003, text: getLocale('Status_Code__PRINTING_TEMPLATE_RELATION_DUPLICATE') },
]
export const map_Status_Code: Enum_Map = {
  20190001: getLocale('Status_Code__INVENTORY_SETTINGS_DUPLICATE_SETTINGS'),
  20190002: getLocale('Status_Code__DUPLICATE_PRINTING_TEMPLATE_NAME'),
  20190003: getLocale('Status_Code__PRINTING_TEMPLATE_RELATION_DUPLICATE'),
}
export const list_PrintingTemplate_Direction: Enum_List_Item[] = [
  { value: 1, text: getLocale('PrintingTemplate_Direction__DIRECTION_HORIZONTAL') },
  { value: 2, text: getLocale('PrintingTemplate_Direction__DIRECTION_VERTICAL') },
]
export const map_PrintingTemplate_Direction: Enum_Map = {
  1: getLocale('PrintingTemplate_Direction__DIRECTION_HORIZONTAL'),
  2: getLocale('PrintingTemplate_Direction__DIRECTION_VERTICAL'),
}
export const list_PrintingTemplate_Type: Enum_List_Item[] = [
  { value: 1, text: getLocale('PrintingTemplate_Type__TYPE_PURCHASE_TASK') },
  { value: 2, text: getLocale('PrintingTemplate_Type__TYPE_INNER_LABEL') },
  { value: 3, text: getLocale('PrintingTemplate_Type__TYPE_OUT_STOCK') },
  { value: 4, text: getLocale('PrintingTemplate_Type__TYPE_IN_STOCK') },
  { value: 5, text: getLocale('PrintingTemplate_Type__TYPE_SORTING') },
  { value: 6, text: getLocale('PrintingTemplate_Type__TYPE_SETTLEMENT') },
  { value: 7, text: getLocale('PrintingTemplate_Type__TYPE_DELIVERY') },
  { value: 8, text: getLocale('PrintingTemplate_Type__TYPE_PURCHASE_SHEET') },
  { value: 9, text: getLocale('PrintingTemplate_Type__TYPE_BOX') },
]
export const map_PrintingTemplate_Type: Enum_Map = {
  1: getLocale('PrintingTemplate_Type__TYPE_PURCHASE_TASK'),
  2: getLocale('PrintingTemplate_Type__TYPE_INNER_LABEL'),
  3: getLocale('PrintingTemplate_Type__TYPE_OUT_STOCK'),
  4: getLocale('PrintingTemplate_Type__TYPE_IN_STOCK'),
  5: getLocale('PrintingTemplate_Type__TYPE_SORTING'),
  6: getLocale('PrintingTemplate_Type__TYPE_SETTLEMENT'),
  7: getLocale('PrintingTemplate_Type__TYPE_DELIVERY'),
  8: getLocale('PrintingTemplate_Type__TYPE_PURCHASE_SHEET'),
  9: getLocale('PrintingTemplate_Type__TYPE_BOX'),
}
export const list_Shop_Type: Enum_List_Item[] = [
  { value: 1, text: getLocale('Shop_Type__TYPE_SOCIAL') },
  { value: 2, text: getLocale('Shop_Type__TYPE_EDUCATION') },
]
export const map_Shop_Type: Enum_Map = {
  1: getLocale('Shop_Type__TYPE_SOCIAL'),
  2: getLocale('Shop_Type__TYPE_EDUCATION'),
}
export const list_Application_Type: Enum_List_Item[] = [
  { value: 1, text: getLocale('Application_Type__TYPE_WECHAT_MINI_PROGRAM') },
  { value: 2, text: getLocale('Application_Type__TYPE_WECHAT_OFFICIAL_ACCOUNT') },
]
export const map_Application_Type: Enum_Map = {
  1: getLocale('Application_Type__TYPE_WECHAT_MINI_PROGRAM'),
  2: getLocale('Application_Type__TYPE_WECHAT_OFFICIAL_ACCOUNT'),
}
export const list_PurchaseSettings_PurchaseTaskPlanValueCalulateMethod: Enum_List_Item[] = [
  { value: 1, text: getLocale('PurchaseSettings_PurchaseTaskPlanValueCalulateMethod__SUGGEST_VALUE') },
  { value: 2, text: getLocale('PurchaseSettings_PurchaseTaskPlanValueCalulateMethod__REQUEST_VALUE') },
]
export const map_PurchaseSettings_PurchaseTaskPlanValueCalulateMethod: Enum_Map = {
  1: getLocale('PurchaseSettings_PurchaseTaskPlanValueCalulateMethod__SUGGEST_VALUE'),
  2: getLocale('PurchaseSettings_PurchaseTaskPlanValueCalulateMethod__REQUEST_VALUE'),
}
export const list_AppointTimeSettings_Type: Enum_List_Item[] = [
  { value: 1, text: getLocale('AppointTimeSettings_Type__NONPROCESSED_PURCHASE_TIME_BEFORE_ORDER_RECV') },
  { value: 2, text: getLocale('AppointTimeSettings_Type__PROCESSED_INGREDIENT_PURCHASE_TIME_BEFORE_ORDER_RECV') },
  { value: 3, text: getLocale('AppointTimeSettings_Type__PROCESSED_PRODUCE_TIME_BEFORE_ORDER_RECV') },
  { value: 4, text: getLocale('AppointTimeSettings_Type__PROCESSED_PACK_TIME_BEFORE_ORDER_RECV') },
  { value: 5, text: getLocale('AppointTimeSettings_Type__PURCHASE_TIME_BEFORE_PRODUCE') },
  { value: 6, text: getLocale('AppointTimeSettings_Type__PRODUCE_TIME_BEFORE_PACK') },
]
export const map_AppointTimeSettings_Type: Enum_Map = {
  1: getLocale('AppointTimeSettings_Type__NONPROCESSED_PURCHASE_TIME_BEFORE_ORDER_RECV'),
  2: getLocale('AppointTimeSettings_Type__PROCESSED_INGREDIENT_PURCHASE_TIME_BEFORE_ORDER_RECV'),
  3: getLocale('AppointTimeSettings_Type__PROCESSED_PRODUCE_TIME_BEFORE_ORDER_RECV'),
  4: getLocale('AppointTimeSettings_Type__PROCESSED_PACK_TIME_BEFORE_ORDER_RECV'),
  5: getLocale('AppointTimeSettings_Type__PURCHASE_TIME_BEFORE_PRODUCE'),
  6: getLocale('AppointTimeSettings_Type__PRODUCE_TIME_BEFORE_PACK'),
}
export const list_PrecisionSettings_Type: Enum_List_Item[] = [
  { value: 1, text: getLocale('PrecisionSettings_Type__PURCHASE') },
]
export const map_PrecisionSettings_Type: Enum_Map = {
  1: getLocale('PrecisionSettings_Type__PURCHASE'),
}
export const list_SortingSettings_SortingNumMethod: Enum_List_Item[] = [
  { value: 1, text: getLocale('SortingSettings_SortingNumMethod__SORTINGNUM_METHOD_ORDER') },
  { value: 2, text: getLocale('SortingSettings_SortingNumMethod__SORTINGNUM_METHOD_ROUTE') },
]
export const map_SortingSettings_SortingNumMethod: Enum_Map = {
  1: getLocale('SortingSettings_SortingNumMethod__SORTINGNUM_METHOD_ORDER'),
  2: getLocale('SortingSettings_SortingNumMethod__SORTINGNUM_METHOD_ROUTE'),
}
export const list_SortingSettings_SortingLockType: Enum_List_Item[] = [
  { value: 1, text: getLocale('SortingSettings_SortingLockType__SORTINGLOCKTYPE_DELIVERYING') },
  { value: 2, text: getLocale('SortingSettings_SortingLockType__SORTINGLOCKTYPE_RECEIVABLE') },
]
export const map_SortingSettings_SortingLockType: Enum_Map = {
  1: getLocale('SortingSettings_SortingLockType__SORTINGLOCKTYPE_DELIVERYING'),
  2: getLocale('SortingSettings_SortingLockType__SORTINGLOCKTYPE_RECEIVABLE'),
}
export const list_InventorySettings_InventoryMethod: Enum_List_Item[] = [
  { value: 1, text: getLocale('InventorySettings_InventoryMethod__INVENTORY_METHOD_FIFO') },
  { value: 2, text: getLocale('InventorySettings_InventoryMethod__INVENTORY_METHOD_WEIGHTED_AVERAGE') },
]
export const map_InventorySettings_InventoryMethod: Enum_Map = {
  1: getLocale('InventorySettings_InventoryMethod__INVENTORY_METHOD_FIFO'),
  2: getLocale('InventorySettings_InventoryMethod__INVENTORY_METHOD_WEIGHTED_AVERAGE'),
}
export const list_ProductionSettings_TaskOutputSource: Enum_List_Item[] = [
  { value: 1, text: getLocale('ProductionSettings_TaskOutputSource__TASKOUTPUT_SOURCE_TYPE_IN') },
  { value: 2, text: getLocale('ProductionSettings_TaskOutputSource__TASKOUTPUT_SOURCE_LAST_PROCESS') },
]
export const map_ProductionSettings_TaskOutputSource: Enum_Map = {
  1: getLocale('ProductionSettings_TaskOutputSource__TASKOUTPUT_SOURCE_TYPE_IN'),
  2: getLocale('ProductionSettings_TaskOutputSource__TASKOUTPUT_SOURCE_LAST_PROCESS'),
}
