// gm_api/src/preference/methods.ts
import { Request } from '../util/request'
import {
  CreateInventorySettingsRequest, CreateInventorySettingsResponse,
  UpdateInventorySettingsRequest, UpdateInventorySettingsResponse,
  GetInventorySettingsRequest, GetInventorySettingsResponse,
  ListPrintingTemplateRequest, ListPrintingTemplateResponse,
  GetPrintingTemplateRequest, GetPrintingTemplateResponse,
  GetDefaultPrintingTemplateRequest, GetDefaultPrintingTemplateResponse,
  CreatePrintingTemplateRequest, CreatePrintingTemplateResponse,
  UpdatePrintingTemplateRequest, UpdatePrintingTemplateResponse,
  DeletePrintingTemplateRequest, DeletePrintingTemplateResponse,
  SelectPrintingTemplateRequest, SelectPrintingTemplateResponse,
  GetPrintingTemplateRelationRequest, GetPrintingTemplateRelationResponse,
  UpdatePrintingTemplateRelationRequest, UpdatePrintingTemplateRelationResponse,
  UpdatePurchaseSettingsRequest, UpdatePurchaseSettingsResponse,
  GetPurchaseSettingsRequest, GetPurchaseSettingsResponse,
  UpdateAppointTimeSettingsRequest, UpdateAppointTimeSettingsResponse,
  GetAppointTimeSettingsRequest, GetAppointTimeSettingsResponse,
  UpdateSortingSettingsRequest, UpdateSortingSettingsResponse,
  GetSortingSettingsRequest, GetSortingSettingsResponse,
  UpdateProductionSettingsRequest, UpdateProductionSettingsResponse,
  GetProductionSettingsRequest, GetProductionSettingsResponse,
  GetManyAppointTimePairRequest, GetManyAppointTimePairResponse,
  GetPrecisionSettingsRequest, GetPrecisionSettingsResponse,
  GetShopRequest, GetShopResponse,
  ListShopRequest, ListShopResponse,
  CreateShopRequest, CreateShopResponse,
  UpdateShopRequest, UpdateShopResponse,
  ListShopBannerRequest, ListShopBannerResponse,
  CreateShopBannerRequest, CreateShopBannerResponse,
  DeleteShopBannerRequest, DeleteShopBannerResponse,
  GetApplicationRelationRequest, GetApplicationRelationResponse,
  ListApplicationRequest, ListApplicationResponse,
} from './types'
export const CreateInventorySettings = (r?: CreateInventorySettingsRequest, codes?: number[]) =>
  Request<CreateInventorySettingsResponse>('/ceres/preference/PreferenceService/CreateInventorySettings')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateInventorySettings.url = '/ceres/preference/PreferenceService/CreateInventorySettings'
export const UpdateInventorySettings = (r?: UpdateInventorySettingsRequest, codes?: number[]) =>
  Request<UpdateInventorySettingsResponse>('/ceres/preference/PreferenceService/UpdateInventorySettings')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateInventorySettings.url = '/ceres/preference/PreferenceService/UpdateInventorySettings'
export const GetInventorySettings = (r?: GetInventorySettingsRequest, codes?: number[]) =>
  Request<GetInventorySettingsResponse>('/ceres/preference/PreferenceService/GetInventorySettings')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetInventorySettings.url = '/ceres/preference/PreferenceService/GetInventorySettings'
export const ListPrintingTemplate = (r?: ListPrintingTemplateRequest, codes?: number[]) =>
  Request<ListPrintingTemplateResponse>('/ceres/preference/PreferenceService/ListPrintingTemplate')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListPrintingTemplate.url = '/ceres/preference/PreferenceService/ListPrintingTemplate'
export const GetPrintingTemplate = (r?: GetPrintingTemplateRequest, codes?: number[]) =>
  Request<GetPrintingTemplateResponse>('/ceres/preference/PreferenceService/GetPrintingTemplate')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetPrintingTemplate.url = '/ceres/preference/PreferenceService/GetPrintingTemplate'
export const GetDefaultPrintingTemplate = (r?: GetDefaultPrintingTemplateRequest, codes?: number[]) =>
  Request<GetDefaultPrintingTemplateResponse>('/ceres/preference/PreferenceService/GetDefaultPrintingTemplate')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetDefaultPrintingTemplate.url = '/ceres/preference/PreferenceService/GetDefaultPrintingTemplate'
export const CreatePrintingTemplate = (r?: CreatePrintingTemplateRequest, codes?: number[]) =>
  Request<CreatePrintingTemplateResponse>('/ceres/preference/PreferenceService/CreatePrintingTemplate')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreatePrintingTemplate.url = '/ceres/preference/PreferenceService/CreatePrintingTemplate'
export const UpdatePrintingTemplate = (r?: UpdatePrintingTemplateRequest, codes?: number[]) =>
  Request<UpdatePrintingTemplateResponse>('/ceres/preference/PreferenceService/UpdatePrintingTemplate')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdatePrintingTemplate.url = '/ceres/preference/PreferenceService/UpdatePrintingTemplate'
export const DeletePrintingTemplate = (r?: DeletePrintingTemplateRequest, codes?: number[]) =>
  Request<DeletePrintingTemplateResponse>('/ceres/preference/PreferenceService/DeletePrintingTemplate')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeletePrintingTemplate.url = '/ceres/preference/PreferenceService/DeletePrintingTemplate'
export const SelectPrintingTemplate = (r?: SelectPrintingTemplateRequest, codes?: number[]) =>
  Request<SelectPrintingTemplateResponse>('/ceres/preference/PreferenceService/SelectPrintingTemplate')
    .code([...(codes || [])])
    .data(r || {})
    .run()
SelectPrintingTemplate.url = '/ceres/preference/PreferenceService/SelectPrintingTemplate'
export const GetPrintingTemplateRelation = (r?: GetPrintingTemplateRelationRequest, codes?: number[]) =>
  Request<GetPrintingTemplateRelationResponse>('/ceres/preference/PreferenceService/GetPrintingTemplateRelation')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetPrintingTemplateRelation.url = '/ceres/preference/PreferenceService/GetPrintingTemplateRelation'
export const UpdatePrintingTemplateRelation = (r?: UpdatePrintingTemplateRelationRequest, codes?: number[]) =>
  Request<UpdatePrintingTemplateRelationResponse>('/ceres/preference/PreferenceService/UpdatePrintingTemplateRelation')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdatePrintingTemplateRelation.url = '/ceres/preference/PreferenceService/UpdatePrintingTemplateRelation'
export const UpdatePurchaseSettings = (r?: UpdatePurchaseSettingsRequest, codes?: number[]) =>
  Request<UpdatePurchaseSettingsResponse>('/ceres/preference/PreferenceService/UpdatePurchaseSettings')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdatePurchaseSettings.url = '/ceres/preference/PreferenceService/UpdatePurchaseSettings'
export const GetPurchaseSettings = (r?: GetPurchaseSettingsRequest, codes?: number[]) =>
  Request<GetPurchaseSettingsResponse>('/ceres/preference/PreferenceService/GetPurchaseSettings')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetPurchaseSettings.url = '/ceres/preference/PreferenceService/GetPurchaseSettings'
export const UpdateAppointTimeSettings = (r?: UpdateAppointTimeSettingsRequest, codes?: number[]) =>
  Request<UpdateAppointTimeSettingsResponse>('/ceres/preference/PreferenceService/UpdateAppointTimeSettings')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateAppointTimeSettings.url = '/ceres/preference/PreferenceService/UpdateAppointTimeSettings'
export const GetAppointTimeSettings = (r?: GetAppointTimeSettingsRequest, codes?: number[]) =>
  Request<GetAppointTimeSettingsResponse>('/ceres/preference/PreferenceService/GetAppointTimeSettings')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetAppointTimeSettings.url = '/ceres/preference/PreferenceService/GetAppointTimeSettings'
export const UpdateSortingSettings = (r?: UpdateSortingSettingsRequest, codes?: number[]) =>
  Request<UpdateSortingSettingsResponse>('/ceres/preference/PreferenceService/UpdateSortingSettings')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateSortingSettings.url = '/ceres/preference/PreferenceService/UpdateSortingSettings'
export const GetSortingSettings = (r?: GetSortingSettingsRequest, codes?: number[]) =>
  Request<GetSortingSettingsResponse>('/ceres/preference/PreferenceService/GetSortingSettings')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetSortingSettings.url = '/ceres/preference/PreferenceService/GetSortingSettings'
export const UpdateProductionSettings = (r?: UpdateProductionSettingsRequest, codes?: number[]) =>
  Request<UpdateProductionSettingsResponse>('/ceres/preference/PreferenceService/UpdateProductionSettings')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateProductionSettings.url = '/ceres/preference/PreferenceService/UpdateProductionSettings'
export const GetProductionSettings = (r?: GetProductionSettingsRequest, codes?: number[]) =>
  Request<GetProductionSettingsResponse>('/ceres/preference/PreferenceService/GetProductionSettings')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetProductionSettings.url = '/ceres/preference/PreferenceService/GetProductionSettings'
export const GetManyAppointTimePair = (r?: GetManyAppointTimePairRequest, codes?: number[]) =>
  Request<GetManyAppointTimePairResponse>('/ceres/preference/PreferenceService/GetManyAppointTimePair')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetManyAppointTimePair.url = '/ceres/preference/PreferenceService/GetManyAppointTimePair'
export const GetPrecisionSettings = (r?: GetPrecisionSettingsRequest, codes?: number[]) =>
  Request<GetPrecisionSettingsResponse>('/ceres/preference/PreferenceService/GetPrecisionSettings')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetPrecisionSettings.url = '/ceres/preference/PreferenceService/GetPrecisionSettings'
export const GetShop = (r?: GetShopRequest, codes?: number[]) =>
  Request<GetShopResponse>('/ceres/preference/PreferenceService/GetShop')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetShop.url = '/ceres/preference/PreferenceService/GetShop'
export const ListShop = (r?: ListShopRequest, codes?: number[]) =>
  Request<ListShopResponse>('/ceres/preference/PreferenceService/ListShop')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListShop.url = '/ceres/preference/PreferenceService/ListShop'
export const CreateShop = (r?: CreateShopRequest, codes?: number[]) =>
  Request<CreateShopResponse>('/ceres/preference/PreferenceService/CreateShop')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateShop.url = '/ceres/preference/PreferenceService/CreateShop'
export const UpdateShop = (r?: UpdateShopRequest, codes?: number[]) =>
  Request<UpdateShopResponse>('/ceres/preference/PreferenceService/UpdateShop')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateShop.url = '/ceres/preference/PreferenceService/UpdateShop'
export const ListShopBanner = (r?: ListShopBannerRequest, codes?: number[]) =>
  Request<ListShopBannerResponse>('/ceres/preference/PreferenceService/ListShopBanner')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListShopBanner.url = '/ceres/preference/PreferenceService/ListShopBanner'
export const CreateShopBanner = (r?: CreateShopBannerRequest, codes?: number[]) =>
  Request<CreateShopBannerResponse>('/ceres/preference/PreferenceService/CreateShopBanner')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateShopBanner.url = '/ceres/preference/PreferenceService/CreateShopBanner'
export const DeleteShopBanner = (r?: DeleteShopBannerRequest, codes?: number[]) =>
  Request<DeleteShopBannerResponse>('/ceres/preference/PreferenceService/DeleteShopBanner')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteShopBanner.url = '/ceres/preference/PreferenceService/DeleteShopBanner'
export const GetApplicationRelation = (r?: GetApplicationRelationRequest, codes?: number[]) =>
  Request<GetApplicationRelationResponse>('/ceres/preference/PreferenceService/GetApplicationRelation')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetApplicationRelation.url = '/ceres/preference/PreferenceService/GetApplicationRelation'
