// gm_api/src/preference/pc.tsx
import React, { FC } from 'react'
import { Select, SelectProps } from '@gm-pc/react'
import {
  SelectList,
  SelectEnum,
  MoreSelectList,
  MapId,
} from '../util/pc_util'
import type {
  MapProps,
  SelectListProps,
  SelectEnumProps,
  MoreSelectListProps,
  MapIdProps,
} from '../util/pc_util'
import {
  list_Status_Code,
  map_Status_Code,
  list_PrintingTemplate_Direction,
  map_PrintingTemplate_Direction,
  list_PrintingTemplate_Type,
  map_PrintingTemplate_Type,
  list_Shop_Type,
  map_Shop_Type,
  list_Application_Type,
  map_Application_Type,
  list_PurchaseSettings_PurchaseTaskPlanValueCalulateMethod,
  map_PurchaseSettings_PurchaseTaskPlanValueCalulateMethod,
  list_AppointTimeSettings_Type,
  map_AppointTimeSettings_Type,
  list_PrecisionSettings_Type,
  map_PrecisionSettings_Type,
  list_SortingSettings_SortingNumMethod,
  map_SortingSettings_SortingNumMethod,
  list_SortingSettings_SortingLockType,
  map_SortingSettings_SortingLockType,
  list_InventorySettings_InventoryMethod,
  map_InventorySettings_InventoryMethod,
  list_ProductionSettings_TaskOutputSource,
  map_ProductionSettings_TaskOutputSource,
  PrintingTemplate,
  ListPrintingTemplate,
  ListPrintingTemplateResponse,
  ShopBanner,
  ListShopBanner,
  ListShopBannerResponse,
  Shop,
  ListShop,
  ListShopResponse,
  Application,
  ListApplication,
  ListApplicationResponse,
} from './index'

export const Select_Status_Code: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Status_Code} />
}

export const Map_Status_Code: FC<MapProps> = ({ value }) => {
  return <>{map_Status_Code[value]}</>
}

export const Select_PrintingTemplate_Direction: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_PrintingTemplate_Direction} />
}

export const Map_PrintingTemplate_Direction: FC<MapProps> = ({ value }) => {
  return <>{map_PrintingTemplate_Direction[value]}</>
}

export const Select_PrintingTemplate_Type: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_PrintingTemplate_Type} />
}

export const Map_PrintingTemplate_Type: FC<MapProps> = ({ value }) => {
  return <>{map_PrintingTemplate_Type[value]}</>
}

export const Select_Shop_Type: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Shop_Type} />
}

export const Map_Shop_Type: FC<MapProps> = ({ value }) => {
  return <>{map_Shop_Type[value]}</>
}

export const Select_Application_Type: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Application_Type} />
}

export const Map_Application_Type: FC<MapProps> = ({ value }) => {
  return <>{map_Application_Type[value]}</>
}

export const Select_PurchaseSettings_PurchaseTaskPlanValueCalulateMethod: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_PurchaseSettings_PurchaseTaskPlanValueCalulateMethod} />
}

export const Map_PurchaseSettings_PurchaseTaskPlanValueCalulateMethod: FC<MapProps> = ({ value }) => {
  return <>{map_PurchaseSettings_PurchaseTaskPlanValueCalulateMethod[value]}</>
}

export const Select_AppointTimeSettings_Type: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_AppointTimeSettings_Type} />
}

export const Map_AppointTimeSettings_Type: FC<MapProps> = ({ value }) => {
  return <>{map_AppointTimeSettings_Type[value]}</>
}

export const Select_PrecisionSettings_Type: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_PrecisionSettings_Type} />
}

export const Map_PrecisionSettings_Type: FC<MapProps> = ({ value }) => {
  return <>{map_PrecisionSettings_Type[value]}</>
}

export const Select_SortingSettings_SortingNumMethod: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_SortingSettings_SortingNumMethod} />
}

export const Map_SortingSettings_SortingNumMethod: FC<MapProps> = ({ value }) => {
  return <>{map_SortingSettings_SortingNumMethod[value]}</>
}

export const Select_SortingSettings_SortingLockType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_SortingSettings_SortingLockType} />
}

export const Map_SortingSettings_SortingLockType: FC<MapProps> = ({ value }) => {
  return <>{map_SortingSettings_SortingLockType[value]}</>
}

export const Select_InventorySettings_InventoryMethod: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_InventorySettings_InventoryMethod} />
}

export const Map_InventorySettings_InventoryMethod: FC<MapProps> = ({ value }) => {
  return <>{map_InventorySettings_InventoryMethod[value]}</>
}

export const Select_ProductionSettings_TaskOutputSource: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_ProductionSettings_TaskOutputSource} />
}

export const Map_ProductionSettings_TaskOutputSource: FC<MapProps> = ({ value }) => {
  return <>{map_ProductionSettings_TaskOutputSource[value]}</>
}

export const Select_PrintingTemplate: FC<SelectListProps<
  PrintingTemplate,
  ListPrintingTemplateResponse
>> = (props) => {
  return (
    <SelectList<PrintingTemplate, ListPrintingTemplateResponse>
      {...props}
      messageName='PrintingTemplate'
      fetchList={ListPrintingTemplate}
    />
  )
}

export const MoreSelect_PrintingTemplate: FC<MoreSelectListProps<
  PrintingTemplate,
  ListPrintingTemplateResponse
>> = (props) => {
  return (
    <MoreSelectList<PrintingTemplate, ListPrintingTemplateResponse>
      {...props}
      messageName='PrintingTemplate'
      fetchList={ListPrintingTemplate}
    />
  )
}

export const Select_ShopBanner: FC<SelectListProps<
  ShopBanner,
  ListShopBannerResponse
>> = (props) => {
  return (
    <SelectList<ShopBanner, ListShopBannerResponse>
      {...props}
      messageName='ShopBanner'
      fetchList={ListShopBanner}
    />
  )
}

export const MoreSelect_ShopBanner: FC<MoreSelectListProps<
  ShopBanner,
  ListShopBannerResponse
>> = (props) => {
  return (
    <MoreSelectList<ShopBanner, ListShopBannerResponse>
      {...props}
      messageName='ShopBanner'
      fetchList={ListShopBanner}
    />
  )
}

export const Select_Shop: FC<SelectListProps<
  Shop,
  ListShopResponse
>> = (props) => {
  return (
    <SelectList<Shop, ListShopResponse>
      {...props}
      messageName='Shop'
      fetchList={ListShop}
    />
  )
}

export const MoreSelect_Shop: FC<MoreSelectListProps<
  Shop,
  ListShopResponse
>> = (props) => {
  return (
    <MoreSelectList<Shop, ListShopResponse>
      {...props}
      messageName='Shop'
      fetchList={ListShop}
    />
  )
}

export const Select_Application: FC<SelectListProps<
  Application,
  ListApplicationResponse
>> = (props) => {
  return (
    <SelectList<Application, ListApplicationResponse>
      {...props}
      messageName='Application'
      fetchList={ListApplication}
    />
  )
}

export const MoreSelect_Application: FC<MoreSelectListProps<
  Application,
  ListApplicationResponse
>> = (props) => {
  return (
    <MoreSelectList<Application, ListApplicationResponse>
      {...props}
      messageName='Application'
      fetchList={ListApplication}
    />
  )
}

export type {
  MapProps,
  SelectListProps,
  MoreSelectListProps,
}
