// gm_api/src/delivery/methods.ts
import { Request } from '../util/request'
import {
  CreateRouteRequest, CreateRouteResponse,
  UpdateRouteRequest, UpdateRouteResponse,
  GetRouteRequest, GetRouteResponse,
  ListRouteRequest, ListRouteResponse,
  DeleteRouteRequest, DeleteRouteResponse,
  GetManyRouteRequest, GetManyRouteResponse,
  GetManyCustomerRouteRequest, GetManyCustomerRouteResponse,
  ListDriverDeliveryTaskRequest, ListDriverDeliveryTaskResponse,
  UpdateDeliveryTaskRequest, UpdateDeliveryTaskResponse,
} from './types'
export const CreateRoute = (r?: CreateRouteRequest, codes?: number[]) =>
  Request<CreateRouteResponse>('/ceres/delivery/DeliveryService/CreateRoute')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateRoute.url = '/ceres/delivery/DeliveryService/CreateRoute'
export const UpdateRoute = (r?: UpdateRouteRequest, codes?: number[]) =>
  Request<UpdateRouteResponse>('/ceres/delivery/DeliveryService/UpdateRoute')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateRoute.url = '/ceres/delivery/DeliveryService/UpdateRoute'
export const GetRoute = (r?: GetRouteRequest, codes?: number[]) =>
  Request<GetRouteResponse>('/ceres/delivery/DeliveryService/GetRoute')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetRoute.url = '/ceres/delivery/DeliveryService/GetRoute'
export const ListRoute = (r?: ListRouteRequest, codes?: number[]) =>
  Request<ListRouteResponse>('/ceres/delivery/DeliveryService/ListRoute')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListRoute.url = '/ceres/delivery/DeliveryService/ListRoute'
export const DeleteRoute = (r?: DeleteRouteRequest, codes?: number[]) =>
  Request<DeleteRouteResponse>('/ceres/delivery/DeliveryService/DeleteRoute')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteRoute.url = '/ceres/delivery/DeliveryService/DeleteRoute'
export const GetManyRoute = (r?: GetManyRouteRequest, codes?: number[]) =>
  Request<GetManyRouteResponse>('/ceres/delivery/DeliveryService/GetManyRoute')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetManyRoute.url = '/ceres/delivery/DeliveryService/GetManyRoute'
export const GetManyCustomerRoute = (r?: GetManyCustomerRouteRequest, codes?: number[]) =>
  Request<GetManyCustomerRouteResponse>('/ceres/delivery/DeliveryService/GetManyCustomerRoute')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetManyCustomerRoute.url = '/ceres/delivery/DeliveryService/GetManyCustomerRoute'
export const ListDriverDeliveryTask = (r?: ListDriverDeliveryTaskRequest, codes?: number[]) =>
  Request<ListDriverDeliveryTaskResponse>('/ceres/delivery/DeliveryService/ListDriverDeliveryTask')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListDriverDeliveryTask.url = '/ceres/delivery/DeliveryService/ListDriverDeliveryTask'
export const UpdateDeliveryTask = (r?: UpdateDeliveryTaskRequest, codes?: number[]) =>
  Request<UpdateDeliveryTaskResponse>('/ceres/delivery/DeliveryService/UpdateDeliveryTask')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateDeliveryTask.url = '/ceres/delivery/DeliveryService/UpdateDeliveryTask'
