// gm_api/src/enterprise/data.ts
import { getLocale } from './lng'

interface Enum_List_Item {
  text: string
  value: any
}

interface Enum_Map {
  [key: string]: string
}
export const list_Status_Code: Enum_List_Item[] = [
  { value: 23330001, text: getLocale('Status_Code__DUPLICATE_ROLE_NAME') },
  { value: 23330002, text: getLocale('Status_Code__DUPLICATE_USER_NAME') },
  { value: 23330004, text: getLocale('Status_Code__DUPLICATE_USER_USERNAME') },
  { value: 23330005, text: getLocale('Status_Code__DUPLICATE_USER_PHONE') },
  { value: 23330003, text: getLocale('Status_Code__DUPLICATE_GROUP_NAME') },
]
export const map_Status_Code: Enum_Map = {
  23330001: getLocale('Status_Code__DUPLICATE_ROLE_NAME'),
  23330002: getLocale('Status_Code__DUPLICATE_USER_NAME'),
  23330004: getLocale('Status_Code__DUPLICATE_USER_USERNAME'),
  23330005: getLocale('Status_Code__DUPLICATE_USER_PHONE'),
  23330003: getLocale('Status_Code__DUPLICATE_GROUP_NAME'),
}
export const list_Group_Type: Enum_List_Item[] = [
  { value: 1, text: getLocale('Group_Type__INTERNAL') },
  { value: 2, text: getLocale('Group_Type__EXTERNAL') },
]
export const map_Group_Type: Enum_Map = {
  1: getLocale('Group_Type__INTERNAL'),
  2: getLocale('Group_Type__EXTERNAL'),
}
export const list_GroupExtraInfo_Source: Enum_List_Item[] = [
  { value: 1, text: getLocale('GroupExtraInfo_Source__SOURCE_PROMOTION') },
  { value: 2, text: getLocale('GroupExtraInfo_Source__SOURCE_REFERRAL') },
  { value: 3, text: getLocale('GroupExtraInfo_Source__SOURCE_SELF_EXPANSION') },
]
export const map_GroupExtraInfo_Source: Enum_Map = {
  1: getLocale('GroupExtraInfo_Source__SOURCE_PROMOTION'),
  2: getLocale('GroupExtraInfo_Source__SOURCE_REFERRAL'),
  3: getLocale('GroupExtraInfo_Source__SOURCE_SELF_EXPANSION'),
}
export const list_GroupExtraInfo_Target: Enum_List_Item[] = [
  { value: 1, text: getLocale('GroupExtraInfo_Target__TARGET_STORES') },
  { value: 2, text: getLocale('GroupExtraInfo_Target__TARGET_RESTAURANT') },
  { value: 3, text: getLocale('GroupExtraInfo_Target__TARGET_HOTEL') },
  { value: 4, text: getLocale('GroupExtraInfo_Target__TARGET_GROUP_MEALS') },
  { value: 5, text: getLocale('GroupExtraInfo_Target__TARGET_NUTRITIONAL_MEALS') },
  { value: 6, text: getLocale('GroupExtraInfo_Target__TARGET_INDIVIDUAL') },
  { value: 7, text: getLocale('GroupExtraInfo_Target__TARGET_RETAIL') },
  { value: 8, text: getLocale('GroupExtraInfo_Target__TARGET_OTHER') },
]
export const map_GroupExtraInfo_Target: Enum_Map = {
  1: getLocale('GroupExtraInfo_Target__TARGET_STORES'),
  2: getLocale('GroupExtraInfo_Target__TARGET_RESTAURANT'),
  3: getLocale('GroupExtraInfo_Target__TARGET_HOTEL'),
  4: getLocale('GroupExtraInfo_Target__TARGET_GROUP_MEALS'),
  5: getLocale('GroupExtraInfo_Target__TARGET_NUTRITIONAL_MEALS'),
  6: getLocale('GroupExtraInfo_Target__TARGET_INDIVIDUAL'),
  7: getLocale('GroupExtraInfo_Target__TARGET_RETAIL'),
  8: getLocale('GroupExtraInfo_Target__TARGET_OTHER'),
}
export const list_GroupExtraInfo_AnnualSalesAmount: Enum_List_Item[] = [
  { value: 1, text: getLocale('GroupExtraInfo_AnnualSalesAmount__ANNUAL_SALES_AMOUNT_0_1000') },
  { value: 2, text: getLocale('GroupExtraInfo_AnnualSalesAmount__ANNUAL_SALES_AMOUNT_1000_3000') },
  { value: 3, text: getLocale('GroupExtraInfo_AnnualSalesAmount__ANNUAL_SALES_AMOUNT_3000_5000') },
  { value: 4, text: getLocale('GroupExtraInfo_AnnualSalesAmount__ANNUAL_SALES_AMOUNT_5000_10000') },
  { value: 5, text: getLocale('GroupExtraInfo_AnnualSalesAmount__ANNUAL_SALES_AMOUNT_10000_INF') },
]
export const map_GroupExtraInfo_AnnualSalesAmount: Enum_Map = {
  1: getLocale('GroupExtraInfo_AnnualSalesAmount__ANNUAL_SALES_AMOUNT_0_1000'),
  2: getLocale('GroupExtraInfo_AnnualSalesAmount__ANNUAL_SALES_AMOUNT_1000_3000'),
  3: getLocale('GroupExtraInfo_AnnualSalesAmount__ANNUAL_SALES_AMOUNT_3000_5000'),
  4: getLocale('GroupExtraInfo_AnnualSalesAmount__ANNUAL_SALES_AMOUNT_5000_10000'),
  5: getLocale('GroupExtraInfo_AnnualSalesAmount__ANNUAL_SALES_AMOUNT_10000_INF'),
}
export const list_GroupExtraInfo_MonthlySalesAmount: Enum_List_Item[] = [
  { value: 1, text: getLocale('GroupExtraInfo_MonthlySalesAmount__MONTHLY_SALES_AMOUNT_0_50') },
  { value: 2, text: getLocale('GroupExtraInfo_MonthlySalesAmount__MONTHLY_SALES_AMOUNT_50_100') },
  { value: 3, text: getLocale('GroupExtraInfo_MonthlySalesAmount__MONTHLY_SALES_AMOUNT_100_200') },
  { value: 4, text: getLocale('GroupExtraInfo_MonthlySalesAmount__MONTHLY_SALES_AMOUNT_200_500') },
  { value: 5, text: getLocale('GroupExtraInfo_MonthlySalesAmount__MONTHLY_SALES_AMOUNT_500_1000') },
  { value: 6, text: getLocale('GroupExtraInfo_MonthlySalesAmount__MONTHLY_SALES_AMOUNT_1000_INF') },
]
export const map_GroupExtraInfo_MonthlySalesAmount: Enum_Map = {
  1: getLocale('GroupExtraInfo_MonthlySalesAmount__MONTHLY_SALES_AMOUNT_0_50'),
  2: getLocale('GroupExtraInfo_MonthlySalesAmount__MONTHLY_SALES_AMOUNT_50_100'),
  3: getLocale('GroupExtraInfo_MonthlySalesAmount__MONTHLY_SALES_AMOUNT_100_200'),
  4: getLocale('GroupExtraInfo_MonthlySalesAmount__MONTHLY_SALES_AMOUNT_200_500'),
  5: getLocale('GroupExtraInfo_MonthlySalesAmount__MONTHLY_SALES_AMOUNT_500_1000'),
  6: getLocale('GroupExtraInfo_MonthlySalesAmount__MONTHLY_SALES_AMOUNT_1000_INF'),
}
export const list_GroupExtraInfo_Type: Enum_List_Item[] = [
  { value: 1, text: getLocale('GroupExtraInfo_Type__TYPE_NORMAL') },
  { value: 2, text: getLocale('GroupExtraInfo_Type__TYPE_KA') },
]
export const map_GroupExtraInfo_Type: Enum_Map = {
  1: getLocale('GroupExtraInfo_Type__TYPE_NORMAL'),
  2: getLocale('GroupExtraInfo_Type__TYPE_KA'),
}
export const list_GroupExtraInfo_Version: Enum_List_Item[] = [
  { value: 1, text: getLocale('GroupExtraInfo_Version__VERSION_STANDARD') },
]
export const map_GroupExtraInfo_Version: Enum_Map = {
  1: getLocale('GroupExtraInfo_Version__VERSION_STANDARD'),
}
export const list_GroupExtraInfo_State: Enum_List_Item[] = [
  { value: 1, text: getLocale('GroupExtraInfo_State__STATE_NORMAL') },
  { value: 2, text: getLocale('GroupExtraInfo_State__STATE_ONLINE') },
  { value: 3, text: getLocale('GroupExtraInfo_State__STATE_OFFLINE') },
  { value: 4, text: getLocale('GroupExtraInfo_State__STATE_STOP') },
]
export const map_GroupExtraInfo_State: Enum_Map = {
  1: getLocale('GroupExtraInfo_State__STATE_NORMAL'),
  2: getLocale('GroupExtraInfo_State__STATE_ONLINE'),
  3: getLocale('GroupExtraInfo_State__STATE_OFFLINE'),
  4: getLocale('GroupExtraInfo_State__STATE_STOP'),
}
export const list_Station_Type: Enum_List_Item[] = [
  { value: 1, text: getLocale('Station_Type__NORMAL') },
]
export const map_Station_Type: Enum_Map = {
  1: getLocale('Station_Type__NORMAL'),
}
export const list_CreditType: Enum_List_Item[] = [
  { value: 1, text: getLocale('CreditType__CREDIT_TYPE_AMOUNT') },
  { value: 3, text: getLocale('CreditType__CREDIT_TYPE_DAILY') },
  { value: 4, text: getLocale('CreditType__CREDIT_TYPE_WEEKLY') },
  { value: 5, text: getLocale('CreditType__CREDIT_TYPE_MONTHLY') },
  { value: 6, text: getLocale('CreditType__CREDIT_TYPE_CUSTOMIZED') },
  { value: 8, text: getLocale('CreditType__CREDIT_TYPE_HALF_MONTHLY') },
]
export const map_CreditType: Enum_Map = {
  1: getLocale('CreditType__CREDIT_TYPE_AMOUNT'),
  3: getLocale('CreditType__CREDIT_TYPE_DAILY'),
  4: getLocale('CreditType__CREDIT_TYPE_WEEKLY'),
  5: getLocale('CreditType__CREDIT_TYPE_MONTHLY'),
  6: getLocale('CreditType__CREDIT_TYPE_CUSTOMIZED'),
  8: getLocale('CreditType__CREDIT_TYPE_HALF_MONTHLY'),
}
export const list_Permission: Enum_List_Item[] = [
  { value: 10, text: getLocale('Permission__PERMISSION_MERCHANDISE_VIEW_QUOTATION') },
  { value: 11, text: getLocale('Permission__PERMISSION_MERCHANDISE_CREATE_QUOTATION') },
  { value: 12, text: getLocale('Permission__PERMISSION_MERCHANDISE_UPDATE_QUOTATION') },
  { value: 13, text: getLocale('Permission__PERMISSION_MERCHANDISE_DELETE_QUOTATION') },
  { value: 20, text: getLocale('Permission__PERMISSION_MERCHANDISE_VIEW_NOT_PACKAGE_SKU_SSU') },
  { value: 21, text: getLocale('Permission__PERMISSION_MERCHANDISE_CREATE_NOT_PACKAGE_SKU_SSU') },
  { value: 22, text: getLocale('Permission__PERMISSION_MERCHANDISE_UPDATE_NOT_PACKAGE_SKU_SSU') },
  { value: 23, text: getLocale('Permission__PERMISSION_MERCHANDISE_DELETE_NOT_PACKAGE_SKU_SSU') },
  { value: 30, text: getLocale('Permission__PERMISSION_MERCHANDISE_VIEW_PACKAGE_SKU_SSU') },
  { value: 31, text: getLocale('Permission__PERMISSION_MERCHANDISE_CREATE_PACKAGE_SKU_SSU') },
  { value: 32, text: getLocale('Permission__PERMISSION_MERCHANDISE_UPDATE_PACKAGE_SKU_SSU') },
  { value: 33, text: getLocale('Permission__PERMISSION_MERCHANDISE_DELETE_PACKAGE_SKU_SSU') },
  { value: 40, text: getLocale('Permission__PERMISSION_PRODUCTION_VIEW_BOM') },
  { value: 41, text: getLocale('Permission__PERMISSION_PRODUCTION_CREATE_BOM') },
  { value: 42, text: getLocale('Permission__PERMISSION_PRODUCTION_UPDATE_BOM') },
  { value: 43, text: getLocale('Permission__PERMISSION_PRODUCTION_DELETE_BOM') },
  { value: 50, text: getLocale('Permission__PERMISSION_MERCHANDISE_VIEW_CATEGORY') },
  { value: 51, text: getLocale('Permission__PERMISSION_MERCHANDISE_CREATE_CATEGORY') },
  { value: 52, text: getLocale('Permission__PERMISSION_MERCHANDISE_UPDATE_CATEGORY') },
  { value: 53, text: getLocale('Permission__PERMISSION_MERCHANDISE_DELETE_CATEGORY') },
  { value: 500, text: getLocale('Permission__PERMISSION_ESHOP_VIEW_MENU') },
  { value: 502, text: getLocale('Permission__PERMISSION_ESHOP_UPDATE_MENU') },
  { value: 504, text: getLocale('Permission__PERMISSION_ESHOP_VIEW_MENU_DETAIL') },
  { value: 505, text: getLocale('Permission__PERMISSION_ESHOP_UPDATE_MENU_DETAIL') },
  { value: 501, text: getLocale('Permission__PERMISSION_ESHOP_CREATE_MENU') },
  { value: 503, text: getLocale('Permission__PERMISSION_ESHOP_DELETE_MENU') },
  { value: 510, text: getLocale('Permission__PERMISSION_ESHOP_VIEW_MENU_PERIOD') },
  { value: 511, text: getLocale('Permission__PERMISSION_ESHOP_CREATE_MENU_PERIOD') },
  { value: 512, text: getLocale('Permission__PERMISSION_ESHOP_UPDATE_MENU_PERIOD') },
  { value: 513, text: getLocale('Permission__PERMISSION_ESHOP_DELETE_MENU_PERIOD') },
  { value: 60, text: getLocale('Permission__PERMISSION_ORDER_VIEW_ORDER') },
  { value: 61, text: getLocale('Permission__PERMISSION_ORDER_CREATE_ORDER') },
  { value: 62, text: getLocale('Permission__PERMISSION_ORDER_UPDATE_ORDER') },
  { value: 63, text: getLocale('Permission__PERMISSION_ORDER_DELETE_ORDER') },
  { value: 64, text: getLocale('Permission__PERMISSION_ORDER_AMEND_ORDER') },
  { value: 65, text: getLocale('Permission__PERMISSION_ORDER_RELEASE_PROCESS_PLAN') },
  { value: 66, text: getLocale('Permission__PERMISSION_ORDER_RELEASE_NONPROCESS_PLAN') },
  { value: 530, text: getLocale('Permission__PERMISSION_ORDER_VIEW_ESHOP_STUDENT_ORDER') },
  { value: 531, text: getLocale('Permission__PERMISSION_ORDER_DELETE_ESHOP_STUDENT_ORDER') },
  { value: 532, text: getLocale('Permission__PERMISSION_ORDER_VIEW_ESHOP_STUFF_ORDER') },
  { value: 533, text: getLocale('Permission__PERMISSION_ORDER_DELETE_ESHOP_STUFF_ORDER') },
  { value: 539, text: getLocale('Permission__PERMISSION_ORDER_VIEW_LEAVE_APPLICATION') },
  { value: 70, text: getLocale('Permission__PERMISSION_PURCHASE_VIEW_PURCHASE_TASK') },
  { value: 71, text: getLocale('Permission__PERMISSION_PURCHASE_CREATE_PURCHASE_TASK') },
  { value: 72, text: getLocale('Permission__PERMISSION_PURCHASE_UPDATE_PURCHASE_TASK') },
  { value: 73, text: getLocale('Permission__PERMISSION_PURCHASE_DELETE_PURCHASE_TASK') },
  { value: 80, text: getLocale('Permission__PERMISSION_PURCHASE_VIEW_PURCHASE_SHEET') },
  { value: 81, text: getLocale('Permission__PERMISSION_PURCHASE_CREATE_PURCHASE_SHEET') },
  { value: 82, text: getLocale('Permission__PERMISSION_PURCHASE_UPDATE_PURCHASE_SHEET') },
  { value: 83, text: getLocale('Permission__PERMISSION_PURCHASE_DELETE_PURCHASE_SHEET') },
  { value: 90, text: getLocale('Permission__PERMISSION_ENTERPRISE_VIEW_PURCHASER') },
  { value: 91, text: getLocale('Permission__PERMISSION_ENTERPRISE_CREATE_PURCHASER') },
  { value: 92, text: getLocale('Permission__PERMISSION_ENTERPRISE_UPDATE_PURCHASER') },
  { value: 93, text: getLocale('Permission__PERMISSION_ENTERPRISE_DELETE_PURCHASER') },
  { value: 100, text: getLocale('Permission__PERMISSION_PURCHASE_VIEW_INQUIRY_PRICE') },
  { value: 101, text: getLocale('Permission__PERMISSION_PURCHASE_IMPORT_INQUIRY_PRICE') },
  { value: 110, text: getLocale('Permission__PERMISSION_PRODUCTION_LIST_PRODUCETASK') },
  { value: 111, text: getLocale('Permission__PERMISSION_PRODUCTION_CREATE_PRODUCETASK') },
  { value: 112, text: getLocale('Permission__PERMISSION_PRODUCTION_UPDATE_PRODUCETASK') },
  { value: 113, text: getLocale('Permission__PERMISSION_PRODUCTION_DELETE_PRODUCETASK') },
  { value: 120, text: getLocale('Permission__PERMISSION_PRODUCTION_LIST_PACKTASK') },
  { value: 121, text: getLocale('Permission__PERMISSION_PRODUCTION_CREATE_PACKTASK') },
  { value: 122, text: getLocale('Permission__PERMISSION_PRODUCTION_UPDATE_PACKTASK') },
  { value: 123, text: getLocale('Permission__PERMISSION_PRODUCTION_DELETE_PACKTASK') },
  { value: 130, text: getLocale('Permission__PERMISSION_PRODUCTION_VIEW_PRODUCEPROGRESS') },
  { value: 131, text: getLocale('Permission__PERMISSION_PRODUCTION_VIEW_PACKPROGRESS') },
  { value: 132, text: getLocale('Permission__PERMISSION_PRODUCTION_VIEW_PRODUCEANALYTICS') },
  { value: 133, text: getLocale('Permission__PERMISSION_PRODUCTION_VIEW_PACKANALYTICS') },
  { value: 134, text: getLocale('Permission__PERMISSION_PRODUCTION_PRINT_BOX_LABLE') },
  { value: 140, text: getLocale('Permission__PERMISSION_DELIVERY_VIEW_TASK') },
  { value: 141, text: getLocale('Permission__PERMISSION_DELIVERY_UPDATE_TASK') },
  { value: 142, text: getLocale('Permission__PERMISSION_DELIVERY_PRINT_TASK') },
  { value: 150, text: getLocale('Permission__PERMISSION_ENTERPRISE_VIEW_DRIVER') },
  { value: 151, text: getLocale('Permission__PERMISSION_ENTERPRISE_CREATE_DRIVER') },
  { value: 152, text: getLocale('Permission__PERMISSION_ENTERPRISE_UPDATE_DRIVER') },
  { value: 153, text: getLocale('Permission__PERMISSION_ENTERPRISE_DELETE_DRIVER') },
  { value: 160, text: getLocale('Permission__PERMISSION_ENTERPRISE_CREATE_DISTRIBUTION_CONTRACTOR') },
  { value: 161, text: getLocale('Permission__PERMISSION_ENTERPRISE_UPDATE_DISTRIBUTION_CONTRACTOR') },
  { value: 162, text: getLocale('Permission__PERMISSION_ENTERPRISE_DELETE_DISTRIBUTION_CONTRACTOR') },
  { value: 170, text: getLocale('Permission__PERMISSION_DELIVERY_VIEW_ROUTE') },
  { value: 171, text: getLocale('Permission__PERMISSION_DELIVERY_CREATE_ROUTE') },
  { value: 172, text: getLocale('Permission__PERMISSION_DELIVERY_UPDATE_ROUTE') },
  { value: 173, text: getLocale('Permission__PERMISSION_DELIVERY_DELETE_ROUTE') },
  { value: 180, text: getLocale('Permission__PERMISSION_INVENTORY_VIEW_PURCHASE_IN_SHEET') },
  { value: 181, text: getLocale('Permission__PERMISSION_INVENTORY_CREATE_PURCHASE_IN_SHEET') },
  { value: 182, text: getLocale('Permission__PERMISSION_INVENTORY_UPDATE_PURCHASE_IN_SHEET') },
  { value: 183, text: getLocale('Permission__PERMISSION_INVENTORY_SUBMIT_PURCHASE_IN_SHEET') },
  { value: 184, text: getLocale('Permission__PERMISSION_INVENTORY_APPROVE_PURCHASE_IN_SHEET') },
  { value: 185, text: getLocale('Permission__PERMISSION_INVENTORY_NOT_APPROVE_PURCHASE_IN_SHEET') },
  { value: 186, text: getLocale('Permission__PERMISSION_INVENTORY_CANCEL_APPROVE_PURCHASE_IN_SHEET') },
  { value: 187, text: getLocale('Permission__PERMISSION_INVENTORY_DELETE_PURCHASE_IN_SHEET') },
  { value: 188, text: getLocale('Permission__PERMISSION_INVENTORY_EXPORT_PURCHASE_IN_SHEET') },
  { value: 189, text: getLocale('Permission__PERMISSION_INVENTORY_PRINT_PURCHASE_IN_SHEET') },
  { value: 200, text: getLocale('Permission__PERMISSION_INVENTORY_VIEW_REFUND_OUT_SHEET') },
  { value: 201, text: getLocale('Permission__PERMISSION_INVENTORY_CREATE_REFUND_OUT_SHEET') },
  { value: 202, text: getLocale('Permission__PERMISSION_INVENTORY_UPDATE_REFUND_OUT_SHEET') },
  { value: 203, text: getLocale('Permission__PERMISSION_INVENTORY_SUBMIT_REFUND_OUT_SHEET') },
  { value: 204, text: getLocale('Permission__PERMISSION_INVENTORY_APPROVE_REFUND_OUT_SHEET') },
  { value: 205, text: getLocale('Permission__PERMISSION_INVENTORY_NOT_APPROVE_REFUND_OUT_SHEET') },
  { value: 206, text: getLocale('Permission__PERMISSION_INVENTORY_CANCEL_APPROVE_REFUND_OUT_SHEET') },
  { value: 207, text: getLocale('Permission__PERMISSION_INVENTORY_DELETE_REFUND_OUT_SHEET') },
  { value: 208, text: getLocale('Permission__PERMISSION_INVENTORY_EXPORT_REFUND_OUT_SHEET') },
  { value: 209, text: getLocale('Permission__PERMISSION_INVENTORY_PRINT_REFUND_OUT_SHEET') },
  { value: 220, text: getLocale('Permission__PERMISSION_INVENTORY_VIEW_SALE_OUT_SHEET') },
  { value: 221, text: getLocale('Permission__PERMISSION_INVENTORY_CREATE_SALE_OUT_SHEET') },
  { value: 222, text: getLocale('Permission__PERMISSION_INVENTORY_UPDATE_SALE_OUT_SHEET') },
  { value: 223, text: getLocale('Permission__PERMISSION_INVENTORY_SUBMIT_SALE_OUT_SHEET') },
  { value: 224, text: getLocale('Permission__PERMISSION_INVENTORY_APPROVE_SALE_OUT_SHEET') },
  { value: 225, text: getLocale('Permission__PERMISSION_INVENTORY_NOT_APPROVE_SALE_OUT_SHEET') },
  { value: 226, text: getLocale('Permission__PERMISSION_INVENTORY_CANCEL_APPROVE_SALE_OUT_SHEET') },
  { value: 227, text: getLocale('Permission__PERMISSION_INVENTORY_DELETE_SALE_OUT_SHEET') },
  { value: 228, text: getLocale('Permission__PERMISSION_INVENTORY_EXPORT_SALE_OUT_SHEET') },
  { value: 229, text: getLocale('Permission__PERMISSION_INVENTORY_PRINT_SALE_OUT_SHEET') },
  { value: 230, text: getLocale('Permission__PERMISSION_INVENTORY_VIEW_PRODUCT_IN') },
  { value: 231, text: getLocale('Permission__PERMISSION_INVENTORY_CREATE_PRODUCT_IN') },
  { value: 232, text: getLocale('Permission__PERMISSION_INVENTORY_UPDATE_PRODUCT_IN') },
  { value: 233, text: getLocale('Permission__PERMISSION_INVENTORY_SUBMIT_PRODUCT_IN') },
  { value: 234, text: getLocale('Permission__PERMISSION_INVENTORY_APPROVE_PRODUCT_IN') },
  { value: 235, text: getLocale('Permission__PERMISSION_INVENTORY_NOT_APPROVE_PRODUCT_IN') },
  { value: 236, text: getLocale('Permission__PERMISSION_INVENTORY_CANCEL_APPROVE_PRODUCT_IN') },
  { value: 237, text: getLocale('Permission__PERMISSION_INVENTORY_DELETE_PRODUCT_IN') },
  { value: 238, text: getLocale('Permission__PERMISSION_INVENTORY_EXPORT_PRODUCT_IN') },
  { value: 239, text: getLocale('Permission__PERMISSION_INVENTORY_PRINT_PRODUCT_IN') },
  { value: 250, text: getLocale('Permission__PERMISSION_INVENTORY_VIEW_MATERIAL_IN_250') },
  { value: 251, text: getLocale('Permission__PERMISSION_INVENTORY_CREATE_MATERIAL_IN_251') },
  { value: 252, text: getLocale('Permission__PERMISSION_INVENTORY_UPDATE_MATERIAL_IN_252') },
  { value: 253, text: getLocale('Permission__PERMISSION_INVENTORY_SUBMIT_MATERIAL_IN_253') },
  { value: 254, text: getLocale('Permission__PERMISSION_INVENTORY_APPROVE_MATERIAL_IN_254') },
  { value: 255, text: getLocale('Permission__PERMISSION_INVENTORY_NOT_APPROVE_MATERIAL_IN_255') },
  { value: 256, text: getLocale('Permission__PERMISSION_INVENTORY_CANCEL_APPROVE_MATERIAL_IN_256') },
  { value: 257, text: getLocale('Permission__PERMISSION_INVENTORY_DELETE_MATERIAL_IN_257') },
  { value: 258, text: getLocale('Permission__PERMISSION_INVENTORY_EXPORT_MATERIAL_IN_258') },
  { value: 259, text: getLocale('Permission__PERMISSION_INVENTORY_PRINT_MATERIAL_IN_259') },
  { value: 270, text: getLocale('Permission__PERMISSION_INVENTORY_VIEW_MATERIAL_OUT') },
  { value: 271, text: getLocale('Permission__PERMISSION_INVENTORY_CREATE_MATERIAL_OUT') },
  { value: 272, text: getLocale('Permission__PERMISSION_INVENTORY_UPDATE_MATERIAL_OUT') },
  { value: 273, text: getLocale('Permission__PERMISSION_INVENTORY_SUBMIT_MATERIAL_OUT') },
  { value: 274, text: getLocale('Permission__PERMISSION_INVENTORY_APPROVE_MATERIAL_OUT') },
  { value: 275, text: getLocale('Permission__PERMISSION_INVENTORY_NOT_APPROVE_MATERIAL_OUT') },
  { value: 276, text: getLocale('Permission__PERMISSION_INVENTORY_CANCEL_APPROVE_MATERIAL_OUT') },
  { value: 277, text: getLocale('Permission__PERMISSION_INVENTORY_DELETE_MATERIAL_OUT') },
  { value: 278, text: getLocale('Permission__PERMISSION_INVENTORY_EXPORT_MATERIAL_OUT') },
  { value: 279, text: getLocale('Permission__PERMISSION_INVENTORY_PRINT_MATERIAL_OUT') },
  { value: 290, text: getLocale('Permission__PERMISSION_INVENTORY_VIEW_OTHER_IN') },
  { value: 291, text: getLocale('Permission__PERMISSION_INVENTORY_CREATE_OTHER_IN') },
  { value: 292, text: getLocale('Permission__PERMISSION_INVENTORY_UPDATE_OTHER_IN') },
  { value: 293, text: getLocale('Permission__PERMISSION_INVENTORY_SUBMIT_OTHER_IN') },
  { value: 294, text: getLocale('Permission__PERMISSION_INVENTORY_APPROVE_OTHER_IN') },
  { value: 295, text: getLocale('Permission__PERMISSION_INVENTORY_NOT_APPROVE_OTHER_IN') },
  { value: 296, text: getLocale('Permission__PERMISSION_INVENTORY_CANCEL_APPROVE_OTHER_IN') },
  { value: 297, text: getLocale('Permission__PERMISSION_INVENTORY_DELETE_OTHER_IN') },
  { value: 298, text: getLocale('Permission__PERMISSION_INVENTORY_EXPORT_OTHER_IN') },
  { value: 299, text: getLocale('Permission__PERMISSION_INVENTORY_PRINT_OTHER_IN') },
  { value: 310, text: getLocale('Permission__PERMISSION_INVENTORY_VIEW_OTHER_OUT') },
  { value: 311, text: getLocale('Permission__PERMISSION_INVENTORY_CREATE_OTHER_OUT') },
  { value: 312, text: getLocale('Permission__PERMISSION_INVENTORY_UPDATE_OTHER_OUT') },
  { value: 313, text: getLocale('Permission__PERMISSION_INVENTORY_SUBMIT_OTHER_OUT') },
  { value: 314, text: getLocale('Permission__PERMISSION_INVENTORY_APPROVE_OTHER_OUT') },
  { value: 315, text: getLocale('Permission__PERMISSION_INVENTORY_NOT_APPROVE_OTHER_OUT') },
  { value: 316, text: getLocale('Permission__PERMISSION_INVENTORY_CANCEL_APPROVE_OTHER_OUT') },
  { value: 317, text: getLocale('Permission__PERMISSION_INVENTORY_DELETE_OTHER_OUT') },
  { value: 318, text: getLocale('Permission__PERMISSION_INVENTORY_EXPORT_OTHER_OUT') },
  { value: 319, text: getLocale('Permission__PERMISSION_INVENTORY_PRINT_OTHER_OUT') },
  { value: 330, text: getLocale('Permission__PERMISSION_INVENTORY_VIEW_ADJUST_SHEET') },
  { value: 340, text: getLocale('Permission__PERMISSION_INVENTORY_VIEW_INVENTORY_SHEET') },
  { value: 341, text: getLocale('Permission__PERMISSION_INVENTORY_CREATE_INVENTORY_SHEET') },
  { value: 342, text: getLocale('Permission__PERMISSION_INVENTORY_UPDATE_INVENTORY_SHEET') },
  { value: 343, text: getLocale('Permission__PERMISSION_INVENTORY_SUBMIT_INVENTORY_SHEET') },
  { value: 344, text: getLocale('Permission__PERMISSION_INVENTORY_APPROVE_INVENTORY_SHEET') },
  { value: 345, text: getLocale('Permission__PERMISSION_INVENTORY_NOT_APPROVE_INVENTORY_SHEET') },
  { value: 346, text: getLocale('Permission__PERMISSION_INVENTORY_CANCEL_APPROVE_INVENTORY_SHEET') },
  { value: 347, text: getLocale('Permission__PERMISSION_INVENTORY_EXPORT_INVENTORY_SHEET') },
  { value: 348, text: getLocale('Permission__PERMISSION_INVENTORY_PRINT_INVENTORY_SHEET') },
  { value: 349, text: getLocale('Permission__PERMISSION_INVENTORY_DELETE_INVENTORY_SHEET') },
  { value: 360, text: getLocale('Permission__PERMISSION_INVENTORY_VIEW_STOCK') },
  { value: 361, text: getLocale('Permission__PERMISSION_INVENTORY_VIEW_STOCK_LIST') },
  { value: 362, text: getLocale('Permission__PERMISSION_INVENTORY_EXPORT_STOCK_LIST') },
  { value: 363, text: getLocale('Permission__PERMISSION_INVENTORY_VIEW_IN_STOCK_LOG') },
  { value: 364, text: getLocale('Permission__PERMISSION_INVENTORY_EXPORT_IN_STOCK_LOG') },
  { value: 365, text: getLocale('Permission__PERMISSION_INVENTORY_VIEW_OUT_STOCK_LOG') },
  { value: 366, text: getLocale('Permission__PERMISSION_INVENTORY_EXPORT_OUT_STOCK_LOG') },
  { value: 367, text: getLocale('Permission__PERMISSION_INVENTORY_VIEW_CHANGE_STOCK_LOG') },
  { value: 368, text: getLocale('Permission__PERMISSION_INVENTORY_EXPORT_CHANGE_STOCK_LOG') },
  { value: 380, text: getLocale('Permission__PERMISSION_ENTERPRISE_VIEW_CUSTOMER') },
  { value: 381, text: getLocale('Permission__PERMISSION_ENTERPRISE_CREATE_CUSTOMER') },
  { value: 382, text: getLocale('Permission__PERMISSION_ENTERPRISE_UPDATE_CUSTOMER') },
  { value: 383, text: getLocale('Permission__PERMISSION_ENTERPRISE_DELETE_CUSTOMER') },
  { value: 390, text: getLocale('Permission__PERMISSION_ENTERPRISE_VIEW_INVITATION_CODE') },
  { value: 520, text: getLocale('Permission__PERMISSION_ENTERPRISE_VIEW_SCHOOL_CLASS') },
  { value: 521, text: getLocale('Permission__PERMISSION_ENTERPRISE_CREATE_SCHOOL_CLASS') },
  { value: 522, text: getLocale('Permission__PERMISSION_ENTERPRISE_UPDATE_SCHOOL_CLASS') },
  { value: 523, text: getLocale('Permission__PERMISSION_ENTERPRISE_DELETE_SCHOOL_CLASS') },
  { value: 540, text: getLocale('Permission__PERMISSION_ENTERPRISE_VIEW_STUDENT') },
  { value: 541, text: getLocale('Permission__PERMISSION_ENTERPRISE_VIEW_STUFF') },
  { value: 391, text: getLocale('Permission__PERMISSION_ENTERPRISE_VIEW_ESHOP_INVITATION_CODE') },
  { value: 550, text: getLocale('Permission__PERMISSION_ANALYTICS_VIEW_MENU_SETTLEMENT') },
  { value: 400, text: getLocale('Permission__PERMISSION_ENTERPRISE_VIEW_SERVICE_PERIOD') },
  { value: 401, text: getLocale('Permission__PERMISSION_ENTERPRISE_CREATE_SERVICE_PERIOD') },
  { value: 402, text: getLocale('Permission__PERMISSION_ENTERPRISE_UPDATE_SERVICE_PERIOD') },
  { value: 403, text: getLocale('Permission__PERMISSION_ENTERPRISE_DELETE_SERVICE_PERIOD') },
  { value: 410, text: getLocale('Permission__PERMISSION_PREFERENCE_VIEW_APPOINT_TIME_SETTINGS') },
  { value: 411, text: getLocale('Permission__PERMISSION_PREFERENCE_UPDATE_APPOINT_TIME_SETTINGS') },
  { value: 420, text: getLocale('Permission__PERMISSION_PREFERENCE_VIEW_PURCHASE_SETTINGS') },
  { value: 421, text: getLocale('Permission__PERMISSION_PREFERENCE_UPDATE_PURCHASE_SETTINGS') },
  { value: 430, text: getLocale('Permission__PERMISSION_PRODUCTION_VIEW_SETTINGS') },
  { value: 431, text: getLocale('Permission__PERMISSION_PRODUCTION_UPDATE_SETTINGS') },
  { value: 440, text: getLocale('Permission__PERMISSION_PREFERENCE_INVENTORY_SETTINGS') },
  { value: 441, text: getLocale('Permission__PERMISSION_PREFERENCE_PRODUCTION_SETTINGS') },
  { value: 450, text: getLocale('Permission__PERMISSION_PREFERENCE_VIEW_IMPORT_TEMPLATE') },
  { value: 451, text: getLocale('Permission__PERMISSION_PREFERENCE_CREATE_IMPORT_TEMPLATE') },
  { value: 452, text: getLocale('Permission__PERMISSION_PREFERENCE_UPDATE_IMPORT_TEMPLATE') },
  { value: 453, text: getLocale('Permission__PERMISSION_PREFERENCE_DELETE_IMPORT_TEMPLATE') },
  { value: 460, text: getLocale('Permission__PERMISSION_PREFERENCE_VIEW_PRINTING_TEMPLATE') },
  { value: 461, text: getLocale('Permission__PERMISSION_PREFERENCE_CREATE_PRINTING_TEMPLATE') },
  { value: 462, text: getLocale('Permission__PERMISSION_PREFERENCE_UPDATE_PRINTING_TEMPLATE') },
  { value: 463, text: getLocale('Permission__PERMISSION_PREFERENCE_DELETE_PRINTING_TEMPLATE') },
  { value: 490, text: getLocale('Permission__PERMISSION_PREFERENCE_UPDATE_SOCIAL_SHOP_LAYOUT') },
  { value: 491, text: getLocale('Permission__PERMISSION_PREFERENCE_UPDATE_EDUCATION_SHOP_LAYOUT') },
  { value: 470, text: getLocale('Permission__PERMISSION_MES_VIEW_PRODUCE') },
  { value: 471, text: getLocale('Permission__PERMISSION_MES_VIEW_PACK') },
  { value: 472, text: getLocale('Permission__PERMISSION_MES_VIEW_SORTING') },
  { value: 480, text: getLocale('Permission__PERMISSION_SORTING_VIEW_SORTING_INFO') },
  { value: 481, text: getLocale('Permission__PERMISSION_SORTING_VIEW_SORTING_TASK') },
  { value: 482, text: getLocale('Permission__PERMISSION_SORTING_EDIT_SORTING_TASK') },
  { value: 483, text: getLocale('Permission__PERMISSION_SORTING_EXPORT_SORTING_PERFORMANCE') },
]
export const map_Permission: Enum_Map = {
  10: getLocale('Permission__PERMISSION_MERCHANDISE_VIEW_QUOTATION'),
  11: getLocale('Permission__PERMISSION_MERCHANDISE_CREATE_QUOTATION'),
  12: getLocale('Permission__PERMISSION_MERCHANDISE_UPDATE_QUOTATION'),
  13: getLocale('Permission__PERMISSION_MERCHANDISE_DELETE_QUOTATION'),
  20: getLocale('Permission__PERMISSION_MERCHANDISE_VIEW_NOT_PACKAGE_SKU_SSU'),
  21: getLocale('Permission__PERMISSION_MERCHANDISE_CREATE_NOT_PACKAGE_SKU_SSU'),
  22: getLocale('Permission__PERMISSION_MERCHANDISE_UPDATE_NOT_PACKAGE_SKU_SSU'),
  23: getLocale('Permission__PERMISSION_MERCHANDISE_DELETE_NOT_PACKAGE_SKU_SSU'),
  30: getLocale('Permission__PERMISSION_MERCHANDISE_VIEW_PACKAGE_SKU_SSU'),
  31: getLocale('Permission__PERMISSION_MERCHANDISE_CREATE_PACKAGE_SKU_SSU'),
  32: getLocale('Permission__PERMISSION_MERCHANDISE_UPDATE_PACKAGE_SKU_SSU'),
  33: getLocale('Permission__PERMISSION_MERCHANDISE_DELETE_PACKAGE_SKU_SSU'),
  40: getLocale('Permission__PERMISSION_PRODUCTION_VIEW_BOM'),
  41: getLocale('Permission__PERMISSION_PRODUCTION_CREATE_BOM'),
  42: getLocale('Permission__PERMISSION_PRODUCTION_UPDATE_BOM'),
  43: getLocale('Permission__PERMISSION_PRODUCTION_DELETE_BOM'),
  50: getLocale('Permission__PERMISSION_MERCHANDISE_VIEW_CATEGORY'),
  51: getLocale('Permission__PERMISSION_MERCHANDISE_CREATE_CATEGORY'),
  52: getLocale('Permission__PERMISSION_MERCHANDISE_UPDATE_CATEGORY'),
  53: getLocale('Permission__PERMISSION_MERCHANDISE_DELETE_CATEGORY'),
  500: getLocale('Permission__PERMISSION_ESHOP_VIEW_MENU'),
  502: getLocale('Permission__PERMISSION_ESHOP_UPDATE_MENU'),
  504: getLocale('Permission__PERMISSION_ESHOP_VIEW_MENU_DETAIL'),
  505: getLocale('Permission__PERMISSION_ESHOP_UPDATE_MENU_DETAIL'),
  501: getLocale('Permission__PERMISSION_ESHOP_CREATE_MENU'),
  503: getLocale('Permission__PERMISSION_ESHOP_DELETE_MENU'),
  510: getLocale('Permission__PERMISSION_ESHOP_VIEW_MENU_PERIOD'),
  511: getLocale('Permission__PERMISSION_ESHOP_CREATE_MENU_PERIOD'),
  512: getLocale('Permission__PERMISSION_ESHOP_UPDATE_MENU_PERIOD'),
  513: getLocale('Permission__PERMISSION_ESHOP_DELETE_MENU_PERIOD'),
  60: getLocale('Permission__PERMISSION_ORDER_VIEW_ORDER'),
  61: getLocale('Permission__PERMISSION_ORDER_CREATE_ORDER'),
  62: getLocale('Permission__PERMISSION_ORDER_UPDATE_ORDER'),
  63: getLocale('Permission__PERMISSION_ORDER_DELETE_ORDER'),
  64: getLocale('Permission__PERMISSION_ORDER_AMEND_ORDER'),
  65: getLocale('Permission__PERMISSION_ORDER_RELEASE_PROCESS_PLAN'),
  66: getLocale('Permission__PERMISSION_ORDER_RELEASE_NONPROCESS_PLAN'),
  530: getLocale('Permission__PERMISSION_ORDER_VIEW_ESHOP_STUDENT_ORDER'),
  531: getLocale('Permission__PERMISSION_ORDER_DELETE_ESHOP_STUDENT_ORDER'),
  532: getLocale('Permission__PERMISSION_ORDER_VIEW_ESHOP_STUFF_ORDER'),
  533: getLocale('Permission__PERMISSION_ORDER_DELETE_ESHOP_STUFF_ORDER'),
  539: getLocale('Permission__PERMISSION_ORDER_VIEW_LEAVE_APPLICATION'),
  70: getLocale('Permission__PERMISSION_PURCHASE_VIEW_PURCHASE_TASK'),
  71: getLocale('Permission__PERMISSION_PURCHASE_CREATE_PURCHASE_TASK'),
  72: getLocale('Permission__PERMISSION_PURCHASE_UPDATE_PURCHASE_TASK'),
  73: getLocale('Permission__PERMISSION_PURCHASE_DELETE_PURCHASE_TASK'),
  80: getLocale('Permission__PERMISSION_PURCHASE_VIEW_PURCHASE_SHEET'),
  81: getLocale('Permission__PERMISSION_PURCHASE_CREATE_PURCHASE_SHEET'),
  82: getLocale('Permission__PERMISSION_PURCHASE_UPDATE_PURCHASE_SHEET'),
  83: getLocale('Permission__PERMISSION_PURCHASE_DELETE_PURCHASE_SHEET'),
  90: getLocale('Permission__PERMISSION_ENTERPRISE_VIEW_PURCHASER'),
  91: getLocale('Permission__PERMISSION_ENTERPRISE_CREATE_PURCHASER'),
  92: getLocale('Permission__PERMISSION_ENTERPRISE_UPDATE_PURCHASER'),
  93: getLocale('Permission__PERMISSION_ENTERPRISE_DELETE_PURCHASER'),
  100: getLocale('Permission__PERMISSION_PURCHASE_VIEW_INQUIRY_PRICE'),
  101: getLocale('Permission__PERMISSION_PURCHASE_IMPORT_INQUIRY_PRICE'),
  110: getLocale('Permission__PERMISSION_PRODUCTION_LIST_PRODUCETASK'),
  111: getLocale('Permission__PERMISSION_PRODUCTION_CREATE_PRODUCETASK'),
  112: getLocale('Permission__PERMISSION_PRODUCTION_UPDATE_PRODUCETASK'),
  113: getLocale('Permission__PERMISSION_PRODUCTION_DELETE_PRODUCETASK'),
  120: getLocale('Permission__PERMISSION_PRODUCTION_LIST_PACKTASK'),
  121: getLocale('Permission__PERMISSION_PRODUCTION_CREATE_PACKTASK'),
  122: getLocale('Permission__PERMISSION_PRODUCTION_UPDATE_PACKTASK'),
  123: getLocale('Permission__PERMISSION_PRODUCTION_DELETE_PACKTASK'),
  130: getLocale('Permission__PERMISSION_PRODUCTION_VIEW_PRODUCEPROGRESS'),
  131: getLocale('Permission__PERMISSION_PRODUCTION_VIEW_PACKPROGRESS'),
  132: getLocale('Permission__PERMISSION_PRODUCTION_VIEW_PRODUCEANALYTICS'),
  133: getLocale('Permission__PERMISSION_PRODUCTION_VIEW_PACKANALYTICS'),
  134: getLocale('Permission__PERMISSION_PRODUCTION_PRINT_BOX_LABLE'),
  140: getLocale('Permission__PERMISSION_DELIVERY_VIEW_TASK'),
  141: getLocale('Permission__PERMISSION_DELIVERY_UPDATE_TASK'),
  142: getLocale('Permission__PERMISSION_DELIVERY_PRINT_TASK'),
  150: getLocale('Permission__PERMISSION_ENTERPRISE_VIEW_DRIVER'),
  151: getLocale('Permission__PERMISSION_ENTERPRISE_CREATE_DRIVER'),
  152: getLocale('Permission__PERMISSION_ENTERPRISE_UPDATE_DRIVER'),
  153: getLocale('Permission__PERMISSION_ENTERPRISE_DELETE_DRIVER'),
  160: getLocale('Permission__PERMISSION_ENTERPRISE_CREATE_DISTRIBUTION_CONTRACTOR'),
  161: getLocale('Permission__PERMISSION_ENTERPRISE_UPDATE_DISTRIBUTION_CONTRACTOR'),
  162: getLocale('Permission__PERMISSION_ENTERPRISE_DELETE_DISTRIBUTION_CONTRACTOR'),
  170: getLocale('Permission__PERMISSION_DELIVERY_VIEW_ROUTE'),
  171: getLocale('Permission__PERMISSION_DELIVERY_CREATE_ROUTE'),
  172: getLocale('Permission__PERMISSION_DELIVERY_UPDATE_ROUTE'),
  173: getLocale('Permission__PERMISSION_DELIVERY_DELETE_ROUTE'),
  180: getLocale('Permission__PERMISSION_INVENTORY_VIEW_PURCHASE_IN_SHEET'),
  181: getLocale('Permission__PERMISSION_INVENTORY_CREATE_PURCHASE_IN_SHEET'),
  182: getLocale('Permission__PERMISSION_INVENTORY_UPDATE_PURCHASE_IN_SHEET'),
  183: getLocale('Permission__PERMISSION_INVENTORY_SUBMIT_PURCHASE_IN_SHEET'),
  184: getLocale('Permission__PERMISSION_INVENTORY_APPROVE_PURCHASE_IN_SHEET'),
  185: getLocale('Permission__PERMISSION_INVENTORY_NOT_APPROVE_PURCHASE_IN_SHEET'),
  186: getLocale('Permission__PERMISSION_INVENTORY_CANCEL_APPROVE_PURCHASE_IN_SHEET'),
  187: getLocale('Permission__PERMISSION_INVENTORY_DELETE_PURCHASE_IN_SHEET'),
  188: getLocale('Permission__PERMISSION_INVENTORY_EXPORT_PURCHASE_IN_SHEET'),
  189: getLocale('Permission__PERMISSION_INVENTORY_PRINT_PURCHASE_IN_SHEET'),
  200: getLocale('Permission__PERMISSION_INVENTORY_VIEW_REFUND_OUT_SHEET'),
  201: getLocale('Permission__PERMISSION_INVENTORY_CREATE_REFUND_OUT_SHEET'),
  202: getLocale('Permission__PERMISSION_INVENTORY_UPDATE_REFUND_OUT_SHEET'),
  203: getLocale('Permission__PERMISSION_INVENTORY_SUBMIT_REFUND_OUT_SHEET'),
  204: getLocale('Permission__PERMISSION_INVENTORY_APPROVE_REFUND_OUT_SHEET'),
  205: getLocale('Permission__PERMISSION_INVENTORY_NOT_APPROVE_REFUND_OUT_SHEET'),
  206: getLocale('Permission__PERMISSION_INVENTORY_CANCEL_APPROVE_REFUND_OUT_SHEET'),
  207: getLocale('Permission__PERMISSION_INVENTORY_DELETE_REFUND_OUT_SHEET'),
  208: getLocale('Permission__PERMISSION_INVENTORY_EXPORT_REFUND_OUT_SHEET'),
  209: getLocale('Permission__PERMISSION_INVENTORY_PRINT_REFUND_OUT_SHEET'),
  220: getLocale('Permission__PERMISSION_INVENTORY_VIEW_SALE_OUT_SHEET'),
  221: getLocale('Permission__PERMISSION_INVENTORY_CREATE_SALE_OUT_SHEET'),
  222: getLocale('Permission__PERMISSION_INVENTORY_UPDATE_SALE_OUT_SHEET'),
  223: getLocale('Permission__PERMISSION_INVENTORY_SUBMIT_SALE_OUT_SHEET'),
  224: getLocale('Permission__PERMISSION_INVENTORY_APPROVE_SALE_OUT_SHEET'),
  225: getLocale('Permission__PERMISSION_INVENTORY_NOT_APPROVE_SALE_OUT_SHEET'),
  226: getLocale('Permission__PERMISSION_INVENTORY_CANCEL_APPROVE_SALE_OUT_SHEET'),
  227: getLocale('Permission__PERMISSION_INVENTORY_DELETE_SALE_OUT_SHEET'),
  228: getLocale('Permission__PERMISSION_INVENTORY_EXPORT_SALE_OUT_SHEET'),
  229: getLocale('Permission__PERMISSION_INVENTORY_PRINT_SALE_OUT_SHEET'),
  230: getLocale('Permission__PERMISSION_INVENTORY_VIEW_PRODUCT_IN'),
  231: getLocale('Permission__PERMISSION_INVENTORY_CREATE_PRODUCT_IN'),
  232: getLocale('Permission__PERMISSION_INVENTORY_UPDATE_PRODUCT_IN'),
  233: getLocale('Permission__PERMISSION_INVENTORY_SUBMIT_PRODUCT_IN'),
  234: getLocale('Permission__PERMISSION_INVENTORY_APPROVE_PRODUCT_IN'),
  235: getLocale('Permission__PERMISSION_INVENTORY_NOT_APPROVE_PRODUCT_IN'),
  236: getLocale('Permission__PERMISSION_INVENTORY_CANCEL_APPROVE_PRODUCT_IN'),
  237: getLocale('Permission__PERMISSION_INVENTORY_DELETE_PRODUCT_IN'),
  238: getLocale('Permission__PERMISSION_INVENTORY_EXPORT_PRODUCT_IN'),
  239: getLocale('Permission__PERMISSION_INVENTORY_PRINT_PRODUCT_IN'),
  250: getLocale('Permission__PERMISSION_INVENTORY_VIEW_MATERIAL_IN_250'),
  251: getLocale('Permission__PERMISSION_INVENTORY_CREATE_MATERIAL_IN_251'),
  252: getLocale('Permission__PERMISSION_INVENTORY_UPDATE_MATERIAL_IN_252'),
  253: getLocale('Permission__PERMISSION_INVENTORY_SUBMIT_MATERIAL_IN_253'),
  254: getLocale('Permission__PERMISSION_INVENTORY_APPROVE_MATERIAL_IN_254'),
  255: getLocale('Permission__PERMISSION_INVENTORY_NOT_APPROVE_MATERIAL_IN_255'),
  256: getLocale('Permission__PERMISSION_INVENTORY_CANCEL_APPROVE_MATERIAL_IN_256'),
  257: getLocale('Permission__PERMISSION_INVENTORY_DELETE_MATERIAL_IN_257'),
  258: getLocale('Permission__PERMISSION_INVENTORY_EXPORT_MATERIAL_IN_258'),
  259: getLocale('Permission__PERMISSION_INVENTORY_PRINT_MATERIAL_IN_259'),
  270: getLocale('Permission__PERMISSION_INVENTORY_VIEW_MATERIAL_OUT'),
  271: getLocale('Permission__PERMISSION_INVENTORY_CREATE_MATERIAL_OUT'),
  272: getLocale('Permission__PERMISSION_INVENTORY_UPDATE_MATERIAL_OUT'),
  273: getLocale('Permission__PERMISSION_INVENTORY_SUBMIT_MATERIAL_OUT'),
  274: getLocale('Permission__PERMISSION_INVENTORY_APPROVE_MATERIAL_OUT'),
  275: getLocale('Permission__PERMISSION_INVENTORY_NOT_APPROVE_MATERIAL_OUT'),
  276: getLocale('Permission__PERMISSION_INVENTORY_CANCEL_APPROVE_MATERIAL_OUT'),
  277: getLocale('Permission__PERMISSION_INVENTORY_DELETE_MATERIAL_OUT'),
  278: getLocale('Permission__PERMISSION_INVENTORY_EXPORT_MATERIAL_OUT'),
  279: getLocale('Permission__PERMISSION_INVENTORY_PRINT_MATERIAL_OUT'),
  290: getLocale('Permission__PERMISSION_INVENTORY_VIEW_OTHER_IN'),
  291: getLocale('Permission__PERMISSION_INVENTORY_CREATE_OTHER_IN'),
  292: getLocale('Permission__PERMISSION_INVENTORY_UPDATE_OTHER_IN'),
  293: getLocale('Permission__PERMISSION_INVENTORY_SUBMIT_OTHER_IN'),
  294: getLocale('Permission__PERMISSION_INVENTORY_APPROVE_OTHER_IN'),
  295: getLocale('Permission__PERMISSION_INVENTORY_NOT_APPROVE_OTHER_IN'),
  296: getLocale('Permission__PERMISSION_INVENTORY_CANCEL_APPROVE_OTHER_IN'),
  297: getLocale('Permission__PERMISSION_INVENTORY_DELETE_OTHER_IN'),
  298: getLocale('Permission__PERMISSION_INVENTORY_EXPORT_OTHER_IN'),
  299: getLocale('Permission__PERMISSION_INVENTORY_PRINT_OTHER_IN'),
  310: getLocale('Permission__PERMISSION_INVENTORY_VIEW_OTHER_OUT'),
  311: getLocale('Permission__PERMISSION_INVENTORY_CREATE_OTHER_OUT'),
  312: getLocale('Permission__PERMISSION_INVENTORY_UPDATE_OTHER_OUT'),
  313: getLocale('Permission__PERMISSION_INVENTORY_SUBMIT_OTHER_OUT'),
  314: getLocale('Permission__PERMISSION_INVENTORY_APPROVE_OTHER_OUT'),
  315: getLocale('Permission__PERMISSION_INVENTORY_NOT_APPROVE_OTHER_OUT'),
  316: getLocale('Permission__PERMISSION_INVENTORY_CANCEL_APPROVE_OTHER_OUT'),
  317: getLocale('Permission__PERMISSION_INVENTORY_DELETE_OTHER_OUT'),
  318: getLocale('Permission__PERMISSION_INVENTORY_EXPORT_OTHER_OUT'),
  319: getLocale('Permission__PERMISSION_INVENTORY_PRINT_OTHER_OUT'),
  330: getLocale('Permission__PERMISSION_INVENTORY_VIEW_ADJUST_SHEET'),
  340: getLocale('Permission__PERMISSION_INVENTORY_VIEW_INVENTORY_SHEET'),
  341: getLocale('Permission__PERMISSION_INVENTORY_CREATE_INVENTORY_SHEET'),
  342: getLocale('Permission__PERMISSION_INVENTORY_UPDATE_INVENTORY_SHEET'),
  343: getLocale('Permission__PERMISSION_INVENTORY_SUBMIT_INVENTORY_SHEET'),
  344: getLocale('Permission__PERMISSION_INVENTORY_APPROVE_INVENTORY_SHEET'),
  345: getLocale('Permission__PERMISSION_INVENTORY_NOT_APPROVE_INVENTORY_SHEET'),
  346: getLocale('Permission__PERMISSION_INVENTORY_CANCEL_APPROVE_INVENTORY_SHEET'),
  347: getLocale('Permission__PERMISSION_INVENTORY_EXPORT_INVENTORY_SHEET'),
  348: getLocale('Permission__PERMISSION_INVENTORY_PRINT_INVENTORY_SHEET'),
  349: getLocale('Permission__PERMISSION_INVENTORY_DELETE_INVENTORY_SHEET'),
  360: getLocale('Permission__PERMISSION_INVENTORY_VIEW_STOCK'),
  361: getLocale('Permission__PERMISSION_INVENTORY_VIEW_STOCK_LIST'),
  362: getLocale('Permission__PERMISSION_INVENTORY_EXPORT_STOCK_LIST'),
  363: getLocale('Permission__PERMISSION_INVENTORY_VIEW_IN_STOCK_LOG'),
  364: getLocale('Permission__PERMISSION_INVENTORY_EXPORT_IN_STOCK_LOG'),
  365: getLocale('Permission__PERMISSION_INVENTORY_VIEW_OUT_STOCK_LOG'),
  366: getLocale('Permission__PERMISSION_INVENTORY_EXPORT_OUT_STOCK_LOG'),
  367: getLocale('Permission__PERMISSION_INVENTORY_VIEW_CHANGE_STOCK_LOG'),
  368: getLocale('Permission__PERMISSION_INVENTORY_EXPORT_CHANGE_STOCK_LOG'),
  380: getLocale('Permission__PERMISSION_ENTERPRISE_VIEW_CUSTOMER'),
  381: getLocale('Permission__PERMISSION_ENTERPRISE_CREATE_CUSTOMER'),
  382: getLocale('Permission__PERMISSION_ENTERPRISE_UPDATE_CUSTOMER'),
  383: getLocale('Permission__PERMISSION_ENTERPRISE_DELETE_CUSTOMER'),
  390: getLocale('Permission__PERMISSION_ENTERPRISE_VIEW_INVITATION_CODE'),
  520: getLocale('Permission__PERMISSION_ENTERPRISE_VIEW_SCHOOL_CLASS'),
  521: getLocale('Permission__PERMISSION_ENTERPRISE_CREATE_SCHOOL_CLASS'),
  522: getLocale('Permission__PERMISSION_ENTERPRISE_UPDATE_SCHOOL_CLASS'),
  523: getLocale('Permission__PERMISSION_ENTERPRISE_DELETE_SCHOOL_CLASS'),
  540: getLocale('Permission__PERMISSION_ENTERPRISE_VIEW_STUDENT'),
  541: getLocale('Permission__PERMISSION_ENTERPRISE_VIEW_STUFF'),
  391: getLocale('Permission__PERMISSION_ENTERPRISE_VIEW_ESHOP_INVITATION_CODE'),
  550: getLocale('Permission__PERMISSION_ANALYTICS_VIEW_MENU_SETTLEMENT'),
  400: getLocale('Permission__PERMISSION_ENTERPRISE_VIEW_SERVICE_PERIOD'),
  401: getLocale('Permission__PERMISSION_ENTERPRISE_CREATE_SERVICE_PERIOD'),
  402: getLocale('Permission__PERMISSION_ENTERPRISE_UPDATE_SERVICE_PERIOD'),
  403: getLocale('Permission__PERMISSION_ENTERPRISE_DELETE_SERVICE_PERIOD'),
  410: getLocale('Permission__PERMISSION_PREFERENCE_VIEW_APPOINT_TIME_SETTINGS'),
  411: getLocale('Permission__PERMISSION_PREFERENCE_UPDATE_APPOINT_TIME_SETTINGS'),
  420: getLocale('Permission__PERMISSION_PREFERENCE_VIEW_PURCHASE_SETTINGS'),
  421: getLocale('Permission__PERMISSION_PREFERENCE_UPDATE_PURCHASE_SETTINGS'),
  430: getLocale('Permission__PERMISSION_PRODUCTION_VIEW_SETTINGS'),
  431: getLocale('Permission__PERMISSION_PRODUCTION_UPDATE_SETTINGS'),
  440: getLocale('Permission__PERMISSION_PREFERENCE_INVENTORY_SETTINGS'),
  441: getLocale('Permission__PERMISSION_PREFERENCE_PRODUCTION_SETTINGS'),
  450: getLocale('Permission__PERMISSION_PREFERENCE_VIEW_IMPORT_TEMPLATE'),
  451: getLocale('Permission__PERMISSION_PREFERENCE_CREATE_IMPORT_TEMPLATE'),
  452: getLocale('Permission__PERMISSION_PREFERENCE_UPDATE_IMPORT_TEMPLATE'),
  453: getLocale('Permission__PERMISSION_PREFERENCE_DELETE_IMPORT_TEMPLATE'),
  460: getLocale('Permission__PERMISSION_PREFERENCE_VIEW_PRINTING_TEMPLATE'),
  461: getLocale('Permission__PERMISSION_PREFERENCE_CREATE_PRINTING_TEMPLATE'),
  462: getLocale('Permission__PERMISSION_PREFERENCE_UPDATE_PRINTING_TEMPLATE'),
  463: getLocale('Permission__PERMISSION_PREFERENCE_DELETE_PRINTING_TEMPLATE'),
  490: getLocale('Permission__PERMISSION_PREFERENCE_UPDATE_SOCIAL_SHOP_LAYOUT'),
  491: getLocale('Permission__PERMISSION_PREFERENCE_UPDATE_EDUCATION_SHOP_LAYOUT'),
  470: getLocale('Permission__PERMISSION_MES_VIEW_PRODUCE'),
  471: getLocale('Permission__PERMISSION_MES_VIEW_PACK'),
  472: getLocale('Permission__PERMISSION_MES_VIEW_SORTING'),
  480: getLocale('Permission__PERMISSION_SORTING_VIEW_SORTING_INFO'),
  481: getLocale('Permission__PERMISSION_SORTING_VIEW_SORTING_TASK'),
  482: getLocale('Permission__PERMISSION_SORTING_EDIT_SORTING_TASK'),
  483: getLocale('Permission__PERMISSION_SORTING_EXPORT_SORTING_PERFORMANCE'),
}
export const list_GroupUser_Type: Enum_List_Item[] = [
  { value: 8, text: getLocale('GroupUser_Type__GROUP_ADMIN') },
  { value: 9, text: getLocale('GroupUser_Type__NORMAL') },
]
export const map_GroupUser_Type: Enum_Map = {
  8: getLocale('GroupUser_Type__GROUP_ADMIN'),
  9: getLocale('GroupUser_Type__NORMAL'),
}
export const list_GroupUser_Status: Enum_List_Item[] = [
  { value: 256, text: getLocale('GroupUser_Status__STATUS_CAN_LOGIN_PURCHASE_APP') },
  { value: 512, text: getLocale('GroupUser_Status__STATUS_CAN_LOGIN_DRIVER_APP') },
]
export const map_GroupUser_Status: Enum_Map = {
  256: getLocale('GroupUser_Status__STATUS_CAN_LOGIN_PURCHASE_APP'),
  512: getLocale('GroupUser_Status__STATUS_CAN_LOGIN_DRIVER_APP'),
}
export const list_Role_Type: Enum_List_Item[] = [
  { value: 1, text: getLocale('Role_Type__NORMAL') },
  { value: 2, text: getLocale('Role_Type__BUILT_IN_DRIVER') },
  { value: 3, text: getLocale('Role_Type__BUILT_IN_PURCHASER') },
  { value: 4, text: getLocale('Role_Type__BUILT_IN_ADMIN') },
]
export const map_Role_Type: Enum_Map = {
  1: getLocale('Role_Type__NORMAL'),
  2: getLocale('Role_Type__BUILT_IN_DRIVER'),
  3: getLocale('Role_Type__BUILT_IN_PURCHASER'),
  4: getLocale('Role_Type__BUILT_IN_ADMIN'),
}
export const list_ServicePeriod_Type: Enum_List_Item[] = [
  { value: 3, text: getLocale('ServicePeriod_Type__TYPE_DEFAULT_MENU_PERIOD') },
  { value: 4, text: getLocale('ServicePeriod_Type__TYPE_CUSTOMIZED_MENU_PERIOD') },
]
export const map_ServicePeriod_Type: Enum_Map = {
  3: getLocale('ServicePeriod_Type__TYPE_DEFAULT_MENU_PERIOD'),
  4: getLocale('ServicePeriod_Type__TYPE_CUSTOMIZED_MENU_PERIOD'),
}
export const list_ChinaVatInvoice_InvoiceType: Enum_List_Item[] = [
  { value: 1, text: getLocale('ChinaVatInvoice_InvoiceType__VAT_NORMAL') },
  { value: 2, text: getLocale('ChinaVatInvoice_InvoiceType__VAT_SPECIAL') },
]
export const map_ChinaVatInvoice_InvoiceType: Enum_Map = {
  1: getLocale('ChinaVatInvoice_InvoiceType__VAT_NORMAL'),
  2: getLocale('ChinaVatInvoice_InvoiceType__VAT_SPECIAL'),
}
export const list_ChinaVatInvoice_TaxpayerType: Enum_List_Item[] = [
  { value: 1, text: getLocale('ChinaVatInvoice_TaxpayerType__TAXPAYER_TYPE_NORMAL') },
  { value: 2, text: getLocale('ChinaVatInvoice_TaxpayerType__TAXPAYER_TYPE_SMALL') },
  { value: 3, text: getLocale('ChinaVatInvoice_TaxpayerType__TAXPAYER_TYPE_NONE') },
]
export const map_ChinaVatInvoice_TaxpayerType: Enum_Map = {
  1: getLocale('ChinaVatInvoice_TaxpayerType__TAXPAYER_TYPE_NORMAL'),
  2: getLocale('ChinaVatInvoice_TaxpayerType__TAXPAYER_TYPE_SMALL'),
  3: getLocale('ChinaVatInvoice_TaxpayerType__TAXPAYER_TYPE_NONE'),
}
export const list_Customer_Status: Enum_List_Item[] = [
  { value: 256, text: getLocale('Customer_Status__STATUS_IS_BILL_TARGET') },
  { value: 512, text: getLocale('Customer_Status__STATUS_IS_SHIP_TARGET') },
]
export const map_Customer_Status: Enum_Map = {
  256: getLocale('Customer_Status__STATUS_IS_BILL_TARGET'),
  512: getLocale('Customer_Status__STATUS_IS_SHIP_TARGET'),
}
export const list_Customer_Type: Enum_List_Item[] = [
  { value: 1, text: getLocale('Customer_Type__TYPE_VIRTUAL') },
  { value: 3, text: getLocale('Customer_Type__TYPE_VIRTUAL_STUDENT') },
  { value: 4, text: getLocale('Customer_Type__TYPE_VIRTUAL_SCHOOL_STAFF') },
  { value: 2, text: getLocale('Customer_Type__TYPE_SOCIAL') },
  { value: 5, text: getLocale('Customer_Type__TYPE_SCHOOL') },
  { value: 6, text: getLocale('Customer_Type__TYPE_CLASS') },
]
export const map_Customer_Type: Enum_Map = {
  1: getLocale('Customer_Type__TYPE_VIRTUAL'),
  3: getLocale('Customer_Type__TYPE_VIRTUAL_STUDENT'),
  4: getLocale('Customer_Type__TYPE_VIRTUAL_SCHOOL_STAFF'),
  2: getLocale('Customer_Type__TYPE_SOCIAL'),
  5: getLocale('Customer_Type__TYPE_SCHOOL'),
  6: getLocale('Customer_Type__TYPE_CLASS'),
}
export const list_Customer_SettlementMethod: Enum_List_Item[] = [
  { value: 1, text: getLocale('Customer_SettlementMethod__SETTLEMENT_METHOD_ADVANCE_PAYMENT') },
  { value: 2, text: getLocale('Customer_SettlementMethod__SETTLEMENT_METHOD_DEFERRED_PAYMENT') },
]
export const map_Customer_SettlementMethod: Enum_Map = {
  1: getLocale('Customer_SettlementMethod__SETTLEMENT_METHOD_ADVANCE_PAYMENT'),
  2: getLocale('Customer_SettlementMethod__SETTLEMENT_METHOD_DEFERRED_PAYMENT'),
}
export const list_InvitationCode_Type: Enum_List_Item[] = [
  { value: 1, text: getLocale('InvitationCode_Type__TEMPORARY') },
  { value: 2, text: getLocale('InvitationCode_Type__FIXED') },
  { value: 3, text: getLocale('InvitationCode_Type__EDUCATION_FIXED') },
]
export const map_InvitationCode_Type: Enum_Map = {
  1: getLocale('InvitationCode_Type__TEMPORARY'),
  2: getLocale('InvitationCode_Type__FIXED'),
  3: getLocale('InvitationCode_Type__EDUCATION_FIXED'),
}
