// gm_api/src/enterprise/methods.ts
import { Request } from '../util/request'
import {
  InitializeRegionRequest, InitializeRegionResponse,
  GetGroupRequest, GetGroupResponse,
  ListGroupRequest, ListGroupResponse,
  CreateGroupRequest, CreateGroupResponse,
  UpdateGroupRequest, UpdateGroupResponse,
  GetGroupExtraInfoRequest, GetGroupExtraInfoResponse,
  UpdateGroupExtraInfoRequest, UpdateGroupExtraInfoResponse,
  ListGroupTimeZoneRequest, ListGroupTimeZoneResponse,
  ListPermissionGroupRequest, ListPermissionGroupResponse,
  GetPermissionGroupRequest, GetPermissionGroupResponse,
  UpdatePermissionGroupRequest, UpdatePermissionGroupResponse,
  GetGroupPermissionRequest, GetGroupPermissionResponse,
  UpdateGroupPermissionRequest, UpdateGroupPermissionResponse,
  GetStationRequest, GetStationResponse,
  ListStationRequest, ListStationResponse,
  CreateStationRequest, CreateStationResponse,
  UpdateStationRequest, UpdateStationResponse,
  UpdateStationAvailableCityRequest, UpdateStationAvailableCityResponse,
  GetSupplierRequest, GetSupplierResponse,
  GetManySupplierRequest, GetManySupplierResponse,
  ListSupplierRequest, ListSupplierResponse,
  CreateSupplierRequest, CreateSupplierResponse,
  UpdateSupplierRequest, UpdateSupplierResponse,
  DeleteSupplierRequest, DeleteSupplierResponse,
  CleanSupplierRequest, CleanSupplierResponse,
  BatchExportSupplierRequest, BatchExportSupplierResponse,
  GetGroupUserRequest, GetGroupUserResponse,
  GetManyGroupUserRequest, GetManyGroupUserResponse,
  ListGroupUserRequest, ListGroupUserResponse,
  CreateGroupUserRequest, CreateGroupUserResponse,
  UpdateGroupUserRequest, UpdateGroupUserResponse,
  UpdateGroupUserPasswordRequest, UpdateGroupUserPasswordResponse,
  DeleteGroupUserRequest, DeleteGroupUserResponse,
  UpdateGroupUserRoleRequest, UpdateGroupUserRoleResponse,
  GetRoleRequest, GetRoleResponse,
  ListRoleRequest, ListRoleResponse,
  CreateRoleRequest, CreateRoleResponse,
  UpdateRoleRequest, UpdateRoleResponse,
  DeleteRoleRequest, DeleteRoleResponse,
  UpdateRolePermissionRequest, UpdateRolePermissionResponse,
  UpdateRoleGroupUserRequest, UpdateRoleGroupUserResponse,
  GetServicePeriodRequest, GetServicePeriodResponse,
  GetManyServicePeriodRequest, GetManyServicePeriodResponse,
  ListServicePeriodRequest, ListServicePeriodResponse,
  CreateServicePeriodRequest, CreateServicePeriodResponse,
  UpdateServicePeriodRequest, UpdateServicePeriodResponse,
  DeleteServicePeriodRequest, DeleteServicePeriodResponse,
  CleanServicePeriodRequest, CleanServicePeriodResponse,
  GetCustomerRequest, GetCustomerResponse,
  GetManyCustomerRequest, GetManyCustomerResponse,
  ListCustomerRequest, ListCustomerResponse,
  CreateCustomerRequest, CreateCustomerResponse,
  CreateCustomerTreeRequest, CreateCustomerTreeResponse,
  UpdateCustomerRequest, UpdateCustomerResponse,
  DeleteCustomerRequest, DeleteCustomerResponse,
  CleanCustomerRequest, CleanCustomerResponse,
  BatchExportCustomerRequest, BatchExportCustomerResponse,
  BatchImportCustomerRequest, BatchImportCustomerResponse,
  GetBatchImportCustomerTemplateRequest, GetBatchImportCustomerTemplateResponse,
  CountNewCustomerRequest, CountNewCustomerResponse,
  GetCustomerUserRequest, GetCustomerUserResponse,
  InternalGetCustomerUserRequest, InternalGetCustomerUserResponse,
  GetManyCustomerUserRequest, GetManyCustomerUserResponse,
  ListCustomerUserRequest, ListCustomerUserResponse,
  CreateCustomerUserRequest, CreateCustomerUserResponse,
  UpdateCustomerUserRequest, UpdateCustomerUserResponse,
  UpdateCustomerUserPasswordRequest, UpdateCustomerUserPasswordResponse,
  DeleteCustomerUserRequest, DeleteCustomerUserResponse,
  CleanCustomerUserRequest, CleanCustomerUserResponse,
  GetCustomerLabelRequest, GetCustomerLabelResponse,
  ListCustomerLabelRequest, ListCustomerLabelResponse,
  CreateCustomerLabelRequest, CreateCustomerLabelResponse,
  UpdateCustomerLabelRequest, UpdateCustomerLabelResponse,
  DeleteCustomerLabelRequest, DeleteCustomerLabelResponse,
  GetDistributionContractorRequest, GetDistributionContractorResponse,
  ListDistributionContractorRequest, ListDistributionContractorResponse,
  CreateDistributionContractorRequest, CreateDistributionContractorResponse,
  UpdateDistributionContractorRequest, UpdateDistributionContractorResponse,
  DeleteDistributionContractorRequest, DeleteDistributionContractorResponse,
  ListCountryRequest, ListCountryResponse,
  ListProvinceRequest, ListProvinceResponse,
  ListCityRequest, ListCityResponse,
  ListDistrictRequest, ListDistrictResponse,
  ListStreetRequest, ListStreetResponse,
  ListInvitationCodeRequest, ListInvitationCodeResponse,
  GetInvitationCodeRequest, GetInvitationCodeResponse,
  CreateInvitationCodeRequest, CreateInvitationCodeResponse,
  DeleteInvitationCodeRequest, DeleteInvitationCodeResponse,
  CleanInvitationCodeRequest, CleanInvitationCodeResponse,
  GetUserInfoRequest, GetUserInfoResponse,
} from './types'

export const GetGroup = (r?: GetGroupRequest, codes?: number[]) =>
  Request<GetGroupResponse>('/ceres/enterprise/EnterpriseService/GetGroup')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetGroup.url = '/ceres/enterprise/EnterpriseService/GetGroup'
export const ListGroup = (r?: ListGroupRequest, codes?: number[]) =>
  Request<ListGroupResponse>('/ceres/enterprise/EnterpriseService/ListGroup')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListGroup.url = '/ceres/enterprise/EnterpriseService/ListGroup'
export const CreateGroup = (r?: CreateGroupRequest, codes?: number[]) =>
  Request<CreateGroupResponse>('/ceres/enterprise/EnterpriseService/CreateGroup')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateGroup.url = '/ceres/enterprise/EnterpriseService/CreateGroup'
export const UpdateGroup = (r?: UpdateGroupRequest, codes?: number[]) =>
  Request<UpdateGroupResponse>('/ceres/enterprise/EnterpriseService/UpdateGroup')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateGroup.url = '/ceres/enterprise/EnterpriseService/UpdateGroup'
export const GetGroupExtraInfo = (r?: GetGroupExtraInfoRequest, codes?: number[]) =>
  Request<GetGroupExtraInfoResponse>('/ceres/enterprise/EnterpriseService/GetGroupExtraInfo')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetGroupExtraInfo.url = '/ceres/enterprise/EnterpriseService/GetGroupExtraInfo'
export const UpdateGroupExtraInfo = (r?: UpdateGroupExtraInfoRequest, codes?: number[]) =>
  Request<UpdateGroupExtraInfoResponse>('/ceres/enterprise/EnterpriseService/UpdateGroupExtraInfo')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateGroupExtraInfo.url = '/ceres/enterprise/EnterpriseService/UpdateGroupExtraInfo'
export const ListGroupTimeZone = (r?: ListGroupTimeZoneRequest, codes?: number[]) =>
  Request<ListGroupTimeZoneResponse>('/ceres/enterprise/EnterpriseService/ListGroupTimeZone')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListGroupTimeZone.url = '/ceres/enterprise/EnterpriseService/ListGroupTimeZone'
export const ListPermissionGroup = (r?: ListPermissionGroupRequest, codes?: number[]) =>
  Request<ListPermissionGroupResponse>('/ceres/enterprise/EnterpriseService/ListPermissionGroup')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListPermissionGroup.url = '/ceres/enterprise/EnterpriseService/ListPermissionGroup'
export const GetPermissionGroup = (r?: GetPermissionGroupRequest, codes?: number[]) =>
  Request<GetPermissionGroupResponse>('/ceres/enterprise/EnterpriseService/GetPermissionGroup')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetPermissionGroup.url = '/ceres/enterprise/EnterpriseService/GetPermissionGroup'
export const UpdatePermissionGroup = (r?: UpdatePermissionGroupRequest, codes?: number[]) =>
  Request<UpdatePermissionGroupResponse>('/ceres/enterprise/EnterpriseService/UpdatePermissionGroup')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdatePermissionGroup.url = '/ceres/enterprise/EnterpriseService/UpdatePermissionGroup'
export const GetGroupPermission = (r?: GetGroupPermissionRequest, codes?: number[]) =>
  Request<GetGroupPermissionResponse>('/ceres/enterprise/EnterpriseService/GetGroupPermission')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetGroupPermission.url = '/ceres/enterprise/EnterpriseService/GetGroupPermission'
export const UpdateGroupPermission = (r?: UpdateGroupPermissionRequest, codes?: number[]) =>
  Request<UpdateGroupPermissionResponse>('/ceres/enterprise/EnterpriseService/UpdateGroupPermission')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateGroupPermission.url = '/ceres/enterprise/EnterpriseService/UpdateGroupPermission'
export const GetStation = (r?: GetStationRequest, codes?: number[]) =>
  Request<GetStationResponse>('/ceres/enterprise/EnterpriseService/GetStation')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetStation.url = '/ceres/enterprise/EnterpriseService/GetStation'
export const ListStation = (r?: ListStationRequest, codes?: number[]) =>
  Request<ListStationResponse>('/ceres/enterprise/EnterpriseService/ListStation')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListStation.url = '/ceres/enterprise/EnterpriseService/ListStation'
export const CreateStation = (r?: CreateStationRequest, codes?: number[]) =>
  Request<CreateStationResponse>('/ceres/enterprise/EnterpriseService/CreateStation')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateStation.url = '/ceres/enterprise/EnterpriseService/CreateStation'
export const UpdateStation = (r?: UpdateStationRequest, codes?: number[]) =>
  Request<UpdateStationResponse>('/ceres/enterprise/EnterpriseService/UpdateStation')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateStation.url = '/ceres/enterprise/EnterpriseService/UpdateStation'
export const UpdateStationAvailableCity = (r?: UpdateStationAvailableCityRequest, codes?: number[]) =>
  Request<UpdateStationAvailableCityResponse>('/ceres/enterprise/EnterpriseService/UpdateStationAvailableCity')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateStationAvailableCity.url = '/ceres/enterprise/EnterpriseService/UpdateStationAvailableCity'
export const GetSupplier = (r?: GetSupplierRequest, codes?: number[]) =>
  Request<GetSupplierResponse>('/ceres/enterprise/EnterpriseService/GetSupplier')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetSupplier.url = '/ceres/enterprise/EnterpriseService/GetSupplier'
export const GetManySupplier = (r?: GetManySupplierRequest, codes?: number[]) =>
  Request<GetManySupplierResponse>('/ceres/enterprise/EnterpriseService/GetManySupplier')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetManySupplier.url = '/ceres/enterprise/EnterpriseService/GetManySupplier'
export const ListSupplier = (r?: ListSupplierRequest, codes?: number[]) =>
  Request<ListSupplierResponse>('/ceres/enterprise/EnterpriseService/ListSupplier')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListSupplier.url = '/ceres/enterprise/EnterpriseService/ListSupplier'
export const CreateSupplier = (r?: CreateSupplierRequest, codes?: number[]) =>
  Request<CreateSupplierResponse>('/ceres/enterprise/EnterpriseService/CreateSupplier')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateSupplier.url = '/ceres/enterprise/EnterpriseService/CreateSupplier'
export const UpdateSupplier = (r?: UpdateSupplierRequest, codes?: number[]) =>
  Request<UpdateSupplierResponse>('/ceres/enterprise/EnterpriseService/UpdateSupplier')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateSupplier.url = '/ceres/enterprise/EnterpriseService/UpdateSupplier'
export const DeleteSupplier = (r?: DeleteSupplierRequest, codes?: number[]) =>
  Request<DeleteSupplierResponse>('/ceres/enterprise/EnterpriseService/DeleteSupplier')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteSupplier.url = '/ceres/enterprise/EnterpriseService/DeleteSupplier'

export const BatchExportSupplier = (r?: BatchExportSupplierRequest, codes?: number[]) =>
  Request<BatchExportSupplierResponse>('/ceres/enterprise/EnterpriseService/BatchExportSupplier')
    .code([...(codes || [])])
    .data(r || {})
    .run()
BatchExportSupplier.url = '/ceres/enterprise/EnterpriseService/BatchExportSupplier'
export const GetGroupUser = (r?: GetGroupUserRequest, codes?: number[]) =>
  Request<GetGroupUserResponse>('/ceres/enterprise/EnterpriseService/GetGroupUser')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetGroupUser.url = '/ceres/enterprise/EnterpriseService/GetGroupUser'
export const GetManyGroupUser = (r?: GetManyGroupUserRequest, codes?: number[]) =>
  Request<GetManyGroupUserResponse>('/ceres/enterprise/EnterpriseService/GetManyGroupUser')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetManyGroupUser.url = '/ceres/enterprise/EnterpriseService/GetManyGroupUser'
export const ListGroupUser = (r?: ListGroupUserRequest, codes?: number[]) =>
  Request<ListGroupUserResponse>('/ceres/enterprise/EnterpriseService/ListGroupUser')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListGroupUser.url = '/ceres/enterprise/EnterpriseService/ListGroupUser'
export const CreateGroupUser = (r?: CreateGroupUserRequest, codes?: number[]) =>
  Request<CreateGroupUserResponse>('/ceres/enterprise/EnterpriseService/CreateGroupUser')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateGroupUser.url = '/ceres/enterprise/EnterpriseService/CreateGroupUser'
export const UpdateGroupUser = (r?: UpdateGroupUserRequest, codes?: number[]) =>
  Request<UpdateGroupUserResponse>('/ceres/enterprise/EnterpriseService/UpdateGroupUser')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateGroupUser.url = '/ceres/enterprise/EnterpriseService/UpdateGroupUser'
export const UpdateGroupUserPassword = (r?: UpdateGroupUserPasswordRequest, codes?: number[]) =>
  Request<UpdateGroupUserPasswordResponse>('/ceres/enterprise/EnterpriseService/UpdateGroupUserPassword')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateGroupUserPassword.url = '/ceres/enterprise/EnterpriseService/UpdateGroupUserPassword'
export const DeleteGroupUser = (r?: DeleteGroupUserRequest, codes?: number[]) =>
  Request<DeleteGroupUserResponse>('/ceres/enterprise/EnterpriseService/DeleteGroupUser')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteGroupUser.url = '/ceres/enterprise/EnterpriseService/DeleteGroupUser'
export const UpdateGroupUserRole = (r?: UpdateGroupUserRoleRequest, codes?: number[]) =>
  Request<UpdateGroupUserRoleResponse>('/ceres/enterprise/EnterpriseService/UpdateGroupUserRole')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateGroupUserRole.url = '/ceres/enterprise/EnterpriseService/UpdateGroupUserRole'
export const GetRole = (r?: GetRoleRequest, codes?: number[]) =>
  Request<GetRoleResponse>('/ceres/enterprise/EnterpriseService/GetRole')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetRole.url = '/ceres/enterprise/EnterpriseService/GetRole'
export const ListRole = (r?: ListRoleRequest, codes?: number[]) =>
  Request<ListRoleResponse>('/ceres/enterprise/EnterpriseService/ListRole')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListRole.url = '/ceres/enterprise/EnterpriseService/ListRole'
export const CreateRole = (r?: CreateRoleRequest, codes?: number[]) =>
  Request<CreateRoleResponse>('/ceres/enterprise/EnterpriseService/CreateRole')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateRole.url = '/ceres/enterprise/EnterpriseService/CreateRole'
export const UpdateRole = (r?: UpdateRoleRequest, codes?: number[]) =>
  Request<UpdateRoleResponse>('/ceres/enterprise/EnterpriseService/UpdateRole')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateRole.url = '/ceres/enterprise/EnterpriseService/UpdateRole'
export const DeleteRole = (r?: DeleteRoleRequest, codes?: number[]) =>
  Request<DeleteRoleResponse>('/ceres/enterprise/EnterpriseService/DeleteRole')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteRole.url = '/ceres/enterprise/EnterpriseService/DeleteRole'
export const UpdateRolePermission = (r?: UpdateRolePermissionRequest, codes?: number[]) =>
  Request<UpdateRolePermissionResponse>('/ceres/enterprise/EnterpriseService/UpdateRolePermission')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateRolePermission.url = '/ceres/enterprise/EnterpriseService/UpdateRolePermission'
export const UpdateRoleGroupUser = (r?: UpdateRoleGroupUserRequest, codes?: number[]) =>
  Request<UpdateRoleGroupUserResponse>('/ceres/enterprise/EnterpriseService/UpdateRoleGroupUser')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateRoleGroupUser.url = '/ceres/enterprise/EnterpriseService/UpdateRoleGroupUser'
export const GetServicePeriod = (r?: GetServicePeriodRequest, codes?: number[]) =>
  Request<GetServicePeriodResponse>('/ceres/enterprise/EnterpriseService/GetServicePeriod')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetServicePeriod.url = '/ceres/enterprise/EnterpriseService/GetServicePeriod'
export const GetManyServicePeriod = (r?: GetManyServicePeriodRequest, codes?: number[]) =>
  Request<GetManyServicePeriodResponse>('/ceres/enterprise/EnterpriseService/GetManyServicePeriod')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetManyServicePeriod.url = '/ceres/enterprise/EnterpriseService/GetManyServicePeriod'
export const ListServicePeriod = (r?: ListServicePeriodRequest, codes?: number[]) =>
  Request<ListServicePeriodResponse>('/ceres/enterprise/EnterpriseService/ListServicePeriod')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListServicePeriod.url = '/ceres/enterprise/EnterpriseService/ListServicePeriod'
export const CreateServicePeriod = (r?: CreateServicePeriodRequest, codes?: number[]) =>
  Request<CreateServicePeriodResponse>('/ceres/enterprise/EnterpriseService/CreateServicePeriod')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateServicePeriod.url = '/ceres/enterprise/EnterpriseService/CreateServicePeriod'
export const UpdateServicePeriod = (r?: UpdateServicePeriodRequest, codes?: number[]) =>
  Request<UpdateServicePeriodResponse>('/ceres/enterprise/EnterpriseService/UpdateServicePeriod')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateServicePeriod.url = '/ceres/enterprise/EnterpriseService/UpdateServicePeriod'
export const DeleteServicePeriod = (r?: DeleteServicePeriodRequest, codes?: number[]) =>
  Request<DeleteServicePeriodResponse>('/ceres/enterprise/EnterpriseService/DeleteServicePeriod')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteServicePeriod.url = '/ceres/enterprise/EnterpriseService/DeleteServicePeriod'

export const GetCustomer = (r?: GetCustomerRequest, codes?: number[]) =>
  Request<GetCustomerResponse>('/ceres/enterprise/EnterpriseService/GetCustomer')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetCustomer.url = '/ceres/enterprise/EnterpriseService/GetCustomer'
export const GetManyCustomer = (r?: GetManyCustomerRequest, codes?: number[]) =>
  Request<GetManyCustomerResponse>('/ceres/enterprise/EnterpriseService/GetManyCustomer')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetManyCustomer.url = '/ceres/enterprise/EnterpriseService/GetManyCustomer'
export const ListCustomer = (r?: ListCustomerRequest, codes?: number[]) =>
  Request<ListCustomerResponse>('/ceres/enterprise/EnterpriseService/ListCustomer')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListCustomer.url = '/ceres/enterprise/EnterpriseService/ListCustomer'
export const CreateCustomer = (r?: CreateCustomerRequest, codes?: number[]) =>
  Request<CreateCustomerResponse>('/ceres/enterprise/EnterpriseService/CreateCustomer')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateCustomer.url = '/ceres/enterprise/EnterpriseService/CreateCustomer'
export const CreateCustomerTree = (r?: CreateCustomerTreeRequest, codes?: number[]) =>
  Request<CreateCustomerTreeResponse>('/ceres/enterprise/EnterpriseService/CreateCustomerTree')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateCustomerTree.url = '/ceres/enterprise/EnterpriseService/CreateCustomerTree'
export const UpdateCustomer = (r?: UpdateCustomerRequest, codes?: number[]) =>
  Request<UpdateCustomerResponse>('/ceres/enterprise/EnterpriseService/UpdateCustomer')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateCustomer.url = '/ceres/enterprise/EnterpriseService/UpdateCustomer'
export const DeleteCustomer = (r?: DeleteCustomerRequest, codes?: number[]) =>
  Request<DeleteCustomerResponse>('/ceres/enterprise/EnterpriseService/DeleteCustomer')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteCustomer.url = '/ceres/enterprise/EnterpriseService/DeleteCustomer'

export const BatchExportCustomer = (r?: BatchExportCustomerRequest, codes?: number[]) =>
  Request<BatchExportCustomerResponse>('/ceres/enterprise/EnterpriseService/BatchExportCustomer')
    .code([...(codes || [])])
    .data(r || {})
    .run()
BatchExportCustomer.url = '/ceres/enterprise/EnterpriseService/BatchExportCustomer'
export const BatchImportCustomer = (r?: BatchImportCustomerRequest, codes?: number[]) =>
  Request<BatchImportCustomerResponse>('/ceres/enterprise/EnterpriseService/BatchImportCustomer')
    .code([...(codes || [])])
    .data(r || {})
    .run()
BatchImportCustomer.url = '/ceres/enterprise/EnterpriseService/BatchImportCustomer'
export const GetBatchImportCustomerTemplate = (r?: GetBatchImportCustomerTemplateRequest, codes?: number[]) =>
  Request<GetBatchImportCustomerTemplateResponse>('/ceres/enterprise/EnterpriseService/GetBatchImportCustomerTemplate')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetBatchImportCustomerTemplate.url = '/ceres/enterprise/EnterpriseService/GetBatchImportCustomerTemplate'
export const CountNewCustomer = (r?: CountNewCustomerRequest, codes?: number[]) =>
  Request<CountNewCustomerResponse>('/ceres/enterprise/EnterpriseService/CountNewCustomer')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CountNewCustomer.url = '/ceres/enterprise/EnterpriseService/CountNewCustomer'
export const GetCustomerUser = (r?: GetCustomerUserRequest, codes?: number[]) =>
  Request<GetCustomerUserResponse>('/ceres/enterprise/EnterpriseService/GetCustomerUser')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetCustomerUser.url = '/ceres/enterprise/EnterpriseService/GetCustomerUser'

export const GetManyCustomerUser = (r?: GetManyCustomerUserRequest, codes?: number[]) =>
  Request<GetManyCustomerUserResponse>('/ceres/enterprise/EnterpriseService/GetManyCustomerUser')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetManyCustomerUser.url = '/ceres/enterprise/EnterpriseService/GetManyCustomerUser'
export const ListCustomerUser = (r?: ListCustomerUserRequest, codes?: number[]) =>
  Request<ListCustomerUserResponse>('/ceres/enterprise/EnterpriseService/ListCustomerUser')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListCustomerUser.url = '/ceres/enterprise/EnterpriseService/ListCustomerUser'
export const CreateCustomerUser = (r?: CreateCustomerUserRequest, codes?: number[]) =>
  Request<CreateCustomerUserResponse>('/ceres/enterprise/EnterpriseService/CreateCustomerUser')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateCustomerUser.url = '/ceres/enterprise/EnterpriseService/CreateCustomerUser'
export const UpdateCustomerUser = (r?: UpdateCustomerUserRequest, codes?: number[]) =>
  Request<UpdateCustomerUserResponse>('/ceres/enterprise/EnterpriseService/UpdateCustomerUser')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateCustomerUser.url = '/ceres/enterprise/EnterpriseService/UpdateCustomerUser'
export const UpdateCustomerUserPassword = (r?: UpdateCustomerUserPasswordRequest, codes?: number[]) =>
  Request<UpdateCustomerUserPasswordResponse>('/ceres/enterprise/EnterpriseService/UpdateCustomerUserPassword')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateCustomerUserPassword.url = '/ceres/enterprise/EnterpriseService/UpdateCustomerUserPassword'
export const DeleteCustomerUser = (r?: DeleteCustomerUserRequest, codes?: number[]) =>
  Request<DeleteCustomerUserResponse>('/ceres/enterprise/EnterpriseService/DeleteCustomerUser')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteCustomerUser.url = '/ceres/enterprise/EnterpriseService/DeleteCustomerUser'

export const GetCustomerLabel = (r?: GetCustomerLabelRequest, codes?: number[]) =>
  Request<GetCustomerLabelResponse>('/ceres/enterprise/EnterpriseService/GetCustomerLabel')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetCustomerLabel.url = '/ceres/enterprise/EnterpriseService/GetCustomerLabel'
export const ListCustomerLabel = (r?: ListCustomerLabelRequest, codes?: number[]) =>
  Request<ListCustomerLabelResponse>('/ceres/enterprise/EnterpriseService/ListCustomerLabel')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListCustomerLabel.url = '/ceres/enterprise/EnterpriseService/ListCustomerLabel'
export const CreateCustomerLabel = (r?: CreateCustomerLabelRequest, codes?: number[]) =>
  Request<CreateCustomerLabelResponse>('/ceres/enterprise/EnterpriseService/CreateCustomerLabel')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateCustomerLabel.url = '/ceres/enterprise/EnterpriseService/CreateCustomerLabel'
export const UpdateCustomerLabel = (r?: UpdateCustomerLabelRequest, codes?: number[]) =>
  Request<UpdateCustomerLabelResponse>('/ceres/enterprise/EnterpriseService/UpdateCustomerLabel')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateCustomerLabel.url = '/ceres/enterprise/EnterpriseService/UpdateCustomerLabel'
export const DeleteCustomerLabel = (r?: DeleteCustomerLabelRequest, codes?: number[]) =>
  Request<DeleteCustomerLabelResponse>('/ceres/enterprise/EnterpriseService/DeleteCustomerLabel')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteCustomerLabel.url = '/ceres/enterprise/EnterpriseService/DeleteCustomerLabel'
export const GetDistributionContractor = (r?: GetDistributionContractorRequest, codes?: number[]) =>
  Request<GetDistributionContractorResponse>('/ceres/enterprise/EnterpriseService/GetDistributionContractor')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetDistributionContractor.url = '/ceres/enterprise/EnterpriseService/GetDistributionContractor'
export const ListDistributionContractor = (r?: ListDistributionContractorRequest, codes?: number[]) =>
  Request<ListDistributionContractorResponse>('/ceres/enterprise/EnterpriseService/ListDistributionContractor')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListDistributionContractor.url = '/ceres/enterprise/EnterpriseService/ListDistributionContractor'
export const CreateDistributionContractor = (r?: CreateDistributionContractorRequest, codes?: number[]) =>
  Request<CreateDistributionContractorResponse>('/ceres/enterprise/EnterpriseService/CreateDistributionContractor')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateDistributionContractor.url = '/ceres/enterprise/EnterpriseService/CreateDistributionContractor'
export const UpdateDistributionContractor = (r?: UpdateDistributionContractorRequest, codes?: number[]) =>
  Request<UpdateDistributionContractorResponse>('/ceres/enterprise/EnterpriseService/UpdateDistributionContractor')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateDistributionContractor.url = '/ceres/enterprise/EnterpriseService/UpdateDistributionContractor'
export const DeleteDistributionContractor = (r?: DeleteDistributionContractorRequest, codes?: number[]) =>
  Request<DeleteDistributionContractorResponse>('/ceres/enterprise/EnterpriseService/DeleteDistributionContractor')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteDistributionContractor.url = '/ceres/enterprise/EnterpriseService/DeleteDistributionContractor'
export const ListCountry = (r?: ListCountryRequest, codes?: number[]) =>
  Request<ListCountryResponse>('/ceres/enterprise/EnterpriseService/ListCountry')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListCountry.url = '/ceres/enterprise/EnterpriseService/ListCountry'
export const ListProvince = (r?: ListProvinceRequest, codes?: number[]) =>
  Request<ListProvinceResponse>('/ceres/enterprise/EnterpriseService/ListProvince')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListProvince.url = '/ceres/enterprise/EnterpriseService/ListProvince'
export const ListCity = (r?: ListCityRequest, codes?: number[]) =>
  Request<ListCityResponse>('/ceres/enterprise/EnterpriseService/ListCity')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListCity.url = '/ceres/enterprise/EnterpriseService/ListCity'
export const ListDistrict = (r?: ListDistrictRequest, codes?: number[]) =>
  Request<ListDistrictResponse>('/ceres/enterprise/EnterpriseService/ListDistrict')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListDistrict.url = '/ceres/enterprise/EnterpriseService/ListDistrict'
export const ListStreet = (r?: ListStreetRequest, codes?: number[]) =>
  Request<ListStreetResponse>('/ceres/enterprise/EnterpriseService/ListStreet')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListStreet.url = '/ceres/enterprise/EnterpriseService/ListStreet'
export const ListInvitationCode = (r?: ListInvitationCodeRequest, codes?: number[]) =>
  Request<ListInvitationCodeResponse>('/ceres/enterprise/EnterpriseService/ListInvitationCode')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListInvitationCode.url = '/ceres/enterprise/EnterpriseService/ListInvitationCode'
export const GetInvitationCode = (r?: GetInvitationCodeRequest, codes?: number[]) =>
  Request<GetInvitationCodeResponse>('/ceres/enterprise/EnterpriseService/GetInvitationCode')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetInvitationCode.url = '/ceres/enterprise/EnterpriseService/GetInvitationCode'
export const CreateInvitationCode = (r?: CreateInvitationCodeRequest, codes?: number[]) =>
  Request<CreateInvitationCodeResponse>('/ceres/enterprise/EnterpriseService/CreateInvitationCode')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateInvitationCode.url = '/ceres/enterprise/EnterpriseService/CreateInvitationCode'
export const DeleteInvitationCode = (r?: DeleteInvitationCodeRequest, codes?: number[]) =>
  Request<DeleteInvitationCodeResponse>('/ceres/enterprise/EnterpriseService/DeleteInvitationCode')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteInvitationCode.url = '/ceres/enterprise/EnterpriseService/DeleteInvitationCode'

export const GetUserInfo = (r?: GetUserInfoRequest, codes?: number[]) =>
  Request<GetUserInfoResponse>('/ceres/enterprise/EnterpriseService/GetUserInfo')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetUserInfo.url = '/ceres/enterprise/EnterpriseService/GetUserInfo'
