// gm_api/src/enterprise/pc.tsx
import React, { FC } from 'react'
import { Select, SelectProps } from '@gm-pc/react'
import {
  SelectList,
  SelectEnum,
  MoreSelectList,
  MapId,
} from '../util/pc_util'
import type {
  MapProps,
  SelectListProps,
  SelectEnumProps,
  MoreSelectListProps,
  MapIdProps,
} from '../util/pc_util'
import {
  list_Status_Code,
  map_Status_Code,
  list_Group_Type,
  map_Group_Type,
  list_GroupExtraInfo_Source,
  map_GroupExtraInfo_Source,
  list_GroupExtraInfo_Target,
  map_GroupExtraInfo_Target,
  list_GroupExtraInfo_AnnualSalesAmount,
  map_GroupExtraInfo_AnnualSalesAmount,
  list_GroupExtraInfo_MonthlySalesAmount,
  map_GroupExtraInfo_MonthlySalesAmount,
  list_GroupExtraInfo_Type,
  map_GroupExtraInfo_Type,
  list_GroupExtraInfo_Version,
  map_GroupExtraInfo_Version,
  list_GroupExtraInfo_State,
  map_GroupExtraInfo_State,
  list_Station_Type,
  map_Station_Type,
  list_CreditType,
  map_CreditType,
  list_Permission,
  map_Permission,
  list_GroupUser_Type,
  map_GroupUser_Type,
  list_GroupUser_Status,
  map_GroupUser_Status,
  list_Role_Type,
  map_Role_Type,
  list_ServicePeriod_Type,
  map_ServicePeriod_Type,
  list_ChinaVatInvoice_InvoiceType,
  map_ChinaVatInvoice_InvoiceType,
  list_ChinaVatInvoice_TaxpayerType,
  map_ChinaVatInvoice_TaxpayerType,
  list_Customer_Status,
  map_Customer_Status,
  list_Customer_Type,
  map_Customer_Type,
  list_Customer_SettlementMethod,
  map_Customer_SettlementMethod,
  list_InvitationCode_Type,
  map_InvitationCode_Type,
  Group,
  ListGroup,
  ListGroupResponse,
  Station,
  ListStation,
  ListStationResponse,
  Supplier,
  ListSupplier,
  ListSupplierResponse,
  DistributionContractor,
  ListDistributionContractor,
  ListDistributionContractorResponse,
  GroupUser,
  ListGroupUser,
  ListGroupUserResponse,
  Role,
  ListRole,
  ListRoleResponse,
  ServicePeriod,
  ListServicePeriod,
  ListServicePeriodResponse,
  Country,
  ListCountry,
  ListCountryResponse,
  Province,
  ListProvince,
  ListProvinceResponse,
  City,
  ListCity,
  ListCityResponse,
  District,
  ListDistrict,
  ListDistrictResponse,
  Street,
  ListStreet,
  ListStreetResponse,
  CustomerLabel,
  ListCustomerLabel,
  ListCustomerLabelResponse,
  Customer,
  ListCustomer,
  ListCustomerResponse,
  CustomerUser,
  ListCustomerUser,
  ListCustomerUserResponse,
  InvitationCode,
  ListInvitationCode,
  ListInvitationCodeResponse,
} from './index'

export const Select_Status_Code: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Status_Code} />
}

export const Map_Status_Code: FC<MapProps> = ({ value }) => {
  return <>{map_Status_Code[value]}</>
}

export const Select_Group_Type: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Group_Type} />
}

export const Map_Group_Type: FC<MapProps> = ({ value }) => {
  return <>{map_Group_Type[value]}</>
}

export const Select_GroupExtraInfo_Source: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_GroupExtraInfo_Source} />
}

export const Map_GroupExtraInfo_Source: FC<MapProps> = ({ value }) => {
  return <>{map_GroupExtraInfo_Source[value]}</>
}

export const Select_GroupExtraInfo_Target: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_GroupExtraInfo_Target} />
}

export const Map_GroupExtraInfo_Target: FC<MapProps> = ({ value }) => {
  return <>{map_GroupExtraInfo_Target[value]}</>
}

export const Select_GroupExtraInfo_AnnualSalesAmount: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_GroupExtraInfo_AnnualSalesAmount} />
}

export const Map_GroupExtraInfo_AnnualSalesAmount: FC<MapProps> = ({ value }) => {
  return <>{map_GroupExtraInfo_AnnualSalesAmount[value]}</>
}

export const Select_GroupExtraInfo_MonthlySalesAmount: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_GroupExtraInfo_MonthlySalesAmount} />
}

export const Map_GroupExtraInfo_MonthlySalesAmount: FC<MapProps> = ({ value }) => {
  return <>{map_GroupExtraInfo_MonthlySalesAmount[value]}</>
}

export const Select_GroupExtraInfo_Type: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_GroupExtraInfo_Type} />
}

export const Map_GroupExtraInfo_Type: FC<MapProps> = ({ value }) => {
  return <>{map_GroupExtraInfo_Type[value]}</>
}

export const Select_GroupExtraInfo_Version: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_GroupExtraInfo_Version} />
}

export const Map_GroupExtraInfo_Version: FC<MapProps> = ({ value }) => {
  return <>{map_GroupExtraInfo_Version[value]}</>
}

export const Select_GroupExtraInfo_State: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_GroupExtraInfo_State} />
}

export const Map_GroupExtraInfo_State: FC<MapProps> = ({ value }) => {
  return <>{map_GroupExtraInfo_State[value]}</>
}

export const Select_Station_Type: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Station_Type} />
}

export const Map_Station_Type: FC<MapProps> = ({ value }) => {
  return <>{map_Station_Type[value]}</>
}

export const Select_CreditType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_CreditType} />
}

export const Map_CreditType: FC<MapProps> = ({ value }) => {
  return <>{map_CreditType[value]}</>
}

export const Select_Permission: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Permission} />
}

export const Map_Permission: FC<MapProps> = ({ value }) => {
  return <>{map_Permission[value]}</>
}

export const Select_GroupUser_Type: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_GroupUser_Type} />
}

export const Map_GroupUser_Type: FC<MapProps> = ({ value }) => {
  return <>{map_GroupUser_Type[value]}</>
}

export const Select_GroupUser_Status: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_GroupUser_Status} />
}

export const Map_GroupUser_Status: FC<MapProps> = ({ value }) => {
  return <>{map_GroupUser_Status[value]}</>
}

export const Select_Role_Type: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Role_Type} />
}

export const Map_Role_Type: FC<MapProps> = ({ value }) => {
  return <>{map_Role_Type[value]}</>
}

export const Select_ServicePeriod_Type: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_ServicePeriod_Type} />
}

export const Map_ServicePeriod_Type: FC<MapProps> = ({ value }) => {
  return <>{map_ServicePeriod_Type[value]}</>
}

export const Select_ChinaVatInvoice_InvoiceType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_ChinaVatInvoice_InvoiceType} />
}

export const Map_ChinaVatInvoice_InvoiceType: FC<MapProps> = ({ value }) => {
  return <>{map_ChinaVatInvoice_InvoiceType[value]}</>
}

export const Select_ChinaVatInvoice_TaxpayerType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_ChinaVatInvoice_TaxpayerType} />
}

export const Map_ChinaVatInvoice_TaxpayerType: FC<MapProps> = ({ value }) => {
  return <>{map_ChinaVatInvoice_TaxpayerType[value]}</>
}

export const Select_Customer_Status: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Customer_Status} />
}

export const Map_Customer_Status: FC<MapProps> = ({ value }) => {
  return <>{map_Customer_Status[value]}</>
}

export const Select_Customer_Type: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Customer_Type} />
}

export const Map_Customer_Type: FC<MapProps> = ({ value }) => {
  return <>{map_Customer_Type[value]}</>
}

export const Select_Customer_SettlementMethod: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Customer_SettlementMethod} />
}

export const Map_Customer_SettlementMethod: FC<MapProps> = ({ value }) => {
  return <>{map_Customer_SettlementMethod[value]}</>
}

export const Select_InvitationCode_Type: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_InvitationCode_Type} />
}

export const Map_InvitationCode_Type: FC<MapProps> = ({ value }) => {
  return <>{map_InvitationCode_Type[value]}</>
}

export const Select_Group: FC<SelectListProps<
  Group,
  ListGroupResponse
>> = (props) => {
  return (
    <SelectList<Group, ListGroupResponse>
      {...props}
      messageName='Group'
      fetchList={ListGroup}
    />
  )
}

export const MoreSelect_Group: FC<MoreSelectListProps<
  Group,
  ListGroupResponse
>> = (props) => {
  return (
    <MoreSelectList<Group, ListGroupResponse>
      {...props}
      messageName='Group'
      fetchList={ListGroup}
    />
  )
}

export const Select_Station: FC<SelectListProps<
  Station,
  ListStationResponse
>> = (props) => {
  return (
    <SelectList<Station, ListStationResponse>
      {...props}
      messageName='Station'
      fetchList={ListStation}
    />
  )
}

export const MoreSelect_Station: FC<MoreSelectListProps<
  Station,
  ListStationResponse
>> = (props) => {
  return (
    <MoreSelectList<Station, ListStationResponse>
      {...props}
      messageName='Station'
      fetchList={ListStation}
    />
  )
}

export const Select_Supplier: FC<SelectListProps<
  Supplier,
  ListSupplierResponse
>> = (props) => {
  return (
    <SelectList<Supplier, ListSupplierResponse>
      {...props}
      messageName='Supplier'
      fetchList={ListSupplier}
    />
  )
}

export const MoreSelect_Supplier: FC<MoreSelectListProps<
  Supplier,
  ListSupplierResponse
>> = (props) => {
  return (
    <MoreSelectList<Supplier, ListSupplierResponse>
      {...props}
      messageName='Supplier'
      fetchList={ListSupplier}
    />
  )
}

export const Select_DistributionContractor: FC<SelectListProps<
  DistributionContractor,
  ListDistributionContractorResponse
>> = (props) => {
  return (
    <SelectList<DistributionContractor, ListDistributionContractorResponse>
      {...props}
      messageName='DistributionContractor'
      fetchList={ListDistributionContractor}
    />
  )
}

export const MoreSelect_DistributionContractor: FC<MoreSelectListProps<
  DistributionContractor,
  ListDistributionContractorResponse
>> = (props) => {
  return (
    <MoreSelectList<DistributionContractor, ListDistributionContractorResponse>
      {...props}
      messageName='DistributionContractor'
      fetchList={ListDistributionContractor}
    />
  )
}

export const Select_GroupUser: FC<SelectListProps<
  GroupUser,
  ListGroupUserResponse
>> = (props) => {
  return (
    <SelectList<GroupUser, ListGroupUserResponse>
      {...props}
      messageName='GroupUser'
      fetchList={ListGroupUser}
    />
  )
}

export const MoreSelect_GroupUser: FC<MoreSelectListProps<
  GroupUser,
  ListGroupUserResponse
>> = (props) => {
  return (
    <MoreSelectList<GroupUser, ListGroupUserResponse>
      {...props}
      messageName='GroupUser'
      fetchList={ListGroupUser}
    />
  )
}

export const Select_Role: FC<SelectListProps<
  Role,
  ListRoleResponse
>> = (props) => {
  return (
    <SelectList<Role, ListRoleResponse>
      {...props}
      messageName='Role'
      fetchList={ListRole}
    />
  )
}

export const MoreSelect_Role: FC<MoreSelectListProps<
  Role,
  ListRoleResponse
>> = (props) => {
  return (
    <MoreSelectList<Role, ListRoleResponse>
      {...props}
      messageName='Role'
      fetchList={ListRole}
    />
  )
}

export const Select_ServicePeriod: FC<SelectListProps<
  ServicePeriod,
  ListServicePeriodResponse
>> = (props) => {
  return (
    <SelectList<ServicePeriod, ListServicePeriodResponse>
      {...props}
      messageName='ServicePeriod'
      fetchList={ListServicePeriod}
    />
  )
}

export const MoreSelect_ServicePeriod: FC<MoreSelectListProps<
  ServicePeriod,
  ListServicePeriodResponse
>> = (props) => {
  return (
    <MoreSelectList<ServicePeriod, ListServicePeriodResponse>
      {...props}
      messageName='ServicePeriod'
      fetchList={ListServicePeriod}
    />
  )
}

export const Select_Country: FC<SelectListProps<
  Country,
  ListCountryResponse
>> = (props) => {
  return (
    <SelectList<Country, ListCountryResponse>
      {...props}
      messageName='Country'
      fetchList={ListCountry}
    />
  )
}

export const MoreSelect_Country: FC<MoreSelectListProps<
  Country,
  ListCountryResponse
>> = (props) => {
  return (
    <MoreSelectList<Country, ListCountryResponse>
      {...props}
      messageName='Country'
      fetchList={ListCountry}
    />
  )
}

export const Select_Province: FC<SelectListProps<
  Province,
  ListProvinceResponse
>> = (props) => {
  return (
    <SelectList<Province, ListProvinceResponse>
      {...props}
      messageName='Province'
      fetchList={ListProvince}
    />
  )
}

export const MoreSelect_Province: FC<MoreSelectListProps<
  Province,
  ListProvinceResponse
>> = (props) => {
  return (
    <MoreSelectList<Province, ListProvinceResponse>
      {...props}
      messageName='Province'
      fetchList={ListProvince}
    />
  )
}

export const Select_City: FC<SelectListProps<
  City,
  ListCityResponse
>> = (props) => {
  return (
    <SelectList<City, ListCityResponse>
      {...props}
      messageName='City'
      fetchList={ListCity}
    />
  )
}

export const MoreSelect_City: FC<MoreSelectListProps<
  City,
  ListCityResponse
>> = (props) => {
  return (
    <MoreSelectList<City, ListCityResponse>
      {...props}
      messageName='City'
      fetchList={ListCity}
    />
  )
}

export const Select_District: FC<SelectListProps<
  District,
  ListDistrictResponse
>> = (props) => {
  return (
    <SelectList<District, ListDistrictResponse>
      {...props}
      messageName='District'
      fetchList={ListDistrict}
    />
  )
}

export const MoreSelect_District: FC<MoreSelectListProps<
  District,
  ListDistrictResponse
>> = (props) => {
  return (
    <MoreSelectList<District, ListDistrictResponse>
      {...props}
      messageName='District'
      fetchList={ListDistrict}
    />
  )
}

export const Select_Street: FC<SelectListProps<
  Street,
  ListStreetResponse
>> = (props) => {
  return (
    <SelectList<Street, ListStreetResponse>
      {...props}
      messageName='Street'
      fetchList={ListStreet}
    />
  )
}

export const MoreSelect_Street: FC<MoreSelectListProps<
  Street,
  ListStreetResponse
>> = (props) => {
  return (
    <MoreSelectList<Street, ListStreetResponse>
      {...props}
      messageName='Street'
      fetchList={ListStreet}
    />
  )
}

export const Select_CustomerLabel: FC<SelectListProps<
  CustomerLabel,
  ListCustomerLabelResponse
>> = (props) => {
  return (
    <SelectList<CustomerLabel, ListCustomerLabelResponse>
      {...props}
      messageName='CustomerLabel'
      fetchList={ListCustomerLabel}
    />
  )
}

export const MoreSelect_CustomerLabel: FC<MoreSelectListProps<
  CustomerLabel,
  ListCustomerLabelResponse
>> = (props) => {
  return (
    <MoreSelectList<CustomerLabel, ListCustomerLabelResponse>
      {...props}
      messageName='CustomerLabel'
      fetchList={ListCustomerLabel}
    />
  )
}

export const Select_Customer: FC<SelectListProps<
  Customer,
  ListCustomerResponse
>> = (props) => {
  return (
    <SelectList<Customer, ListCustomerResponse>
      {...props}
      messageName='Customer'
      fetchList={ListCustomer}
    />
  )
}

export const MoreSelect_Customer: FC<MoreSelectListProps<
  Customer,
  ListCustomerResponse
>> = (props) => {
  return (
    <MoreSelectList<Customer, ListCustomerResponse>
      {...props}
      messageName='Customer'
      fetchList={ListCustomer}
    />
  )
}

export const Select_CustomerUser: FC<SelectListProps<
  CustomerUser,
  ListCustomerUserResponse
>> = (props) => {
  return (
    <SelectList<CustomerUser, ListCustomerUserResponse>
      {...props}
      messageName='CustomerUser'
      fetchList={ListCustomerUser}
    />
  )
}

export const MoreSelect_CustomerUser: FC<MoreSelectListProps<
  CustomerUser,
  ListCustomerUserResponse
>> = (props) => {
  return (
    <MoreSelectList<CustomerUser, ListCustomerUserResponse>
      {...props}
      messageName='CustomerUser'
      fetchList={ListCustomerUser}
    />
  )
}

export const Select_InvitationCode: FC<SelectListProps<
  InvitationCode,
  ListInvitationCodeResponse
>> = (props) => {
  return (
    <SelectList<InvitationCode, ListInvitationCodeResponse>
      {...props}
      messageName='InvitationCode'
      fetchList={ListInvitationCode}
    />
  )
}

export const MoreSelect_InvitationCode: FC<MoreSelectListProps<
  InvitationCode,
  ListInvitationCodeResponse
>> = (props) => {
  return (
    <MoreSelectList<InvitationCode, ListInvitationCodeResponse>
      {...props}
      messageName='InvitationCode'
      fetchList={ListInvitationCode}
    />
  )
}

export const MapId_Group: FC<MapIdProps<Group, ListGroupResponse>> = (
  props,
) => {
  return (
    <MapId<Group, ListGroupResponse>
      {...props}
      packageName='enterprise'
      messageName='Group'
      fetchList={ListGroup}
    />
  )
}

export const MapId_GroupUser: FC<MapIdProps<GroupUser, ListGroupUserResponse>> = (
  props,
) => {
  return (
    <MapId<GroupUser, ListGroupUserResponse>
      {...props}
      packageName='enterprise'
      messageName='GroupUser'
      fetchList={ListGroupUser}
    />
  )
}

export const MapId_Role: FC<MapIdProps<Role, ListRoleResponse>> = (
  props,
) => {
  return (
    <MapId<Role, ListRoleResponse>
      {...props}
      packageName='enterprise'
      messageName='Role'
      fetchList={ListRole}
    />
  )
}

export const MapId_ServicePeriod: FC<MapIdProps<ServicePeriod, ListServicePeriodResponse>> = (
  props,
) => {
  return (
    <MapId<ServicePeriod, ListServicePeriodResponse>
      {...props}
      packageName='enterprise'
      messageName='ServicePeriod'
      fetchList={ListServicePeriod}
    />
  )
}

export const MapId_Country: FC<MapIdProps<Country, ListCountryResponse>> = (
  props,
) => {
  return (
    <MapId<Country, ListCountryResponse>
      {...props}
      packageName='enterprise'
      messageName='Country'
      fetchList={ListCountry}
    />
  )
}

export const MapId_Province: FC<MapIdProps<Province, ListProvinceResponse>> = (
  props,
) => {
  return (
    <MapId<Province, ListProvinceResponse>
      {...props}
      packageName='enterprise'
      messageName='Province'
      fetchList={ListProvince}
    />
  )
}

export const MapId_City: FC<MapIdProps<City, ListCityResponse>> = (
  props,
) => {
  return (
    <MapId<City, ListCityResponse>
      {...props}
      packageName='enterprise'
      messageName='City'
      fetchList={ListCity}
    />
  )
}

export const MapId_District: FC<MapIdProps<District, ListDistrictResponse>> = (
  props,
) => {
  return (
    <MapId<District, ListDistrictResponse>
      {...props}
      packageName='enterprise'
      messageName='District'
      fetchList={ListDistrict}
    />
  )
}

export const MapId_Street: FC<MapIdProps<Street, ListStreetResponse>> = (
  props,
) => {
  return (
    <MapId<Street, ListStreetResponse>
      {...props}
      packageName='enterprise'
      messageName='Street'
      fetchList={ListStreet}
    />
  )
}

export const MapId_Customer: FC<MapIdProps<Customer, ListCustomerResponse>> = (
  props,
) => {
  return (
    <MapId<Customer, ListCustomerResponse>
      {...props}
      packageName='enterprise'
      messageName='Customer'
      fetchList={ListCustomer}
    />
  )
}

export const MapId_CustomerUser: FC<MapIdProps<CustomerUser, ListCustomerUserResponse>> = (
  props,
) => {
  return (
    <MapId<CustomerUser, ListCustomerUserResponse>
      {...props}
      packageName='enterprise'
      messageName='CustomerUser'
      fetchList={ListCustomerUser}
    />
  )
}

export type {
  MapProps,
  SelectListProps,
  MoreSelectListProps,
}
