// gm_api/src/enterprise/types.ts
import * as account from '../account'
import * as common from '../common'
import * as dao from '../dao'
import * as idgenerator from '../idgenerator'
export enum Status_Code {
  CODE_UNSPECIFIED = 0,
  DUPLICATE_ROLE_NAME = 23330001,
  DUPLICATE_USER_NAME = 23330002,
  DUPLICATE_USER_USERNAME = 23330004,
  DUPLICATE_USER_PHONE = 23330005,
  DUPLICATE_GROUP_NAME = 23330003,
}
export interface Status_DuplicateRoleNameDetail {
  role_id?: string
}
export interface Status_DuplicateUserNameDetail {
  user_id?: string
}
export interface Status_DuplicateUserUsernameDetail {
  user_id?: string
}
export interface Status_DuplicateUserPhoneDetail {
  user_id?: string
}
export interface Status {
}
export enum Group_Type {
  TYPE_UNSPECIFIED = 0,
  INTERNAL = 1,
  EXTERNAL = 2,
}
export interface Group {
  group_id: string
  create_time?: string
  update_time?: string
  is_valid?: boolean
  name: string
  address?: Address
  time_zone?: string
  type: Group_Type
}
export interface ReqCreateGroup {
  // group_id: string
  // create_time?: string
  // update_time?: string
  is_valid?: boolean
  name: string
  address?: Address
  time_zone?: string
  type: Group_Type
}
export enum GroupExtraInfo_Source {
  SOURCE_UNSPECIFIED = 0,
  SOURCE_PROMOTION = 1,
  SOURCE_REFERRAL = 2,
  SOURCE_SELF_EXPANSION = 3,
}
export enum GroupExtraInfo_Target {
  TARGET_UNSPECIFIED = 0,
  TARGET_STORES = 1,
  TARGET_RESTAURANT = 2,
  TARGET_HOTEL = 3,
  TARGET_GROUP_MEALS = 4,
  TARGET_NUTRITIONAL_MEALS = 5,
  TARGET_INDIVIDUAL = 6,
  TARGET_RETAIL = 7,
  TARGET_OTHER = 8,
}
export enum GroupExtraInfo_AnnualSalesAmount {
  ANNUAL_SALES_AMOUNT_UNSPECIFIED = 0,
  ANNUAL_SALES_AMOUNT_0_1000 = 1,
  ANNUAL_SALES_AMOUNT_1000_3000 = 2,
  ANNUAL_SALES_AMOUNT_3000_5000 = 3,
  ANNUAL_SALES_AMOUNT_5000_10000 = 4,
  ANNUAL_SALES_AMOUNT_10000_INF = 5,
}
export enum GroupExtraInfo_MonthlySalesAmount {
  MONTHLY_SALES_AMOUNT_UNSPECIFIED = 0,
  MONTHLY_SALES_AMOUNT_0_50 = 1,
  MONTHLY_SALES_AMOUNT_50_100 = 2,
  MONTHLY_SALES_AMOUNT_100_200 = 3,
  MONTHLY_SALES_AMOUNT_200_500 = 4,
  MONTHLY_SALES_AMOUNT_500_1000 = 5,
  MONTHLY_SALES_AMOUNT_1000_INF = 6,
}
export enum GroupExtraInfo_Type {
  TYPE_UNSPECIFIED = 0,
  TYPE_NORMAL = 1,
  TYPE_KA = 2,
}
export enum GroupExtraInfo_Version {
  VERSION_UNSPECIFIED = 0,
  VERSION_STANDARD = 1,
}
export enum GroupExtraInfo_State {
  STATE_UNSPECIFIED = 0,
  STATE_NORMAL = 1,
  STATE_ONLINE = 2,
  STATE_OFFLINE = 3,
  STATE_STOP = 4,
}
export interface GroupExtraInfo_Attrs {
  targets?: GroupExtraInfo_Target[]
}
export interface GroupExtraInfo {
  group_id: string
  create_time?: string
  update_time?: string
  attrs?: GroupExtraInfo_Attrs
  address?: Address
  source?: GroupExtraInfo_Source
  annual_sales_amount?: GroupExtraInfo_AnnualSalesAmount
  monthly_sales_amount?: GroupExtraInfo_MonthlySalesAmount
  type?: GroupExtraInfo_Type
  version?: GroupExtraInfo_Version
  state?: GroupExtraInfo_State
  saler?: string
  contract_amount?: string
  maintenance_amount?: string
  remark?: string
  up_time?: string
  downline_time?: string
  renewal_time?: string
  handover_time?: string
}
export enum Station_Type {
  TYPE_UNSPECIFIED = 0,
  NORMAL = 1,
}
export interface Station_Attrs {
  available_city_ids?: string[]
}
export interface Station {
  station_id: string
  create_time?: string
  update_time?: string
  group_id?: string
  default_quotation_id?: string
  attrs?: Station_Attrs
  name: string
  description?: string
  is_valid?: boolean
  type: Station_Type
  address: Address
}
export interface ReqCreateStation {
  // station_id: string
  // create_time?: string
  // update_time?: string
  // group_id?: string
  default_quotation_id?: string
  attrs?: Station_Attrs
  name: string
  description?: string
  is_valid?: boolean
  type: Station_Type
  address: Address
}
export enum CreditType {
  CREDIT_TYPE_UNSPECIFIED = 0,
  CREDIT_TYPE_AMOUNT = 1,
  CREDIT_TYPE_DAILY = 3,
  CREDIT_TYPE_WEEKLY = 4,
  CREDIT_TYPE_MONTHLY = 5,
  CREDIT_TYPE_CUSTOMIZED = 6,
  CREDIT_TYPE_HALF_MONTHLY = 8,
}
export interface Supplier_Attrs {
  china_vat_invoice?: ChinaVatInvoice
  qualification_images?: common.Image[]
  available_category_ids?: string[]
}
export interface Supplier {
  supplier_id: string
  status?: string
  create_time?: string
  update_time?: string
  delete_time?: string
  group_id?: string
  default_purchase_id?: string
  name: string
  description?: string
  is_valid?: boolean
  address?: Address
  credit_type?: CreditType
  phone?: string
  customized_code: string
  attrs?: Supplier_Attrs
}
export interface ReqCreateSupplier {
  // supplier_id: string
  status?: string
  // create_time?: string
  // update_time?: string
  // delete_time?: string
  // group_id?: string
  default_purchase_id?: string
  name: string
  description?: string
  is_valid?: boolean
  address?: Address
  credit_type?: CreditType
  phone?: string
  customized_code: string
  attrs?: Supplier_Attrs
}
export interface DistributionContractor {
  distribution_contractor_id: string
  create_time?: string
  update_time?: string
  delete_time?: string
  group_id?: string
  name: string
}
export interface ReqCreateDistributionContractor {
  // distribution_contractor_id: string
  // create_time?: string
  // update_time?: string
  // delete_time?: string
  // group_id?: string
  name: string
}
export enum Permission {
  PERMISSION_UNSPECIFIED = 0,
  PERMISSION_MERCHANDISE_VIEW_QUOTATION = 10,
  PERMISSION_MERCHANDISE_CREATE_QUOTATION = 11,
  PERMISSION_MERCHANDISE_UPDATE_QUOTATION = 12,
  PERMISSION_MERCHANDISE_DELETE_QUOTATION = 13,
  PERMISSION_MERCHANDISE_VIEW_NOT_PACKAGE_SKU_SSU = 20,
  PERMISSION_MERCHANDISE_CREATE_NOT_PACKAGE_SKU_SSU = 21,
  PERMISSION_MERCHANDISE_UPDATE_NOT_PACKAGE_SKU_SSU = 22,
  PERMISSION_MERCHANDISE_DELETE_NOT_PACKAGE_SKU_SSU = 23,
  PERMISSION_MERCHANDISE_VIEW_PACKAGE_SKU_SSU = 30,
  PERMISSION_MERCHANDISE_CREATE_PACKAGE_SKU_SSU = 31,
  PERMISSION_MERCHANDISE_UPDATE_PACKAGE_SKU_SSU = 32,
  PERMISSION_MERCHANDISE_DELETE_PACKAGE_SKU_SSU = 33,
  PERMISSION_PRODUCTION_VIEW_BOM = 40,
  PERMISSION_PRODUCTION_CREATE_BOM = 41,
  PERMISSION_PRODUCTION_UPDATE_BOM = 42,
  PERMISSION_PRODUCTION_DELETE_BOM = 43,
  PERMISSION_MERCHANDISE_VIEW_CATEGORY = 50,
  PERMISSION_MERCHANDISE_CREATE_CATEGORY = 51,
  PERMISSION_MERCHANDISE_UPDATE_CATEGORY = 52,
  PERMISSION_MERCHANDISE_DELETE_CATEGORY = 53,
  PERMISSION_ESHOP_VIEW_MENU = 500,
  PERMISSION_ESHOP_UPDATE_MENU = 502,
  PERMISSION_ESHOP_VIEW_MENU_DETAIL = 504,
  PERMISSION_ESHOP_UPDATE_MENU_DETAIL = 505,
  PERMISSION_ESHOP_CREATE_MENU = 501,
  PERMISSION_ESHOP_DELETE_MENU = 503,
  PERMISSION_ESHOP_VIEW_MENU_PERIOD = 510,
  PERMISSION_ESHOP_CREATE_MENU_PERIOD = 511,
  PERMISSION_ESHOP_UPDATE_MENU_PERIOD = 512,
  PERMISSION_ESHOP_DELETE_MENU_PERIOD = 513,
  PERMISSION_ORDER_VIEW_ORDER = 60,
  PERMISSION_ORDER_CREATE_ORDER = 61,
  PERMISSION_ORDER_UPDATE_ORDER = 62,
  PERMISSION_ORDER_DELETE_ORDER = 63,
  PERMISSION_ORDER_AMEND_ORDER = 64,
  PERMISSION_ORDER_RELEASE_PROCESS_PLAN = 65,
  PERMISSION_ORDER_RELEASE_NONPROCESS_PLAN = 66,
  PERMISSION_ORDER_VIEW_ESHOP_STUDENT_ORDER = 530,
  PERMISSION_ORDER_DELETE_ESHOP_STUDENT_ORDER = 531,
  PERMISSION_ORDER_VIEW_ESHOP_STUFF_ORDER = 532,
  PERMISSION_ORDER_DELETE_ESHOP_STUFF_ORDER = 533,
  PERMISSION_ORDER_VIEW_LEAVE_APPLICATION = 539,
  PERMISSION_PURCHASE_VIEW_PURCHASE_TASK = 70,
  PERMISSION_PURCHASE_CREATE_PURCHASE_TASK = 71,
  PERMISSION_PURCHASE_UPDATE_PURCHASE_TASK = 72,
  PERMISSION_PURCHASE_DELETE_PURCHASE_TASK = 73,
  PERMISSION_PURCHASE_VIEW_PURCHASE_SHEET = 80,
  PERMISSION_PURCHASE_CREATE_PURCHASE_SHEET = 81,
  PERMISSION_PURCHASE_UPDATE_PURCHASE_SHEET = 82,
  PERMISSION_PURCHASE_DELETE_PURCHASE_SHEET = 83,
  PERMISSION_ENTERPRISE_VIEW_PURCHASER = 90,
  PERMISSION_ENTERPRISE_CREATE_PURCHASER = 91,
  PERMISSION_ENTERPRISE_UPDATE_PURCHASER = 92,
  PERMISSION_ENTERPRISE_DELETE_PURCHASER = 93,
  PERMISSION_PURCHASE_VIEW_INQUIRY_PRICE = 100,
  PERMISSION_PURCHASE_IMPORT_INQUIRY_PRICE = 101,
  PERMISSION_PRODUCTION_LIST_PRODUCETASK = 110,
  PERMISSION_PRODUCTION_CREATE_PRODUCETASK = 111,
  PERMISSION_PRODUCTION_UPDATE_PRODUCETASK = 112,
  PERMISSION_PRODUCTION_DELETE_PRODUCETASK = 113,
  PERMISSION_PRODUCTION_LIST_PACKTASK = 120,
  PERMISSION_PRODUCTION_CREATE_PACKTASK = 121,
  PERMISSION_PRODUCTION_UPDATE_PACKTASK = 122,
  PERMISSION_PRODUCTION_DELETE_PACKTASK = 123,
  PERMISSION_PRODUCTION_VIEW_PRODUCEPROGRESS = 130,
  PERMISSION_PRODUCTION_VIEW_PACKPROGRESS = 131,
  PERMISSION_PRODUCTION_VIEW_PRODUCEANALYTICS = 132,
  PERMISSION_PRODUCTION_VIEW_PACKANALYTICS = 133,
  PERMISSION_PRODUCTION_PRINT_BOX_LABLE = 134,
  PERMISSION_DELIVERY_VIEW_TASK = 140,
  PERMISSION_DELIVERY_UPDATE_TASK = 141,
  PERMISSION_DELIVERY_PRINT_TASK = 142,
  PERMISSION_ENTERPRISE_VIEW_DRIVER = 150,
  PERMISSION_ENTERPRISE_CREATE_DRIVER = 151,
  PERMISSION_ENTERPRISE_UPDATE_DRIVER = 152,
  PERMISSION_ENTERPRISE_DELETE_DRIVER = 153,
  PERMISSION_ENTERPRISE_CREATE_DISTRIBUTION_CONTRACTOR = 160,
  PERMISSION_ENTERPRISE_UPDATE_DISTRIBUTION_CONTRACTOR = 161,
  PERMISSION_ENTERPRISE_DELETE_DISTRIBUTION_CONTRACTOR = 162,
  PERMISSION_DELIVERY_VIEW_ROUTE = 170,
  PERMISSION_DELIVERY_CREATE_ROUTE = 171,
  PERMISSION_DELIVERY_UPDATE_ROUTE = 172,
  PERMISSION_DELIVERY_DELETE_ROUTE = 173,
  PERMISSION_INVENTORY_VIEW_PURCHASE_IN_SHEET = 180,
  PERMISSION_INVENTORY_CREATE_PURCHASE_IN_SHEET = 181,
  PERMISSION_INVENTORY_UPDATE_PURCHASE_IN_SHEET = 182,
  PERMISSION_INVENTORY_SUBMIT_PURCHASE_IN_SHEET = 183,
  PERMISSION_INVENTORY_APPROVE_PURCHASE_IN_SHEET = 184,
  PERMISSION_INVENTORY_NOT_APPROVE_PURCHASE_IN_SHEET = 185,
  PERMISSION_INVENTORY_CANCEL_APPROVE_PURCHASE_IN_SHEET = 186,
  PERMISSION_INVENTORY_DELETE_PURCHASE_IN_SHEET = 187,
  PERMISSION_INVENTORY_EXPORT_PURCHASE_IN_SHEET = 188,
  PERMISSION_INVENTORY_PRINT_PURCHASE_IN_SHEET = 189,
  PERMISSION_INVENTORY_VIEW_REFUND_OUT_SHEET = 200,
  PERMISSION_INVENTORY_CREATE_REFUND_OUT_SHEET = 201,
  PERMISSION_INVENTORY_UPDATE_REFUND_OUT_SHEET = 202,
  PERMISSION_INVENTORY_SUBMIT_REFUND_OUT_SHEET = 203,
  PERMISSION_INVENTORY_APPROVE_REFUND_OUT_SHEET = 204,
  PERMISSION_INVENTORY_NOT_APPROVE_REFUND_OUT_SHEET = 205,
  PERMISSION_INVENTORY_CANCEL_APPROVE_REFUND_OUT_SHEET = 206,
  PERMISSION_INVENTORY_DELETE_REFUND_OUT_SHEET = 207,
  PERMISSION_INVENTORY_EXPORT_REFUND_OUT_SHEET = 208,
  PERMISSION_INVENTORY_PRINT_REFUND_OUT_SHEET = 209,
  PERMISSION_INVENTORY_VIEW_SALE_OUT_SHEET = 220,
  PERMISSION_INVENTORY_CREATE_SALE_OUT_SHEET = 221,
  PERMISSION_INVENTORY_UPDATE_SALE_OUT_SHEET = 222,
  PERMISSION_INVENTORY_SUBMIT_SALE_OUT_SHEET = 223,
  PERMISSION_INVENTORY_APPROVE_SALE_OUT_SHEET = 224,
  PERMISSION_INVENTORY_NOT_APPROVE_SALE_OUT_SHEET = 225,
  PERMISSION_INVENTORY_CANCEL_APPROVE_SALE_OUT_SHEET = 226,
  PERMISSION_INVENTORY_DELETE_SALE_OUT_SHEET = 227,
  PERMISSION_INVENTORY_EXPORT_SALE_OUT_SHEET = 228,
  PERMISSION_INVENTORY_PRINT_SALE_OUT_SHEET = 229,
  PERMISSION_INVENTORY_VIEW_PRODUCT_IN = 230,
  PERMISSION_INVENTORY_CREATE_PRODUCT_IN = 231,
  PERMISSION_INVENTORY_UPDATE_PRODUCT_IN = 232,
  PERMISSION_INVENTORY_SUBMIT_PRODUCT_IN = 233,
  PERMISSION_INVENTORY_APPROVE_PRODUCT_IN = 234,
  PERMISSION_INVENTORY_NOT_APPROVE_PRODUCT_IN = 235,
  PERMISSION_INVENTORY_CANCEL_APPROVE_PRODUCT_IN = 236,
  PERMISSION_INVENTORY_DELETE_PRODUCT_IN = 237,
  PERMISSION_INVENTORY_EXPORT_PRODUCT_IN = 238,
  PERMISSION_INVENTORY_PRINT_PRODUCT_IN = 239,
  PERMISSION_INVENTORY_VIEW_MATERIAL_IN_250 = 250,
  PERMISSION_INVENTORY_CREATE_MATERIAL_IN_251 = 251,
  PERMISSION_INVENTORY_UPDATE_MATERIAL_IN_252 = 252,
  PERMISSION_INVENTORY_SUBMIT_MATERIAL_IN_253 = 253,
  PERMISSION_INVENTORY_APPROVE_MATERIAL_IN_254 = 254,
  PERMISSION_INVENTORY_NOT_APPROVE_MATERIAL_IN_255 = 255,
  PERMISSION_INVENTORY_CANCEL_APPROVE_MATERIAL_IN_256 = 256,
  PERMISSION_INVENTORY_DELETE_MATERIAL_IN_257 = 257,
  PERMISSION_INVENTORY_EXPORT_MATERIAL_IN_258 = 258,
  PERMISSION_INVENTORY_PRINT_MATERIAL_IN_259 = 259,
  PERMISSION_INVENTORY_VIEW_MATERIAL_OUT = 270,
  PERMISSION_INVENTORY_CREATE_MATERIAL_OUT = 271,
  PERMISSION_INVENTORY_UPDATE_MATERIAL_OUT = 272,
  PERMISSION_INVENTORY_SUBMIT_MATERIAL_OUT = 273,
  PERMISSION_INVENTORY_APPROVE_MATERIAL_OUT = 274,
  PERMISSION_INVENTORY_NOT_APPROVE_MATERIAL_OUT = 275,
  PERMISSION_INVENTORY_CANCEL_APPROVE_MATERIAL_OUT = 276,
  PERMISSION_INVENTORY_DELETE_MATERIAL_OUT = 277,
  PERMISSION_INVENTORY_EXPORT_MATERIAL_OUT = 278,
  PERMISSION_INVENTORY_PRINT_MATERIAL_OUT = 279,
  PERMISSION_INVENTORY_VIEW_OTHER_IN = 290,
  PERMISSION_INVENTORY_CREATE_OTHER_IN = 291,
  PERMISSION_INVENTORY_UPDATE_OTHER_IN = 292,
  PERMISSION_INVENTORY_SUBMIT_OTHER_IN = 293,
  PERMISSION_INVENTORY_APPROVE_OTHER_IN = 294,
  PERMISSION_INVENTORY_NOT_APPROVE_OTHER_IN = 295,
  PERMISSION_INVENTORY_CANCEL_APPROVE_OTHER_IN = 296,
  PERMISSION_INVENTORY_DELETE_OTHER_IN = 297,
  PERMISSION_INVENTORY_EXPORT_OTHER_IN = 298,
  PERMISSION_INVENTORY_PRINT_OTHER_IN = 299,
  PERMISSION_INVENTORY_VIEW_OTHER_OUT = 310,
  PERMISSION_INVENTORY_CREATE_OTHER_OUT = 311,
  PERMISSION_INVENTORY_UPDATE_OTHER_OUT = 312,
  PERMISSION_INVENTORY_SUBMIT_OTHER_OUT = 313,
  PERMISSION_INVENTORY_APPROVE_OTHER_OUT = 314,
  PERMISSION_INVENTORY_NOT_APPROVE_OTHER_OUT = 315,
  PERMISSION_INVENTORY_CANCEL_APPROVE_OTHER_OUT = 316,
  PERMISSION_INVENTORY_DELETE_OTHER_OUT = 317,
  PERMISSION_INVENTORY_EXPORT_OTHER_OUT = 318,
  PERMISSION_INVENTORY_PRINT_OTHER_OUT = 319,
  PERMISSION_INVENTORY_VIEW_ADJUST_SHEET = 330,
  PERMISSION_INVENTORY_VIEW_INVENTORY_SHEET = 340,
  PERMISSION_INVENTORY_CREATE_INVENTORY_SHEET = 341,
  PERMISSION_INVENTORY_UPDATE_INVENTORY_SHEET = 342,
  PERMISSION_INVENTORY_SUBMIT_INVENTORY_SHEET = 343,
  PERMISSION_INVENTORY_APPROVE_INVENTORY_SHEET = 344,
  PERMISSION_INVENTORY_NOT_APPROVE_INVENTORY_SHEET = 345,
  PERMISSION_INVENTORY_CANCEL_APPROVE_INVENTORY_SHEET = 346,
  PERMISSION_INVENTORY_EXPORT_INVENTORY_SHEET = 347,
  PERMISSION_INVENTORY_PRINT_INVENTORY_SHEET = 348,
  PERMISSION_INVENTORY_DELETE_INVENTORY_SHEET = 349,
  PERMISSION_INVENTORY_VIEW_STOCK = 360,
  PERMISSION_INVENTORY_VIEW_STOCK_LIST = 361,
  PERMISSION_INVENTORY_EXPORT_STOCK_LIST = 362,
  PERMISSION_INVENTORY_VIEW_IN_STOCK_LOG = 363,
  PERMISSION_INVENTORY_EXPORT_IN_STOCK_LOG = 364,
  PERMISSION_INVENTORY_VIEW_OUT_STOCK_LOG = 365,
  PERMISSION_INVENTORY_EXPORT_OUT_STOCK_LOG = 366,
  PERMISSION_INVENTORY_VIEW_CHANGE_STOCK_LOG = 367,
  PERMISSION_INVENTORY_EXPORT_CHANGE_STOCK_LOG = 368,
  PERMISSION_ENTERPRISE_VIEW_CUSTOMER = 380,
  PERMISSION_ENTERPRISE_CREATE_CUSTOMER = 381,
  PERMISSION_ENTERPRISE_UPDATE_CUSTOMER = 382,
  PERMISSION_ENTERPRISE_DELETE_CUSTOMER = 383,
  PERMISSION_ENTERPRISE_VIEW_INVITATION_CODE = 390,
  PERMISSION_ENTERPRISE_VIEW_SCHOOL_CLASS = 520,
  PERMISSION_ENTERPRISE_CREATE_SCHOOL_CLASS = 521,
  PERMISSION_ENTERPRISE_UPDATE_SCHOOL_CLASS = 522,
  PERMISSION_ENTERPRISE_DELETE_SCHOOL_CLASS = 523,
  PERMISSION_ENTERPRISE_VIEW_STUDENT = 540,
  PERMISSION_ENTERPRISE_VIEW_STUFF = 541,
  PERMISSION_ENTERPRISE_VIEW_ESHOP_INVITATION_CODE = 391,
  PERMISSION_ANALYTICS_VIEW_MENU_SETTLEMENT = 550,
  PERMISSION_ENTERPRISE_VIEW_SERVICE_PERIOD = 400,
  PERMISSION_ENTERPRISE_CREATE_SERVICE_PERIOD = 401,
  PERMISSION_ENTERPRISE_UPDATE_SERVICE_PERIOD = 402,
  PERMISSION_ENTERPRISE_DELETE_SERVICE_PERIOD = 403,
  PERMISSION_PREFERENCE_VIEW_APPOINT_TIME_SETTINGS = 410,
  PERMISSION_PREFERENCE_UPDATE_APPOINT_TIME_SETTINGS = 411,
  PERMISSION_PREFERENCE_VIEW_PURCHASE_SETTINGS = 420,
  PERMISSION_PREFERENCE_UPDATE_PURCHASE_SETTINGS = 421,
  PERMISSION_PRODUCTION_VIEW_SETTINGS = 430,
  PERMISSION_PRODUCTION_UPDATE_SETTINGS = 431,
  PERMISSION_PREFERENCE_INVENTORY_SETTINGS = 440,
  PERMISSION_PREFERENCE_PRODUCTION_SETTINGS = 441,
  PERMISSION_PREFERENCE_VIEW_IMPORT_TEMPLATE = 450,
  PERMISSION_PREFERENCE_CREATE_IMPORT_TEMPLATE = 451,
  PERMISSION_PREFERENCE_UPDATE_IMPORT_TEMPLATE = 452,
  PERMISSION_PREFERENCE_DELETE_IMPORT_TEMPLATE = 453,
  PERMISSION_PREFERENCE_VIEW_PRINTING_TEMPLATE = 460,
  PERMISSION_PREFERENCE_CREATE_PRINTING_TEMPLATE = 461,
  PERMISSION_PREFERENCE_UPDATE_PRINTING_TEMPLATE = 462,
  PERMISSION_PREFERENCE_DELETE_PRINTING_TEMPLATE = 463,
  PERMISSION_PREFERENCE_UPDATE_SOCIAL_SHOP_LAYOUT = 490,
  PERMISSION_PREFERENCE_UPDATE_EDUCATION_SHOP_LAYOUT = 491,
  PERMISSION_MES_VIEW_PRODUCE = 470,
  PERMISSION_MES_VIEW_PACK = 471,
  PERMISSION_MES_VIEW_SORTING = 472,
  PERMISSION_SORTING_VIEW_SORTING_INFO = 480,
  PERMISSION_SORTING_VIEW_SORTING_TASK = 481,
  PERMISSION_SORTING_EDIT_SORTING_TASK = 482,
  PERMISSION_SORTING_EXPORT_SORTING_PERFORMANCE = 483,
}
export enum GroupUser_Type {
  UNSPECIFIED = 0,
  GROUP_ADMIN = 8,
  NORMAL = 9,
}
export enum GroupUser_Status {
  STATUS_UNSPECIFIED = 0,
  STATUS_CAN_LOGIN_PURCHASE_APP = 256,
  STATUS_CAN_LOGIN_DRIVER_APP = 512,
}
export interface GroupUser_Attrs {
  visable_all_stations?: boolean
  visable_station_ids?: string[]
  permission_ids?: string[]
  bounded_customer_ids?: string[]
  distribution_contractor_ids?: string[]
  car_license_plate_number?: string
  id_number?: string
}
export interface GroupUser {
  group_user_id: string
  status?: string
  create_time?: string
  update_time?: string
  delete_time?: string
  group_id?: string
  account_id?: string
  distribution_contractor_id?: string
  name: string
  is_valid?: boolean
  type: GroupUser_Type
  attrs?: GroupUser_Attrs
  username?: string
  phone?: string
  email?: string
}
export interface ReqCreateGroupUser {
  // group_user_id: string
  status?: string
  // create_time?: string
  // update_time?: string
  // delete_time?: string
  // group_id?: string
  account_id?: string
  distribution_contractor_id?: string
  name: string
  is_valid?: boolean
  type: GroupUser_Type
  attrs?: GroupUser_Attrs
  username?: string
  phone?: string
  email?: string
}
export enum Role_Type {
  UNSPECIFIED = 0,
  NORMAL = 1,
  BUILT_IN_DRIVER = 2,
  BUILT_IN_PURCHASER = 3,
  BUILT_IN_ADMIN = 4,
}
export interface Role {
  role_id: string
  create_time?: string
  update_time?: string
  delete_time?: string
  group_id?: string
  name: string
  description?: string
  type: Role_Type
}
export interface ReqCreateRole {
  // role_id: string
  // create_time?: string
  // update_time?: string
  // delete_time?: string
  // group_id?: string
  name: string
  description?: string
  type: Role_Type
}
export interface GroupPermissionRelation {
  relation_id: string
  create_time?: string
  update_time?: string
  delete_time?: string
  group_id?: string
  permission_id?: string
}
export interface RolePermissionRelation {
  relation_id: string
  create_time?: string
  update_time?: string
  delete_time?: string
  role_id?: string
  permission_id?: string
  group_id?: string
}
export interface GroupUserRoleRelation {
  relation_id: string
  create_time?: string
  update_time?: string
  delete_time?: string
  role_id?: string
  group_user_id?: string
  group_id?: string
}
export enum ServicePeriod_Type {
  TYPE_UNSPECIFIED = 0,
  TYPE_DEFAULT_MENU_PERIOD = 3,
  TYPE_CUSTOMIZED_MENU_PERIOD = 4,
}
export interface ServicePeriod {
  service_period_id: string
  revision?: string
  create_time?: string
  update_time?: string
  delete_time?: string
  group_id?: string
  menu_period_group_id?: string
  name: string
  description?: string
  type?: ServicePeriod_Type
  order_create_min_time: string
  order_create_max_time: string
  order_receive_min_time: string
  order_receive_max_time: string
  order_receive_min_date: string
  order_receive_max_date: string
}
export interface ReqCreateServicePeriod {
  // service_period_id: string
  revision?: string
  // create_time?: string
  // update_time?: string
  // delete_time?: string
  // group_id?: string
  menu_period_group_id?: string
  name: string
  description?: string
  type?: ServicePeriod_Type
  order_create_min_time: string
  order_create_max_time: string
  order_receive_min_time: string
  order_receive_max_time: string
  order_receive_min_date: string
  order_receive_max_date: string
}
export interface Country {
  country_id: string
  create_time?: string
  update_time?: string
  delete_time?: string
  iso_code: string
  currency_iso_code: string
}
export interface Province {
  province_id: string
  revision?: string
  create_time?: string
  update_time?: string
  delete_time?: string
  country_id: string
  gov_code?: string
  local_name: string
}
export interface City {
  city_id: string
  revision?: string
  create_time?: string
  update_time?: string
  delete_time?: string
  country_id: string
  province_id: string
  gov_code?: string
  local_name: string
}
export interface District {
  district_id: string
  revision?: string
  create_time?: string
  update_time?: string
  delete_time?: string
  country_id: string
  city_id: string
  gov_code?: string
  local_name: string
}
export interface Street {
  street_id: string
  revision?: string
  create_time?: string
  update_time?: string
  delete_time?: string
  country_id: string
  district_id: string
  gov_code?: string
  local_name: string
}
export interface Address {
  city_id?: string
  district_id?: string
  street_id?: string
  receiver?: string
  phone?: string
  address?: string
  poi_title?: string
  poi_address?: string
  geotag?: Geotag
  is_default?: boolean
}
export enum ChinaVatInvoice_InvoiceType {
  UNSPECIFIED = 0,
  VAT_NORMAL = 1,
  VAT_SPECIAL = 2,
}
export enum ChinaVatInvoice_TaxpayerType {
  TAXPAYER_TYPE_UNSPECIFIED = 0,
  TAXPAYER_TYPE_NORMAL = 1,
  TAXPAYER_TYPE_SMALL = 2,
  TAXPAYER_TYPE_NONE = 3,
}
export interface ChinaVatInvoice {
  company_name?: string
  company_address?: string
  company_phone?: string
  invoice_type?: ChinaVatInvoice_InvoiceType
  taxpayer_identity_number?: string
  bank_name?: string
  bank_account?: string
  bank_card_owner_name?: string
  max_invoice_amount?: string
  business_license_number?: string
  financial_contact_name?: string
  financial_contact_phone?: string
  taxpayer_type?: ChinaVatInvoice_TaxpayerType
}
export interface Geotag {
  latitude?: string
  longitude?: string
}
export interface CustomerLabelRelation {
  relation_id: string
  create_time?: string
  update_time?: string
  delete_time?: string
  group_id?: string
  customer_id?: string
  customer_label_id?: string
}
export interface CustomerLabel {
  customer_label_id: string
  revision?: string
  create_time?: string
  update_time?: string
  delete_time?: string
  group_id?: string
  name: string
}
export interface ReqCreateCustomerLabel {
  // customer_label_id: string
  revision?: string
  // create_time?: string
  // update_time?: string
  // delete_time?: string
  // group_id?: string
  name: string
}
export enum Customer_Status {
  STATUS_UNSPECIFIED = 0,
  STATUS_IS_BILL_TARGET = 256,
  STATUS_IS_SHIP_TARGET = 512,
}
export interface Customer_Attrs {
  addresses?: Address[]
  parent_ids?: string[]
  children_number?: number
  menu_period_relation?: { [key: string]: string }
}
export enum Customer_Type {
  TYPE_UNSPECIFIED = 0,
  TYPE_VIRTUAL = 1,
  TYPE_VIRTUAL_STUDENT = 3,
  TYPE_VIRTUAL_SCHOOL_STAFF = 4,
  TYPE_SOCIAL = 2,
  TYPE_SCHOOL = 5,
  TYPE_CLASS = 6,
}
export enum Customer_SettlementMethod {
  SETTLEMENT_METHOD_UNSPECIFIED = 0,
  SETTLEMENT_METHOD_ADVANCE_PAYMENT = 1,
  SETTLEMENT_METHOD_DEFERRED_PAYMENT = 2,
}
export interface Customer_Settlement {
  weekly_or_monthly_credit_start_day?: number
  weekly_or_monthly_credit_settle_day?: number
  customized_credit_start_time?: number
  customized_credit_proied_days?: number
  customized_credit_deley_days?: number
  china_vat_invoice?: ChinaVatInvoice
}
export interface Customer {
  customer_id: string
  status?: string
  create_time?: string
  update_time?: string
  delete_time?: string
  group_id?: string
  parent_id?: string
  create_group_user_id?: string
  sales_group_user_id?: string
  bound_station_id?: string
  name: string
  remark?: string
  customized_code?: string
  type: Customer_Type
  is_reviewed?: boolean
  is_in_whitelist?: boolean
  is_frozen?: boolean
  level?: number
  child_number?: number
  settlement?: Customer_Settlement
  settlement_method?: Customer_SettlementMethod
  credit_type?: CreditType
  phone?: string
  gender?: common.Gender
  parent_phone?: string
  parent_name?: string
  city_id?: string
  district_id?: string
  street_id?: string
  attrs?: Customer_Attrs
}
export interface ReqCreateCustomer {
  // customer_id: string
  status?: string
  // create_time?: string
  // update_time?: string
  // delete_time?: string
  // group_id?: string
  parent_id?: string
  create_group_user_id?: string
  sales_group_user_id?: string
  bound_station_id?: string
  name: string
  remark?: string
  customized_code?: string
  type: Customer_Type
  is_reviewed?: boolean
  is_in_whitelist?: boolean
  is_frozen?: boolean
  level?: number
  child_number?: number
  settlement?: Customer_Settlement
  settlement_method?: Customer_SettlementMethod
  credit_type?: CreditType
  phone?: string
  gender?: common.Gender
  parent_phone?: string
  parent_name?: string
  city_id?: string
  district_id?: string
  street_id?: string
  attrs?: Customer_Attrs
}
export interface CustomerUser {
  customer_user_id: string
  create_time?: string
  update_time?: string
  delete_time?: string
  group_id?: string
  customer_id?: string
  account_id?: string
  name?: string
  username?: string
  phone?: string
  email?: string
}
export interface ReqCreateCustomerUser {
  // customer_user_id: string
  // create_time?: string
  // update_time?: string
  // delete_time?: string
  // group_id?: string
  customer_id?: string
  account_id?: string
  name?: string
  username?: string
  phone?: string
  email?: string
}
export interface CustomerServicePeriodRelation {
  relation_id: string
  create_time?: string
  update_time?: string
  delete_time?: string
  group_id?: string
  customer_id?: string
  service_period_id?: string
}
export enum InvitationCode_Type {
  TYPE_UNSPECIFIED = 0,
  TEMPORARY = 1,
  FIXED = 2,
  EDUCATION_FIXED = 3,
}
export interface InvitationCode {
  invitation_code_id: string
  create_time?: string
  update_time?: string
  delete_time?: string
  group_id?: string
  service_period_id?: string
  quotation_id?: string
  type: InvitationCode_Type
  used?: boolean
  root_customer_id?: string
  text?: string
}
export interface ReqCreateInvitationCode {
  // invitation_code_id: string
  // create_time?: string
  // update_time?: string
  // delete_time?: string
  // group_id?: string
  service_period_id?: string
  quotation_id?: string
  type: InvitationCode_Type
  used?: boolean
  root_customer_id?: string
  text?: string
}
export interface CreateAccountParams {
  username?: string
  password?: string
  phone?: string
  email?: string
  wechat_openid?: string
  wechat_unionid?: string
  wechat_nickname?: string
  wechat_avatar?: string
}
export interface UpdateAccountParams {
  password: string
}
export interface UpdateCurrentAccountParams {
  password: string
  origin_password: string
}
export interface InitializeRegionRequest {
  testing?: boolean
}
export interface InitializeRegionResponse {
}
export interface GetGroupRequest {
  group_id: string
}
export interface GetGroupResponse {
  group: Group
}
export interface ListGroupRequest {
  group_ids?: string[]
  paging: common.PagingParams
}
export interface ListGroupResponse {
  groups: Group[]
  paging: common.PagingResult
}
export interface CreateGroupRequest {
  group: ReqCreateGroup
}
export interface CreateGroupResponse {
  group: Group
}
export interface UpdateGroupRequest {
  group: Group
}
export interface UpdateGroupResponse {
  group: Group
}
export interface UpdateGroupExtraInfoRequest {
  group_extra_info: GroupExtraInfo
}
export interface UpdateGroupExtraInfoResponse {
  group_extra_info: GroupExtraInfo
}
export interface GetGroupExtraInfoRequest {
  group_id?: string
}
export interface GetGroupExtraInfoResponse {
  group_extra_info: GroupExtraInfo
}
export interface ListGroupTimeZoneRequest {
}
export interface ListGroupTimeZoneResponse {
  time_zones?: { [key: string]: string }
}
export interface ListPermissionGroupRequest {
  permissions: Permission[]
}
export interface ListPermissionGroupResponse {
  permissions?: { [key: string]: common.Uint64Set }
}
export interface GetPermissionGroupRequest {
  permission?: Permission
}
export interface GetPermissionGroupResponse {
  group_ids?: string[]
}
export interface UpdatePermissionGroupRequest {
  permission?: Permission
  group_ids?: string[]
}
export interface UpdatePermissionGroupResponse {
}
export interface GetGroupPermissionRequest {
  group_id?: string
}
export interface GetGroupPermissionResponse {
  permissions?: Permission[]
}
export interface UpdateGroupPermissionRequest {
  group_id?: string
  permissions?: Permission[]
}
export interface UpdateGroupPermissionResponse {
}
export interface GetStationRequest {
  station_id: string
}
export interface GetStationResponse {
  station: Station
}
export interface ListStationRequest {
  group_id?: string
  paging: common.PagingParams
}
export interface ListStationResponse {
  stations: Station[]
  paging: common.PagingResult
}
export interface CreateStationRequest {
  station?: ReqCreateStation
}
export interface CreateStationResponse {
  station: Station
}
export interface UpdateStationRequest {
  station?: Station
}
export interface UpdateStationResponse {
  station: Station
}
export interface UpdateStationAvailableCityRequest {
  station_id?: string
  available_city_ids?: string[]
}
export interface UpdateStationAvailableCityResponse {
  station?: Station
}
export interface GetSupplierRequest {
  supplier_id?: string
  customized_code?: string
}
export interface GetSupplierResponse {
  supplier: Supplier
}
export interface GetManySupplierRequest {
  supplier_ids?: string[]
}
export interface GetManySupplierResponse {
  suppliers: { [key: string]: Supplier }
}
export interface ListSupplierRequest {
  paging: common.PagingParams
  q?: string
}
export interface ListSupplierResponse {
  suppliers: Supplier[]
  paging: common.PagingResult
}
export interface CreateSupplierRequest {
  supplier?: ReqCreateSupplier
}
export interface CreateSupplierResponse {
  supplier: Supplier
}
export interface UpdateSupplierRequest {
  supplier?: Supplier
}
export interface UpdateSupplierResponse {
  supplier: Supplier
}
export interface DeleteSupplierRequest {
  supplier_id?: string
}
export interface DeleteSupplierResponse {
  supplier: Supplier
}
export interface CleanSupplierRequest {
}
export interface CleanSupplierResponse {
}
export interface BatchExportSupplierRequest {
  q?: string
}
export interface BatchExportSupplierResponse {
  task_id?: string
}
export interface GetGroupUserRequest {
  group_user_id?: string
  account_id?: string
  username?: string
  need_distribution_contractor?: boolean
  need_roles?: boolean
  need_permissions?: boolean
}
export interface GetGroupUserResponse {
  group_user: GroupUser
  distribution_contractor?: DistributionContractor
  roles?: { [key: string]: Role }
  permissions?: Permission[]
}
export interface GetManyGroupUserRequest {
  group_user_ids?: string[]
}
export interface GetManyGroupUserResponse {
  group_users: { [key: string]: GroupUser }
}
export interface ListGroupUserRequest {
  group_user_ids?: string[]
  account_ids?: string[]
  role_types?: Role_Type[]
  q?: string
  is_valid?: common.Filters_Bool
  need_roles?: boolean
  need_distribution_contractor?: boolean
  paging: common.PagingParams
}
export interface ListGroupUserResponse {
  group_users: GroupUser[]
  role_relation?: { [key: string]: common.Uint64Set }
  roles?: { [key: string]: Role }
  distribution_contractors?: { [key: string]: DistributionContractor }
  paging: common.PagingResult
}
export interface CreateGroupUserRequest {
  group_user: ReqCreateGroupUser
  account: CreateAccountParams
  bind_to_driver_role?: boolean
  bind_to_purchaser_role?: boolean
  role_ids?: string[]
}
export interface CreateGroupUserResponse {
  group_user: GroupUser
}
export interface UpdateGroupUserRequest {
  group_user: GroupUser
  account?: UpdateAccountParams
}
export interface UpdateGroupUserPasswordRequest {
  account: UpdateCurrentAccountParams
}
export interface UpdateGroupUserPasswordResponse {
}
export interface UpdateGroupUserResponse {
  group_user: GroupUser
}
export interface DeleteGroupUserRequest {
  group_user_id: string
}
export interface DeleteGroupUserResponse {
  group_user: GroupUser
}
export interface UpdateGroupUserRoleRequest {
  group_user_id?: string
  role_ids?: string[]
}
export interface UpdateGroupUserRoleResponse {
}
export interface GetRoleRequest {
  role_id: string
  need_permissions?: boolean
  need_group_users?: boolean
}
export interface GetRoleResponse {
  role: Role
  permissions?: Permission[]
  group_users?: GroupUser[]
}
export interface ListRoleRequest {
  station_ids?: string[]
  role_ids?: string[]
  q?: string
  types?: Role_Type[]
  need_group_users?: boolean
  need_permissions?: boolean
  paging: common.PagingParams
}
export interface ListRoleResponse_PermissionSet {
  permissions?: Permission[]
}
export interface ListRoleResponse {
  roles: Role[]
  group_user_relation?: { [key: string]: common.Uint64Set }
  group_users?: { [key: string]: GroupUser }
  permissions?: { [key: string]: ListRoleResponse_PermissionSet }
  paging: common.PagingResult
}
export interface CreateRoleRequest {
  role: ReqCreateRole
}
export interface CreateRoleResponse {
  role: Role
}
export interface UpdateRoleRequest {
  role: Role
}
export interface UpdateRoleResponse {
  role: Role
}
export interface DeleteRoleRequest {
  role_id: string
}
export interface DeleteRoleResponse {
  role: Role
}
export interface UpdateRolePermissionRequest {
  role_id?: string
  permissions?: Permission[]
}
export interface UpdateRolePermissionResponse {
}
export interface UpdateRoleGroupUserRequest {
  role_id?: string
  group_user_ids?: string[]
}
export interface UpdateRoleGroupUserResponse {
}
export interface GetServicePeriodRequest {
  service_period_id: string
  revision?: string
}
export interface GetServicePeriodResponse {
  service_period: ServicePeriod
}
export interface GetManyServicePeriodRequest {
  service_period_ids?: string[]
}
export interface GetManyServicePeriodResponse {
  service_periods: { [key: string]: ServicePeriod }
}
export interface ListServicePeriodRequest {
  group_id?: string
  station_id?: string
  service_period_ids?: string[]
  service_period_type: ServicePeriod_Type
  paging: common.PagingParams
}
export interface ListServicePeriodResponse {
  service_periods: ServicePeriod[]
  paging: common.PagingResult
}
export interface CreateServicePeriodRequest {
  service_period: ReqCreateServicePeriod
}
export interface CreateServicePeriodResponse {
  service_period: ServicePeriod
}
export interface UpdateServicePeriodRequest {
  service_period: ServicePeriod
}
export interface UpdateServicePeriodResponse {
  service_period: ServicePeriod
}
export interface DeleteServicePeriodRequest {
  service_period_id: string
}
export interface DeleteServicePeriodResponse {
  service_period: ServicePeriod
}
export interface CleanServicePeriodRequest {
}
export interface CleanServicePeriodResponse {
}
export interface GetCustomerRequest {
  customer_id?: string
  customer_customized_code?: string
  need_quotations?: boolean
  need_customer_label?: boolean
  need_service_period?: boolean
  need_menus?: boolean
}
export interface GetCustomerResponse {
  customer: Customer
  quotation_relations?: QuotationCustomerRelation[]
  menu_relations?: MenuCustomerRelation[]
  customer_labels?: { [key: string]: CustomerLabel }
  service_periods?: { [key: string]: ServicePeriod }
}
export interface GetManyCustomerRequest {
  customer_ids?: string[]
}
export interface GetManyCustomerResponse {
  customers: { [key: string]: Customer }
}
export interface ListCustomerRequest {
  paging: common.PagingParams
  need_quotations?: boolean
  need_group_users?: boolean
  need_service_periods?: boolean
  need_deleted?: boolean
  need_customer_label?: boolean
  need_child_customers?: boolean
  need_menus?: boolean
  is_frozen?: common.Filters_Bool
  is_in_whitelist?: common.Filters_Bool
  credit_types?: CreditType[]
  quotation_ids?: string[]
  service_period_ids?: string[]
  create_group_user_ids?: string[]
  sales_group_user_ids?: string[]
  customer_label_ids?: string[]
  q?: string
  parent_ids?: string[]
  level?: number
  is_bill_target?: common.Filters_Bool
  is_ship_target?: common.Filters_Bool
  city_ids?: string[]
  district_ids?: string[]
  street_ids?: string[]
  customer_ids?: string[]
  type?: Customer_Type
}
export interface ListCustomerResponse {
  customers: Customer[]
  quotation_relations?: QuotationCustomerRelation[]
  menu_relations?: MenuCustomerRelation[]
  service_period_relation?: { [key: string]: common.Uint64Set }
  service_periods?: { [key: string]: ServicePeriod }
  customer_label_relation?: { [key: string]: common.Uint64Set }
  customer_labels?: { [key: string]: CustomerLabel }
  group_users?: { [key: string]: GroupUser }
  child_customer_relation?: { [key: string]: common.Uint64Set }
  child_customers: { [key: string]: Customer }
  paging: common.PagingResult
}
export interface CreateCustomerRequest {
  customer: ReqCreateCustomer
  invitation_code_id?: string
  quotation_ids?: string[]
  service_period_ids?: string[]
  customer_label_ids?: string[]
}
export interface CreateCustomerResponse {
  customer: Customer
}
export interface UpdateCustomerRequest {
  customer: Customer
  quotation_ids?: string[]
  service_period_ids?: string[]
  customer_label_ids?: string[]
}
export interface UpdateCustomerResponse {
  customer: Customer
}
export interface DeleteCustomerRequest {
  customer_id: string
}
export interface DeleteCustomerResponse {
  customer: Customer
}
export interface CleanCustomerRequest {
}
export interface CleanCustomerResponse {
}
export interface BatchExportCustomerRequest {
  customer_type: Customer_Type
}
export interface BatchExportCustomerResponse {
  task_id?: string
}
export interface BatchImportCustomerRequest {
  customer_type: Customer_Type
  file_url: string
}
export interface BatchImportCustomerResponse {
  task_id?: string
}
export interface GetBatchImportCustomerTemplateRequest {
  customer_type: Customer_Type
}
export interface GetBatchImportCustomerTemplateResponse {
  file_url: string
}
export interface CountNewCustomerRequest {
  begin_datetime: string
  end_datetime: string
}
export interface CountNewCustomerResponse_Range {
  begin_datetime?: string
  end_datetime?: string
  count?: number
}
export interface CountNewCustomerResponse {
  ranges?: CountNewCustomerResponse_Range[]
}
export interface QuotationCustomerRelation {
  station_id?: string
  customer_id?: string
  quotation_id?: string
}
export interface MenuCustomerRelation {
  station_id?: string
  customer_id?: string
  menu_id?: string
  customer_type?: Customer_Type
}
export interface CreateCustomerTreeRequest {
  root_customer_request: CreateCustomerRequest
  child_customer_request?: CreateCustomerRequest
  root_customer_user_request?: CreateCustomerUserRequest
}
export interface CreateCustomerTreeResponse {
  root_customer_response?: CreateCustomerResponse
  child_customer_response?: CreateCustomerResponse
  root_customer_user_response?: CreateCustomerUserResponse
}
export interface GetCustomerUserRequest {
  customer_user_id?: string
  account_id?: string
  need_account?: boolean
}
export interface GetCustomerUserResponse {
  customer_user: CustomerUser
  account?: account.Account
}
export interface InternalGetCustomerUserRequest {
  account_id: string
  group_id: string
}
export interface InternalGetCustomerUserResponse {
  customer_user?: CustomerUser
}
export interface GetManyCustomerUserRequest {
  customer_user_ids?: string[]
}
export interface GetManyCustomerUserResponse {
  customer_users: { [key: string]: CustomerUser }
}
export interface ListCustomerUserRequest {
  group_id: string
  customer_ids?: string[]
  account_ids?: string[]
  customer_user_ids?: string[]
  q?: string
  paging: common.PagingParams
}
export interface ListCustomerUserResponse {
  customer_users: CustomerUser[]
  paging: common.PagingResult
}
export interface CreateCustomerUserRequest {
  customer_user: ReqCreateCustomerUser
  account?: CreateAccountParams
}
export interface CreateCustomerUserResponse {
  customer_user: CustomerUser
}
export interface UpdateCustomerUserRequest {
  customer_user: CustomerUser
  account?: UpdateAccountParams
}
export interface UpdateCustomerUserResponse {
  customer_user: CustomerUser
}
export interface UpdateCustomerUserPasswordRequest {
  account: UpdateCurrentAccountParams
}
export interface UpdateCustomerUserPasswordResponse {
}
export interface DeleteCustomerUserRequest {
  customer_user_id: string
}
export interface DeleteCustomerUserResponse {
  customer_user: CustomerUser
}
export interface CleanCustomerUserRequest {
}
export interface CleanCustomerUserResponse {
}
export interface GetCustomerLabelRequest {
  customer_label_id?: string
}
export interface GetCustomerLabelResponse {
  customer_label: CustomerLabel
}
export interface ListCustomerLabelRequest {
  q?: string
  paging: common.PagingParams
}
export interface ListCustomerLabelResponse {
  customer_labels: CustomerLabel[]
  paging: common.PagingResult
}
export interface CreateCustomerLabelRequest {
  customer_label?: ReqCreateCustomerLabel
}
export interface CreateCustomerLabelResponse {
  customer_label: CustomerLabel
}
export interface UpdateCustomerLabelRequest {
  customer_label?: CustomerLabel
}
export interface UpdateCustomerLabelResponse {
  customer_label: CustomerLabel
}
export interface DeleteCustomerLabelRequest {
  customer_label_id?: string
}
export interface DeleteCustomerLabelResponse {
  customer_label: CustomerLabel
}
export interface GetDistributionContractorRequest {
  distribution_contractor_id: string
  need_group_user_ids?: boolean
}
export interface GetDistributionContractorResponse {
  distribution_contractor: DistributionContractor
  group_user_ids?: string[]
}
export interface ListDistributionContractorRequest {
  group_id?: string
  need_group_users?: boolean
  q?: string
  paging: common.PagingParams
}
export interface ListDistributionContractorResponse {
  distribution_contractors: DistributionContractor[]
  group_user_ids?: { [key: string]: common.Uint64Set }
  group_users?: { [key: string]: GroupUser }
  paging: common.PagingResult
}
export interface CreateDistributionContractorRequest {
  distribution_contractor: ReqCreateDistributionContractor
}
export interface CreateDistributionContractorResponse {
  distribution_contractor: DistributionContractor
}
export interface UpdateDistributionContractorRequest {
  distribution_contractor?: DistributionContractor
  group_user_ids?: string[]
}
export interface UpdateDistributionContractorResponse {
  distribution_contractor: DistributionContractor
}
export interface DeleteDistributionContractorRequest {
  distribution_contractor_id: string
}
export interface DeleteDistributionContractorResponse {
  distribution_contractor: DistributionContractor
}
export interface ListCountryRequest {
  country_ids?: string[]
}
export interface ListCountryResponse {
  countries: Country[]
}
export interface ListProvinceRequest {
  country_id?: string
  province_ids?: string[]
}
export interface ListProvinceResponse {
  provinces: Province[]
}
export interface ListCityRequest {
  province_ids?: string[]
  city_ids?: string[]
}
export interface ListCityResponse {
  cities: City[]
}
export interface ListDistrictRequest {
  city_ids?: string[]
  district_ids?: string[]
}
export interface ListDistrictResponse {
  districts: District[]
}
export interface ListStreetRequest {
  district_ids?: string[]
  street_ids?: string[]
}
export interface ListStreetResponse {
  streets: Street[]
}
export interface ListInvitationCodeRequest {
  paging: common.PagingParams
  invitation_code_types?: InvitationCode_Type[]
}
export interface ListInvitationCodeResponse {
  invitation_codes: InvitationCode[]
  paging: common.PagingResult
}
export interface GetInvitationCodeRequest {
  invitation_code_id?: string
  text?: string
}
export interface GetInvitationCodeResponse {
  invitation_code: InvitationCode
}
export interface CreateInvitationCodeRequest {
  invitation_code: ReqCreateInvitationCode
}
export interface CreateInvitationCodeResponse {
  invitation_code: InvitationCode
}
export interface DeleteInvitationCodeRequest {
  invitation_code_id: string
}
export interface DeleteInvitationCodeResponse {
  invitation_code: InvitationCode
}
export interface CleanInvitationCodeRequest {
}
export interface CleanInvitationCodeResponse {
}
export interface GetUserInfoRequest {
}
export interface GetUserInfoResponse_UserInfo {
  client_id?: string
  account_id?: string
  group_id?: string
  station_id?: string
  group?: Group
  group_user?: GroupUser
  customer?: Customer
  customer_user?: CustomerUser
}
export interface GetUserInfoResponse {
  user_info: GetUserInfoResponse_UserInfo
}
