// gm_api/src/idgenerator/pc.tsx
import React, { FC } from 'react'
import { Select, SelectProps } from '@gm-pc/react'
import {
  SelectList,
  SelectEnum,
  MoreSelectList,
  MapId,
} from '../util/pc_util'
import type {
  MapProps,
  SelectListProps,
  SelectEnumProps,
  MoreSelectListProps,
  MapIdProps,
} from '../util/pc_util'
import {
  list_SerialConfig_Type,
  map_SerialConfig_Type,
  list_SerialConfig_SegmentationType,
  map_SerialConfig_SegmentationType,
} from './index'

export const Select_SerialConfig_Type: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_SerialConfig_Type} />
}

export const Map_SerialConfig_Type: FC<MapProps> = ({ value }) => {
  return <>{map_SerialConfig_Type[value]}</>
}

export const Select_SerialConfig_SegmentationType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_SerialConfig_SegmentationType} />
}

export const Map_SerialConfig_SegmentationType: FC<MapProps> = ({ value }) => {
  return <>{map_SerialConfig_SegmentationType[value]}</>
}

export type {
  MapProps,
  SelectListProps,
  MoreSelectListProps,
}
