// gm_api/src/idgenerator/types.ts
import * as common from '../common'
export enum SerialConfig_Type {
  TYPE_UNSPECIFIED = 0,
  TYPE_ORDER_ORDER = 100,
  TYPE_ORDER_ESHOP = 101,
  TYPE_STOCK_SHEET_PURCHASE_IN = 201,
  TYPE_STOCK_SHEET_PRODUCT_IN = 202,
  TYPE_STOCK_SHEET_MATERIAL_IN = 203,
  TYPE_STOCK_SHEET_REFUND_IN = 204,
  TYPE_STOCK_SHEET_TRANSFER_IN = 205,
  TYPE_STOCK_SHEET_OTHER_IN = 206,
  TYPE_STOCK_SHEET_SALE_OUT = 207,
  TYPE_STOCK_SHEET_MATERIAL_OUT = 208,
  TYPE_STOCK_SHEET_REFUND_OUT = 209,
  TYPE_STOCK_SHEET_OTHER_OUT = 210,
  TYPE_STOCK_SHEET_INVENTORY = 211,
  TYPE_STOCK_SHEET_TURNOVER_LOAN = 212,
  TYPE_STOCK_SHEET_TURNOVER_REVERT = 213,
  TYPE_ADJUST_SHEET = 299,
  TYPE_PURCHASE_TASK = 300,
  TYPE_PURCHASE_SHEET = 301,
  TYPE_PRODUCTION_TASK_PACK = 320,
  TYPE_PRODUCTION_TASK_PRODUCT = 321,
  TYPE_PRODUCTION_TASK_SHEET = 322,
  TYPE_PRODUCTION_TASK_WEIGHT = 323,
  TYPE_FINANCE_SETTLE = 400,
  TYPE_FINANCE_AMOUNT_DISCOUNT = 401,
  TYPE_FINANCE_TRANSACTION_FLOW = 402,
  TYPE_FINANCE_ACCOUNT_BALANCE = 403,
  TYPE_FINANCE_CUSTOMER_SETTLE = 404,
  TYPE_FINANCE_SUPPLIER_SETTLE = 405,
  TYPE_AFTER_SALE_ORDER = 500,
}
export enum SerialConfig_SegmentationType {
  SEGMENTATION_TYPE_UNSPECIFIED = 0,
  SEGMENTATION_TYPE_YYYYMMDD = 1,
  SEGMENTATION_TYPE_YYMMDD = 2,
}
export interface SerialConfig {
  scope?: string
  serial_type?: SerialConfig_Type
  segmentation_type?: SerialConfig_SegmentationType
  timezone?: string
  prefix?: string
  sequence_width?: number
  sequence_min?: number
  sequence_max?: number
}
export interface SerialSegmentation {
  scope?: string
  serial_type?: SerialConfig_Type
  segmentation?: number
  sequence?: number
}
export interface GenerateIdRequest {
  model_type?: common.Model_Type
}
export interface GenerateIdResponse {
  id?: string
}
export interface GenerateIdsRequest {
  model_type?: common.Model_Type
  count?: number
}
export interface GenerateIdsResponse {
  ids?: string[]
}
export interface GenerateSnRequest {
  scope?: string
  serial_type?: SerialConfig_Type
  timestamp?: string
}
export interface GenerateSnResponse {
  sn?: string
}
export interface GenerateSnsRequest {
  scope?: string
  serial_type?: SerialConfig_Type
  timestamp?: string
  count?: number
}
export interface GenerateSnsResponse {
  sns?: string[]
}
export interface GenerateIdSnRequest {
  id_request?: GenerateIdRequest
  sn_request?: GenerateSnRequest
}
export interface GenerateIdSnResponse {
  id_response?: GenerateIdResponse
  sn_response?: GenerateSnResponse
}
export interface UpdateSerialConfigRequest {
  config?: SerialConfig
}
export interface UpdateSerialConfigResponse {
}
