// gm_api/src/orderlogic/data.ts
import { getLocale } from './lng'

interface Enum_List_Item {
  text: string
  value: any
}

interface Enum_Map {
  [key: string]: string
}
export const list_Status_Code: Enum_List_Item[] = [
  { value: 20200001, text: getLocale('Status_Code__CODE_NO_SSU_SYSTEMKEY') },
  { value: 20200002, text: getLocale('Status_Code__CODE_NO_CUSTOMER_SYSTEMKEY') },
  { value: 20200003, text: getLocale('Status_Code__CODE_NO_QUANTITY_SYSTEMKEY') },
  { value: 20200004, text: getLocale('Status_Code__CODE_NO_ORDER_TO_PRODUCTION') },
  { value: 20200005, text: getLocale('Status_Code__CODE_TASK_CONCURRENCY_LIMIT') },
]
export const map_Status_Code: Enum_Map = {
  20200001: getLocale('Status_Code__CODE_NO_SSU_SYSTEMKEY'),
  20200002: getLocale('Status_Code__CODE_NO_CUSTOMER_SYSTEMKEY'),
  20200003: getLocale('Status_Code__CODE_NO_QUANTITY_SYSTEMKEY'),
  20200004: getLocale('Status_Code__CODE_NO_ORDER_TO_PRODUCTION'),
  20200005: getLocale('Status_Code__CODE_TASK_CONCURRENCY_LIMIT'),
}
export const list_OrderImportTemplete_Type: Enum_List_Item[] = [
  { value: 1, text: getLocale('OrderImportTemplete_Type__TYPE_SYSTEM') },
  { value: 2, text: getLocale('OrderImportTemplete_Type__TYPE_CUSTOMIZE') },
]
export const map_OrderImportTemplete_Type: Enum_Map = {
  1: getLocale('OrderImportTemplete_Type__TYPE_SYSTEM'),
  2: getLocale('OrderImportTemplete_Type__TYPE_CUSTOMIZE'),
}
export const list_OrderImportTemplete_SystemKey: Enum_List_Item[] = [
  { value: 1, text: getLocale('OrderImportTemplete_SystemKey__SYSTEMKEY_SSU_NAME') },
  { value: 2, text: getLocale('OrderImportTemplete_SystemKey__SYSTEMKEY_SSU_CUSTOMIZE_CODE') },
  { value: 3, text: getLocale('OrderImportTemplete_SystemKey__SYSTEMKEY_SSU_REMARK') },
  { value: 4, text: getLocale('OrderImportTemplete_SystemKey__SYSTEMKEY_UNITPRICE') },
  { value: 5, text: getLocale('OrderImportTemplete_SystemKey__SYSTEMKEY_CUSTOMER_CUSTOMIZE_CODE') },
  { value: 6, text: getLocale('OrderImportTemplete_SystemKey__SYSTEMKEY_CUSTOMERNAME') },
  { value: 7, text: getLocale('OrderImportTemplete_SystemKey__SYSTEMKEY_QUANTITY') },
  { value: 8, text: getLocale('OrderImportTemplete_SystemKey__SYSTEMKEY_ORDERNO') },
]
export const map_OrderImportTemplete_SystemKey: Enum_Map = {
  1: getLocale('OrderImportTemplete_SystemKey__SYSTEMKEY_SSU_NAME'),
  2: getLocale('OrderImportTemplete_SystemKey__SYSTEMKEY_SSU_CUSTOMIZE_CODE'),
  3: getLocale('OrderImportTemplete_SystemKey__SYSTEMKEY_SSU_REMARK'),
  4: getLocale('OrderImportTemplete_SystemKey__SYSTEMKEY_UNITPRICE'),
  5: getLocale('OrderImportTemplete_SystemKey__SYSTEMKEY_CUSTOMER_CUSTOMIZE_CODE'),
  6: getLocale('OrderImportTemplete_SystemKey__SYSTEMKEY_CUSTOMERNAME'),
  7: getLocale('OrderImportTemplete_SystemKey__SYSTEMKEY_QUANTITY'),
  8: getLocale('OrderImportTemplete_SystemKey__SYSTEMKEY_ORDERNO'),
}
