// gm_api/src/orderlogic/methods.ts
import { Request } from '../util/request'
import {
  GetCartOrderRequest, GetCartOrderResponse,
  UpdateCartOrderRequest, UpdateCartOrderResponse,
  CreateBshopOrderRequest, CreateBshopOrderResponse,
  CreateOrderTempleteRequest, CreateOrderTempleteResponse,
  UpdateOrderTempleteRequest, UpdateOrderTempleteResponse,
  DeleteOrderTempleteRequest, DeleteOrderTempleteResponse,
  GetOrderTempleteRequest, GetOrderTempleteResponse,
  ListOrderTempleteRequest, ListOrderTempleteResponse,
  UploadOrderTempleteRequest, UploadOrderTempleteResponse,
  BatchUpdateOrderStateRequest, BatchUpdateOrderStateResponse,
  BatchCreateOrderRequest, BatchCreateOrderResponse,
  BatchUpdateOrderPriceRequest, BatchUpdateOrderPriceResponse,
  BatchSyncPriceToOrderRequest, BatchSyncPriceToOrderResponse,
  BatchUpdateOrderDriverRequest, BatchUpdateOrderDriverResponse,
  AutoUpdateOrderDriverRequest, AutoUpdateOrderDriverResponse,
  ListOrderWithRelationRequest, ListOrderWithRelationResponse,
  ListOrderDetailWithRelationRequest, ListOrderDetailWithRelationResponse,
  SyncPurchaseTaskFromOrderRequest, SyncPurchaseTaskFromOrderResponse,
  DispatchProductionTaskFromOrderRequest, DispatchProductionTaskFromOrderResponse,
  ExportProductionPurchaseTaskRequest, ExportProductionPurchaseTaskResponse,
  ExportOrderRequest, ExportOrderResponse,
  ExportSortingPerformanceRequest, ExportSortingPerformanceResponse,
  BatchSyncSaleOutStockSheetFromOrderRequest, BatchSyncSaleOutStockSheetFromOrderResponse,
  BatchScanReceiptRequest, BatchScanReceiptResponse,
} from './types'
export const GetCartOrder = (r?: GetCartOrderRequest, codes?: number[]) =>
  Request<GetCartOrderResponse>('/ceres/orderlogic/OrderLogicService/GetCartOrder')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetCartOrder.url = '/ceres/orderlogic/OrderLogicService/GetCartOrder'
export const UpdateCartOrder = (r?: UpdateCartOrderRequest, codes?: number[]) =>
  Request<UpdateCartOrderResponse>('/ceres/orderlogic/OrderLogicService/UpdateCartOrder')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateCartOrder.url = '/ceres/orderlogic/OrderLogicService/UpdateCartOrder'
export const CreateBshopOrder = (r?: CreateBshopOrderRequest, codes?: number[]) =>
  Request<CreateBshopOrderResponse>('/ceres/orderlogic/OrderLogicService/CreateBshopOrder')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateBshopOrder.url = '/ceres/orderlogic/OrderLogicService/CreateBshopOrder'
export const CreateOrderTemplete = (r?: CreateOrderTempleteRequest, codes?: number[]) =>
  Request<CreateOrderTempleteResponse>('/ceres/orderlogic/OrderLogicService/CreateOrderTemplete')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateOrderTemplete.url = '/ceres/orderlogic/OrderLogicService/CreateOrderTemplete'
export const UpdateOrderTemplete = (r?: UpdateOrderTempleteRequest, codes?: number[]) =>
  Request<UpdateOrderTempleteResponse>('/ceres/orderlogic/OrderLogicService/UpdateOrderTemplete')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateOrderTemplete.url = '/ceres/orderlogic/OrderLogicService/UpdateOrderTemplete'
export const DeleteOrderTemplete = (r?: DeleteOrderTempleteRequest, codes?: number[]) =>
  Request<DeleteOrderTempleteResponse>('/ceres/orderlogic/OrderLogicService/DeleteOrderTemplete')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteOrderTemplete.url = '/ceres/orderlogic/OrderLogicService/DeleteOrderTemplete'
export const GetOrderTemplete = (r?: GetOrderTempleteRequest, codes?: number[]) =>
  Request<GetOrderTempleteResponse>('/ceres/orderlogic/OrderLogicService/GetOrderTemplete')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetOrderTemplete.url = '/ceres/orderlogic/OrderLogicService/GetOrderTemplete'
export const ListOrderTemplete = (r?: ListOrderTempleteRequest, codes?: number[]) =>
  Request<ListOrderTempleteResponse>('/ceres/orderlogic/OrderLogicService/ListOrderTemplete')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListOrderTemplete.url = '/ceres/orderlogic/OrderLogicService/ListOrderTemplete'
export const UploadOrderTemplete = (r?: UploadOrderTempleteRequest, codes?: number[]) =>
  Request<UploadOrderTempleteResponse>('/ceres/orderlogic/OrderLogicService/UploadOrderTemplete')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UploadOrderTemplete.url = '/ceres/orderlogic/OrderLogicService/UploadOrderTemplete'
export const BatchUpdateOrderState = (r?: BatchUpdateOrderStateRequest, codes?: number[]) =>
  Request<BatchUpdateOrderStateResponse>('/ceres/orderlogic/OrderLogicService/BatchUpdateOrderState')
    .code([...(codes || [])])
    .data(r || {})
    .run()
BatchUpdateOrderState.url = '/ceres/orderlogic/OrderLogicService/BatchUpdateOrderState'
export const BatchCreateOrder = (r?: BatchCreateOrderRequest, codes?: number[]) =>
  Request<BatchCreateOrderResponse>('/ceres/orderlogic/OrderLogicService/BatchCreateOrder')
    .code([...(codes || [])])
    .data(r || {})
    .run()
BatchCreateOrder.url = '/ceres/orderlogic/OrderLogicService/BatchCreateOrder'
export const BatchUpdateOrderPrice = (r?: BatchUpdateOrderPriceRequest, codes?: number[]) =>
  Request<BatchUpdateOrderPriceResponse>('/ceres/orderlogic/OrderLogicService/BatchUpdateOrderPrice')
    .code([...(codes || [])])
    .data(r || {})
    .run()
BatchUpdateOrderPrice.url = '/ceres/orderlogic/OrderLogicService/BatchUpdateOrderPrice'
export const BatchSyncPriceToOrder = (r?: BatchSyncPriceToOrderRequest, codes?: number[]) =>
  Request<BatchSyncPriceToOrderResponse>('/ceres/orderlogic/OrderLogicService/BatchSyncPriceToOrder')
    .code([...(codes || [])])
    .data(r || {})
    .run()
BatchSyncPriceToOrder.url = '/ceres/orderlogic/OrderLogicService/BatchSyncPriceToOrder'
export const BatchUpdateOrderDriver = (r?: BatchUpdateOrderDriverRequest, codes?: number[]) =>
  Request<BatchUpdateOrderDriverResponse>('/ceres/orderlogic/OrderLogicService/BatchUpdateOrderDriver')
    .code([...(codes || [])])
    .data(r || {})
    .run()
BatchUpdateOrderDriver.url = '/ceres/orderlogic/OrderLogicService/BatchUpdateOrderDriver'
export const AutoUpdateOrderDriver = (r?: AutoUpdateOrderDriverRequest, codes?: number[]) =>
  Request<AutoUpdateOrderDriverResponse>('/ceres/orderlogic/OrderLogicService/AutoUpdateOrderDriver')
    .code([...(codes || [])])
    .data(r || {})
    .run()
AutoUpdateOrderDriver.url = '/ceres/orderlogic/OrderLogicService/AutoUpdateOrderDriver'
export const ListOrderWithRelation = (r?: ListOrderWithRelationRequest, codes?: number[]) =>
  Request<ListOrderWithRelationResponse>('/ceres/orderlogic/OrderLogicService/ListOrderWithRelation')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListOrderWithRelation.url = '/ceres/orderlogic/OrderLogicService/ListOrderWithRelation'
export const ListOrderDetailWithRelation = (r?: ListOrderDetailWithRelationRequest, codes?: number[]) =>
  Request<ListOrderDetailWithRelationResponse>('/ceres/orderlogic/OrderLogicService/ListOrderDetailWithRelation')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListOrderDetailWithRelation.url = '/ceres/orderlogic/OrderLogicService/ListOrderDetailWithRelation'
export const SyncPurchaseTaskFromOrder = (r?: SyncPurchaseTaskFromOrderRequest, codes?: number[]) =>
  Request<SyncPurchaseTaskFromOrderResponse>('/ceres/orderlogic/OrderLogicService/SyncPurchaseTaskFromOrder')
    .code([...(codes || [])])
    .data(r || {})
    .run()
SyncPurchaseTaskFromOrder.url = '/ceres/orderlogic/OrderLogicService/SyncPurchaseTaskFromOrder'
export const DispatchProductionTaskFromOrder = (r?: DispatchProductionTaskFromOrderRequest, codes?: number[]) =>
  Request<DispatchProductionTaskFromOrderResponse>('/ceres/orderlogic/OrderLogicService/DispatchProductionTaskFromOrder')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DispatchProductionTaskFromOrder.url = '/ceres/orderlogic/OrderLogicService/DispatchProductionTaskFromOrder'
export const ExportProductionPurchaseTask = (r?: ExportProductionPurchaseTaskRequest, codes?: number[]) =>
  Request<ExportProductionPurchaseTaskResponse>('/ceres/orderlogic/OrderLogicService/ExportProductionPurchaseTask')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ExportProductionPurchaseTask.url = '/ceres/orderlogic/OrderLogicService/ExportProductionPurchaseTask'
export const ExportOrder = (r?: ExportOrderRequest, codes?: number[]) =>
  Request<ExportOrderResponse>('/ceres/orderlogic/OrderLogicService/ExportOrder')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ExportOrder.url = '/ceres/orderlogic/OrderLogicService/ExportOrder'
export const ExportSortingPerformance = (r?: ExportSortingPerformanceRequest, codes?: number[]) =>
  Request<ExportSortingPerformanceResponse>('/ceres/orderlogic/OrderLogicService/ExportSortingPerformance')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ExportSortingPerformance.url = '/ceres/orderlogic/OrderLogicService/ExportSortingPerformance'
export const BatchSyncSaleOutStockSheetFromOrder = (r?: BatchSyncSaleOutStockSheetFromOrderRequest, codes?: number[]) =>
  Request<BatchSyncSaleOutStockSheetFromOrderResponse>('/ceres/orderlogic/OrderLogicService/BatchSyncSaleOutStockSheetFromOrder')
    .code([...(codes || [])])
    .data(r || {})
    .run()
BatchSyncSaleOutStockSheetFromOrder.url = '/ceres/orderlogic/OrderLogicService/BatchSyncSaleOutStockSheetFromOrder'
export const BatchScanReceipt = (r?: BatchScanReceiptRequest, codes?: number[]) =>
  Request<BatchScanReceiptResponse>('/ceres/orderlogic/OrderLogicService/BatchScanReceipt')
    .code([...(codes || [])])
    .data(r || {})
    .run()
BatchScanReceipt.url = '/ceres/orderlogic/OrderLogicService/BatchScanReceipt'
