// gm_api/src/orderlogic/pc.tsx
import React, { FC } from 'react'
import { Select, SelectProps } from '@gm-pc/react'
import {
  SelectList,
  SelectEnum,
  MoreSelectList,
  MapId,
} from '../util/pc_util'
import type {
  MapProps,
  SelectListProps,
  SelectEnumProps,
  MoreSelectListProps,
  MapIdProps,
} from '../util/pc_util'
import {
  list_Status_Code,
  map_Status_Code,
  list_OrderImportTemplete_Type,
  map_OrderImportTemplete_Type,
  list_OrderImportTemplete_SystemKey,
  map_OrderImportTemplete_SystemKey,
} from './index'

export const Select_Status_Code: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Status_Code} />
}

export const Map_Status_Code: FC<MapProps> = ({ value }) => {
  return <>{map_Status_Code[value]}</>
}

export const Select_OrderImportTemplete_Type: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_OrderImportTemplete_Type} />
}

export const Map_OrderImportTemplete_Type: FC<MapProps> = ({ value }) => {
  return <>{map_OrderImportTemplete_Type[value]}</>
}

export const Select_OrderImportTemplete_SystemKey: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_OrderImportTemplete_SystemKey} />
}

export const Map_OrderImportTemplete_SystemKey: FC<MapProps> = ({ value }) => {
  return <>{map_OrderImportTemplete_SystemKey[value]}</>
}

export type {
  MapProps,
  SelectListProps,
  MoreSelectListProps,
}
