// gm_api/src/initialization/types.ts
export interface InitializeSystemRequest {
  testing?: boolean
}
export interface InitializeSystemResponse {
}
export enum InitializeTestingGroupRequest_StopAfter {
  STOP_AFTER_UNSPECIFIED = 0,
  STOP_AFTER_SYSTEM = 1,
  STOP_AFTER_ENTERPRISE = 2,
  STOP_AFTER_MERCHANDISE = 3,
  STOP_AFTER_PRODUCTION = 4,
  STOP_AFTER_BOM = 5,
  STOP_AFTER_MENU = 6,
  STOP_AFTER_ORDER = 7,
  STOP_AFTER_LEAVE_APPLICATION = 8,
}
export interface InitializeTestingGroupRequest {
  more?: boolean
  eshop_order_size?: number
  stop_after?: InitializeTestingGroupRequest_StopAfter
}
export interface InitializeTestingGroupResponse {
  group_id?: string
  admin_group_user_username?: string
  admin_group_user_phone?: string
  admin_group_user_password?: string
  driver_group_user_username?: string
  driver_group_user_password?: string
  purchase1_group_user_username?: string
  purchase1_group_user_password?: string
  purchase2_group_user_username?: string
  purchase2_group_user_password?: string
  customer_user_username?: string
  customer_user_phone?: string
  customer_user_password?: string
}
export interface CleanGroupTicketRequest {
  group_id?: string
}
export interface CleanGroupTicketResponse {
  task_id?: string
}
export interface CleanGroupDataRequest {
  group_id?: string
}
export interface CleanGroupDataResponse {
  task_id?: string
}
