// gm_api/src/order/data.ts
import { getLocale } from './lng'

interface Enum_List_Item {
  text: string
  value: any
}

interface Enum_Map {
  [key: string]: string
}
export const list_Status_Code: Enum_List_Item[] = [
  { value: 20480001, text: getLocale('Status_Code__CODE_FAILURE_VERSION') },
  { value: 20480002, text: getLocale('Status_Code__CODE_CALCULATE_ERROR_UNIT') },
  { value: 20480003, text: getLocale('Status_Code__CODE_CLOSED_ORDER') },
  { value: 20480004, text: getLocale('Status_Code__CODE_DELETED_ORDER') },
  { value: 20480005, text: getLocale('Status_Code__CODE_ERROR_STATE_ORDER') },
  { value: 20480006, text: getLocale('Status_Code__CODE_ERROR_PAYSTATE_ORDER') },
  { value: 20480007, text: getLocale('Status_Code__CODE_ERROR_TIME_ZONE') },
  { value: 20480008, text: getLocale('Status_Code__CODE_ERROR_RECEIVE_TIME') },
  { value: 20480009, text: getLocale('Status_Code__CODE_NO_ORDER_DETAIL') },
  { value: 20480010, text: getLocale('Status_Code__CODE_NOT_REVIEWED_CUSTOMER') },
  { value: 20480011, text: getLocale('Status_Code__CODE_IS_FROZEN_CUSTOMER') },
  { value: 20480013, text: getLocale('Status_Code__CODE_ORDER_DUPLICATE_MENU_SSU') },
  { value: 20480014, text: getLocale('Status_Code__CODE_OVER_LIMIT_SKU_QUANTITY') },
  { value: 20480015, text: getLocale('Status_Code__CODE_ERROR_SSU_PRICE') },
  { value: 20480016, text: getLocale('Status_Code__CODE_NO_ORDER') },
  { value: 20480017, text: getLocale('Status_Code__CODE_CANNOT_ORDER_CUSTOMER') },
  { value: 20480018, text: getLocale('Status_Code__CODE_DOWNSSU') },
  { value: 20480019, text: getLocale('Status_Code__CODE_ERROR_INVENTORY_SSU') },
  { value: 20480020, text: getLocale('Status_Code__CODE_NOT_FOUND_UNIT') },
  { value: 20480021, text: getLocale('Status_Code__CODE_ERROR_SSU') },
  { value: 20480022, text: getLocale('Status_Code__CODE_LEAST_SSU') },
  { value: 20480023, text: getLocale('Status_Code__CODE_DEBT_CUSTOMER') },
  { value: 20480024, text: getLocale('Status_Code__CODE_NO_RELATION_CUSTOMER_AND_MENU') },
  { value: 20480025, text: getLocale('Status_Code__CODE_NO_RELATION_MENU_AND_SSU') },
  { value: 20480027, text: getLocale('Status_Code__CODE_NO_RELATION_CUSTOMER_AND_SERVICEPERIOD') },
  { value: 20480028, text: getLocale('Status_Code__CODE_NO_RELATION_CUSTOMER_AND_MENUPERIOD') },
  { value: 20480029, text: getLocale('Status_Code__CODE_NOT_STUDENT_SCHOOLSTAFF_TYPE') },
  { value: 20480030, text: getLocale('Status_Code__CODE_ERROR_CUSTOMER_TYPE') },
  { value: 20480031, text: getLocale('Status_Code__CODE_IS_PRODUCTED_ORDER') },
  { value: 20480032, text: getLocale('Status_Code__CODE_JOINED_SETTLE_SHEET') },
]
export const map_Status_Code: Enum_Map = {
  20480001: getLocale('Status_Code__CODE_FAILURE_VERSION'),
  20480002: getLocale('Status_Code__CODE_CALCULATE_ERROR_UNIT'),
  20480003: getLocale('Status_Code__CODE_CLOSED_ORDER'),
  20480004: getLocale('Status_Code__CODE_DELETED_ORDER'),
  20480005: getLocale('Status_Code__CODE_ERROR_STATE_ORDER'),
  20480006: getLocale('Status_Code__CODE_ERROR_PAYSTATE_ORDER'),
  20480007: getLocale('Status_Code__CODE_ERROR_TIME_ZONE'),
  20480008: getLocale('Status_Code__CODE_ERROR_RECEIVE_TIME'),
  20480009: getLocale('Status_Code__CODE_NO_ORDER_DETAIL'),
  20480010: getLocale('Status_Code__CODE_NOT_REVIEWED_CUSTOMER'),
  20480011: getLocale('Status_Code__CODE_IS_FROZEN_CUSTOMER'),
  20480013: getLocale('Status_Code__CODE_ORDER_DUPLICATE_MENU_SSU'),
  20480014: getLocale('Status_Code__CODE_OVER_LIMIT_SKU_QUANTITY'),
  20480015: getLocale('Status_Code__CODE_ERROR_SSU_PRICE'),
  20480016: getLocale('Status_Code__CODE_NO_ORDER'),
  20480017: getLocale('Status_Code__CODE_CANNOT_ORDER_CUSTOMER'),
  20480018: getLocale('Status_Code__CODE_DOWNSSU'),
  20480019: getLocale('Status_Code__CODE_ERROR_INVENTORY_SSU'),
  20480020: getLocale('Status_Code__CODE_NOT_FOUND_UNIT'),
  20480021: getLocale('Status_Code__CODE_ERROR_SSU'),
  20480022: getLocale('Status_Code__CODE_LEAST_SSU'),
  20480023: getLocale('Status_Code__CODE_DEBT_CUSTOMER'),
  20480024: getLocale('Status_Code__CODE_NO_RELATION_CUSTOMER_AND_MENU'),
  20480025: getLocale('Status_Code__CODE_NO_RELATION_MENU_AND_SSU'),
  20480027: getLocale('Status_Code__CODE_NO_RELATION_CUSTOMER_AND_SERVICEPERIOD'),
  20480028: getLocale('Status_Code__CODE_NO_RELATION_CUSTOMER_AND_MENUPERIOD'),
  20480029: getLocale('Status_Code__CODE_NOT_STUDENT_SCHOOLSTAFF_TYPE'),
  20480030: getLocale('Status_Code__CODE_ERROR_CUSTOMER_TYPE'),
  20480031: getLocale('Status_Code__CODE_IS_PRODUCTED_ORDER'),
  20480032: getLocale('Status_Code__CODE_JOINED_SETTLE_SHEET'),
}
export const list_Order_Type: Enum_List_Item[] = [
  { value: 1, text: getLocale('Order_Type__TYPE_DELICATESSEN') },
  { value: 2, text: getLocale('Order_Type__TYPE_PRODUCT_FROM_DELICATESSEN') },
  { value: 3, text: getLocale('Order_Type__TYPE_PRODUCT') },
  { value: 4, text: getLocale('Order_Type__TYPE_PRODUCT_FROM_STATION') },
]
export const map_Order_Type: Enum_Map = {
  1: getLocale('Order_Type__TYPE_DELICATESSEN'),
  2: getLocale('Order_Type__TYPE_PRODUCT_FROM_DELICATESSEN'),
  3: getLocale('Order_Type__TYPE_PRODUCT'),
  4: getLocale('Order_Type__TYPE_PRODUCT_FROM_STATION'),
}
export const list_Order_State: Enum_List_Item[] = [
  { value: 1, text: getLocale('Order_State__STATE_WAITING_SORT') },
  { value: 2, text: getLocale('Order_State__STATE_SORTING') },
  { value: 3, text: getLocale('Order_State__STATE_DELIVERYING') },
  { value: 4, text: getLocale('Order_State__STATE_RECEIVABLE') },
  { value: 5, text: getLocale('Order_State__STATE_NOT_PRODUCE') },
  { value: 6, text: getLocale('Order_State__STATE_PRODUCED') },
  { value: 1024, text: getLocale('Order_State__STATE_DELETE') },
]
export const map_Order_State: Enum_Map = {
  1: getLocale('Order_State__STATE_WAITING_SORT'),
  2: getLocale('Order_State__STATE_SORTING'),
  3: getLocale('Order_State__STATE_DELIVERYING'),
  4: getLocale('Order_State__STATE_RECEIVABLE'),
  5: getLocale('Order_State__STATE_NOT_PRODUCE'),
  6: getLocale('Order_State__STATE_PRODUCED'),
  1024: getLocale('Order_State__STATE_DELETE'),
}
export const list_Order_PayState: Enum_List_Item[] = [
  { value: 1, text: getLocale('Order_PayState__PAYSTATE_NOTPAY') },
  { value: 2, text: getLocale('Order_PayState__PAYSTATE_PARTPAY') },
  { value: 3, text: getLocale('Order_PayState__PAYSTATE_PAID') },
  { value: 4, text: getLocale('Order_PayState__PAYSTATE_REFUND') },
  { value: 5, text: getLocale('Order_PayState__PAYSTATE_CLOSED') },
]
export const map_Order_PayState: Enum_Map = {
  1: getLocale('Order_PayState__PAYSTATE_NOTPAY'),
  2: getLocale('Order_PayState__PAYSTATE_PARTPAY'),
  3: getLocale('Order_PayState__PAYSTATE_PAID'),
  4: getLocale('Order_PayState__PAYSTATE_REFUND'),
  5: getLocale('Order_PayState__PAYSTATE_CLOSED'),
}
export const list_Order_UserType: Enum_List_Item[] = [
  { value: 1, text: getLocale('Order_UserType__USERTYPE_CUSTOMER_USER') },
  { value: 2, text: getLocale('Order_UserType__USERTYPE_GROUP_USER') },
]
export const map_Order_UserType: Enum_Map = {
  1: getLocale('Order_UserType__USERTYPE_CUSTOMER_USER'),
  2: getLocale('Order_UserType__USERTYPE_GROUP_USER'),
}
export const list_Order_OrderOp: Enum_List_Item[] = [
  { value: 1, text: getLocale('Order_OrderOp__ORDER_NORMAL') },
  { value: 2, text: getLocale('Order_OrderOp__ORDER_AMEND') },
]
export const map_Order_OrderOp: Enum_Map = {
  1: getLocale('Order_OrderOp__ORDER_NORMAL'),
  2: getLocale('Order_OrderOp__ORDER_AMEND'),
}
export const list_Order_Status: Enum_List_Item[] = [
  { value: 256, text: getLocale('Order_Status__STATUS_IS_FREEZE') },
  { value: 512, text: getLocale('Order_Status__STATUS_IS_PRINT') },
  { value: 1024, text: getLocale('Order_Status__STATUS_IS_CREATE_STOCK_SHEET') },
  { value: 2048, text: getLocale('Order_Status__STATUS_SORTING_IS_FINISH') },
  { value: 4096, text: getLocale('Order_Status__STATUS_IS_SIGNED') },
]
export const map_Order_Status: Enum_Map = {
  256: getLocale('Order_Status__STATUS_IS_FREEZE'),
  512: getLocale('Order_Status__STATUS_IS_PRINT'),
  1024: getLocale('Order_Status__STATUS_IS_CREATE_STOCK_SHEET'),
  2048: getLocale('Order_Status__STATUS_SORTING_IS_FINISH'),
  4096: getLocale('Order_Status__STATUS_IS_SIGNED'),
}
export const list_SortingStatus: Enum_List_Item[] = [
  { value: 1, text: getLocale('SortingStatus__SORTINGSTATUS_UNWEIGHT') },
  { value: 2, text: getLocale('SortingStatus__SORTINGSTATUS_WEIGHTED') },
  { value: 3, text: getLocale('SortingStatus__SORTINGSTATUS_OUTOFSTOCK') },
]
export const map_SortingStatus: Enum_Map = {
  1: getLocale('SortingStatus__SORTINGSTATUS_UNWEIGHT'),
  2: getLocale('SortingStatus__SORTINGSTATUS_WEIGHTED'),
  3: getLocale('SortingStatus__SORTINGSTATUS_OUTOFSTOCK'),
}
export const list_OrderDetail_Sync: Enum_List_Item[] = [
  { value: 1, text: getLocale('OrderDetail_Sync__SYNC_PURCHASE') },
  { value: 2, text: getLocale('OrderDetail_Sync__SYNC_PRODUCTION') },
]
export const map_OrderDetail_Sync: Enum_Map = {
  1: getLocale('OrderDetail_Sync__SYNC_PURCHASE'),
  2: getLocale('OrderDetail_Sync__SYNC_PRODUCTION'),
}
export const list_OrderDetail_Status: Enum_List_Item[] = [
  { value: 256, text: getLocale('OrderDetail_Status__STATUS_IS_CREATE_PURCHASE_TASK') },
  { value: 512, text: getLocale('OrderDetail_Status__STATUS_IS_CREATE_PRODUCTION_TASK') },
  { value: 1024, text: getLocale('OrderDetail_Status__STATUS_SORTING_TASK_IS_PRINT') },
  { value: 2048, text: getLocale('OrderDetail_Status__STATUS_SORTING_TASK_PART_SORT') },
]
export const map_OrderDetail_Status: Enum_Map = {
  256: getLocale('OrderDetail_Status__STATUS_IS_CREATE_PURCHASE_TASK'),
  512: getLocale('OrderDetail_Status__STATUS_IS_CREATE_PRODUCTION_TASK'),
  1024: getLocale('OrderDetail_Status__STATUS_SORTING_TASK_IS_PRINT'),
  2048: getLocale('OrderDetail_Status__STATUS_SORTING_TASK_PART_SORT'),
}
export const list_OrderDetail_State: Enum_List_Item[] = [
  { value: 1024, text: getLocale('OrderDetail_State__STATE_DELETE') },
]
export const map_OrderDetail_State: Enum_Map = {
  1024: getLocale('OrderDetail_State__STATE_DELETE'),
}
export const list_ListOrderRequest_PagingField: Enum_List_Item[] = [
  { value: 1, text: getLocale('ListOrderRequest_PagingField__STATE') },
  { value: 2, text: getLocale('ListOrderRequest_PagingField__ORDER_TIME') },
  { value: 3, text: getLocale('ListOrderRequest_PagingField__RECEIVE_TIME') },
]
export const map_ListOrderRequest_PagingField: Enum_Map = {
  1: getLocale('ListOrderRequest_PagingField__STATE'),
  2: getLocale('ListOrderRequest_PagingField__ORDER_TIME'),
  3: getLocale('ListOrderRequest_PagingField__RECEIVE_TIME'),
}
export const list_ListEshopOrderRequest_PagingField: Enum_List_Item[] = [
  { value: 1, text: getLocale('ListEshopOrderRequest_PagingField__RECEIVE_TIME') },
  { value: 2, text: getLocale('ListEshopOrderRequest_PagingField__ORDER_TIME') },
]
export const map_ListEshopOrderRequest_PagingField: Enum_Map = {
  1: getLocale('ListEshopOrderRequest_PagingField__RECEIVE_TIME'),
  2: getLocale('ListEshopOrderRequest_PagingField__ORDER_TIME'),
}
export const list_IdType: Enum_List_Item[] = [
  { value: 1, text: getLocale('IdType__CUSTOMER_ID') },
  { value: 2, text: getLocale('IdType__QUOTATION_ID') },
  { value: 3, text: getLocale('IdType__SKU_ID') },
  { value: 4, text: getLocale('IdType__GROUP_USER_ID') },
  { value: 5, text: getLocale('IdType__CUSTOMER_USER_ID') },
  { value: 6, text: getLocale('IdType__DRIVER_ID') },
  { value: 7, text: getLocale('IdType__BOM_SNAPSHOT_ID') },
  { value: 8, text: getLocale('IdType__SALES_GROUP_USER_ID') },
  { value: 9, text: getLocale('IdType__SERVICE_PERIOD_ID') },
]
export const map_IdType: Enum_Map = {
  1: getLocale('IdType__CUSTOMER_ID'),
  2: getLocale('IdType__QUOTATION_ID'),
  3: getLocale('IdType__SKU_ID'),
  4: getLocale('IdType__GROUP_USER_ID'),
  5: getLocale('IdType__CUSTOMER_USER_ID'),
  6: getLocale('IdType__DRIVER_ID'),
  7: getLocale('IdType__BOM_SNAPSHOT_ID'),
  8: getLocale('IdType__SALES_GROUP_USER_ID'),
  9: getLocale('IdType__SERVICE_PERIOD_ID'),
}
