// gm_api/src/order/methods.ts
import { Request } from '../util/request'
import {
  GetOrderRequest, GetOrderResponse,
  GetMergeOrderRequest, GetMergeOrderResponse,
  CreateOrderRequest, CreateOrderResponse,
  BatchCreateOrderRequest, BatchCreateOrderResponse,
  UpdateOrderDetailRequest, UpdateOrderDetailResponse,
  UpdateOrderDetailByOrderDetailRequest, UpdateOrderDetailByOrderDetailResponse,
  DeleteOrderDetailRequest, DeleteOrderDetailResponse,
  UpdateOrderDetailPartFieldRequest, UpdateOrderDetailPartFieldResponse,
  UpdateOrderPartFieldRequest, UpdateOrderPartFieldResponse,
  UpdateOrderRequest, UpdateOrderResponse,
  DeleteOrderRequest, DeleteOrderResponse,
  ListOrderRequest, ListOrderResponse,
  GetManyOrderRequest, GetManyOrderResponse,
  ListOrderDetailRequest, ListOrderDetailResponse,
  ListOrderDetailGroupBySsuRequest, ListOrderDetailGroupBySsuResponse,
  ListOrderAfterSaleRequest, ListOrderAfterSaleResponse,
  GetOrderDetailRequest, GetOrderDetailResponse,
  UpdateOrderAfterSaleRequest, UpdateOrderAfterSaleResponse,
  UpdateOrderSortingDetailRequest, UpdateOrderSortingDetailResponse,
  GetOrderSkuCacheKeyRequest, GetOrderSkuCacheKeyResponse,
  GetOrderSkuCacheRequest, GetOrderSkuCacheResponse,
  CreateOrderSkuCacheRequest, CreateOrderSkuCacheResponse,
  DeleteOrderSkuCacheRequest, DeleteOrderSkuCacheResponse,
  GetEshopOrderRequest, GetEshopOrderResponse,
  CreateEshopOrderRequest, CreateEshopOrderResponse,
  BatchCreateEshopOrderRequest, BatchCreateEshopOrderResponse,
  UpdateEshopOrderRequest, UpdateEshopOrderResponse,
  UpdateEshopOrderDetailRequest, UpdateEshopOrderDetailResponse,
  ListEshopOrderRequest, ListEshopOrderResponse,
  CreateOrderFromEshopRequest, CreateOrderFromEshopResponse,
  CheckCustomerStatusRequest, CheckCustomerStatusResponse,
  SysSyncOrderRequest, SysSyncOrderResponse,
  SysSyncOrderDetailRequest, SysSyncOrderDetailResponse,
  UpdateCustomerLastUsedDriverRequest, UpdateCustomerLastUsedDriverResponse,
  GetManyCustomerLastUsedDriverRequest, GetManyCustomerLastUsedDriverResponse,
  UpdateOrderBySettleRequest, UpdateOrderBySettleResponse,
} from './types'
export const GetOrder = (r?: GetOrderRequest, codes?: number[]) =>
  Request<GetOrderResponse>('/ceres/order/OrderService/GetOrder')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetOrder.url = '/ceres/order/OrderService/GetOrder'
export const GetMergeOrder = (r?: GetMergeOrderRequest, codes?: number[]) =>
  Request<GetMergeOrderResponse>('/ceres/order/OrderService/GetMergeOrder')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetMergeOrder.url = '/ceres/order/OrderService/GetMergeOrder'
export const CreateOrder = (r?: CreateOrderRequest, codes?: number[]) =>
  Request<CreateOrderResponse>('/ceres/order/OrderService/CreateOrder')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateOrder.url = '/ceres/order/OrderService/CreateOrder'
export const BatchCreateOrder = (r?: BatchCreateOrderRequest, codes?: number[]) =>
  Request<BatchCreateOrderResponse>('/ceres/order/OrderService/BatchCreateOrder')
    .code([...(codes || [])])
    .data(r || {})
    .run()
BatchCreateOrder.url = '/ceres/order/OrderService/BatchCreateOrder'
export const UpdateOrderDetail = (r?: UpdateOrderDetailRequest, codes?: number[]) =>
  Request<UpdateOrderDetailResponse>('/ceres/order/OrderService/UpdateOrderDetail')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateOrderDetail.url = '/ceres/order/OrderService/UpdateOrderDetail'
export const UpdateOrderDetailByOrderDetail = (r?: UpdateOrderDetailByOrderDetailRequest, codes?: number[]) =>
  Request<UpdateOrderDetailByOrderDetailResponse>('/ceres/order/OrderService/UpdateOrderDetailByOrderDetail')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateOrderDetailByOrderDetail.url = '/ceres/order/OrderService/UpdateOrderDetailByOrderDetail'
export const DeleteOrderDetail = (r?: DeleteOrderDetailRequest, codes?: number[]) =>
  Request<DeleteOrderDetailResponse>('/ceres/order/OrderService/DeleteOrderDetail')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteOrderDetail.url = '/ceres/order/OrderService/DeleteOrderDetail'
export const UpdateOrderDetailPartField = (r?: UpdateOrderDetailPartFieldRequest, codes?: number[]) =>
  Request<UpdateOrderDetailPartFieldResponse>('/ceres/order/OrderService/UpdateOrderDetailPartField')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateOrderDetailPartField.url = '/ceres/order/OrderService/UpdateOrderDetailPartField'
export const UpdateOrderPartField = (r?: UpdateOrderPartFieldRequest, codes?: number[]) =>
  Request<UpdateOrderPartFieldResponse>('/ceres/order/OrderService/UpdateOrderPartField')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateOrderPartField.url = '/ceres/order/OrderService/UpdateOrderPartField'
export const UpdateOrder = (r?: UpdateOrderRequest, codes?: number[]) =>
  Request<UpdateOrderResponse>('/ceres/order/OrderService/UpdateOrder')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateOrder.url = '/ceres/order/OrderService/UpdateOrder'
export const DeleteOrder = (r?: DeleteOrderRequest, codes?: number[]) =>
  Request<DeleteOrderResponse>('/ceres/order/OrderService/DeleteOrder')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteOrder.url = '/ceres/order/OrderService/DeleteOrder'
export const ListOrder = (r?: ListOrderRequest, codes?: number[]) =>
  Request<ListOrderResponse>('/ceres/order/OrderService/ListOrder')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListOrder.url = '/ceres/order/OrderService/ListOrder'
export const GetManyOrder = (r?: GetManyOrderRequest, codes?: number[]) =>
  Request<GetManyOrderResponse>('/ceres/order/OrderService/GetManyOrder')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetManyOrder.url = '/ceres/order/OrderService/GetManyOrder'
export const ListOrderDetail = (r?: ListOrderDetailRequest, codes?: number[]) =>
  Request<ListOrderDetailResponse>('/ceres/order/OrderService/ListOrderDetail')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListOrderDetail.url = '/ceres/order/OrderService/ListOrderDetail'
export const ListOrderDetailGroupBySsu = (r?: ListOrderDetailGroupBySsuRequest, codes?: number[]) =>
  Request<ListOrderDetailGroupBySsuResponse>('/ceres/order/OrderService/ListOrderDetailGroupBySsu')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListOrderDetailGroupBySsu.url = '/ceres/order/OrderService/ListOrderDetailGroupBySsu'
export const ListOrderAfterSale = (r?: ListOrderAfterSaleRequest, codes?: number[]) =>
  Request<ListOrderAfterSaleResponse>('/ceres/order/OrderService/ListOrderAfterSale')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListOrderAfterSale.url = '/ceres/order/OrderService/ListOrderAfterSale'
export const GetOrderDetail = (r?: GetOrderDetailRequest, codes?: number[]) =>
  Request<GetOrderDetailResponse>('/ceres/order/OrderService/GetOrderDetail')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetOrderDetail.url = '/ceres/order/OrderService/GetOrderDetail'
export const UpdateOrderAfterSale = (r?: UpdateOrderAfterSaleRequest, codes?: number[]) =>
  Request<UpdateOrderAfterSaleResponse>('/ceres/order/OrderService/UpdateOrderAfterSale')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateOrderAfterSale.url = '/ceres/order/OrderService/UpdateOrderAfterSale'
export const UpdateOrderSortingDetail = (r?: UpdateOrderSortingDetailRequest, codes?: number[]) =>
  Request<UpdateOrderSortingDetailResponse>('/ceres/order/OrderService/UpdateOrderSortingDetail')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateOrderSortingDetail.url = '/ceres/order/OrderService/UpdateOrderSortingDetail'
export const GetOrderSkuCacheKey = (r?: GetOrderSkuCacheKeyRequest, codes?: number[]) =>
  Request<GetOrderSkuCacheKeyResponse>('/ceres/order/OrderService/GetOrderSkuCacheKey')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetOrderSkuCacheKey.url = '/ceres/order/OrderService/GetOrderSkuCacheKey'
export const GetOrderSkuCache = (r?: GetOrderSkuCacheRequest, codes?: number[]) =>
  Request<GetOrderSkuCacheResponse>('/ceres/order/OrderService/GetOrderSkuCache')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetOrderSkuCache.url = '/ceres/order/OrderService/GetOrderSkuCache'
export const CreateOrderSkuCache = (r?: CreateOrderSkuCacheRequest, codes?: number[]) =>
  Request<CreateOrderSkuCacheResponse>('/ceres/order/OrderService/CreateOrderSkuCache')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateOrderSkuCache.url = '/ceres/order/OrderService/CreateOrderSkuCache'
export const DeleteOrderSkuCache = (r?: DeleteOrderSkuCacheRequest, codes?: number[]) =>
  Request<DeleteOrderSkuCacheResponse>('/ceres/order/OrderService/DeleteOrderSkuCache')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteOrderSkuCache.url = '/ceres/order/OrderService/DeleteOrderSkuCache'
export const GetEshopOrder = (r?: GetEshopOrderRequest, codes?: number[]) =>
  Request<GetEshopOrderResponse>('/ceres/order/OrderService/GetEshopOrder')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetEshopOrder.url = '/ceres/order/OrderService/GetEshopOrder'
export const CreateEshopOrder = (r?: CreateEshopOrderRequest, codes?: number[]) =>
  Request<CreateEshopOrderResponse>('/ceres/order/OrderService/CreateEshopOrder')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateEshopOrder.url = '/ceres/order/OrderService/CreateEshopOrder'
export const BatchCreateEshopOrder = (r?: BatchCreateEshopOrderRequest, codes?: number[]) =>
  Request<BatchCreateEshopOrderResponse>('/ceres/order/OrderService/BatchCreateEshopOrder')
    .code([...(codes || [])])
    .data(r || {})
    .run()
BatchCreateEshopOrder.url = '/ceres/order/OrderService/BatchCreateEshopOrder'
export const UpdateEshopOrder = (r?: UpdateEshopOrderRequest, codes?: number[]) =>
  Request<UpdateEshopOrderResponse>('/ceres/order/OrderService/UpdateEshopOrder')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateEshopOrder.url = '/ceres/order/OrderService/UpdateEshopOrder'
export const UpdateEshopOrderDetail = (r?: UpdateEshopOrderDetailRequest, codes?: number[]) =>
  Request<UpdateEshopOrderDetailResponse>('/ceres/order/OrderService/UpdateEshopOrderDetail')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateEshopOrderDetail.url = '/ceres/order/OrderService/UpdateEshopOrderDetail'
export const ListEshopOrder = (r?: ListEshopOrderRequest, codes?: number[]) =>
  Request<ListEshopOrderResponse>('/ceres/order/OrderService/ListEshopOrder')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListEshopOrder.url = '/ceres/order/OrderService/ListEshopOrder'
export const CreateOrderFromEshop = (r?: CreateOrderFromEshopRequest, codes?: number[]) =>
  Request<CreateOrderFromEshopResponse>('/ceres/order/OrderService/CreateOrderFromEshop')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateOrderFromEshop.url = '/ceres/order/OrderService/CreateOrderFromEshop'
export const CheckCustomerStatus = (r?: CheckCustomerStatusRequest, codes?: number[]) =>
  Request<CheckCustomerStatusResponse>('/ceres/order/OrderService/CheckCustomerStatus')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CheckCustomerStatus.url = '/ceres/order/OrderService/CheckCustomerStatus'
export const SysSyncOrder = (r?: SysSyncOrderRequest, codes?: number[]) =>
  Request<SysSyncOrderResponse>('/ceres/order/OrderService/SysSyncOrder')
    .code([...(codes || [])])
    .data(r || {})
    .run()
SysSyncOrder.url = '/ceres/order/OrderService/SysSyncOrder'
export const SysSyncOrderDetail = (r?: SysSyncOrderDetailRequest, codes?: number[]) =>
  Request<SysSyncOrderDetailResponse>('/ceres/order/OrderService/SysSyncOrderDetail')
    .code([...(codes || [])])
    .data(r || {})
    .run()
SysSyncOrderDetail.url = '/ceres/order/OrderService/SysSyncOrderDetail'
export const UpdateCustomerLastUsedDriver = (r?: UpdateCustomerLastUsedDriverRequest, codes?: number[]) =>
  Request<UpdateCustomerLastUsedDriverResponse>('/ceres/order/OrderService/UpdateCustomerLastUsedDriver')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateCustomerLastUsedDriver.url = '/ceres/order/OrderService/UpdateCustomerLastUsedDriver'
export const GetManyCustomerLastUsedDriver = (r?: GetManyCustomerLastUsedDriverRequest, codes?: number[]) =>
  Request<GetManyCustomerLastUsedDriverResponse>('/ceres/order/OrderService/GetManyCustomerLastUsedDriver')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetManyCustomerLastUsedDriver.url = '/ceres/order/OrderService/GetManyCustomerLastUsedDriver'
