// gm_api/src/order/pc.tsx
import React, { FC } from 'react'
import { Select, SelectProps } from '@gm-pc/react'
import {
  SelectList,
  SelectEnum,
  MoreSelectList,
  MapId,
} from '../util/pc_util'
import type {
  MapProps,
  SelectListProps,
  SelectEnumProps,
  MoreSelectListProps,
  MapIdProps,
} from '../util/pc_util'
import {
  list_Status_Code,
  map_Status_Code,
  list_Order_Type,
  map_Order_Type,
  list_Order_State,
  map_Order_State,
  list_Order_PayState,
  map_Order_PayState,
  list_Order_UserType,
  map_Order_UserType,
  list_Order_OrderOp,
  map_Order_OrderOp,
  list_Order_Status,
  map_Order_Status,
  list_SortingStatus,
  map_SortingStatus,
  list_OrderDetail_Sync,
  map_OrderDetail_Sync,
  list_OrderDetail_Status,
  map_OrderDetail_Status,
  list_OrderDetail_State,
  map_OrderDetail_State,
  list_ListOrderRequest_PagingField,
  map_ListOrderRequest_PagingField,
  list_ListEshopOrderRequest_PagingField,
  map_ListEshopOrderRequest_PagingField,
  list_IdType,
  map_IdType,
  Order,
  ListOrder,
  ListOrderResponse,
  OrderDetail,
  ListOrderDetail,
  ListOrderDetailResponse,
} from './index'

export const Select_Status_Code: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Status_Code} />
}

export const Map_Status_Code: FC<MapProps> = ({ value }) => {
  return <>{map_Status_Code[value]}</>
}

export const Select_Order_Type: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Order_Type} />
}

export const Map_Order_Type: FC<MapProps> = ({ value }) => {
  return <>{map_Order_Type[value]}</>
}

export const Select_Order_State: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Order_State} />
}

export const Map_Order_State: FC<MapProps> = ({ value }) => {
  return <>{map_Order_State[value]}</>
}

export const Select_Order_PayState: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Order_PayState} />
}

export const Map_Order_PayState: FC<MapProps> = ({ value }) => {
  return <>{map_Order_PayState[value]}</>
}

export const Select_Order_UserType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Order_UserType} />
}

export const Map_Order_UserType: FC<MapProps> = ({ value }) => {
  return <>{map_Order_UserType[value]}</>
}

export const Select_Order_OrderOp: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Order_OrderOp} />
}

export const Map_Order_OrderOp: FC<MapProps> = ({ value }) => {
  return <>{map_Order_OrderOp[value]}</>
}

export const Select_Order_Status: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Order_Status} />
}

export const Map_Order_Status: FC<MapProps> = ({ value }) => {
  return <>{map_Order_Status[value]}</>
}

export const Select_SortingStatus: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_SortingStatus} />
}

export const Map_SortingStatus: FC<MapProps> = ({ value }) => {
  return <>{map_SortingStatus[value]}</>
}

export const Select_OrderDetail_Sync: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_OrderDetail_Sync} />
}

export const Map_OrderDetail_Sync: FC<MapProps> = ({ value }) => {
  return <>{map_OrderDetail_Sync[value]}</>
}

export const Select_OrderDetail_Status: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_OrderDetail_Status} />
}

export const Map_OrderDetail_Status: FC<MapProps> = ({ value }) => {
  return <>{map_OrderDetail_Status[value]}</>
}

export const Select_OrderDetail_State: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_OrderDetail_State} />
}

export const Map_OrderDetail_State: FC<MapProps> = ({ value }) => {
  return <>{map_OrderDetail_State[value]}</>
}

export const Select_ListOrderRequest_PagingField: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_ListOrderRequest_PagingField} />
}

export const Map_ListOrderRequest_PagingField: FC<MapProps> = ({ value }) => {
  return <>{map_ListOrderRequest_PagingField[value]}</>
}

export const Select_ListEshopOrderRequest_PagingField: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_ListEshopOrderRequest_PagingField} />
}

export const Map_ListEshopOrderRequest_PagingField: FC<MapProps> = ({ value }) => {
  return <>{map_ListEshopOrderRequest_PagingField[value]}</>
}

export const Select_IdType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_IdType} />
}

export const Map_IdType: FC<MapProps> = ({ value }) => {
  return <>{map_IdType[value]}</>
}

export const Select_Order: FC<SelectListProps<
  Order,
  ListOrderResponse
>> = (props) => {
  return (
    <SelectList<Order, ListOrderResponse>
      {...props}
      messageName='Order'
      fetchList={ListOrder}
    />
  )
}

export const MoreSelect_Order: FC<MoreSelectListProps<
  Order,
  ListOrderResponse
>> = (props) => {
  return (
    <MoreSelectList<Order, ListOrderResponse>
      {...props}
      messageName='Order'
      fetchList={ListOrder}
    />
  )
}

export const Select_OrderDetail: FC<SelectListProps<
  OrderDetail,
  ListOrderDetailResponse
>> = (props) => {
  return (
    <SelectList<OrderDetail, ListOrderDetailResponse>
      {...props}
      messageName='OrderDetail'
      fetchList={ListOrderDetail}
    />
  )
}

export const MoreSelect_OrderDetail: FC<MoreSelectListProps<
  OrderDetail,
  ListOrderDetailResponse
>> = (props) => {
  return (
    <MoreSelectList<OrderDetail, ListOrderDetailResponse>
      {...props}
      messageName='OrderDetail'
      fetchList={ListOrderDetail}
    />
  )
}

export type {
  MapProps,
  SelectListProps,
  MoreSelectListProps,
}
