// gm_api/src/driverapp/types.ts
import * as common from '../common'
import * as order from '../order'
import * as enterprise from '../enterprise'
import * as account from '../account'
import * as dao from '../dao'
import * as idgenerator from '../idgenerator'
import * as merchandise from '../merchandise'
import * as asynctask from '../asynctask'
import * as oauth from '../oauth'
import * as inventory from '../inventory'
import * as preference from '../preference'
export interface GetTaskInfoRequest {
}
export interface GetTaskInfoResponse_MonthTaskInfo {
  delivery_task_count?: string
  delivery_customer_count?: string
  delivery_task_amount?: string
}
export interface GetTaskInfoResponse {
  to_delivery_task_count?: number
  today_recived_task_count?: number
  month_task_info?: GetTaskInfoResponse_MonthTaskInfo
}
export interface GetDriverInfoRequest {
}
export interface GetDriverInfoResponse_BaseInfo {
  roles?: enterprise.Role[]
  stations?: enterprise.Station[]
  logo?: common.Image
}
export interface GetDriverInfoResponse {
  driver?: enterprise.GroupUser
  distribution_contractor?: enterprise.DistributionContractor
  base_info?: GetDriverInfoResponse_BaseInfo
}
export enum ListTaskRequest_TaskStatus {
  UNSPECIFIED = 0,
  TODELIVERY = 1,
}
export interface ListTaskRequest {
  task_status?: ListTaskRequest_TaskStatus
  receive_begin_date: string
  receive_end_date: string
  paging: common.PagingParams
}
export interface ListTaskResponse {
  tasks?: order.Order[]
  customers?: { [key: string]: enterprise.Customer }
  quotations?: { [key: string]: merchandise.Quotation }
  paging: common.PagingResult
}
