// gm_api/src/wechat/methods.ts
import { Request } from '../util/request'
import {
  GetTokenRequest, GetTokenResponse,
  GetTicketRequest, GetTicketResponse,
  GetSnsAccessTokenRequest, GetSnsAccessTokenResponse,
  GetSnsUserInfoRequest, GetSnsUserInfoResponse,
  GetWxaSessionRequest, GetWxaSessionResponse,
  GetWxaUserInfoRequest, GetWxaUserInfoResponse,
  GetWxaPhoneNumberRequest, GetWxaPhoneNumberResponse,
  EncodeWxaSceneRequest, EncodeWxaSceneResponse,
  DecodeWxaSceneRequest, DecodeWxaSceneResponse,
  GetWxaCodeUnlimitRequest, GetWxaCodeUnlimitResponse,
  AddWxaFormIdRequest, AddWxaFormIdResponse,
  GetWxaCustomMessageResponderRequest, GetWxaCustomMessageResponderResponse,
  SetWxaCustomMessageResponderRequest, SetWxaCustomMessageResponderResponse,
  RecvWxaCustomMessageRequest, RecvWxaCustomMessageResponse,
  SendWxaCustomMessageRequest, SendWxaCustomMessageResponse,
} from './types'
export const GetToken = (r?: GetTokenRequest, codes?: number[]) =>
  Request<GetTokenResponse>('/ceres/wechat/WechatService/GetToken')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetToken.url = '/ceres/wechat/WechatService/GetToken'
export const GetTicket = (r?: GetTicketRequest, codes?: number[]) =>
  Request<GetTicketResponse>('/ceres/wechat/WechatService/GetTicket')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetTicket.url = '/ceres/wechat/WechatService/GetTicket'
export const GetSnsAccessToken = (r?: GetSnsAccessTokenRequest, codes?: number[]) =>
  Request<GetSnsAccessTokenResponse>('/ceres/wechat/WechatService/GetSnsAccessToken')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetSnsAccessToken.url = '/ceres/wechat/WechatService/GetSnsAccessToken'
export const GetSnsUserInfo = (r?: GetSnsUserInfoRequest, codes?: number[]) =>
  Request<GetSnsUserInfoResponse>('/ceres/wechat/WechatService/GetSnsUserInfo')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetSnsUserInfo.url = '/ceres/wechat/WechatService/GetSnsUserInfo'
export const GetWxaSession = (r?: GetWxaSessionRequest, codes?: number[]) =>
  Request<GetWxaSessionResponse>('/ceres/wechat/WechatService/GetWxaSession')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetWxaSession.url = '/ceres/wechat/WechatService/GetWxaSession'
export const GetWxaUserInfo = (r?: GetWxaUserInfoRequest, codes?: number[]) =>
  Request<GetWxaUserInfoResponse>('/ceres/wechat/WechatService/GetWxaUserInfo')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetWxaUserInfo.url = '/ceres/wechat/WechatService/GetWxaUserInfo'
export const GetWxaPhoneNumber = (r?: GetWxaPhoneNumberRequest, codes?: number[]) =>
  Request<GetWxaPhoneNumberResponse>('/ceres/wechat/WechatService/GetWxaPhoneNumber')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetWxaPhoneNumber.url = '/ceres/wechat/WechatService/GetWxaPhoneNumber'
export const EncodeWxaScene = (r?: EncodeWxaSceneRequest, codes?: number[]) =>
  Request<EncodeWxaSceneResponse>('/ceres/wechat/WechatService/EncodeWxaScene')
    .code([...(codes || [])])
    .data(r || {})
    .run()
EncodeWxaScene.url = '/ceres/wechat/WechatService/EncodeWxaScene'
export const DecodeWxaScene = (r?: DecodeWxaSceneRequest, codes?: number[]) =>
  Request<DecodeWxaSceneResponse>('/ceres/wechat/WechatService/DecodeWxaScene')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DecodeWxaScene.url = '/ceres/wechat/WechatService/DecodeWxaScene'
export const GetWxaCodeUnlimit = (r?: GetWxaCodeUnlimitRequest, codes?: number[]) =>
  Request<GetWxaCodeUnlimitResponse>('/ceres/wechat/WechatService/GetWxaCodeUnlimit')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetWxaCodeUnlimit.url = '/ceres/wechat/WechatService/GetWxaCodeUnlimit'
export const AddWxaFormId = (r?: AddWxaFormIdRequest, codes?: number[]) =>
  Request<AddWxaFormIdResponse>('/ceres/wechat/WechatService/AddWxaFormId')
    .code([...(codes || [])])
    .data(r || {})
    .run()
AddWxaFormId.url = '/ceres/wechat/WechatService/AddWxaFormId'
export const GetWxaCustomMessageResponder = (r?: GetWxaCustomMessageResponderRequest, codes?: number[]) =>
  Request<GetWxaCustomMessageResponderResponse>('/ceres/wechat/WechatService/GetWxaCustomMessageResponder')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetWxaCustomMessageResponder.url = '/ceres/wechat/WechatService/GetWxaCustomMessageResponder'
export const SetWxaCustomMessageResponder = (r?: SetWxaCustomMessageResponderRequest, codes?: number[]) =>
  Request<SetWxaCustomMessageResponderResponse>('/ceres/wechat/WechatService/SetWxaCustomMessageResponder')
    .code([...(codes || [])])
    .data(r || {})
    .run()
SetWxaCustomMessageResponder.url = '/ceres/wechat/WechatService/SetWxaCustomMessageResponder'
export const RecvWxaCustomMessage = (r?: RecvWxaCustomMessageRequest, codes?: number[]) =>
  Request<RecvWxaCustomMessageResponse>('/ceres/wechat/WechatService/RecvWxaCustomMessage')
    .code([...(codes || [])])
    .data(r || {})
    .run()
RecvWxaCustomMessage.url = '/ceres/wechat/WechatService/RecvWxaCustomMessage'
export const SendWxaCustomMessage = (r?: SendWxaCustomMessageRequest, codes?: number[]) =>
  Request<SendWxaCustomMessageResponse>('/ceres/wechat/WechatService/SendWxaCustomMessage')
    .code([...(codes || [])])
    .data(r || {})
    .run()
SendWxaCustomMessage.url = '/ceres/wechat/WechatService/SendWxaCustomMessage'
