// gm_api/src/wechat/pc.tsx
import React, { FC } from 'react'
import { Select, SelectProps } from '@gm-pc/react'
import {
  SelectList,
  SelectEnum,
  MoreSelectList,
  MapId,
} from '../util/pc_util'
import type {
  MapProps,
  SelectListProps,
  SelectEnumProps,
  MoreSelectListProps,
  MapIdProps,
} from '../util/pc_util'
import {
  list_Payment_Status,
  map_Payment_Status,
  list_SendWxaTemplateRequest_TemplateId,
  map_SendWxaTemplateRequest_TemplateId,
} from './index'

export const Select_Payment_Status: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Payment_Status} />
}

export const Map_Payment_Status: FC<MapProps> = ({ value }) => {
  return <>{map_Payment_Status[value]}</>
}

export const Select_SendWxaTemplateRequest_TemplateId: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_SendWxaTemplateRequest_TemplateId} />
}

export const Map_SendWxaTemplateRequest_TemplateId: FC<MapProps> = ({ value }) => {
  return <>{map_SendWxaTemplateRequest_TemplateId[value]}</>
}

export type {
  MapProps,
  SelectListProps,
  MoreSelectListProps,
}
