// gm_api/src/wechat/types.ts
import * as common from '../common'
export interface WXApp {
  app_id?: string
  name?: string
  secret?: string
  pay_key?: string
  pay_mch_id?: string
  pay_notify_url?: string
  pay_cert_pem?: string
  pay_key_pem?: string
  token?: WXToken
  ticket?: WXTicket
}
export interface WXHeader {
  errcode?: number
  errmsg?: string
}
export interface WXToken {
  access_token?: string
  expires_in?: string
}
export interface WXTicket {
  ticket?: string
  expires_in?: string
}
export interface SnsAccessToken {
  access_token?: string
  expires_in?: number
  refresh_token?: string
  openid?: string
  scope?: string
  unionid?: string
}
export interface SnsUserInfo {
  openid?: string
  nickname?: string
  sex?: number
  province?: string
  city?: string
  country?: string
  headimgurl?: string
  privilege?: string[]
  unionid?: string
  language?: string
}
export interface WxaSession {
  openid?: string
  session_key?: string
  unionid?: string
}
export interface WxaUserInfo {
  openid?: string
  nickname?: string
  gender?: number
  province?: string
  city?: string
  country?: string
  avatar_url?: string
  unionid?: string
  language?: string
}
export interface WxaPhoneNumber {
  phone_number?: string
  pure_phone_number?: string
  country_code?: string
}
export interface WxaFormId {
  form_id?: string
  status?: string
  revision?: string
  created_at?: string
  updated_at?: string
  app_id?: number
  account_id?: string
  openid?: string
}
export interface WxaTemplateMsg {
  msg_id?: string
  status?: string
  revision?: string
  created_at?: string
  updated_at?: string
  app_id?: number
  account_id?: string
  template_id?: string
  sender_id?: string
}
export interface WxaTemplate {
  id?: string
  app_id?: string
  template_id?: string
  keywords?: { [key: string]: string }
}
export interface WxaMedia {
  type?: string
  media_id?: string
  created_at?: number
}
export interface WxaCustomMessageRecvType {
  to_user_name?: string
  from_user_name?: string
  create_time?: string
  msg_id: string
  msg_type?: string
  content?: string
  pic_url?: string
  media_id?: string
  title?: string
  ppp_id?: string
  page_path?: string
  thumb_url?: string
  thumb_media_id?: string
  event?: string
  session_from?: string
}
export interface WxaCustomMessageSendType_Text {
  content?: string
}
export interface WxaCustomMessageSendType_Image {
  media_id?: string
}
export interface WxaCustomMessageSendType_Link {
  title?: string
  description?: string
  url?: string
  thumb_url?: string
}
export interface WxaCustomMessageSendType_Miniprogrampage {
  title?: string
  pagepath?: string
  thumb_media_id?: string
}
export interface WxaCustomMessageSendType {
  touser?: string
  msgtype?: string
  text?: WxaCustomMessageSendType_Text
  image?: WxaCustomMessageSendType_Image
  link?: WxaCustomMessageSendType_Link
  miniprogrampage?: WxaCustomMessageSendType_Miniprogrampage
}
export enum WxaCustomMessageResponder_Rule_Status {
  STATUS_UNSPECIFIED = 0,
  STATUS_DEFAULT = 1,
  STATUS_REGEX = 2,
}
export interface WxaCustomMessageResponder_Rule {
  status?: string
  content?: string
  response?: WxaCustomMessageSendType
}
export interface WxaCustomMessageResponder_Responder {
  rules?: WxaCustomMessageResponder_Rule[]
}
export interface WxaCustomMessageResponder {
  responder_id?: string
  status?: string
  revision?: string
  created_at?: string
  updated_at?: string
  app_id?: number
  responder?: WxaCustomMessageResponder_Responder
}
export enum Payment_Status {
  STATUS_UNSPECIFIED = 0,
  STATUS_REFUNDED = 256,
}
export interface Payment {
  payment_id?: string
  status?: string
  revision?: string
  created_at?: string
  updated_at?: string
  app_id?: number
  account_id?: string
  mch_id?: string
  body?: string
  nonce_str?: string
  out_trade_no?: string
  total_fee?: string
  spbill_create_ip?: string
  notify_url?: string
  trade_type?: string
  openid?: string
  sign_type?: string
  prepay_id?: string
  code_url?: string
  transaction_id?: string
  trade_state?: string
  refund_fee?: string
  out_refund_no?: string
  refund_desc?: string
  refunded_at?: string
}
export interface Transfer {
  transfer_id?: string
  status?: string
  revision?: string
  created_at?: string
  updated_at?: string
  app_id?: number
  account_id?: string
  reference_id?: string
  mch_id?: string
  description?: string
  partner_trade_no?: string
  amount?: string
  openid?: string
  spbill_create_ip?: string
  transfer_state?: string
  check_name?: string
  nonce_str?: string
  payment_no?: string
  payment_time?: string
}
export interface GetTokenRequest {
  app_id?: string
  refresh?: boolean
}
export interface GetTokenResponse {
  token?: WXToken
}
export interface GetTicketRequest {
  app_id?: string
  refresh?: boolean
}
export interface GetTicketResponse {
  ticket?: WXTicket
}
export interface GetSnsAccessTokenRequest {
  app_id?: string
  code?: string
}
export interface GetSnsAccessTokenResponse {
  access_token: SnsAccessToken
}
export interface GetSnsUserInfoRequest {
  app_id?: string
  access_token?: string
  openid?: string
  lang?: string
}
export interface GetSnsUserInfoResponse {
  user_info: SnsUserInfo
}
export interface GetWxaSessionRequest {
  app_id?: string
  code?: string
}
export interface GetWxaSessionResponse {
  session: WxaSession
}
export interface GetWxaUserInfoRequest {
  app_id?: string
  openid?: string
  encrypted_data?: string
  iv?: string
}
export interface GetWxaUserInfoResponse {
  user_info: WxaUserInfo
}
export interface GetWxaPhoneNumberRequest {
  app_id?: string
  openid?: string
  encrypted_data?: string
  iv?: string
}
export interface GetWxaPhoneNumberResponse {
  phone_number: WxaPhoneNumber
}
export interface EncodeWxaSceneRequest {
  values?: string[]
}
export interface EncodeWxaSceneResponse {
  scene?: string
}
export interface DecodeWxaSceneRequest {
  scene?: string
}
export interface DecodeWxaSceneResponse {
  values?: string[]
}
export interface GetWxaCodeUnlimitRequest {
  app_id?: string
  scene?: string
  page?: string
  width?: number
}
export interface GetWxaCodeUnlimitResponse {
  image?: bytes
}
export interface AddWxaFormIdRequest {
  form_id?: WxaFormId
}
export interface AddWxaFormIdResponse {
  form_id?: WxaFormId
}
export enum SendWxaTemplateRequest_TemplateId {
  TEMPLATE_ID_UNSPECIFIED = 0,
}
export interface SendWxaTemplateRequest {
  app_id?: number
  account_id?: string
  template_id?: string
  page?: string
  keywords?: { [key: string]: common.StringSet }
  emphasis_keyword?: string
  interval?: string
  sender_id?: string
}
export interface SendWxaTemplateResponse {
  msg?: WxaTemplateMsg
}
export interface GetWxaCustomMessageResponderRequest {
  responder_id?: string
}
export interface GetWxaCustomMessageResponderResponse {
  responder: WxaCustomMessageResponder
}
export interface SetWxaCustomMessageResponderRequest {
  responder?: WxaCustomMessageResponder
}
export interface SetWxaCustomMessageResponderResponse {
  responder?: WxaCustomMessageResponder
}
export interface RecvWxaCustomMessageRequest {
  message?: WxaCustomMessageRecvType
}
export interface RecvWxaCustomMessageResponse {
  message?: WxaCustomMessageSendType
}
export interface SendWxaCustomMessageRequest {
  message?: WxaCustomMessageSendType
}
export interface SendWxaCustomMessageResponse {
  message?: WxaCustomMessageSendType
}
export interface AddPaymentRequest {
  payment?: Payment
}
export interface AddPaymentResponse_Params {
  time_stamp?: string
  nonce_str?: string
  pkg?: string
  sign_type?: string
  pay_sign?: string
}
export interface AddPaymentResponse {
  payment?: Payment
  params?: AddPaymentResponse_Params
}
export interface SetPaymentRequest {
  payment?: Payment
}
export interface SetPaymentResponse {
  payment?: Payment
}
export interface GetPaymentRequest {
  payment_id?: string
}
export interface GetPaymentResponse {
  payment: Payment
}
export interface ListPaymentRequest {
  account_id?: string
}
export interface ListPaymentResponse {
  payments: Payment[]
  count?: number
  has_next?: boolean
}
export interface TransferRequest {
  transfer?: Transfer
}
export interface TransferResponse {
  transfer?: Transfer
}
export interface RefundRequest {
  payment_id?: string
  refund_fee?: string
  refund_desc_note_id?: string
  auto?: boolean
}
export interface RefundResponse {
  payment?: Payment
}
