// gm_api/src/analytics/types.ts
import * as common from '../common'
import * as enterprise from '../enterprise'
import * as account from '../account'
import * as dao from '../dao'
import * as idgenerator from '../idgenerator'
import * as order from '../order'
import * as merchandise from '../merchandise'
import * as asynctask from '../asynctask'
import * as oauth from '../oauth'
import * as inventory from '../inventory'
import * as preference from '../preference'
import * as production from '../production'
export interface DataSequence {
  model_type?: common.Model_Type
  partition_id?: string
  sequence?: string
  update_time?: string
}
export interface PatchRequest {
  model_type?: common.Model_Type
  created_ids?: string[]
  updated_ids?: string[]
  deleted_ids?: string[]
}
export interface PatchResponse {
}
export interface FieldExpr {
  name?: string
  expr?: string
}
export interface FieldValue {
  name?: string
  value?: string
}
export interface ModelValue {
  field_values?: FieldValue[]
  kv?: { [key: string]: string }
}
export interface QueryExpr {
  model_type?: common.Model_Type
  filter_string?: string
  filter_arguments?: FieldExpr[]
  query_fields?: FieldExpr[]
  group_by_fields?: FieldExpr[]
  order_by_fields?: FieldExpr[]
  offset?: number
  limit?: number
}
export interface QueryData {
  model_values?: ModelValue[]
  update_time?: string
  count?: number
  has_more?: boolean
}
export interface QueryRequest {
  expr?: QueryExpr
}
export interface QueryResponse {
  data?: QueryData
}
export interface SysQueryRequest {
  expr?: QueryExpr
}
export interface SysQueryResponse {
  data?: QueryData
}
export interface SysResetRequest {
  model_type?: common.Model_Type
}
export interface SysResetResponse {
}
export interface Example {
  example_id: string
  status?: string
  revision?: string
  create_time?: string
  update_time?: string
  delete_time?: string
  group_id?: string
  customer_id?: string
  price?: string
  quantity?: string
}
export interface Order {
  order_id: string
  status?: string
  revision?: string
  create_time?: string
  update_time?: string
  delete_time?: string
  group_id?: string
  station_id?: string
  creator_id?: string
  bill_customer_id?: string
  receive_customer_id?: string
  driver_id?: string
  service_period_id?: string
  quotation_id?: string
  customer_type?: enterprise.Customer_Type
  order_price?: string
  freight_price?: string
  aftersale_price?: string
  outstock_price?: string
  coupon_price?: string
  manual_adjust_price?: string
  sale_price?: string
  paid_amount?: string
  refund_amount?: string
  actual_amount?: string
  receive_time?: string
  outstock_time?: string
  order_time?: string
  received_time?: string
  time_zone?: string
  serial_no?: string
  state?: order.Order_State
  pay_state?: order.Order_PayState
  has_after_sale?: boolean
  order_type?: order.Order_Type
  creator_id_type?: order.Order_UserType
  app_type?: common.App_Type
  order_op?: order.Order_OrderOp
  sorting_total_count?: number
  sorting_weight_count?: number
  sorting_out_stock_count?: number
  sorting_unweight_count?: number
  city_id?: string
  district_id?: string
  street_id?: string
  customer_id_l1?: string
  customer_id_l2?: string
  customer_id_l3?: string
  customer_id_l4?: string
  customer_id_l5?: string
}
export interface OrderDetail {
  order_detail_id: string
  status?: string
  revision?: string
  create_time?: string
  update_time?: string
  delete_time?: string
  group_id?: string
  order_id?: string
  sku_id?: string
  bom_id?: string
  bom_revision?: string
  sorting_operater_id?: string
  sorting_mark_id?: string
  order_price?: string
  aftersale_price?: string
  outstock_price?: string
  manual_adjust_price?: string
  sale_price?: string
  ssu_id?: string
  sorting_status?: order.SortingStatus
  purchase_task_id?: string
  state?: order.OrderDetail_State
  unit_id?: string
  ssu_customize_code?: string
  order_input1_unit_id?: string
  order_input1_quantity?: string
  order_input1_price?: string
  order_calculate_unit_id?: string
  order_calculate_quantity?: string
  order_calculate_price?: string
  outstock_input1_unit_id?: string
  outstock_input1_quantity?: string
  outstock_input1_price?: string
  outstock_calculate_unit_id?: string
  outstock_calculate_quantity?: string
  outstock_calculate_price?: string
  sorting_input1_unit_id?: string
  sorting_input1_quantity?: string
  sorting_input1_price?: string
  sorting_calculate_unit_id?: string
  sorting_calculate_quantity?: string
  sorting_calculate_price?: string
}
export interface Task {
  task_id: string
  create_time?: string
  update_time?: string
  delete_time?: string
  group_id?: string
  station_id?: string
  sku_id: string
  sku_name?: string
  unit_id: string
  bom_id?: string
  bom_name?: string
  bom_revision?: string
  serial_no?: string
  batch?: string
  delivery_time?: string
  release_time?: string
  finish_time?: string
  state?: production.Task_State
  type?: production.Task_Type
  order_amount: string
  plan_amount: string
  output_amount: string
  base_unit_id?: string
  base_unit_output_amount: string
  source?: production.Task_Source
  target_customer_id?: string
  last_process_id?: string
  input_receive_total_price?: string
  input_return_total_price?: string
  input_actual_usage_total_price?: string
}
export interface TaskInput {
  task_input_id: string
  create_time?: string
  update_time?: string
  delete_time?: string
  group_id?: string
  station_id?: string
  task_id?: string
  sku_id?: string
  sku_name?: string
  unit_id?: string
  plan_usage_amount: string
  actual_usage_amount: string
  receive_amount: string
  return_amount: string
  towards?: production.TaskInput_Towards
  receive_total_price?: string
  return_total_price?: string
  actual_usage_total_price?: string
  bom_id?: string
}
export interface TaskOutput {
  task_output_id: string
  create_time?: string
  update_time?: string
  delete_time?: string
  group_id?: string
  station_id?: string
  sku_id: string
  sku_name?: string
  type?: production.OutputType
  state?: production.TaskOutput_State
  unit_id: string
  output_amount: string
  base_unit_id: string
  base_unit_output_amount: string
  task_id?: string
  task_serial_no?: string
  task_type?: production.Task_Type
  delivery_time?: string
  release_time?: string
  bom_id?: string
}
