// gm_api/src/operation/pc.tsx
import React, { FC } from 'react'
import { Select, SelectProps } from '@gm-pc/react'
import {
  SelectList,
  SelectEnum,
  MoreSelectList,
  MapId,
} from '../util/pc_util'
import type {
  MapProps,
  SelectListProps,
  SelectEnumProps,
  MoreSelectListProps,
  MapIdProps,
} from '../util/pc_util'
import {
  list_Operation_Type,
  map_Operation_Type,
  Operation,
  ListOperation,
  ListOperationResponse,
} from './index'

export const Select_Operation_Type: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Operation_Type} />
}

export const Map_Operation_Type: FC<MapProps> = ({ value }) => {
  return <>{map_Operation_Type[value]}</>
}

export const Select_Operation: FC<SelectListProps<
  Operation,
  ListOperationResponse
>> = (props) => {
  return (
    <SelectList<Operation, ListOperationResponse>
      {...props}
      messageName='Operation'
      fetchList={ListOperation}
    />
  )
}

export const MoreSelect_Operation: FC<MoreSelectListProps<
  Operation,
  ListOperationResponse
>> = (props) => {
  return (
    <MoreSelectList<Operation, ListOperationResponse>
      {...props}
      messageName='Operation'
      fetchList={ListOperation}
    />
  )
}

export type {
  MapProps,
  SelectListProps,
  MoreSelectListProps,
}
