// gm_api/src/operation/types.ts
import * as common from '../common'
export enum Operation_Type {
  TYPE_UNSPECIFIED = 0,
  TYPE_ORDER_ORDER_CREATE = 21000,
  TYPE_ORDER_ORDER_UPDATE = 21001,
  TYPE_ORDER_ORDER_DELETE = 21002,
}
export interface Operation_Operand_Field {
  name?: string
  value1?: string
  value2?: string
}
export interface Operation_Operand {
  fields?: Operation_Operand_Field[]
}
export interface Operation {
  operation_id: string
  status?: string
  revision?: string
  create_time?: string
  update_time?: string
  group_id?: string
  model_type?: common.Model_Type
  type?: Operation_Type
  operator_id?: string
  operator_name?: string
  operand_id?: string
  operand_name?: string
  operand?: Operation_Operand
}
export interface ReqCreateOperation {
  // operation_id: string
  status?: string
  revision?: string
  // create_time?: string
  // update_time?: string
  // group_id?: string
  model_type?: common.Model_Type
  type?: Operation_Type
  operator_id?: string
  operator_name?: string
  operand_id?: string
  operand_name?: string
  operand?: Operation_Operand
}
export interface CreateOperationRequest {
  operation?: ReqCreateOperation
}
export interface CreateOperationResponse {
  operation: Operation
}
export interface ListOperationRequest {
  group_id?: string
  model_type?: common.Model_Type
  type?: Operation_Type
  operator_id?: string
  operand_id?: string
}
export interface ListOperationResponse {
  operations: Operation[]
}
