// gm_api/src/production/data.ts
import { getLocale } from './lng'

interface Enum_List_Item {
  text: string
  value: any
}

interface Enum_Map {
  [key: string]: string
}
export const list_Status_Code: Enum_List_Item[] = [
  { value: 85700001, text: getLocale('Status_Code__BUILTIN_PROCESS_TEMPLATE') },
  { value: 85700002, text: getLocale('Status_Code__DUPLICATE_NAME') },
  { value: 85700003, text: getLocale('Status_Code__DUPLICATE_CODE') },
]
export const map_Status_Code: Enum_Map = {
  85700001: getLocale('Status_Code__BUILTIN_PROCESS_TEMPLATE'),
  85700002: getLocale('Status_Code__DUPLICATE_NAME'),
  85700003: getLocale('Status_Code__DUPLICATE_CODE'),
}
export const list_Attr_Type: Enum_List_Item[] = [
  { value: 1, text: getLocale('Attr_Type__TYPE_TEXT') },
  { value: 2, text: getLocale('Attr_Type__TYPE_OPTIONS') },
]
export const map_Attr_Type: Enum_Map = {
  1: getLocale('Attr_Type__TYPE_TEXT'),
  2: getLocale('Attr_Type__TYPE_OPTIONS'),
}
export const list_OpType: Enum_List_Item[] = [
  { value: 1, text: getLocale('OpType__OPTYPE_CREATE') },
  { value: 2, text: getLocale('OpType__OPTYPE_UPDATE') },
]
export const map_OpType: Enum_Map = {
  1: getLocale('OpType__OPTYPE_CREATE'),
  2: getLocale('OpType__OPTYPE_UPDATE'),
}
export const list_BomType: Enum_List_Item[] = [
  { value: 1, text: getLocale('BomType__BOM_TYPE_PRODUCE') },
  { value: 2, text: getLocale('BomType__BOM_TYPE_PACK') },
]
export const map_BomType: Enum_Map = {
  1: getLocale('BomType__BOM_TYPE_PRODUCE'),
  2: getLocale('BomType__BOM_TYPE_PACK'),
}
export const list_OutputType: Enum_List_Item[] = [
  { value: 1, text: getLocale('OutputType__OUTPUT_TYPE_MAIN') },
  { value: 2, text: getLocale('OutputType__OUTPUT_TYPE_EXTRA') },
]
export const map_OutputType: Enum_Map = {
  1: getLocale('OutputType__OUTPUT_TYPE_MAIN'),
  2: getLocale('OutputType__OUTPUT_TYPE_EXTRA'),
}
export const list_ProcessType_Status: Enum_List_Item[] = [
  { value: 256, text: getLocale('ProcessType_Status__STATUS_DEFAULT') },
]
export const map_ProcessType_Status: Enum_Map = {
  256: getLocale('ProcessType_Status__STATUS_DEFAULT'),
}
export const list_ProcessTemplate_Status: Enum_List_Item[] = [
  { value: 256, text: getLocale('ProcessTemplate_Status__STATUS_DEFAULT') },
]
export const map_ProcessTemplate_Status: Enum_Map = {
  256: getLocale('ProcessTemplate_Status__STATUS_DEFAULT'),
}
export const list_Bom_Status: Enum_List_Item[] = [
  { value: 256, text: getLocale('Bom_Status__STATUS_COMBINED') },
  { value: 512, text: getLocale('Bom_Status__STATUS_DEFAULT') },
]
export const map_Bom_Status: Enum_Map = {
  256: getLocale('Bom_Status__STATUS_COMBINED'),
  512: getLocale('Bom_Status__STATUS_DEFAULT'),
}
export const list_Task_State: Enum_List_Item[] = [
  { value: 1, text: getLocale('Task_State__STATE_PREPARE') },
  { value: 2, text: getLocale('Task_State__STATE_STARTED') },
  { value: 3, text: getLocale('Task_State__STATE_FINISHED') },
]
export const map_Task_State: Enum_Map = {
  1: getLocale('Task_State__STATE_PREPARE'),
  2: getLocale('Task_State__STATE_STARTED'),
  3: getLocale('Task_State__STATE_FINISHED'),
}
export const list_Task_Type: Enum_List_Item[] = [
  { value: 1, text: getLocale('Task_Type__TYPE_PRODUCE') },
  { value: 2, text: getLocale('Task_Type__TYPE_PACK') },
]
export const map_Task_Type: Enum_Map = {
  1: getLocale('Task_Type__TYPE_PRODUCE'),
  2: getLocale('Task_Type__TYPE_PACK'),
}
export const list_Task_Source: Enum_List_Item[] = [
  { value: 1, text: getLocale('Task_Source__SOURCE_ORDER') },
  { value: 2, text: getLocale('Task_Source__SOURCE_PLAN') },
  { value: 3, text: getLocale('Task_Source__SOURCE_PACK_TASK') },
]
export const map_Task_Source: Enum_Map = {
  1: getLocale('Task_Source__SOURCE_ORDER'),
  2: getLocale('Task_Source__SOURCE_PLAN'),
  3: getLocale('Task_Source__SOURCE_PACK_TASK'),
}
export const list_Task_Operation: Enum_List_Item[] = [
  { value: 1, text: getLocale('Task_Operation__OPERATION_RELEASE') },
  { value: 2, text: getLocale('Task_Operation__OPERATION_FINISH') },
  { value: 3, text: getLocale('Task_Operation__OPERATION_MERGE') },
  { value: 4, text: getLocale('Task_Operation__OPERATION_DELETE') },
  { value: 5, text: getLocale('Task_Operation__OPERATION_MODIFY_PLAN_AMOUNT') },
  { value: 6, text: getLocale('Task_Operation__OPERATION_MODIFY_OUTPUT_AMOUNT') },
  { value: 7, text: getLocale('Task_Operation__OPERATION_FINISH_AND_OUTPUT') },
]
export const map_Task_Operation: Enum_Map = {
  1: getLocale('Task_Operation__OPERATION_RELEASE'),
  2: getLocale('Task_Operation__OPERATION_FINISH'),
  3: getLocale('Task_Operation__OPERATION_MERGE'),
  4: getLocale('Task_Operation__OPERATION_DELETE'),
  5: getLocale('Task_Operation__OPERATION_MODIFY_PLAN_AMOUNT'),
  6: getLocale('Task_Operation__OPERATION_MODIFY_OUTPUT_AMOUNT'),
  7: getLocale('Task_Operation__OPERATION_FINISH_AND_OUTPUT'),
}
export const list_Task_TimeType: Enum_List_Item[] = [
  { value: 1, text: getLocale('Task_TimeType__TIME_TYPE_CREATE') },
  { value: 2, text: getLocale('Task_TimeType__TIME_TYPE_DELIVERY') },
  { value: 3, text: getLocale('Task_TimeType__TIME_TYPE_RELEASE') },
]
export const map_Task_TimeType: Enum_Map = {
  1: getLocale('Task_TimeType__TIME_TYPE_CREATE'),
  2: getLocale('Task_TimeType__TIME_TYPE_DELIVERY'),
  3: getLocale('Task_TimeType__TIME_TYPE_RELEASE'),
}
export const list_Task_OutputSource: Enum_List_Item[] = [
  { value: 1, text: getLocale('Task_OutputSource__OUTPUT_SOURCE_TYPE_IN') },
  { value: 2, text: getLocale('Task_OutputSource__OUTPUT_SOURCE_LAST_PROCESS') },
]
export const map_Task_OutputSource: Enum_Map = {
  1: getLocale('Task_OutputSource__OUTPUT_SOURCE_TYPE_IN'),
  2: getLocale('Task_OutputSource__OUTPUT_SOURCE_LAST_PROCESS'),
}
export const list_TaskInput_Towards: Enum_List_Item[] = [
  { value: 1, text: getLocale('TaskInput_Towards__TOWARDS_PRODUCTION') },
  { value: 2, text: getLocale('TaskInput_Towards__TOWARDS_PURCHASE') },
]
export const map_TaskInput_Towards: Enum_Map = {
  1: getLocale('TaskInput_Towards__TOWARDS_PRODUCTION'),
  2: getLocale('TaskInput_Towards__TOWARDS_PURCHASE'),
}
export const list_TaskOutput_State: Enum_List_Item[] = [
  { value: 1, text: getLocale('TaskOutput_State__STATE_NOT_SUBMITTED') },
  { value: 2, text: getLocale('TaskOutput_State__STATE_SUBMITTED') },
]
export const map_TaskOutput_State: Enum_Map = {
  1: getLocale('TaskOutput_State__STATE_NOT_SUBMITTED'),
  2: getLocale('TaskOutput_State__STATE_SUBMITTED'),
}
export const list_TaskProcess_State: Enum_List_Item[] = [
  { value: 1, text: getLocale('TaskProcess_State__STATE_PREPARE') },
  { value: 2, text: getLocale('TaskProcess_State__STATE_STARTED') },
  { value: 3, text: getLocale('TaskProcess_State__STATE_FINISHED') },
]
export const map_TaskProcess_State: Enum_Map = {
  1: getLocale('TaskProcess_State__STATE_PREPARE'),
  2: getLocale('TaskProcess_State__STATE_STARTED'),
  3: getLocale('TaskProcess_State__STATE_FINISHED'),
}
export const list_TaskProcess_Operation: Enum_List_Item[] = [
  { value: 1, text: getLocale('TaskProcess_Operation__OPERATION_MODIFY_PROCESSOR') },
  { value: 2, text: getLocale('TaskProcess_Operation__OPERATION_START') },
  { value: 3, text: getLocale('TaskProcess_Operation__OPERATION_FINISH') },
  { value: 4, text: getLocale('TaskProcess_Operation__OPERATION_MODIFY_OUTPUT_AMOUNT') },
  { value: 5, text: getLocale('TaskProcess_Operation__OPERATION_MODIFY_ACTUAL_AMOUNT') },
  { value: 6, text: getLocale('TaskProcess_Operation__OPERATION_FINISH_AND_OUTPUT') },
]
export const map_TaskProcess_Operation: Enum_Map = {
  1: getLocale('TaskProcess_Operation__OPERATION_MODIFY_PROCESSOR'),
  2: getLocale('TaskProcess_Operation__OPERATION_START'),
  3: getLocale('TaskProcess_Operation__OPERATION_FINISH'),
  4: getLocale('TaskProcess_Operation__OPERATION_MODIFY_OUTPUT_AMOUNT'),
  5: getLocale('TaskProcess_Operation__OPERATION_MODIFY_ACTUAL_AMOUNT'),
  6: getLocale('TaskProcess_Operation__OPERATION_FINISH_AND_OUTPUT'),
}
export const list_TaskWeight_State: Enum_List_Item[] = [
  { value: 1, text: getLocale('TaskWeight_State__STATE_NOT_SUBMITTED') },
  { value: 2, text: getLocale('TaskWeight_State__STATE_SUBMITTED') },
]
export const map_TaskWeight_State: Enum_Map = {
  1: getLocale('TaskWeight_State__STATE_NOT_SUBMITTED'),
  2: getLocale('TaskWeight_State__STATE_SUBMITTED'),
}
export const list_TaskSheet_Type: Enum_List_Item[] = [
  { value: 1, text: getLocale('TaskSheet_Type__TYPE_PRODUCT') },
  { value: 2, text: getLocale('TaskSheet_Type__TYPE_MATERIAL') },
]
export const map_TaskSheet_Type: Enum_Map = {
  1: getLocale('TaskSheet_Type__TYPE_PRODUCT'),
  2: getLocale('TaskSheet_Type__TYPE_MATERIAL'),
}
export const list_UpdateTaskInputRequest_Type: Enum_List_Item[] = [
  { value: 1, text: getLocale('UpdateTaskInputRequest_Type__TYPE_RECEIVE') },
  { value: 2, text: getLocale('UpdateTaskInputRequest_Type__TYPE_RETURN') },
]
export const map_UpdateTaskInputRequest_Type: Enum_Map = {
  1: getLocale('UpdateTaskInputRequest_Type__TYPE_RECEIVE'),
  2: getLocale('UpdateTaskInputRequest_Type__TYPE_RETURN'),
}
export const list_AsyncTaskCreateParameter_Source: Enum_List_Item[] = [
  { value: 1, text: getLocale('AsyncTaskCreateParameter_Source__SOURCE_PLAN_PACK') },
  { value: 2, text: getLocale('AsyncTaskCreateParameter_Source__SOURCE_PLAN_PRODUCTION') },
  { value: 3, text: getLocale('AsyncTaskCreateParameter_Source__SOURCE_ORDER_DISPATCH') },
]
export const map_AsyncTaskCreateParameter_Source: Enum_Map = {
  1: getLocale('AsyncTaskCreateParameter_Source__SOURCE_PLAN_PACK'),
  2: getLocale('AsyncTaskCreateParameter_Source__SOURCE_PLAN_PRODUCTION'),
  3: getLocale('AsyncTaskCreateParameter_Source__SOURCE_ORDER_DISPATCH'),
}
