// gm_api/src/production/pc.tsx
import React, { FC } from 'react'
import { Select, SelectProps } from '@gm-pc/react'
import {
  SelectList,
  SelectEnum,
  MoreSelectList,
  MapId,
} from '../util/pc_util'
import type {
  MapProps,
  SelectListProps,
  SelectEnumProps,
  MoreSelectListProps,
  MapIdProps,
} from '../util/pc_util'
import {
  list_Status_Code,
  map_Status_Code,
  list_Attr_Type,
  map_Attr_Type,
  list_OpType,
  map_OpType,
  list_BomType,
  map_BomType,
  list_OutputType,
  map_OutputType,
  list_ProcessType_Status,
  map_ProcessType_Status,
  list_ProcessTemplate_Status,
  map_ProcessTemplate_Status,
  list_Bom_Status,
  map_Bom_Status,
  list_Task_State,
  map_Task_State,
  list_Task_Type,
  map_Task_Type,
  list_Task_Source,
  map_Task_Source,
  list_Task_Operation,
  map_Task_Operation,
  list_Task_TimeType,
  map_Task_TimeType,
  list_Task_OutputSource,
  map_Task_OutputSource,
  list_TaskInput_Towards,
  map_TaskInput_Towards,
  list_TaskOutput_State,
  map_TaskOutput_State,
  list_TaskProcess_State,
  map_TaskProcess_State,
  list_TaskProcess_Operation,
  map_TaskProcess_Operation,
  list_TaskWeight_State,
  map_TaskWeight_State,
  list_TaskSheet_Type,
  map_TaskSheet_Type,
  list_UpdateTaskInputRequest_Type,
  map_UpdateTaskInputRequest_Type,
  list_AsyncTaskCreateParameter_Source,
  map_AsyncTaskCreateParameter_Source,
  ProcessType,
  ListProcessType,
  ListProcessTypeResponse,
  ProcessTemplate,
  ListProcessTemplate,
  ListProcessTemplateResponse,
  Processor,
  ListProcessor,
  ListProcessorResponse,
  Task,
  ListTask,
  ListTaskResponse,
  TaskOutput,
  ListTaskOutput,
  ListTaskOutputResponse,
  TaskWeight,
  ListTaskWeight,
  ListTaskWeightResponse,
} from './index'

export const Select_Status_Code: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Status_Code} />
}

export const Map_Status_Code: FC<MapProps> = ({ value }) => {
  return <>{map_Status_Code[value]}</>
}

export const Select_Attr_Type: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Attr_Type} />
}

export const Map_Attr_Type: FC<MapProps> = ({ value }) => {
  return <>{map_Attr_Type[value]}</>
}

export const Select_OpType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_OpType} />
}

export const Map_OpType: FC<MapProps> = ({ value }) => {
  return <>{map_OpType[value]}</>
}

export const Select_BomType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_BomType} />
}

export const Map_BomType: FC<MapProps> = ({ value }) => {
  return <>{map_BomType[value]}</>
}

export const Select_OutputType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_OutputType} />
}

export const Map_OutputType: FC<MapProps> = ({ value }) => {
  return <>{map_OutputType[value]}</>
}

export const Select_ProcessType_Status: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_ProcessType_Status} />
}

export const Map_ProcessType_Status: FC<MapProps> = ({ value }) => {
  return <>{map_ProcessType_Status[value]}</>
}

export const Select_ProcessTemplate_Status: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_ProcessTemplate_Status} />
}

export const Map_ProcessTemplate_Status: FC<MapProps> = ({ value }) => {
  return <>{map_ProcessTemplate_Status[value]}</>
}

export const Select_Bom_Status: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Bom_Status} />
}

export const Map_Bom_Status: FC<MapProps> = ({ value }) => {
  return <>{map_Bom_Status[value]}</>
}

export const Select_Task_State: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Task_State} />
}

export const Map_Task_State: FC<MapProps> = ({ value }) => {
  return <>{map_Task_State[value]}</>
}

export const Select_Task_Type: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Task_Type} />
}

export const Map_Task_Type: FC<MapProps> = ({ value }) => {
  return <>{map_Task_Type[value]}</>
}

export const Select_Task_Source: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Task_Source} />
}

export const Map_Task_Source: FC<MapProps> = ({ value }) => {
  return <>{map_Task_Source[value]}</>
}

export const Select_Task_Operation: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Task_Operation} />
}

export const Map_Task_Operation: FC<MapProps> = ({ value }) => {
  return <>{map_Task_Operation[value]}</>
}

export const Select_Task_TimeType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Task_TimeType} />
}

export const Map_Task_TimeType: FC<MapProps> = ({ value }) => {
  return <>{map_Task_TimeType[value]}</>
}

export const Select_Task_OutputSource: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Task_OutputSource} />
}

export const Map_Task_OutputSource: FC<MapProps> = ({ value }) => {
  return <>{map_Task_OutputSource[value]}</>
}

export const Select_TaskInput_Towards: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_TaskInput_Towards} />
}

export const Map_TaskInput_Towards: FC<MapProps> = ({ value }) => {
  return <>{map_TaskInput_Towards[value]}</>
}

export const Select_TaskOutput_State: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_TaskOutput_State} />
}

export const Map_TaskOutput_State: FC<MapProps> = ({ value }) => {
  return <>{map_TaskOutput_State[value]}</>
}

export const Select_TaskProcess_State: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_TaskProcess_State} />
}

export const Map_TaskProcess_State: FC<MapProps> = ({ value }) => {
  return <>{map_TaskProcess_State[value]}</>
}

export const Select_TaskProcess_Operation: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_TaskProcess_Operation} />
}

export const Map_TaskProcess_Operation: FC<MapProps> = ({ value }) => {
  return <>{map_TaskProcess_Operation[value]}</>
}

export const Select_TaskWeight_State: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_TaskWeight_State} />
}

export const Map_TaskWeight_State: FC<MapProps> = ({ value }) => {
  return <>{map_TaskWeight_State[value]}</>
}

export const Select_TaskSheet_Type: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_TaskSheet_Type} />
}

export const Map_TaskSheet_Type: FC<MapProps> = ({ value }) => {
  return <>{map_TaskSheet_Type[value]}</>
}

export const Select_UpdateTaskInputRequest_Type: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_UpdateTaskInputRequest_Type} />
}

export const Map_UpdateTaskInputRequest_Type: FC<MapProps> = ({ value }) => {
  return <>{map_UpdateTaskInputRequest_Type[value]}</>
}

export const Select_AsyncTaskCreateParameter_Source: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_AsyncTaskCreateParameter_Source} />
}

export const Map_AsyncTaskCreateParameter_Source: FC<MapProps> = ({ value }) => {
  return <>{map_AsyncTaskCreateParameter_Source[value]}</>
}

export const Select_ProcessType: FC<SelectListProps<
  ProcessType,
  ListProcessTypeResponse
>> = (props) => {
  return (
    <SelectList<ProcessType, ListProcessTypeResponse>
      {...props}
      messageName='ProcessType'
      fetchList={ListProcessType}
    />
  )
}

export const MoreSelect_ProcessType: FC<MoreSelectListProps<
  ProcessType,
  ListProcessTypeResponse
>> = (props) => {
  return (
    <MoreSelectList<ProcessType, ListProcessTypeResponse>
      {...props}
      messageName='ProcessType'
      fetchList={ListProcessType}
    />
  )
}

export const Select_ProcessTemplate: FC<SelectListProps<
  ProcessTemplate,
  ListProcessTemplateResponse
>> = (props) => {
  return (
    <SelectList<ProcessTemplate, ListProcessTemplateResponse>
      {...props}
      messageName='ProcessTemplate'
      fetchList={ListProcessTemplate}
    />
  )
}

export const MoreSelect_ProcessTemplate: FC<MoreSelectListProps<
  ProcessTemplate,
  ListProcessTemplateResponse
>> = (props) => {
  return (
    <MoreSelectList<ProcessTemplate, ListProcessTemplateResponse>
      {...props}
      messageName='ProcessTemplate'
      fetchList={ListProcessTemplate}
    />
  )
}

export const Select_Processor: FC<SelectListProps<
  Processor,
  ListProcessorResponse
>> = (props) => {
  return (
    <SelectList<Processor, ListProcessorResponse>
      {...props}
      messageName='Processor'
      fetchList={ListProcessor}
    />
  )
}

export const MoreSelect_Processor: FC<MoreSelectListProps<
  Processor,
  ListProcessorResponse
>> = (props) => {
  return (
    <MoreSelectList<Processor, ListProcessorResponse>
      {...props}
      messageName='Processor'
      fetchList={ListProcessor}
    />
  )
}

export const Select_Task: FC<SelectListProps<
  Task,
  ListTaskResponse
>> = (props) => {
  return (
    <SelectList<Task, ListTaskResponse>
      {...props}
      messageName='Task'
      fetchList={ListTask}
    />
  )
}

export const MoreSelect_Task: FC<MoreSelectListProps<
  Task,
  ListTaskResponse
>> = (props) => {
  return (
    <MoreSelectList<Task, ListTaskResponse>
      {...props}
      messageName='Task'
      fetchList={ListTask}
    />
  )
}

export const Select_TaskOutput: FC<SelectListProps<
  TaskOutput,
  ListTaskOutputResponse
>> = (props) => {
  return (
    <SelectList<TaskOutput, ListTaskOutputResponse>
      {...props}
      messageName='TaskOutput'
      fetchList={ListTaskOutput}
    />
  )
}

export const MoreSelect_TaskOutput: FC<MoreSelectListProps<
  TaskOutput,
  ListTaskOutputResponse
>> = (props) => {
  return (
    <MoreSelectList<TaskOutput, ListTaskOutputResponse>
      {...props}
      messageName='TaskOutput'
      fetchList={ListTaskOutput}
    />
  )
}

export const Select_TaskWeight: FC<SelectListProps<
  TaskWeight,
  ListTaskWeightResponse
>> = (props) => {
  return (
    <SelectList<TaskWeight, ListTaskWeightResponse>
      {...props}
      messageName='TaskWeight'
      fetchList={ListTaskWeight}
    />
  )
}

export const MoreSelect_TaskWeight: FC<MoreSelectListProps<
  TaskWeight,
  ListTaskWeightResponse
>> = (props) => {
  return (
    <MoreSelectList<TaskWeight, ListTaskWeightResponse>
      {...props}
      messageName='TaskWeight'
      fetchList={ListTaskWeight}
    />
  )
}

export const MapId_ProcessTemplate: FC<MapIdProps<ProcessTemplate, ListProcessTemplateResponse>> = (
  props,
) => {
  return (
    <MapId<ProcessTemplate, ListProcessTemplateResponse>
      {...props}
      packageName='production'
      messageName='ProcessTemplate'
      fetchList={ListProcessTemplate}
    />
  )
}

export const MapId_Processor: FC<MapIdProps<Processor, ListProcessorResponse>> = (
  props,
) => {
  return (
    <MapId<Processor, ListProcessorResponse>
      {...props}
      packageName='production'
      messageName='Processor'
      fetchList={ListProcessor}
    />
  )
}

export const MapId_Task: FC<MapIdProps<Task, ListTaskResponse>> = (
  props,
) => {
  return (
    <MapId<Task, ListTaskResponse>
      {...props}
      packageName='production'
      messageName='Task'
      fetchList={ListTask}
    />
  )
}

export type {
  MapProps,
  SelectListProps,
  MoreSelectListProps,
}
