// gm_api/src/common/data.ts
import { getLocale } from './lng'

interface Enum_List_Item {
  text: string
  value: any
}

interface Enum_Map {
  [key: string]: string
}
export const list_Status_Code: Enum_List_Item[] = [
  { value: 1, text: getLocale('Status_Code__CANCELED') },
  { value: 2, text: getLocale('Status_Code__UNKNOWN') },
  { value: 3, text: getLocale('Status_Code__INVALID_ARGUMENT') },
  { value: 4, text: getLocale('Status_Code__DEADLINE_EXCEEDED') },
  { value: 5, text: getLocale('Status_Code__NOT_FOUND') },
  { value: 6, text: getLocale('Status_Code__ALREADY_EXISTS') },
  { value: 7, text: getLocale('Status_Code__PERMISSION_DENIED') },
  { value: 8, text: getLocale('Status_Code__RESOURCE_EXHAUSTED') },
  { value: 9, text: getLocale('Status_Code__FAILED_PRECONDITION') },
  { value: 10, text: getLocale('Status_Code__ABORTED') },
  { value: 11, text: getLocale('Status_Code__OUT_OF_RANGE') },
  { value: 12, text: getLocale('Status_Code__UNIMPLEMENTED') },
  { value: 13, text: getLocale('Status_Code__INTERNAL') },
  { value: 14, text: getLocale('Status_Code__UNAVAILABLE') },
  { value: 15, text: getLocale('Status_Code__DATA_LOSS') },
  { value: 16, text: getLocale('Status_Code__UNAUTHENTICATED') },
  { value: 1001, text: getLocale('Status_Code__DATABASE_ERROR') },
  { value: 1002, text: getLocale('Status_Code__ASYNC_ERROR') },
  { value: 1003, text: getLocale('Status_Code__CACHE_NOT_EXISTED') },
  { value: 1004, text: getLocale('Status_Code__DATA_EXPIRED') },
  { value: 1005, text: getLocale('Status_Code__DUPLICATE_CUSTOMIZED_CODE') },
  { value: 1006, text: getLocale('Status_Code__REDIS_ERROR') },
  { value: 1007, text: getLocale('Status_Code__REDIS_LOCKED') },
]
export const map_Status_Code: Enum_Map = {
  1: getLocale('Status_Code__CANCELED'),
  2: getLocale('Status_Code__UNKNOWN'),
  3: getLocale('Status_Code__INVALID_ARGUMENT'),
  4: getLocale('Status_Code__DEADLINE_EXCEEDED'),
  5: getLocale('Status_Code__NOT_FOUND'),
  6: getLocale('Status_Code__ALREADY_EXISTS'),
  7: getLocale('Status_Code__PERMISSION_DENIED'),
  8: getLocale('Status_Code__RESOURCE_EXHAUSTED'),
  9: getLocale('Status_Code__FAILED_PRECONDITION'),
  10: getLocale('Status_Code__ABORTED'),
  11: getLocale('Status_Code__OUT_OF_RANGE'),
  12: getLocale('Status_Code__UNIMPLEMENTED'),
  13: getLocale('Status_Code__INTERNAL'),
  14: getLocale('Status_Code__UNAVAILABLE'),
  15: getLocale('Status_Code__DATA_LOSS'),
  16: getLocale('Status_Code__UNAUTHENTICATED'),
  1001: getLocale('Status_Code__DATABASE_ERROR'),
  1002: getLocale('Status_Code__ASYNC_ERROR'),
  1003: getLocale('Status_Code__CACHE_NOT_EXISTED'),
  1004: getLocale('Status_Code__DATA_EXPIRED'),
  1005: getLocale('Status_Code__DUPLICATE_CUSTOMIZED_CODE'),
  1006: getLocale('Status_Code__REDIS_ERROR'),
  1007: getLocale('Status_Code__REDIS_LOCKED'),
}
export const list_Model_Type: Enum_List_Item[] = [
  { value: 1, text: getLocale('Model_Type__TYPE_ACCOUNT_ACCOUNT') },
  { value: 2, text: getLocale('Model_Type__TYPE_PRODUCTION_PROCESSTYPE') },
  { value: 3, text: getLocale('Model_Type__TYPE_PRODUCTION_PROCESSTEMPLATE') },
  { value: 4, text: getLocale('Model_Type__TYPE_PRODUCTION_PROCESSOR') },
  { value: 5, text: getLocale('Model_Type__TYPE_PRODUCTION_BOM') },
  { value: 6, text: getLocale('Model_Type__TYPE_PRODUCTION_TASK') },
  { value: 7, text: getLocale('Model_Type__TYPE_PRODUCTION_TASKINPUT') },
  { value: 8, text: getLocale('Model_Type__TYPE_PRODUCTION_TASKPROCESS') },
  { value: 47, text: getLocale('Model_Type__TYPE_PRODUCTION_TASKWEIGHT') },
  { value: 66, text: getLocale('Model_Type__TYPE_PRODUCTION_TASKOUTPUT') },
  { value: 9, text: getLocale('Model_Type__TYPE_INVENTORY_STOCKSHEET') },
  { value: 10, text: getLocale('Model_Type__TYPE_INVENTORY_STOCKLOG') },
  { value: 11, text: getLocale('Model_Type__TYPE_INVENTORY_BATCHLOG') },
  { value: 12, text: getLocale('Model_Type__TYPE_INVENTORY_SHELF') },
  { value: 13, text: getLocale('Model_Type__TYPE_INVENTORY_BATCH') },
  { value: 14, text: getLocale('Model_Type__TYPE_INVENTORY_SKUUNITSTOCK') },
  { value: 15, text: getLocale('Model_Type__TYPE_INVENTORY_SKUSTOCK') },
  { value: 16, text: getLocale('Model_Type__TYPE_INVENTORY_STATEMENT') },
  { value: 17, text: getLocale('Model_Type__TYPE_INVENTORY_ADJUSTSHEET') },
  { value: 18, text: getLocale('Model_Type__TYPE_INVENTORY_ADJUSTLOG') },
  { value: 19, text: getLocale('Model_Type__TYPE_INVENTORY_CUSTOMERTURNOVER') },
  { value: 20, text: getLocale('Model_Type__TYPE_INVENTORY_REPLACE') },
  { value: 21, text: getLocale('Model_Type__TYPE_ORDER_ORDER') },
  { value: 22, text: getLocale('Model_Type__TYPE_ORDER_ORDER_DETAIL') },
  { value: 23, text: getLocale('Model_Type__TYPE_ORDER_ORDER_IMPORT_TEMPLETE') },
  { value: 48, text: getLocale('Model_Type__TYPE_ORDER_ESHOP_ORDER') },
  { value: 49, text: getLocale('Model_Type__TYPE_ORDER_ESHOP_ORDER_DETAIL') },
  { value: 70, text: getLocale('Model_Type__TYPE_ORDER_ESHOP_MENU_PERIOD_GROUP') },
  { value: 71, text: getLocale('Model_Type__TYPE_ORDER_ESHOP_LEAVE_APPLICATION') },
  { value: 72, text: getLocale('Model_Type__TYPE_ORDER_ESHOP_MENU') },
  { value: 73, text: getLocale('Model_Type__TYPE_ORDER_ESHOP_MENU_DETAIL') },
  { value: 24, text: getLocale('Model_Type__TYPE_ENTERPRISE_GROUP') },
  { value: 25, text: getLocale('Model_Type__TYPE_ENTERPRISE_GROUP_USER') },
  { value: 26, text: getLocale('Model_Type__TYPE_ENTERPRISE_STATION') },
  { value: 27, text: getLocale('Model_Type__TYPE_ENTERPRISE_CUSTOMER') },
  { value: 28, text: getLocale('Model_Type__TYPE_ENTERPRISE_CUSTOMER_USER') },
  { value: 45, text: getLocale('Model_Type__TYPE_ENTERPRISE_CUSTOMER_LABEL') },
  { value: 29, text: getLocale('Model_Type__TYPE_ENTERPRISE_SERVICE_PERIOD') },
  { value: 30, text: getLocale('Model_Type__TYPE_ENTERPRISE_SUPPLIER') },
  { value: 31, text: getLocale('Model_Type__TYPE_ENTERPRISE_DISTRIBUTION_CONTRACTOR') },
  { value: 32, text: getLocale('Model_Type__TYPE_ENTERPRISE_ROLE') },
  { value: 34, text: getLocale('Model_Type__TYPE_ENTERPRISE_INVITATION_CODE') },
  { value: 46, text: getLocale('Model_Type__TYPE_ENTERPRISE_SHOP_DIY_SETTING') },
  { value: 35, text: getLocale('Model_Type__TYPE_PURCHASE_TASK') },
  { value: 36, text: getLocale('Model_Type__TYPE_PURCHASE_SHEET') },
  { value: 37, text: getLocale('Model_Type__TYPE_PURCHASE_BATCH') },
  { value: 38, text: getLocale('Model_Type__TYPE_PURCHASE_INQUIRY_PRICE') },
  { value: 50, text: getLocale('Model_Type__TYPE_PURCHASE_SRC_RELATION') },
  { value: 39, text: getLocale('Model_Type__TYPE_OPERATION_OPERATION') },
  { value: 40, text: getLocale('Model_Type__TYPE_MERCHANDISE_CATEGORY') },
  { value: 41, text: getLocale('Model_Type__TYPE_MERCHANDISE_SPU') },
  { value: 42, text: getLocale('Model_Type__TYPE_MERCHANDISE_SKU') },
  { value: 43, text: getLocale('Model_Type__TYPE_MERCHANDISE_QUOTATION') },
  { value: 100, text: getLocale('Model_Type__TYPE_MERCHANDISE_FAVORITE') },
  { value: 101, text: getLocale('Model_Type__TYPE_MERCHANDISE_FAVORITE_SSU') },
  { value: 44, text: getLocale('Model_Type__TYPE_ASYNCTASK_TASK') },
  { value: 51, text: getLocale('Model_Type__TYPE_SORTING_TAG') },
  { value: 52, text: getLocale('Model_Type__TYPE_SORTING_SKU_GROUP') },
  { value: 53, text: getLocale('Model_Type__TYPE_SORTING_BASKET') },
  { value: 65, text: getLocale('Model_Type__TYPE_SORTING_SKU_GROUP_SKU') },
  { value: 54, text: getLocale('Model_Type__TYPE_ROUTE') },
  { value: 55, text: getLocale('Model_Type__TYPE_MENU') },
  { value: 56, text: getLocale('Model_Type__TYPE_MENU_DETAIL_ID') },
  { value: 57, text: getLocale('Model_Type__TYPE_PREFERENCE_PRINTING_TEMPLATE') },
  { value: 58, text: getLocale('Model_Type__TYPE_PREFERENCE_INNER_LABEL_PRINTING_TEMPLATE') },
  { value: 59, text: getLocale('Model_Type__TYPE_PREFERENCE_SHOP') },
  { value: 60, text: getLocale('Model_Type__TYPE_PREFERENCE_SHOP_LAYOUT_SETTINGS') },
  { value: 67, text: getLocale('Model_Type__TYPE_PREFERENCE_SHOP_BANNER') },
  { value: 61, text: getLocale('Model_Type__TYPE_PURCHASE_SETTINGS') },
  { value: 62, text: getLocale('Model_Type__TYPE_INVENTORY_SETTINGS') },
  { value: 63, text: getLocale('Model_Type__TYPE_APPOINT_TIME_SETTINGS') },
  { value: 64, text: getLocale('Model_Type__TYPE_PRECISION_SETTINGS') },
  { value: 68, text: getLocale('Model_Type__TYPE_SORTING_SETTINGS') },
  { value: 74, text: getLocale('Model_Type__TYPE_PRODUCTION_SETTINGS') },
  { value: 80, text: getLocale('Model_Type__TYPE_AFTER_SALE_ORDER') },
  { value: 81, text: getLocale('Model_Type__TYPE_AFTER_SALE_ORDER_DETAIL') },
  { value: 90, text: getLocale('Model_Type__TYPE_BOSS_PERMISSION_SET') },
  { value: 200, text: getLocale('Model_Type__TYPE_DATA_SYNC_SEQUENCE') },
  { value: 300, text: getLocale('Model_Type__TYPE_FINANCE_SETTLE') },
  { value: 301, text: getLocale('Model_Type__TYPE_FINANCE_AMOUNT_DISCOUNT') },
  { value: 302, text: getLocale('Model_Type__TYPE_FINANCE_TRANSACTION_FLOW') },
  { value: 303, text: getLocale('Model_Type__TYPE_FINANCE_ACCOUNT_BALANCE') },
  { value: 1000000, text: getLocale('Model_Type__TYPE_FAKE_ANALYTICS_EXAMPLE') },
  { value: 1000001, text: getLocale('Model_Type__TYPE_FAKE_ANALYTICS_EXAMPLE_TYPE_B') },
]
export const map_Model_Type: Enum_Map = {
  1: getLocale('Model_Type__TYPE_ACCOUNT_ACCOUNT'),
  2: getLocale('Model_Type__TYPE_PRODUCTION_PROCESSTYPE'),
  3: getLocale('Model_Type__TYPE_PRODUCTION_PROCESSTEMPLATE'),
  4: getLocale('Model_Type__TYPE_PRODUCTION_PROCESSOR'),
  5: getLocale('Model_Type__TYPE_PRODUCTION_BOM'),
  6: getLocale('Model_Type__TYPE_PRODUCTION_TASK'),
  7: getLocale('Model_Type__TYPE_PRODUCTION_TASKINPUT'),
  8: getLocale('Model_Type__TYPE_PRODUCTION_TASKPROCESS'),
  47: getLocale('Model_Type__TYPE_PRODUCTION_TASKWEIGHT'),
  66: getLocale('Model_Type__TYPE_PRODUCTION_TASKOUTPUT'),
  9: getLocale('Model_Type__TYPE_INVENTORY_STOCKSHEET'),
  10: getLocale('Model_Type__TYPE_INVENTORY_STOCKLOG'),
  11: getLocale('Model_Type__TYPE_INVENTORY_BATCHLOG'),
  12: getLocale('Model_Type__TYPE_INVENTORY_SHELF'),
  13: getLocale('Model_Type__TYPE_INVENTORY_BATCH'),
  14: getLocale('Model_Type__TYPE_INVENTORY_SKUUNITSTOCK'),
  15: getLocale('Model_Type__TYPE_INVENTORY_SKUSTOCK'),
  16: getLocale('Model_Type__TYPE_INVENTORY_STATEMENT'),
  17: getLocale('Model_Type__TYPE_INVENTORY_ADJUSTSHEET'),
  18: getLocale('Model_Type__TYPE_INVENTORY_ADJUSTLOG'),
  19: getLocale('Model_Type__TYPE_INVENTORY_CUSTOMERTURNOVER'),
  20: getLocale('Model_Type__TYPE_INVENTORY_REPLACE'),
  21: getLocale('Model_Type__TYPE_ORDER_ORDER'),
  22: getLocale('Model_Type__TYPE_ORDER_ORDER_DETAIL'),
  23: getLocale('Model_Type__TYPE_ORDER_ORDER_IMPORT_TEMPLETE'),
  48: getLocale('Model_Type__TYPE_ORDER_ESHOP_ORDER'),
  49: getLocale('Model_Type__TYPE_ORDER_ESHOP_ORDER_DETAIL'),
  70: getLocale('Model_Type__TYPE_ORDER_ESHOP_MENU_PERIOD_GROUP'),
  71: getLocale('Model_Type__TYPE_ORDER_ESHOP_LEAVE_APPLICATION'),
  72: getLocale('Model_Type__TYPE_ORDER_ESHOP_MENU'),
  73: getLocale('Model_Type__TYPE_ORDER_ESHOP_MENU_DETAIL'),
  24: getLocale('Model_Type__TYPE_ENTERPRISE_GROUP'),
  25: getLocale('Model_Type__TYPE_ENTERPRISE_GROUP_USER'),
  26: getLocale('Model_Type__TYPE_ENTERPRISE_STATION'),
  27: getLocale('Model_Type__TYPE_ENTERPRISE_CUSTOMER'),
  28: getLocale('Model_Type__TYPE_ENTERPRISE_CUSTOMER_USER'),
  45: getLocale('Model_Type__TYPE_ENTERPRISE_CUSTOMER_LABEL'),
  29: getLocale('Model_Type__TYPE_ENTERPRISE_SERVICE_PERIOD'),
  30: getLocale('Model_Type__TYPE_ENTERPRISE_SUPPLIER'),
  31: getLocale('Model_Type__TYPE_ENTERPRISE_DISTRIBUTION_CONTRACTOR'),
  32: getLocale('Model_Type__TYPE_ENTERPRISE_ROLE'),
  34: getLocale('Model_Type__TYPE_ENTERPRISE_INVITATION_CODE'),
  46: getLocale('Model_Type__TYPE_ENTERPRISE_SHOP_DIY_SETTING'),
  35: getLocale('Model_Type__TYPE_PURCHASE_TASK'),
  36: getLocale('Model_Type__TYPE_PURCHASE_SHEET'),
  37: getLocale('Model_Type__TYPE_PURCHASE_BATCH'),
  38: getLocale('Model_Type__TYPE_PURCHASE_INQUIRY_PRICE'),
  50: getLocale('Model_Type__TYPE_PURCHASE_SRC_RELATION'),
  39: getLocale('Model_Type__TYPE_OPERATION_OPERATION'),
  40: getLocale('Model_Type__TYPE_MERCHANDISE_CATEGORY'),
  41: getLocale('Model_Type__TYPE_MERCHANDISE_SPU'),
  42: getLocale('Model_Type__TYPE_MERCHANDISE_SKU'),
  43: getLocale('Model_Type__TYPE_MERCHANDISE_QUOTATION'),
  100: getLocale('Model_Type__TYPE_MERCHANDISE_FAVORITE'),
  101: getLocale('Model_Type__TYPE_MERCHANDISE_FAVORITE_SSU'),
  44: getLocale('Model_Type__TYPE_ASYNCTASK_TASK'),
  51: getLocale('Model_Type__TYPE_SORTING_TAG'),
  52: getLocale('Model_Type__TYPE_SORTING_SKU_GROUP'),
  53: getLocale('Model_Type__TYPE_SORTING_BASKET'),
  65: getLocale('Model_Type__TYPE_SORTING_SKU_GROUP_SKU'),
  54: getLocale('Model_Type__TYPE_ROUTE'),
  55: getLocale('Model_Type__TYPE_MENU'),
  56: getLocale('Model_Type__TYPE_MENU_DETAIL_ID'),
  57: getLocale('Model_Type__TYPE_PREFERENCE_PRINTING_TEMPLATE'),
  58: getLocale('Model_Type__TYPE_PREFERENCE_INNER_LABEL_PRINTING_TEMPLATE'),
  59: getLocale('Model_Type__TYPE_PREFERENCE_SHOP'),
  60: getLocale('Model_Type__TYPE_PREFERENCE_SHOP_LAYOUT_SETTINGS'),
  67: getLocale('Model_Type__TYPE_PREFERENCE_SHOP_BANNER'),
  61: getLocale('Model_Type__TYPE_PURCHASE_SETTINGS'),
  62: getLocale('Model_Type__TYPE_INVENTORY_SETTINGS'),
  63: getLocale('Model_Type__TYPE_APPOINT_TIME_SETTINGS'),
  64: getLocale('Model_Type__TYPE_PRECISION_SETTINGS'),
  68: getLocale('Model_Type__TYPE_SORTING_SETTINGS'),
  74: getLocale('Model_Type__TYPE_PRODUCTION_SETTINGS'),
  80: getLocale('Model_Type__TYPE_AFTER_SALE_ORDER'),
  81: getLocale('Model_Type__TYPE_AFTER_SALE_ORDER_DETAIL'),
  90: getLocale('Model_Type__TYPE_BOSS_PERMISSION_SET'),
  200: getLocale('Model_Type__TYPE_DATA_SYNC_SEQUENCE'),
  300: getLocale('Model_Type__TYPE_FINANCE_SETTLE'),
  301: getLocale('Model_Type__TYPE_FINANCE_AMOUNT_DISCOUNT'),
  302: getLocale('Model_Type__TYPE_FINANCE_TRANSACTION_FLOW'),
  303: getLocale('Model_Type__TYPE_FINANCE_ACCOUNT_BALANCE'),
  1000000: getLocale('Model_Type__TYPE_FAKE_ANALYTICS_EXAMPLE'),
  1000001: getLocale('Model_Type__TYPE_FAKE_ANALYTICS_EXAMPLE_TYPE_B'),
}
export const list_Filters_Bool: Enum_List_Item[] = [
  { value: 1, text: getLocale('Filters_Bool__TRUE') },
  { value: 2, text: getLocale('Filters_Bool__FALSE') },
]
export const map_Filters_Bool: Enum_Map = {
  1: getLocale('Filters_Bool__TRUE'),
  2: getLocale('Filters_Bool__FALSE'),
}
export const list_App_Type: Enum_List_Item[] = [
  { value: 1, text: getLocale('App_Type__TYPE_STATION') },
  { value: 2, text: getLocale('App_Type__TYPE_BSHOP') },
  { value: 3, text: getLocale('App_Type__TYPE_CSHOP') },
  { value: 4, text: getLocale('App_Type__TYPE_ESHOP') },
  { value: 5, text: getLocale('App_Type__TYPE_CMD') },
]
export const map_App_Type: Enum_Map = {
  1: getLocale('App_Type__TYPE_STATION'),
  2: getLocale('App_Type__TYPE_BSHOP'),
  3: getLocale('App_Type__TYPE_CSHOP'),
  4: getLocale('App_Type__TYPE_ESHOP'),
  5: getLocale('App_Type__TYPE_CMD'),
}
export const list_Image_Type: Enum_List_Item[] = [
  { value: 1, text: getLocale('Image_Type__TYPE_QINIU') },
  { value: 2, text: getLocale('Image_Type__TYPE_TENCENT_COS') },
]
export const map_Image_Type: Enum_Map = {
  1: getLocale('Image_Type__TYPE_QINIU'),
  2: getLocale('Image_Type__TYPE_TENCENT_COS'),
}
export const list_Gender: Enum_List_Item[] = [
  { value: 1, text: getLocale('Gender__GENDER_FEMALE') },
  { value: 2, text: getLocale('Gender__GENDER_MALE') },
]
export const map_Gender: Enum_Map = {
  1: getLocale('Gender__GENDER_FEMALE'),
  2: getLocale('Gender__GENDER_MALE'),
}
