// gm_api/src/common/pc.tsx
import React, { FC } from 'react'
import { Select, SelectProps } from '@gm-pc/react'
import {
  SelectList,
  SelectEnum,
  MoreSelectList,
  MapId,
} from '../util/pc_util'
import type {
  MapProps,
  SelectListProps,
  SelectEnumProps,
  MoreSelectListProps,
  MapIdProps,
} from '../util/pc_util'
import {
  list_Status_Code,
  map_Status_Code,
  list_Model_Type,
  map_Model_Type,
  list_Filters_Bool,
  map_Filters_Bool,
  list_App_Type,
  map_App_Type,
  list_Image_Type,
  map_Image_Type,
  list_Gender,
  map_Gender,
} from './index'

export const Select_Status_Code: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Status_Code} />
}

export const Map_Status_Code: FC<MapProps> = ({ value }) => {
  return <>{map_Status_Code[value]}</>
}

export const Select_Model_Type: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Model_Type} />
}

export const Map_Model_Type: FC<MapProps> = ({ value }) => {
  return <>{map_Model_Type[value]}</>
}

export const Select_Filters_Bool: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Filters_Bool} />
}

export const Map_Filters_Bool: FC<MapProps> = ({ value }) => {
  return <>{map_Filters_Bool[value]}</>
}

export const Select_App_Type: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_App_Type} />
}

export const Map_App_Type: FC<MapProps> = ({ value }) => {
  return <>{map_App_Type[value]}</>
}

export const Select_Image_Type: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Image_Type} />
}

export const Map_Image_Type: FC<MapProps> = ({ value }) => {
  return <>{map_Image_Type[value]}</>
}

export const Select_Gender: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Gender} />
}

export const Map_Gender: FC<MapProps> = ({ value }) => {
  return <>{map_Gender[value]}</>
}

export type {
  MapProps,
  SelectListProps,
  MoreSelectListProps,
}
