// gm_api/src/common/types.ts
export enum Status_Code {
  OK = 0,
  CANCELED = 1,
  UNKNOWN = 2,
  INVALID_ARGUMENT = 3,
  DEADLINE_EXCEEDED = 4,
  NOT_FOUND = 5,
  ALREADY_EXISTS = 6,
  PERMISSION_DENIED = 7,
  RESOURCE_EXHAUSTED = 8,
  FAILED_PRECONDITION = 9,
  ABORTED = 10,
  OUT_OF_RANGE = 11,
  UNIMPLEMENTED = 12,
  INTERNAL = 13,
  UNAVAILABLE = 14,
  DATA_LOSS = 15,
  UNAUTHENTICATED = 16,
  DATABASE_ERROR = 1001,
  ASYNC_ERROR = 1002,
  CACHE_NOT_EXISTED = 1003,
  DATA_EXPIRED = 1004,
  DUPLICATE_CUSTOMIZED_CODE = 1005,
  REDIS_ERROR = 1006,
  REDIS_LOCKED = 1007,
}
export interface Status_InvalidArgumentDetail {
  reason?: string
}
export interface Status {
}
export interface Uint64Set {
  values?: string[]
}
export interface StringSet {
  values?: string[]
}
export enum Model_Type {
  TYPE_UNSPECIFIED = 0,
  TYPE_ACCOUNT_ACCOUNT = 1,
  TYPE_PRODUCTION_PROCESSTYPE = 2,
  TYPE_PRODUCTION_PROCESSTEMPLATE = 3,
  TYPE_PRODUCTION_PROCESSOR = 4,
  TYPE_PRODUCTION_BOM = 5,
  TYPE_PRODUCTION_TASK = 6,
  TYPE_PRODUCTION_TASKINPUT = 7,
  TYPE_PRODUCTION_TASKPROCESS = 8,
  TYPE_PRODUCTION_TASKWEIGHT = 47,
  TYPE_PRODUCTION_TASKOUTPUT = 66,
  TYPE_INVENTORY_STOCKSHEET = 9,
  TYPE_INVENTORY_STOCKLOG = 10,
  TYPE_INVENTORY_BATCHLOG = 11,
  TYPE_INVENTORY_SHELF = 12,
  TYPE_INVENTORY_BATCH = 13,
  TYPE_INVENTORY_SKUUNITSTOCK = 14,
  TYPE_INVENTORY_SKUSTOCK = 15,
  TYPE_INVENTORY_STATEMENT = 16,
  TYPE_INVENTORY_ADJUSTSHEET = 17,
  TYPE_INVENTORY_ADJUSTLOG = 18,
  TYPE_INVENTORY_CUSTOMERTURNOVER = 19,
  TYPE_INVENTORY_REPLACE = 20,
  TYPE_ORDER_ORDER = 21,
  TYPE_ORDER_ORDER_DETAIL = 22,
  TYPE_ORDER_ORDER_IMPORT_TEMPLETE = 23,
  TYPE_ORDER_ESHOP_ORDER = 48,
  TYPE_ORDER_ESHOP_ORDER_DETAIL = 49,
  TYPE_ORDER_ESHOP_MENU_PERIOD_GROUP = 70,
  TYPE_ORDER_ESHOP_LEAVE_APPLICATION = 71,
  TYPE_ORDER_ESHOP_MENU = 72,
  TYPE_ORDER_ESHOP_MENU_DETAIL = 73,
  TYPE_ENTERPRISE_GROUP = 24,
  TYPE_ENTERPRISE_GROUP_USER = 25,
  TYPE_ENTERPRISE_STATION = 26,
  TYPE_ENTERPRISE_CUSTOMER = 27,
  TYPE_ENTERPRISE_CUSTOMER_USER = 28,
  TYPE_ENTERPRISE_CUSTOMER_LABEL = 45,
  TYPE_ENTERPRISE_SERVICE_PERIOD = 29,
  TYPE_ENTERPRISE_SUPPLIER = 30,
  TYPE_ENTERPRISE_DISTRIBUTION_CONTRACTOR = 31,
  TYPE_ENTERPRISE_ROLE = 32,
  TYPE_ENTERPRISE_INVITATION_CODE = 34,
  TYPE_ENTERPRISE_SHOP_DIY_SETTING = 46,
  TYPE_PURCHASE_TASK = 35,
  TYPE_PURCHASE_SHEET = 36,
  TYPE_PURCHASE_BATCH = 37,
  TYPE_PURCHASE_INQUIRY_PRICE = 38,
  TYPE_PURCHASE_SRC_RELATION = 50,
  TYPE_OPERATION_OPERATION = 39,
  TYPE_MERCHANDISE_CATEGORY = 40,
  TYPE_MERCHANDISE_SPU = 41,
  TYPE_MERCHANDISE_SKU = 42,
  TYPE_MERCHANDISE_QUOTATION = 43,
  TYPE_MERCHANDISE_FAVORITE = 100,
  TYPE_MERCHANDISE_FAVORITE_SSU = 101,
  TYPE_ASYNCTASK_TASK = 44,
  TYPE_SORTING_TAG = 51,
  TYPE_SORTING_SKU_GROUP = 52,
  TYPE_SORTING_BASKET = 53,
  TYPE_SORTING_SKU_GROUP_SKU = 65,
  TYPE_ROUTE = 54,
  TYPE_MENU = 55,
  TYPE_MENU_DETAIL_ID = 56,
  TYPE_PREFERENCE_PRINTING_TEMPLATE = 57,
  TYPE_PREFERENCE_INNER_LABEL_PRINTING_TEMPLATE = 58,
  TYPE_PREFERENCE_SHOP = 59,
  TYPE_PREFERENCE_SHOP_LAYOUT_SETTINGS = 60,
  TYPE_PREFERENCE_SHOP_BANNER = 67,
  TYPE_PURCHASE_SETTINGS = 61,
  TYPE_INVENTORY_SETTINGS = 62,
  TYPE_APPOINT_TIME_SETTINGS = 63,
  TYPE_PRECISION_SETTINGS = 64,
  TYPE_SORTING_SETTINGS = 68,
  TYPE_PRODUCTION_SETTINGS = 74,
  TYPE_AFTER_SALE_ORDER = 80,
  TYPE_AFTER_SALE_ORDER_DETAIL = 81,
  TYPE_BOSS_PERMISSION_SET = 90,
  TYPE_DATA_SYNC_SEQUENCE = 200,
  TYPE_FINANCE_SETTLE = 300,
  TYPE_FINANCE_AMOUNT_DISCOUNT = 301,
  TYPE_FINANCE_TRANSACTION_FLOW = 302,
  TYPE_FINANCE_ACCOUNT_BALANCE = 303,
  TYPE_FAKE_ANALYTICS_EXAMPLE = 1000000,
  TYPE_FAKE_ANALYTICS_EXAMPLE_TYPE_B = 1000001,
}
export interface Model {
}
export enum Filters_Bool {
  ALL = 0,
  TRUE = 1,
  FALSE = 2,
}
export interface Filters {
}
export enum App_Type {
  TYPE_UNSPECIFIED = 0,
  TYPE_STATION = 1,
  TYPE_BSHOP = 2,
  TYPE_CSHOP = 3,
  TYPE_ESHOP = 4,
  TYPE_CMD = 5,
}
export interface App {
}
export enum Image_Type {
  TYPE_UNSPECIFIED = 0,
  TYPE_QINIU = 1,
  TYPE_TENCENT_COS = 2,
}
export interface Image {
  type: Image_Type
  height?: number
  weight?: number
  path: string
}
export interface SortBy {
  field?: number
  desc?: boolean
}
export interface PagingParams {
  offset?: number
  sequence?: string
  all?: boolean
  limit: number
  need_count?: boolean
}
export interface PagingResult {
  has_more?: boolean
  count?: number
}
export interface SyncOptions {
  model_type?: Model_Type
  partition?: string
  sequence_min?: string
  sequence_max?: string
  limit: number
}
export enum Gender {
  GENDER_UNSPECIFIED = 0,
  GENDER_FEMALE = 1,
  GENDER_MALE = 2,
}
