// gm_api/example.json.js
// 以 merchandise.proto 为例
const json = {
  nested: {
    // 当前的包
    ceres: {
      nested: {
        // 当前模块
        merchandise: {
          // 模块的 options
          options: {},
          // 这下面有两种 service 和 message
          nested: {
            // service
            MerchandiseService: {
              methods: {
                // rpc
                GetCategory: {
                  requestType: 'GetCategoryRequest',
                  responseType: 'GetCategoryResponse',
                  options: {
                    // url
                    '(google.api.http).post': '/merchandise/GetCategory',
                    // 和后台约定的业务成功 code，比如下单合单code
                    '(common.options.annotation).codes': 'XXXXX',
                  },
                },
              },
            },
            // message 类型
            Sku: {
              // 字段
              fields: {
                sku_id: {
                  // 类型
                  type: 'uint64',
                  // 位置
                  id: 1,
                  // 各种各样的 validate，具体见 proto
                  options: {
                    '(validate.rules).message.required': true,
                  },
                },
                ssu_price: {
                  // 存在即 map 类型 { [key: string]: string }
                  keyType: 'uint64',
                  type: 'string',
                  id: 20,
                },
                customer_ids: {
                  // repeated 数组类型
                  rule: 'repeated',
                  type: 'uint64',
                  id: 21,
                },
                paging: {
                  // 来自 common.proto
                  type: 'common.PagingParams',
                  id: 5,
                  options: {
                    '(validate.rules).message.required': true,
                  },
                },
              },
              // 子类型
              nested: {
                // 同理类型
              },
            },
          },
        },
        // 会用到的其他模块
        common: {},
      },
    },
    // google 包。 同 validate
    google: {},
  },
}

export default json
