// gm_api/src/inventory/data.ts
import { getLocale } from './lng'

interface Enum_List_Item {
  text: string
  value: any
}

interface Enum_Map {
  [key: string]: string
}
export const list_ChangeType: Enum_List_Item[] = [
  { value: 1, text: getLocale('ChangeType__CHANGE_TYPE_ADD') },
  { value: 2, text: getLocale('ChangeType__CHANGE_TYPE_SUB') },
  { value: 3, text: getLocale('ChangeType__CHANGE_TYPE_SET') },
]
export const map_ChangeType: Enum_Map = {
  1: getLocale('ChangeType__CHANGE_TYPE_ADD'),
  2: getLocale('ChangeType__CHANGE_TYPE_SUB'),
  3: getLocale('ChangeType__CHANGE_TYPE_SET'),
}
export const list_OperateType: Enum_List_Item[] = [
  { value: 100, text: getLocale('OperateType__OPERATE_TYPE_PURCHASE_IN') },
  { value: 101, text: getLocale('OperateType__OPERATE_TYPE_PRODUCT_IN') },
  { value: 102, text: getLocale('OperateType__OPERATE_TYPE_MATERIAL_IN') },
  { value: 103, text: getLocale('OperateType__OPERATE_TYPE_REFUND_IN') },
  { value: 104, text: getLocale('OperateType__OPERATE_TYPE_TRANSFER_IN') },
  { value: 105, text: getLocale('OperateType__OPERATE_TYPE_OTHER_IN') },
  { value: 106, text: getLocale('OperateType__OPERATE_TYPE_SALE_OUT_ROLL_BACK') },
  { value: 111, text: getLocale('OperateType__OPERATE_TYPE_REFUND_OUT_ROLL_BACK') },
  { value: 109, text: getLocale('OperateType__OPERATE_TYPE_OTHER_OUT_ROLL_BACK') },
  { value: 112, text: getLocale('OperateType__OPERATE_TYPE_MATERIAL_OUT_ROLL_BACK') },
  { value: 113, text: getLocale('OperateType__OPERATE_TYPE_TURNOVER_LOAN') },
  { value: 114, text: getLocale('OperateType__OPERATE_TYPE_TURNOVER_REVERT') },
  { value: 501, text: getLocale('OperateType__OPERATE_TYPE_SALE_OUT') },
  { value: 502, text: getLocale('OperateType__OPERATE_TYPE_MATERIAL_OUT') },
  { value: 503, text: getLocale('OperateType__OPERATE_TYPE_REFUND_OUT') },
  { value: 504, text: getLocale('OperateType__OPERATE_TYPE_OTHER_OUT') },
  { value: 505, text: getLocale('OperateType__OPERATE_TYPE_PURCHASE_IN_ROLL_BACK') },
  { value: 507, text: getLocale('OperateType__OPERATE_TYPE_PRODUCT_IN_ROLL_BACK') },
  { value: 509, text: getLocale('OperateType__OPERATE_TYPE_MATERIAL_IN_ROLL_BACK') },
  { value: 511, text: getLocale('OperateType__OPERATE_TYPE_OTHER_IN_ROLL_BACK') },
  { value: 1000, text: getLocale('OperateType__OPERATE_TYPE_LOSS') },
  { value: 1001, text: getLocale('OperateType__OPERATE_TYPE_INCREASE') },
  { value: 1003, text: getLocale('OperateType__OPERATE_TYPE_CHECK') },
  { value: 900, text: getLocale('OperateType__OPERATE_TYPE_REPLACE_OUT') },
  { value: 901, text: getLocale('OperateType__OPERATE_TYPE_REPLACE_IN') },
  { value: 1100, text: getLocale('OperateType__OPERATE_TYPE_VIRTUAL_IN') },
  { value: 1101, text: getLocale('OperateType__OPERATE_TYPE_VIRTUAL_IN_ROLL_BACK') },
  { value: 1, text: getLocale('OperateType__OPERATE_TYPE_ALL') },
  { value: 2, text: getLocale('OperateType__BASE_UPDATE') },
]
export const map_OperateType: Enum_Map = {
  100: getLocale('OperateType__OPERATE_TYPE_PURCHASE_IN'),
  101: getLocale('OperateType__OPERATE_TYPE_PRODUCT_IN'),
  102: getLocale('OperateType__OPERATE_TYPE_MATERIAL_IN'),
  103: getLocale('OperateType__OPERATE_TYPE_REFUND_IN'),
  104: getLocale('OperateType__OPERATE_TYPE_TRANSFER_IN'),
  105: getLocale('OperateType__OPERATE_TYPE_OTHER_IN'),
  106: getLocale('OperateType__OPERATE_TYPE_SALE_OUT_ROLL_BACK'),
  111: getLocale('OperateType__OPERATE_TYPE_REFUND_OUT_ROLL_BACK'),
  109: getLocale('OperateType__OPERATE_TYPE_OTHER_OUT_ROLL_BACK'),
  112: getLocale('OperateType__OPERATE_TYPE_MATERIAL_OUT_ROLL_BACK'),
  113: getLocale('OperateType__OPERATE_TYPE_TURNOVER_LOAN'),
  114: getLocale('OperateType__OPERATE_TYPE_TURNOVER_REVERT'),
  501: getLocale('OperateType__OPERATE_TYPE_SALE_OUT'),
  502: getLocale('OperateType__OPERATE_TYPE_MATERIAL_OUT'),
  503: getLocale('OperateType__OPERATE_TYPE_REFUND_OUT'),
  504: getLocale('OperateType__OPERATE_TYPE_OTHER_OUT'),
  505: getLocale('OperateType__OPERATE_TYPE_PURCHASE_IN_ROLL_BACK'),
  507: getLocale('OperateType__OPERATE_TYPE_PRODUCT_IN_ROLL_BACK'),
  509: getLocale('OperateType__OPERATE_TYPE_MATERIAL_IN_ROLL_BACK'),
  511: getLocale('OperateType__OPERATE_TYPE_OTHER_IN_ROLL_BACK'),
  1000: getLocale('OperateType__OPERATE_TYPE_LOSS'),
  1001: getLocale('OperateType__OPERATE_TYPE_INCREASE'),
  1003: getLocale('OperateType__OPERATE_TYPE_CHECK'),
  900: getLocale('OperateType__OPERATE_TYPE_REPLACE_OUT'),
  901: getLocale('OperateType__OPERATE_TYPE_REPLACE_IN'),
  1100: getLocale('OperateType__OPERATE_TYPE_VIRTUAL_IN'),
  1101: getLocale('OperateType__OPERATE_TYPE_VIRTUAL_IN_ROLL_BACK'),
  1: getLocale('OperateType__OPERATE_TYPE_ALL'),
  2: getLocale('OperateType__BASE_UPDATE'),
}
export const list_Status_Code: Enum_List_Item[] = [
  { value: 20770001, text: getLocale('Status_Code__STOCK_ERROR_STATUS_UNKNOWN') },
  { value: 20770002, text: getLocale('Status_Code__SHEET_STATUS_INVALID') },
  { value: 20770003, text: getLocale('Status_Code__CROSS_UPDATE') },
  { value: 20770004, text: getLocale('Status_Code__AUTHENTICATION_ERROR') },
  { value: 20770005, text: getLocale('Status_Code__DUPLICATE_SHELF_NAME') },
  { value: 20770006, text: getLocale('Status_Code__LACK_OF_STOCK') },
  { value: 20770007, text: getLocale('Status_Code__FAIL_GET_ID') },
  { value: 20770008, text: getLocale('Status_Code__MERCHANDISE_NOT_FOUND') },
  { value: 20770009, text: getLocale('Status_Code__REVERT_TURNOVER_RESULT_LESS_THTAN_ZERO') },
  { value: 20770010, text: getLocale('Status_Code__REDISLOCK_BUSY') },
  { value: 20770011, text: getLocale('Status_Code__REDISLOCK_ERROR') },
  { value: 20770012, text: getLocale('Status_Code__COST_ALLOCATION_MONEY_ZERO') },
  { value: 20770013, text: getLocale('Status_Code__CANCEL_USED_BATCH') },
  { value: 20770014, text: getLocale('Status_Code__STOCK_DATA_ERROR') },
  { value: 20770015, text: getLocale('Status_Code__STOCK_SHEET_TOTAL_PRICE_NEGATIVE') },
  { value: 20770016, text: getLocale('Status_Code__PARENT_SHELF_STOCK_OPERATE') },
  { value: 20770017, text: getLocale('Status_Code__SHELF_DELETED_OR_NOT_EXISTS') },
  { value: 20770018, text: getLocale('Status_Code__BATCH_DELETED_OR_NOT_EXISTS') },
  { value: 20770019, text: getLocale('Status_Code__UPDATE_FUTURE_BATCH') },
  { value: 20770020, text: getLocale('Status_Code__FUTURE_OPERATE_ERROR') },
  { value: 20770021, text: getLocale('Status_Code__CHECK_STOCK_DISTINCT_SIGN') },
  { value: 20770022, text: getLocale('Status_Code__VIRTUAL_BATCH_HAD_BEEN_REPLACED') },
  { value: 20770023, text: getLocale('Status_Code__REPLECE_STOCK_SUBMIT_TIME_TOO_LATE') },
  { value: 20770024, text: getLocale('Status_Code__REVERT_NO_BATCH_MATCH') },
  { value: 20770025, text: getLocale('Status_Code__CANCEL_PAID_SHEET') },
]
export const map_Status_Code: Enum_Map = {
  20770001: getLocale('Status_Code__STOCK_ERROR_STATUS_UNKNOWN'),
  20770002: getLocale('Status_Code__SHEET_STATUS_INVALID'),
  20770003: getLocale('Status_Code__CROSS_UPDATE'),
  20770004: getLocale('Status_Code__AUTHENTICATION_ERROR'),
  20770005: getLocale('Status_Code__DUPLICATE_SHELF_NAME'),
  20770006: getLocale('Status_Code__LACK_OF_STOCK'),
  20770007: getLocale('Status_Code__FAIL_GET_ID'),
  20770008: getLocale('Status_Code__MERCHANDISE_NOT_FOUND'),
  20770009: getLocale('Status_Code__REVERT_TURNOVER_RESULT_LESS_THTAN_ZERO'),
  20770010: getLocale('Status_Code__REDISLOCK_BUSY'),
  20770011: getLocale('Status_Code__REDISLOCK_ERROR'),
  20770012: getLocale('Status_Code__COST_ALLOCATION_MONEY_ZERO'),
  20770013: getLocale('Status_Code__CANCEL_USED_BATCH'),
  20770014: getLocale('Status_Code__STOCK_DATA_ERROR'),
  20770015: getLocale('Status_Code__STOCK_SHEET_TOTAL_PRICE_NEGATIVE'),
  20770016: getLocale('Status_Code__PARENT_SHELF_STOCK_OPERATE'),
  20770017: getLocale('Status_Code__SHELF_DELETED_OR_NOT_EXISTS'),
  20770018: getLocale('Status_Code__BATCH_DELETED_OR_NOT_EXISTS'),
  20770019: getLocale('Status_Code__UPDATE_FUTURE_BATCH'),
  20770020: getLocale('Status_Code__FUTURE_OPERATE_ERROR'),
  20770021: getLocale('Status_Code__CHECK_STOCK_DISTINCT_SIGN'),
  20770022: getLocale('Status_Code__VIRTUAL_BATCH_HAD_BEEN_REPLACED'),
  20770023: getLocale('Status_Code__REPLECE_STOCK_SUBMIT_TIME_TOO_LATE'),
  20770024: getLocale('Status_Code__REVERT_NO_BATCH_MATCH'),
  20770025: getLocale('Status_Code__CANCEL_PAID_SHEET'),
}
export const list_Batch_BatchType: Enum_List_Item[] = [
  { value: 1, text: getLocale('Batch_BatchType__BATCH_TYPE_CONST') },
  { value: 2, text: getLocale('Batch_BatchType__BATCH_TYPE_TMP') },
]
export const map_Batch_BatchType: Enum_Map = {
  1: getLocale('Batch_BatchType__BATCH_TYPE_CONST'),
  2: getLocale('Batch_BatchType__BATCH_TYPE_TMP'),
}
export const list_PendingType: Enum_List_Item[] = [
  { value: 1, text: getLocale('PendingType__PENDING_TYPE_PRODUCT_IN') },
  { value: 2, text: getLocale('PendingType__PENDING_TYPE_MATERIAL_OUT') },
  { value: 3, text: getLocale('PendingType__PENDING_TYPE_MATERIAL_IN') },
  { value: 4, text: getLocale('PendingType__PENDING_TYPE_PURCHASE_IN') },
  { value: 5, text: getLocale('PendingType__PENDING_TYPE_ORDER') },
  { value: 6, text: getLocale('PendingType__PENDING_TYPE_VIRTUAL') },
]
export const map_PendingType: Enum_Map = {
  1: getLocale('PendingType__PENDING_TYPE_PRODUCT_IN'),
  2: getLocale('PendingType__PENDING_TYPE_MATERIAL_OUT'),
  3: getLocale('PendingType__PENDING_TYPE_MATERIAL_IN'),
  4: getLocale('PendingType__PENDING_TYPE_PURCHASE_IN'),
  5: getLocale('PendingType__PENDING_TYPE_ORDER'),
  6: getLocale('PendingType__PENDING_TYPE_VIRTUAL'),
}
export const list_SkuUnitStock_Config: Enum_List_Item[] = [
  { value: 1, text: getLocale('SkuUnitStock_Config__SALE_STOCK_NO_CONFIG') },
  { value: 2, text: getLocale('SkuUnitStock_Config__SALE_STOCK_AVAILABLE_STOCK') },
  { value: 3, text: getLocale('SkuUnitStock_Config__SALE_STOCK_VIRTUAL_STOCK') },
]
export const map_SkuUnitStock_Config: Enum_Map = {
  1: getLocale('SkuUnitStock_Config__SALE_STOCK_NO_CONFIG'),
  2: getLocale('SkuUnitStock_Config__SALE_STOCK_AVAILABLE_STOCK'),
  3: getLocale('SkuUnitStock_Config__SALE_STOCK_VIRTUAL_STOCK'),
}
export const list_SkuStock_Status: Enum_List_Item[] = [
  { value: 256, text: getLocale('SkuStock_Status__STATUS_IS_REMAINING') },
  { value: 512, text: getLocale('SkuStock_Status__STATUS_IS_VIRTUAL_STOCK') },
]
export const map_SkuStock_Status: Enum_Map = {
  256: getLocale('SkuStock_Status__STATUS_IS_REMAINING'),
  512: getLocale('SkuStock_Status__STATUS_IS_VIRTUAL_STOCK'),
}
export const list_CostAllocation_Type: Enum_List_Item[] = [
  { value: 1, text: getLocale('CostAllocation_Type__TYPE_PRICE') },
  { value: 2, text: getLocale('CostAllocation_Type__TYPE_QUANTITY') },
]
export const map_CostAllocation_Type: Enum_Map = {
  1: getLocale('CostAllocation_Type__TYPE_PRICE'),
  2: getLocale('CostAllocation_Type__TYPE_QUANTITY'),
}
export const list_CostAllocation_Reason: Enum_List_Item[] = [
  { value: 1, text: getLocale('CostAllocation_Reason__REASON_TRANSPORT') },
  { value: 2, text: getLocale('CostAllocation_Reason__REASON_OTHER') },
]
export const map_CostAllocation_Reason: Enum_Map = {
  1: getLocale('CostAllocation_Reason__REASON_TRANSPORT'),
  2: getLocale('CostAllocation_Reason__REASON_OTHER'),
}
export const list_CostAllocation_MoneyType: Enum_List_Item[] = [
  { value: 1, text: getLocale('CostAllocation_MoneyType__MONEY_ADD') },
  { value: 2, text: getLocale('CostAllocation_MoneyType__MONEY_SUB') },
]
export const map_CostAllocation_MoneyType: Enum_Map = {
  1: getLocale('CostAllocation_MoneyType__MONEY_ADD'),
  2: getLocale('CostAllocation_MoneyType__MONEY_SUB'),
}
export const list_StockSheet_SheetType: Enum_List_Item[] = [
  { value: 100, text: getLocale('StockSheet_SheetType__SHEET_TYPE_PURCHASE_IN') },
  { value: 110, text: getLocale('StockSheet_SheetType__SHEET_TYPE_PRODUCT_IN') },
  { value: 120, text: getLocale('StockSheet_SheetType__SHEET_TYPE_MATERIAL_IN') },
  { value: 130, text: getLocale('StockSheet_SheetType__SHEET_TYPE_REFUND_IN') },
  { value: 140, text: getLocale('StockSheet_SheetType__SHEET_TYPE_TRANSFER_IN') },
  { value: 150, text: getLocale('StockSheet_SheetType__SHEET_TYPE_OTHER_IN') },
  { value: 300, text: getLocale('StockSheet_SheetType__SHEET_TYPE_SALE_OUT') },
  { value: 310, text: getLocale('StockSheet_SheetType__SHEET_TYPE_MATERIAL_OUT') },
  { value: 320, text: getLocale('StockSheet_SheetType__SHEET_TYPE_REFUND_OUT') },
  { value: 330, text: getLocale('StockSheet_SheetType__SHEET_TYPE_OTHER_OUT') },
  { value: 600, text: getLocale('StockSheet_SheetType__SHEET_TYPE_INVENTORY') },
  { value: 400, text: getLocale('StockSheet_SheetType__SHEET_TYPE_TRANSFER') },
  { value: 700, text: getLocale('StockSheet_SheetType__SHEET_TYPE_TURNOVER_LOAN') },
  { value: 800, text: getLocale('StockSheet_SheetType__SHEET_TYPE_TURNOVER_REVERT') },
  { value: 900, text: getLocale('StockSheet_SheetType__SHEET_TYPE_REPLACE') },
]
export const map_StockSheet_SheetType: Enum_Map = {
  100: getLocale('StockSheet_SheetType__SHEET_TYPE_PURCHASE_IN'),
  110: getLocale('StockSheet_SheetType__SHEET_TYPE_PRODUCT_IN'),
  120: getLocale('StockSheet_SheetType__SHEET_TYPE_MATERIAL_IN'),
  130: getLocale('StockSheet_SheetType__SHEET_TYPE_REFUND_IN'),
  140: getLocale('StockSheet_SheetType__SHEET_TYPE_TRANSFER_IN'),
  150: getLocale('StockSheet_SheetType__SHEET_TYPE_OTHER_IN'),
  300: getLocale('StockSheet_SheetType__SHEET_TYPE_SALE_OUT'),
  310: getLocale('StockSheet_SheetType__SHEET_TYPE_MATERIAL_OUT'),
  320: getLocale('StockSheet_SheetType__SHEET_TYPE_REFUND_OUT'),
  330: getLocale('StockSheet_SheetType__SHEET_TYPE_OTHER_OUT'),
  600: getLocale('StockSheet_SheetType__SHEET_TYPE_INVENTORY'),
  400: getLocale('StockSheet_SheetType__SHEET_TYPE_TRANSFER'),
  700: getLocale('StockSheet_SheetType__SHEET_TYPE_TURNOVER_LOAN'),
  800: getLocale('StockSheet_SheetType__SHEET_TYPE_TURNOVER_REVERT'),
  900: getLocale('StockSheet_SheetType__SHEET_TYPE_REPLACE'),
}
export const list_StockSheet_SheetStatus: Enum_List_Item[] = [
  { value: 1, text: getLocale('StockSheet_SheetStatus__SHEET_STATUS_NOT_SUBMITTED') },
  { value: 2, text: getLocale('StockSheet_SheetStatus__SHEET_STATUS_SUBMITTED') },
  { value: 3, text: getLocale('StockSheet_SheetStatus__SHEET_STATUS_NOT_APPROVED') },
  { value: 4, text: getLocale('StockSheet_SheetStatus__SHEET_STATUS_APPROVED') },
  { value: 5, text: getLocale('StockSheet_SheetStatus__SHEET_STATUS_CANCEL_APPROVAL') },
  { value: 6, text: getLocale('StockSheet_SheetStatus__SHEET_STATUS_DELETED') },
  { value: 7, text: getLocale('StockSheet_SheetStatus__SHEET_STATUS_PAID') },
]
export const map_StockSheet_SheetStatus: Enum_Map = {
  1: getLocale('StockSheet_SheetStatus__SHEET_STATUS_NOT_SUBMITTED'),
  2: getLocale('StockSheet_SheetStatus__SHEET_STATUS_SUBMITTED'),
  3: getLocale('StockSheet_SheetStatus__SHEET_STATUS_NOT_APPROVED'),
  4: getLocale('StockSheet_SheetStatus__SHEET_STATUS_APPROVED'),
  5: getLocale('StockSheet_SheetStatus__SHEET_STATUS_CANCEL_APPROVAL'),
  6: getLocale('StockSheet_SheetStatus__SHEET_STATUS_DELETED'),
  7: getLocale('StockSheet_SheetStatus__SHEET_STATUS_PAID'),
}
export const list_StockSheet_TargetType: Enum_List_Item[] = [
  { value: 1, text: getLocale('StockSheet_TargetType__SHEET_TARGET_TYPE_CUSTOMER') },
  { value: 2, text: getLocale('StockSheet_TargetType__SHEET_TARGET_TYPE_SUPPLIER') },
  { value: 3, text: getLocale('StockSheet_TargetType__SHEET_TARGET_TYPE_OTHER') },
]
export const map_StockSheet_TargetType: Enum_Map = {
  1: getLocale('StockSheet_TargetType__SHEET_TARGET_TYPE_CUSTOMER'),
  2: getLocale('StockSheet_TargetType__SHEET_TARGET_TYPE_SUPPLIER'),
  3: getLocale('StockSheet_TargetType__SHEET_TARGET_TYPE_OTHER'),
}
export const list_StockSheet_Status: Enum_List_Item[] = [
  { value: 256, text: getLocale('StockSheet_Status__STATUS_REPLACE') },
  { value: 512, text: getLocale('StockSheet_Status__STATUS_PRINTED') },
]
export const map_StockSheet_Status: Enum_Map = {
  256: getLocale('StockSheet_Status__STATUS_REPLACE'),
  512: getLocale('StockSheet_Status__STATUS_PRINTED'),
}
export const list_StockSheet_PayStatus: Enum_List_Item[] = [
  { value: 1, text: getLocale('StockSheet_PayStatus__PAY_STATUS_NOT_PAID') },
  { value: 2, text: getLocale('StockSheet_PayStatus__PAY_STATUS_READY_TO_PAY') },
  { value: 3, text: getLocale('StockSheet_PayStatus__PAY_STATUS_PAID') },
]
export const map_StockSheet_PayStatus: Enum_Map = {
  1: getLocale('StockSheet_PayStatus__PAY_STATUS_NOT_PAID'),
  2: getLocale('StockSheet_PayStatus__PAY_STATUS_READY_TO_PAY'),
  3: getLocale('StockSheet_PayStatus__PAY_STATUS_PAID'),
}
export const list_StockSheet_RelatedSheetType: Enum_List_Item[] = [
  { value: 2, text: getLocale('StockSheet_RelatedSheetType__RELATED_SHEET_TYPE_ORDER') },
  { value: 3, text: getLocale('StockSheet_RelatedSheetType__RELATED_SHEET_TYPE_PURCHASE_SHEET') },
  { value: 4, text: getLocale('StockSheet_RelatedSheetType__RELATED_SHEET_TYPE_STOCK_SHEET') },
  { value: 5, text: getLocale('StockSheet_RelatedSheetType__RELATED_SHEET_TYPE_PRODUCTION') },
]
export const map_StockSheet_RelatedSheetType: Enum_Map = {
  2: getLocale('StockSheet_RelatedSheetType__RELATED_SHEET_TYPE_ORDER'),
  3: getLocale('StockSheet_RelatedSheetType__RELATED_SHEET_TYPE_PURCHASE_SHEET'),
  4: getLocale('StockSheet_RelatedSheetType__RELATED_SHEET_TYPE_STOCK_SHEET'),
  5: getLocale('StockSheet_RelatedSheetType__RELATED_SHEET_TYPE_PRODUCTION'),
}
export const list_AdjustSheet_AdjustSheetStatus: Enum_List_Item[] = [
  { value: 1, text: getLocale('AdjustSheet_AdjustSheetStatus__ADJUST_SHEET_STATUS_NOT_SUBMITTED') },
  { value: 2, text: getLocale('AdjustSheet_AdjustSheetStatus__ADJUST_SHEET_STATUS_ADJUSTING') },
  { value: 3, text: getLocale('AdjustSheet_AdjustSheetStatus__ADJUST_SHEET_STATUS_DONE') },
  { value: 4, text: getLocale('AdjustSheet_AdjustSheetStatus__ADJUST_SHEET_STATUS_FAIL') },
]
export const map_AdjustSheet_AdjustSheetStatus: Enum_Map = {
  1: getLocale('AdjustSheet_AdjustSheetStatus__ADJUST_SHEET_STATUS_NOT_SUBMITTED'),
  2: getLocale('AdjustSheet_AdjustSheetStatus__ADJUST_SHEET_STATUS_ADJUSTING'),
  3: getLocale('AdjustSheet_AdjustSheetStatus__ADJUST_SHEET_STATUS_DONE'),
  4: getLocale('AdjustSheet_AdjustSheetStatus__ADJUST_SHEET_STATUS_FAIL'),
}
export const list_TimeType: Enum_List_Item[] = [
  { value: 1, text: getLocale('TimeType__TIME_TYPE_CREATE') },
  { value: 2, text: getLocale('TimeType__TIME_TYPE_SUBMIT') },
  { value: 3, text: getLocale('TimeType__TIME_TYPE_COMMIT') },
  { value: 11, text: getLocale('TimeType__TIME_TYPE_ORDER_RECEIVE') },
]
export const map_TimeType: Enum_Map = {
  1: getLocale('TimeType__TIME_TYPE_CREATE'),
  2: getLocale('TimeType__TIME_TYPE_SUBMIT'),
  3: getLocale('TimeType__TIME_TYPE_COMMIT'),
  11: getLocale('TimeType__TIME_TYPE_ORDER_RECEIVE'),
}
export const list_RequestDetail: Enum_List_Item[] = [
  { value: 256, text: getLocale('RequestDetail__REQUEST_DETAIL_SKUS') },
  { value: 512, text: getLocale('RequestDetail__REQUEST_DETAIL_SHELVES') },
]
export const map_RequestDetail: Enum_Map = {
  256: getLocale('RequestDetail__REQUEST_DETAIL_SKUS'),
  512: getLocale('RequestDetail__REQUEST_DETAIL_SHELVES'),
}
