// gm_api/src/inventory/methods.ts
import { Request } from '../util/request'
import {
  CreateStockSheetRequest, CreateStockSheetResponse,
  UpdateStockSheetRequest, UpdateStockSheetResponse,
  CreateReplaceStockSheetRequest, CreateReplaceStockSheetResponse,
  DeleteStockSheetRequest, DeleteStockSheetResponse,
  GetStockSheetRequest, GetStockSheetResponse,
  ListStockSheetRequest, ListStockSheetResponse,
  BatchUpdateStockSheetPayStatusRequest, BatchUpdateStockSheetPayStatusResponse,
  CleanStockSheetRequest, CleanStockSheetResponse,
  BatchSyncTurnoverLoansRequest, BatchSyncTurnoverLoansResponse,
  GetManyStockSheetRequest, GetManyStockSheetResponse,
  BatchUpdateStockSheetRequest, BatchUpdateStockSheetResponse,
  SyncStockSheetRequest, SyncStockSheetResponse,
  CreateZeroCheckSheetRequest, CreateZeroCheckSheetResponse,
  CreateBatchRequest, CreateBatchResponse,
  UpdateBatchRequest, UpdateBatchResponse,
  DeleteBatchRequest, DeleteBatchResponse,
  GetBatchRequest, GetBatchResponse,
  ListBatchRequest, ListBatchResponse,
  CleanBatchRequest, CleanBatchResponse,
  CreateShelfRequest, CreateShelfResponse,
  UpdateShelfRequest, UpdateShelfResponse,
  DeleteShelfRequest, DeleteShelfResponse,
  GetShelfRequest, GetShelfResponse,
  ListShelfRequest, ListShelfResponse,
  CleanShelfRequest, CleanShelfResponse,
  CreateBatchLogRequest, CreateBatchLogResponse,
  UpdateBatchLogRequest, UpdateBatchLogResponse,
  DeleteBatchLogRequest, DeleteBatchLogResponse,
  GetBatchLogRequest, GetBatchLogResponse,
  ListBatchLogRequest, ListBatchLogResponse,
  CleanBatchLogRequest, CleanBatchLogResponse,
  CreateStockLogRequest, CreateStockLogResponse,
  UpdateStockLogRequest, UpdateStockLogResponse,
  DeleteStockLogRequest, DeleteStockLogResponse,
  CleanStockLogRequest, CleanStockLogResponse,
  GetStockLogRequest, GetStockLogResponse,
  ListStockLogRequest, ListStockLogResponse,
  ExportStockLogRequest, ExportStockLogResponse,
  ExportStockSheetRequest, ExportStockSheetResponse,
  CreateSkuStockRequest, CreateSkuStockResponse,
  UpdateSkuStockRequest, UpdateSkuStockResponse,
  DeleteSkuStockRequest, DeleteSkuStockResponse,
  ListSkuStockRequest, ListSkuStockResponse,
  ListVirtualStockRequest, ListVirtualStockResponse,
  GetSkuStockRequest, GetSkuStockResponse,
  ExportSkuStockRequest, ExportSkuStockResponse,
  GetManySkuStockRequest, GetManySkuStockResponse,
  CleanSkuStockRequest, CleanSkuStockResponse,
  CreateSkuUnitStockRequest, CreateSkuUnitStockResponse,
  UpdateSkuUnitStockRequest, UpdateSkuUnitStockResponse,
  DeleteSkuUnitStockRequest, DeleteSkuUnitStockResponse,
  ListSkuUnitStockRequest, ListSkuUnitStockResponse,
  GetSkuUnitStockRequest, GetSkuUnitStockResponse,
  CleanSkuUnitStockRequest, CleanSkuUnitStockResponse,
  UpdatePendingStockRequest, UpdatePendingStockResponse,
  ListPendingStockRequest, ListPendingStockResponse,
  CreateCustomerTurnoverRequest, CreateCustomerTurnoverResponse,
  UpdateCustomerTurnoverRequest, UpdateCustomerTurnoverResponse,
  DeleteCustomerTurnoverRequest, DeleteCustomerTurnoverResponse,
  GetCustomerTurnoverRequest, GetCustomerTurnoverResponse,
  ListCustomerTurnoverRequest, ListCustomerTurnoverResponse,
  ExportCustomerTurnoverRequest, ExportCustomerTurnoverResponse,
  CleanCustomerTurnoverRequest, CleanCustomerTurnoverResponse,
  UpdateOrderStockRequest, UpdateOrderStockResponse,
  UpdateSkuUnitStockConfigRequest, UpdateSkuUnitStockConfigResponse,
  CreateAdjustSheetRequest, CreateAdjustSheetResponse,
  UpdateAdjustSheetRequest, UpdateAdjustSheetResponse,
  ListAdjustSheetRequest, ListAdjustSheetResponse,
  DeleteAdjustSheetRequest, DeleteAdjustSheetResponse,
  GetAdjustSheetRequest, GetAdjustSheetResponse,
  CleanAdjustSheetRequest, CleanAdjustSheetResponse,
  AsyncCommitAdjustSheetRequest, AsyncCommitAdjustSheetResponse,
  CheckOutStockRequest, CheckOutStockResponse,
  CreateReplaceBatchRequest, CreateReplaceBatchResponse,
  ListProductionTaskCostRequest, ListProductionTaskCostResponse,
} from './types'
export const CreateStockSheet = (r?: CreateStockSheetRequest, codes?: number[]) =>
  Request<CreateStockSheetResponse>('/ceres/inventory/InventoryService/CreateStockSheet')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateStockSheet.url = '/ceres/inventory/InventoryService/CreateStockSheet'
export const UpdateStockSheet = (r?: UpdateStockSheetRequest, codes?: number[]) =>
  Request<UpdateStockSheetResponse>('/ceres/inventory/InventoryService/UpdateStockSheet')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateStockSheet.url = '/ceres/inventory/InventoryService/UpdateStockSheet'
export const CreateReplaceStockSheet = (r?: CreateReplaceStockSheetRequest, codes?: number[]) =>
  Request<CreateReplaceStockSheetResponse>('/ceres/inventory/InventoryService/CreateReplaceStockSheet')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateReplaceStockSheet.url = '/ceres/inventory/InventoryService/CreateReplaceStockSheet'
export const DeleteStockSheet = (r?: DeleteStockSheetRequest, codes?: number[]) =>
  Request<DeleteStockSheetResponse>('/ceres/inventory/InventoryService/DeleteStockSheet')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteStockSheet.url = '/ceres/inventory/InventoryService/DeleteStockSheet'
export const GetStockSheet = (r?: GetStockSheetRequest, codes?: number[]) =>
  Request<GetStockSheetResponse>('/ceres/inventory/InventoryService/GetStockSheet')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetStockSheet.url = '/ceres/inventory/InventoryService/GetStockSheet'
export const ListStockSheet = (r?: ListStockSheetRequest, codes?: number[]) =>
  Request<ListStockSheetResponse>('/ceres/inventory/InventoryService/ListStockSheet')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListStockSheet.url = '/ceres/inventory/InventoryService/ListStockSheet'
export const BatchUpdateStockSheetPayStatus = (r?: BatchUpdateStockSheetPayStatusRequest, codes?: number[]) =>
  Request<BatchUpdateStockSheetPayStatusResponse>('/ceres/inventory/InventoryService/BatchUpdateStockSheetPayStatus')
    .code([...(codes || [])])
    .data(r || {})
    .run()
BatchUpdateStockSheetPayStatus.url = '/ceres/inventory/InventoryService/BatchUpdateStockSheetPayStatus'

export const BatchSyncTurnoverLoans = (r?: BatchSyncTurnoverLoansRequest, codes?: number[]) =>
  Request<BatchSyncTurnoverLoansResponse>('/ceres/inventory/InventoryService/BatchSyncTurnoverLoans')
    .code([...(codes || [])])
    .data(r || {})
    .run()
BatchSyncTurnoverLoans.url = '/ceres/inventory/InventoryService/BatchSyncTurnoverLoans'
export const GetManyStockSheet = (r?: GetManyStockSheetRequest, codes?: number[]) =>
  Request<GetManyStockSheetResponse>('/ceres/inventory/InventoryService/GetManyStockSheet')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetManyStockSheet.url = '/ceres/inventory/InventoryService/GetManyStockSheet'
export const BatchUpdateStockSheet = (r?: BatchUpdateStockSheetRequest, codes?: number[]) =>
  Request<BatchUpdateStockSheetResponse>('/ceres/inventory/InventoryService/BatchUpdateStockSheet')
    .code([...(codes || [])])
    .data(r || {})
    .run()
BatchUpdateStockSheet.url = '/ceres/inventory/InventoryService/BatchUpdateStockSheet'
export const SyncStockSheet = (r?: SyncStockSheetRequest, codes?: number[]) =>
  Request<SyncStockSheetResponse>('/ceres/inventory/InventoryService/SyncStockSheet')
    .code([...(codes || [])])
    .data(r || {})
    .run()
SyncStockSheet.url = '/ceres/inventory/InventoryService/SyncStockSheet'
export const CreateZeroCheckSheet = (r?: CreateZeroCheckSheetRequest, codes?: number[]) =>
  Request<CreateZeroCheckSheetResponse>('/ceres/inventory/InventoryService/CreateZeroCheckSheet')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateZeroCheckSheet.url = '/ceres/inventory/InventoryService/CreateZeroCheckSheet'
export const CreateBatch = (r?: CreateBatchRequest, codes?: number[]) =>
  Request<CreateBatchResponse>('/ceres/inventory/InventoryService/CreateBatch')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateBatch.url = '/ceres/inventory/InventoryService/CreateBatch'
export const UpdateBatch = (r?: UpdateBatchRequest, codes?: number[]) =>
  Request<UpdateBatchResponse>('/ceres/inventory/InventoryService/UpdateBatch')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateBatch.url = '/ceres/inventory/InventoryService/UpdateBatch'
export const DeleteBatch = (r?: DeleteBatchRequest, codes?: number[]) =>
  Request<DeleteBatchResponse>('/ceres/inventory/InventoryService/DeleteBatch')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteBatch.url = '/ceres/inventory/InventoryService/DeleteBatch'
export const GetBatch = (r?: GetBatchRequest, codes?: number[]) =>
  Request<GetBatchResponse>('/ceres/inventory/InventoryService/GetBatch')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetBatch.url = '/ceres/inventory/InventoryService/GetBatch'
export const ListBatch = (r?: ListBatchRequest, codes?: number[]) =>
  Request<ListBatchResponse>('/ceres/inventory/InventoryService/ListBatch')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListBatch.url = '/ceres/inventory/InventoryService/ListBatch'

export const CreateShelf = (r?: CreateShelfRequest, codes?: number[]) =>
  Request<CreateShelfResponse>('/ceres/inventory/InventoryService/CreateShelf')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateShelf.url = '/ceres/inventory/InventoryService/CreateShelf'
export const UpdateShelf = (r?: UpdateShelfRequest, codes?: number[]) =>
  Request<UpdateShelfResponse>('/ceres/inventory/InventoryService/UpdateShelf')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateShelf.url = '/ceres/inventory/InventoryService/UpdateShelf'
export const DeleteShelf = (r?: DeleteShelfRequest, codes?: number[]) =>
  Request<DeleteShelfResponse>('/ceres/inventory/InventoryService/DeleteShelf')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteShelf.url = '/ceres/inventory/InventoryService/DeleteShelf'
export const GetShelf = (r?: GetShelfRequest, codes?: number[]) =>
  Request<GetShelfResponse>('/ceres/inventory/InventoryService/GetShelf')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetShelf.url = '/ceres/inventory/InventoryService/GetShelf'
export const ListShelf = (r?: ListShelfRequest, codes?: number[]) =>
  Request<ListShelfResponse>('/ceres/inventory/InventoryService/ListShelf')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListShelf.url = '/ceres/inventory/InventoryService/ListShelf'

export const CreateBatchLog = (r?: CreateBatchLogRequest, codes?: number[]) =>
  Request<CreateBatchLogResponse>('/ceres/inventory/InventoryService/CreateBatchLog')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateBatchLog.url = '/ceres/inventory/InventoryService/CreateBatchLog'
export const UpdateBatchLog = (r?: UpdateBatchLogRequest, codes?: number[]) =>
  Request<UpdateBatchLogResponse>('/ceres/inventory/InventoryService/UpdateBatchLog')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateBatchLog.url = '/ceres/inventory/InventoryService/UpdateBatchLog'
export const DeleteBatchLog = (r?: DeleteBatchLogRequest, codes?: number[]) =>
  Request<DeleteBatchLogResponse>('/ceres/inventory/InventoryService/DeleteBatchLog')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteBatchLog.url = '/ceres/inventory/InventoryService/DeleteBatchLog'
export const GetBatchLog = (r?: GetBatchLogRequest, codes?: number[]) =>
  Request<GetBatchLogResponse>('/ceres/inventory/InventoryService/GetBatchLog')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetBatchLog.url = '/ceres/inventory/InventoryService/GetBatchLog'
export const ListBatchLog = (r?: ListBatchLogRequest, codes?: number[]) =>
  Request<ListBatchLogResponse>('/ceres/inventory/InventoryService/ListBatchLog')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListBatchLog.url = '/ceres/inventory/InventoryService/ListBatchLog'

export const CreateStockLog = (r?: CreateStockLogRequest, codes?: number[]) =>
  Request<CreateStockLogResponse>('/ceres/inventory/InventoryService/CreateStockLog')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateStockLog.url = '/ceres/inventory/InventoryService/CreateStockLog'
export const UpdateStockLog = (r?: UpdateStockLogRequest, codes?: number[]) =>
  Request<UpdateStockLogResponse>('/ceres/inventory/InventoryService/UpdateStockLog')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateStockLog.url = '/ceres/inventory/InventoryService/UpdateStockLog'
export const DeleteStockLog = (r?: DeleteStockLogRequest, codes?: number[]) =>
  Request<DeleteStockLogResponse>('/ceres/inventory/InventoryService/DeleteStockLog')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteStockLog.url = '/ceres/inventory/InventoryService/DeleteStockLog'

export const GetStockLog = (r?: GetStockLogRequest, codes?: number[]) =>
  Request<GetStockLogResponse>('/ceres/inventory/InventoryService/GetStockLog')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetStockLog.url = '/ceres/inventory/InventoryService/GetStockLog'
export const ListStockLog = (r?: ListStockLogRequest, codes?: number[]) =>
  Request<ListStockLogResponse>('/ceres/inventory/InventoryService/ListStockLog')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListStockLog.url = '/ceres/inventory/InventoryService/ListStockLog'
export const ExportStockLog = (r?: ExportStockLogRequest, codes?: number[]) =>
  Request<ExportStockLogResponse>('/ceres/inventory/InventoryService/ExportStockLog')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ExportStockLog.url = '/ceres/inventory/InventoryService/ExportStockLog'
export const ExportStockSheet = (r?: ExportStockSheetRequest, codes?: number[]) =>
  Request<ExportStockSheetResponse>('/ceres/inventory/InventoryService/ExportStockSheet')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ExportStockSheet.url = '/ceres/inventory/InventoryService/ExportStockSheet'
export const CreateSkuStock = (r?: CreateSkuStockRequest, codes?: number[]) =>
  Request<CreateSkuStockResponse>('/ceres/inventory/InventoryService/CreateSkuStock')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateSkuStock.url = '/ceres/inventory/InventoryService/CreateSkuStock'
export const UpdateSkuStock = (r?: UpdateSkuStockRequest, codes?: number[]) =>
  Request<UpdateSkuStockResponse>('/ceres/inventory/InventoryService/UpdateSkuStock')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateSkuStock.url = '/ceres/inventory/InventoryService/UpdateSkuStock'
export const DeleteSkuStock = (r?: DeleteSkuStockRequest, codes?: number[]) =>
  Request<DeleteSkuStockResponse>('/ceres/inventory/InventoryService/DeleteSkuStock')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteSkuStock.url = '/ceres/inventory/InventoryService/DeleteSkuStock'
export const ListSkuStock = (r?: ListSkuStockRequest, codes?: number[]) =>
  Request<ListSkuStockResponse>('/ceres/inventory/InventoryService/ListSkuStock')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListSkuStock.url = '/ceres/inventory/InventoryService/ListSkuStock'
export const ListVirtualStock = (r?: ListVirtualStockRequest, codes?: number[]) =>
  Request<ListVirtualStockResponse>('/ceres/inventory/InventoryService/ListVirtualStock')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListVirtualStock.url = '/ceres/inventory/InventoryService/ListVirtualStock'
export const GetSkuStock = (r?: GetSkuStockRequest, codes?: number[]) =>
  Request<GetSkuStockResponse>('/ceres/inventory/InventoryService/GetSkuStock')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetSkuStock.url = '/ceres/inventory/InventoryService/GetSkuStock'
export const ExportSkuStock = (r?: ExportSkuStockRequest, codes?: number[]) =>
  Request<ExportSkuStockResponse>('/ceres/inventory/InventoryService/ExportSkuStock')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ExportSkuStock.url = '/ceres/inventory/InventoryService/ExportSkuStock'
export const GetManySkuStock = (r?: GetManySkuStockRequest, codes?: number[]) =>
  Request<GetManySkuStockResponse>('/ceres/inventory/InventoryService/GetManySkuStock')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetManySkuStock.url = '/ceres/inventory/InventoryService/GetManySkuStock'

export const CreateSkuUnitStock = (r?: CreateSkuUnitStockRequest, codes?: number[]) =>
  Request<CreateSkuUnitStockResponse>('/ceres/inventory/InventoryService/CreateSkuUnitStock')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateSkuUnitStock.url = '/ceres/inventory/InventoryService/CreateSkuUnitStock'
export const UpdateSkuUnitStock = (r?: UpdateSkuUnitStockRequest, codes?: number[]) =>
  Request<UpdateSkuUnitStockResponse>('/ceres/inventory/InventoryService/UpdateSkuUnitStock')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateSkuUnitStock.url = '/ceres/inventory/InventoryService/UpdateSkuUnitStock'
export const DeleteSkuUnitStock = (r?: DeleteSkuUnitStockRequest, codes?: number[]) =>
  Request<DeleteSkuUnitStockResponse>('/ceres/inventory/InventoryService/DeleteSkuUnitStock')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteSkuUnitStock.url = '/ceres/inventory/InventoryService/DeleteSkuUnitStock'
export const ListSkuUnitStock = (r?: ListSkuUnitStockRequest, codes?: number[]) =>
  Request<ListSkuUnitStockResponse>('/ceres/inventory/InventoryService/ListSkuUnitStock')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListSkuUnitStock.url = '/ceres/inventory/InventoryService/ListSkuUnitStock'
export const GetSkuUnitStock = (r?: GetSkuUnitStockRequest, codes?: number[]) =>
  Request<GetSkuUnitStockResponse>('/ceres/inventory/InventoryService/GetSkuUnitStock')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetSkuUnitStock.url = '/ceres/inventory/InventoryService/GetSkuUnitStock'

export const UpdatePendingStock = (r?: UpdatePendingStockRequest, codes?: number[]) =>
  Request<UpdatePendingStockResponse>('/ceres/inventory/InventoryService/UpdatePendingStock')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdatePendingStock.url = '/ceres/inventory/InventoryService/UpdatePendingStock'
export const ListPendingStock = (r?: ListPendingStockRequest, codes?: number[]) =>
  Request<ListPendingStockResponse>('/ceres/inventory/InventoryService/ListPendingStock')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListPendingStock.url = '/ceres/inventory/InventoryService/ListPendingStock'
export const CreateCustomerTurnover = (r?: CreateCustomerTurnoverRequest, codes?: number[]) =>
  Request<CreateCustomerTurnoverResponse>('/ceres/inventory/InventoryService/CreateCustomerTurnover')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateCustomerTurnover.url = '/ceres/inventory/InventoryService/CreateCustomerTurnover'
export const UpdateCustomerTurnover = (r?: UpdateCustomerTurnoverRequest, codes?: number[]) =>
  Request<UpdateCustomerTurnoverResponse>('/ceres/inventory/InventoryService/UpdateCustomerTurnover')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateCustomerTurnover.url = '/ceres/inventory/InventoryService/UpdateCustomerTurnover'
export const DeleteCustomerTurnover = (r?: DeleteCustomerTurnoverRequest, codes?: number[]) =>
  Request<DeleteCustomerTurnoverResponse>('/ceres/inventory/InventoryService/DeleteCustomerTurnover')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteCustomerTurnover.url = '/ceres/inventory/InventoryService/DeleteCustomerTurnover'
export const GetCustomerTurnover = (r?: GetCustomerTurnoverRequest, codes?: number[]) =>
  Request<GetCustomerTurnoverResponse>('/ceres/inventory/InventoryService/GetCustomerTurnover')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetCustomerTurnover.url = '/ceres/inventory/InventoryService/GetCustomerTurnover'
export const ListCustomerTurnover = (r?: ListCustomerTurnoverRequest, codes?: number[]) =>
  Request<ListCustomerTurnoverResponse>('/ceres/inventory/InventoryService/ListCustomerTurnover')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListCustomerTurnover.url = '/ceres/inventory/InventoryService/ListCustomerTurnover'
export const ExportCustomerTurnover = (r?: ExportCustomerTurnoverRequest, codes?: number[]) =>
  Request<ExportCustomerTurnoverResponse>('/ceres/inventory/InventoryService/ExportCustomerTurnover')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ExportCustomerTurnover.url = '/ceres/inventory/InventoryService/ExportCustomerTurnover'

export const UpdateOrderStock = (r?: UpdateOrderStockRequest, codes?: number[]) =>
  Request<UpdateOrderStockResponse>('/ceres/inventory/InventoryService/UpdateOrderStock')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateOrderStock.url = '/ceres/inventory/InventoryService/UpdateOrderStock'
export const UpdateSkuUnitStockConfig = (r?: UpdateSkuUnitStockConfigRequest, codes?: number[]) =>
  Request<UpdateSkuUnitStockConfigResponse>('/ceres/inventory/InventoryService/UpdateSkuUnitStockConfig')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateSkuUnitStockConfig.url = '/ceres/inventory/InventoryService/UpdateSkuUnitStockConfig'
export const CreateAdjustSheet = (r?: CreateAdjustSheetRequest, codes?: number[]) =>
  Request<CreateAdjustSheetResponse>('/ceres/inventory/InventoryService/CreateAdjustSheet')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateAdjustSheet.url = '/ceres/inventory/InventoryService/CreateAdjustSheet'
export const UpdateAdjustSheet = (r?: UpdateAdjustSheetRequest, codes?: number[]) =>
  Request<UpdateAdjustSheetResponse>('/ceres/inventory/InventoryService/UpdateAdjustSheet')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateAdjustSheet.url = '/ceres/inventory/InventoryService/UpdateAdjustSheet'
export const ListAdjustSheet = (r?: ListAdjustSheetRequest, codes?: number[]) =>
  Request<ListAdjustSheetResponse>('/ceres/inventory/InventoryService/ListAdjustSheet')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListAdjustSheet.url = '/ceres/inventory/InventoryService/ListAdjustSheet'
export const DeleteAdjustSheet = (r?: DeleteAdjustSheetRequest, codes?: number[]) =>
  Request<DeleteAdjustSheetResponse>('/ceres/inventory/InventoryService/DeleteAdjustSheet')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteAdjustSheet.url = '/ceres/inventory/InventoryService/DeleteAdjustSheet'
export const GetAdjustSheet = (r?: GetAdjustSheetRequest, codes?: number[]) =>
  Request<GetAdjustSheetResponse>('/ceres/inventory/InventoryService/GetAdjustSheet')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetAdjustSheet.url = '/ceres/inventory/InventoryService/GetAdjustSheet'

export const AsyncCommitAdjustSheet = (r?: AsyncCommitAdjustSheetRequest, codes?: number[]) =>
  Request<AsyncCommitAdjustSheetResponse>('/ceres/inventory/InventoryService/AsyncCommitAdjustSheet')
    .code([...(codes || [])])
    .data(r || {})
    .run()
AsyncCommitAdjustSheet.url = '/ceres/inventory/InventoryService/AsyncCommitAdjustSheet'
export const CheckOutStock = (r?: CheckOutStockRequest, codes?: number[]) =>
  Request<CheckOutStockResponse>('/ceres/inventory/InventoryService/CheckOutStock')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CheckOutStock.url = '/ceres/inventory/InventoryService/CheckOutStock'
export const CreateReplaceBatch = (r?: CreateReplaceBatchRequest, codes?: number[]) =>
  Request<CreateReplaceBatchResponse>('/ceres/inventory/InventoryService/CreateReplaceBatch')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateReplaceBatch.url = '/ceres/inventory/InventoryService/CreateReplaceBatch'
export const ListProductionTaskCost = (r?: ListProductionTaskCostRequest, codes?: number[]) =>
  Request<ListProductionTaskCostResponse>('/ceres/inventory/InventoryService/ListProductionTaskCost')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListProductionTaskCost.url = '/ceres/inventory/InventoryService/ListProductionTaskCost'
