// gm_api/src/inventory/pc.tsx
import React, { FC } from 'react'
import { Select, SelectProps } from '@gm-pc/react'
import {
  SelectList,
  SelectEnum,
  MoreSelectList,
  MapId,
} from '../util/pc_util'
import type {
  MapProps,
  SelectListProps,
  SelectEnumProps,
  MoreSelectListProps,
  MapIdProps,
} from '../util/pc_util'
import {
  list_ChangeType,
  map_ChangeType,
  list_OperateType,
  map_OperateType,
  list_Status_Code,
  map_Status_Code,
  list_Batch_BatchType,
  map_Batch_BatchType,
  list_PendingType,
  map_PendingType,
  list_SkuUnitStock_Config,
  map_SkuUnitStock_Config,
  list_SkuStock_Status,
  map_SkuStock_Status,
  list_CostAllocation_Type,
  map_CostAllocation_Type,
  list_CostAllocation_Reason,
  map_CostAllocation_Reason,
  list_CostAllocation_MoneyType,
  map_CostAllocation_MoneyType,
  list_StockSheet_SheetType,
  map_StockSheet_SheetType,
  list_StockSheet_SheetStatus,
  map_StockSheet_SheetStatus,
  list_StockSheet_TargetType,
  map_StockSheet_TargetType,
  list_StockSheet_Status,
  map_StockSheet_Status,
  list_StockSheet_PayStatus,
  map_StockSheet_PayStatus,
  list_StockSheet_RelatedSheetType,
  map_StockSheet_RelatedSheetType,
  list_AdjustSheet_AdjustSheetStatus,
  map_AdjustSheet_AdjustSheetStatus,
  list_TimeType,
  map_TimeType,
  list_RequestDetail,
  map_RequestDetail,
  Shelf,
  ListShelf,
  ListShelfResponse,
  Batch,
  ListBatch,
  ListBatchResponse,
  BatchLog,
  ListBatchLog,
  ListBatchLogResponse,
  SkuUnitStock,
  ListSkuUnitStock,
  ListSkuUnitStockResponse,
  StockSheet,
  ListStockSheet,
  ListStockSheetResponse,
  StockLog,
  ListStockLog,
  ListStockLogResponse,
  AdjustSheet,
  ListAdjustSheet,
  ListAdjustSheetResponse,
} from './index'

export const Select_ChangeType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_ChangeType} />
}

export const Map_ChangeType: FC<MapProps> = ({ value }) => {
  return <>{map_ChangeType[value]}</>
}

export const Select_OperateType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_OperateType} />
}

export const Map_OperateType: FC<MapProps> = ({ value }) => {
  return <>{map_OperateType[value]}</>
}

export const Select_Status_Code: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Status_Code} />
}

export const Map_Status_Code: FC<MapProps> = ({ value }) => {
  return <>{map_Status_Code[value]}</>
}

export const Select_Batch_BatchType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Batch_BatchType} />
}

export const Map_Batch_BatchType: FC<MapProps> = ({ value }) => {
  return <>{map_Batch_BatchType[value]}</>
}

export const Select_PendingType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_PendingType} />
}

export const Map_PendingType: FC<MapProps> = ({ value }) => {
  return <>{map_PendingType[value]}</>
}

export const Select_SkuUnitStock_Config: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_SkuUnitStock_Config} />
}

export const Map_SkuUnitStock_Config: FC<MapProps> = ({ value }) => {
  return <>{map_SkuUnitStock_Config[value]}</>
}

export const Select_SkuStock_Status: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_SkuStock_Status} />
}

export const Map_SkuStock_Status: FC<MapProps> = ({ value }) => {
  return <>{map_SkuStock_Status[value]}</>
}

export const Select_CostAllocation_Type: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_CostAllocation_Type} />
}

export const Map_CostAllocation_Type: FC<MapProps> = ({ value }) => {
  return <>{map_CostAllocation_Type[value]}</>
}

export const Select_CostAllocation_Reason: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_CostAllocation_Reason} />
}

export const Map_CostAllocation_Reason: FC<MapProps> = ({ value }) => {
  return <>{map_CostAllocation_Reason[value]}</>
}

export const Select_CostAllocation_MoneyType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_CostAllocation_MoneyType} />
}

export const Map_CostAllocation_MoneyType: FC<MapProps> = ({ value }) => {
  return <>{map_CostAllocation_MoneyType[value]}</>
}

export const Select_StockSheet_SheetType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_StockSheet_SheetType} />
}

export const Map_StockSheet_SheetType: FC<MapProps> = ({ value }) => {
  return <>{map_StockSheet_SheetType[value]}</>
}

export const Select_StockSheet_SheetStatus: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_StockSheet_SheetStatus} />
}

export const Map_StockSheet_SheetStatus: FC<MapProps> = ({ value }) => {
  return <>{map_StockSheet_SheetStatus[value]}</>
}

export const Select_StockSheet_TargetType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_StockSheet_TargetType} />
}

export const Map_StockSheet_TargetType: FC<MapProps> = ({ value }) => {
  return <>{map_StockSheet_TargetType[value]}</>
}

export const Select_StockSheet_Status: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_StockSheet_Status} />
}

export const Map_StockSheet_Status: FC<MapProps> = ({ value }) => {
  return <>{map_StockSheet_Status[value]}</>
}

export const Select_StockSheet_PayStatus: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_StockSheet_PayStatus} />
}

export const Map_StockSheet_PayStatus: FC<MapProps> = ({ value }) => {
  return <>{map_StockSheet_PayStatus[value]}</>
}

export const Select_StockSheet_RelatedSheetType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_StockSheet_RelatedSheetType} />
}

export const Map_StockSheet_RelatedSheetType: FC<MapProps> = ({ value }) => {
  return <>{map_StockSheet_RelatedSheetType[value]}</>
}

export const Select_AdjustSheet_AdjustSheetStatus: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_AdjustSheet_AdjustSheetStatus} />
}

export const Map_AdjustSheet_AdjustSheetStatus: FC<MapProps> = ({ value }) => {
  return <>{map_AdjustSheet_AdjustSheetStatus[value]}</>
}

export const Select_TimeType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_TimeType} />
}

export const Map_TimeType: FC<MapProps> = ({ value }) => {
  return <>{map_TimeType[value]}</>
}

export const Select_RequestDetail: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_RequestDetail} />
}

export const Map_RequestDetail: FC<MapProps> = ({ value }) => {
  return <>{map_RequestDetail[value]}</>
}

export const Select_Shelf: FC<SelectListProps<
  Shelf,
  ListShelfResponse
>> = (props) => {
  return (
    <SelectList<Shelf, ListShelfResponse>
      {...props}
      messageName='Shelf'
      fetchList={ListShelf}
    />
  )
}

export const MoreSelect_Shelf: FC<MoreSelectListProps<
  Shelf,
  ListShelfResponse
>> = (props) => {
  return (
    <MoreSelectList<Shelf, ListShelfResponse>
      {...props}
      messageName='Shelf'
      fetchList={ListShelf}
    />
  )
}

export const Select_Batch: FC<SelectListProps<
  Batch,
  ListBatchResponse
>> = (props) => {
  return (
    <SelectList<Batch, ListBatchResponse>
      {...props}
      messageName='Batch'
      fetchList={ListBatch}
    />
  )
}

export const MoreSelect_Batch: FC<MoreSelectListProps<
  Batch,
  ListBatchResponse
>> = (props) => {
  return (
    <MoreSelectList<Batch, ListBatchResponse>
      {...props}
      messageName='Batch'
      fetchList={ListBatch}
    />
  )
}

export const Select_BatchLog: FC<SelectListProps<
  BatchLog,
  ListBatchLogResponse
>> = (props) => {
  return (
    <SelectList<BatchLog, ListBatchLogResponse>
      {...props}
      messageName='BatchLog'
      fetchList={ListBatchLog}
    />
  )
}

export const MoreSelect_BatchLog: FC<MoreSelectListProps<
  BatchLog,
  ListBatchLogResponse
>> = (props) => {
  return (
    <MoreSelectList<BatchLog, ListBatchLogResponse>
      {...props}
      messageName='BatchLog'
      fetchList={ListBatchLog}
    />
  )
}

export const Select_SkuUnitStock: FC<SelectListProps<
  SkuUnitStock,
  ListSkuUnitStockResponse
>> = (props) => {
  return (
    <SelectList<SkuUnitStock, ListSkuUnitStockResponse>
      {...props}
      messageName='SkuUnitStock'
      fetchList={ListSkuUnitStock}
    />
  )
}

export const MoreSelect_SkuUnitStock: FC<MoreSelectListProps<
  SkuUnitStock,
  ListSkuUnitStockResponse
>> = (props) => {
  return (
    <MoreSelectList<SkuUnitStock, ListSkuUnitStockResponse>
      {...props}
      messageName='SkuUnitStock'
      fetchList={ListSkuUnitStock}
    />
  )
}

export const Select_StockSheet: FC<SelectListProps<
  StockSheet,
  ListStockSheetResponse
>> = (props) => {
  return (
    <SelectList<StockSheet, ListStockSheetResponse>
      {...props}
      messageName='StockSheet'
      fetchList={ListStockSheet}
    />
  )
}

export const MoreSelect_StockSheet: FC<MoreSelectListProps<
  StockSheet,
  ListStockSheetResponse
>> = (props) => {
  return (
    <MoreSelectList<StockSheet, ListStockSheetResponse>
      {...props}
      messageName='StockSheet'
      fetchList={ListStockSheet}
    />
  )
}

export const Select_StockLog: FC<SelectListProps<
  StockLog,
  ListStockLogResponse
>> = (props) => {
  return (
    <SelectList<StockLog, ListStockLogResponse>
      {...props}
      messageName='StockLog'
      fetchList={ListStockLog}
    />
  )
}

export const MoreSelect_StockLog: FC<MoreSelectListProps<
  StockLog,
  ListStockLogResponse
>> = (props) => {
  return (
    <MoreSelectList<StockLog, ListStockLogResponse>
      {...props}
      messageName='StockLog'
      fetchList={ListStockLog}
    />
  )
}

export const Select_AdjustSheet: FC<SelectListProps<
  AdjustSheet,
  ListAdjustSheetResponse
>> = (props) => {
  return (
    <SelectList<AdjustSheet, ListAdjustSheetResponse>
      {...props}
      messageName='AdjustSheet'
      fetchList={ListAdjustSheet}
    />
  )
}

export const MoreSelect_AdjustSheet: FC<MoreSelectListProps<
  AdjustSheet,
  ListAdjustSheetResponse
>> = (props) => {
  return (
    <MoreSelectList<AdjustSheet, ListAdjustSheetResponse>
      {...props}
      messageName='AdjustSheet'
      fetchList={ListAdjustSheet}
    />
  )
}

export const MapId_Shelf: FC<MapIdProps<Shelf, ListShelfResponse>> = (
  props,
) => {
  return (
    <MapId<Shelf, ListShelfResponse>
      {...props}
      packageName='inventory'
      messageName='Shelf'
      fetchList={ListShelf}
    />
  )
}

export type {
  MapProps,
  SelectListProps,
  MoreSelectListProps,
}
