// gm_api/src/asynctask/data.ts
import { getLocale } from './lng'

interface Enum_List_Item {
  text: string
  value: any
}

interface Enum_Map {
  [key: string]: string
}
export const list_Task_Category: Enum_List_Item[] = [
  { value: 1, text: getLocale('Task_Category__CATEGORY_EXPORT') },
  { value: 999, text: getLocale('Task_Category__CATEGORY_OTHER') },
]
export const map_Task_Category: Enum_Map = {
  1: getLocale('Task_Category__CATEGORY_EXPORT'),
  999: getLocale('Task_Category__CATEGORY_OTHER'),
}
export const list_Task_Type: Enum_List_Item[] = [
  { value: 1, text: getLocale('Task_Type__TYPE_ASYNCTASK_FOO_TEST_NORMAL') },
  { value: 2, text: getLocale('Task_Type__TYPE_ASYNCTASK_FOO_TEST_MAX_FAILURE') },
  { value: 1000, text: getLocale('Task_Type__TYPE_ORDER_ORDER_IMPORT') },
  { value: 1001, text: getLocale('Task_Type__TYPE_ORDER_ORDERSTATE_UPDATE') },
  { value: 1002, text: getLocale('Task_Type__TYPE_ORDER_ORDEROUTSTOCK_UPDATE') },
  { value: 1003, text: getLocale('Task_Type__TYPE_ORDER_ORDERSSUPRICE_UPDATE') },
  { value: 1004, text: getLocale('Task_Type__TYPE_ORDER_SSUPRICE_SYNC') },
  { value: 1005, text: getLocale('Task_Type__TYPE_ORDER_ORDER_CREATE') },
  { value: 1006, text: getLocale('Task_Type__TYPE_ORDER_ORDER_EXPORT') },
  { value: 1007, text: getLocale('Task_Type__TYPE_ORDER_ORDERDRIVER_UPDATE') },
  { value: 1008, text: getLocale('Task_Type__TYPE_ORDER_ORDERDRIVER_AUTOUPDATE') },
  { value: 1009, text: getLocale('Task_Type__TYPE_ORDER_ORDER_SUBMIT_PURCHASE_TASK') },
  { value: 1010, text: getLocale('Task_Type__TYPE_ORDER_SCAN_RECEIPT') },
  { value: 1100, text: getLocale('Task_Type__TYPE_PRODUCTION_TASK_CREATE') },
  { value: 1101, text: getLocale('Task_Type__TYPE_PRODUCTION_PURCHASE_SKU_EXPORT') },
  { value: 1102, text: getLocale('Task_Type__TYPE_PRODUCTION_TASK_EXPORT') },
  { value: 1103, text: getLocale('Task_Type__TYPE_PRODUCTION_BOM_EXPORT') },
  { value: 1104, text: getLocale('Task_Type__TYPE_PRODUCTION_TASK_DATA_EXPORT') },
  { value: 1105, text: getLocale('Task_Type__TYPE_PRODUCTION_BOM_IMPORT') },
  { value: 1106, text: getLocale('Task_Type__TYPE_PRODUCTION_PROCESS_TEMPLATE_EXPORT') },
  { value: 1200, text: getLocale('Task_Type__TYPE_MERCHANDISE_SKU_EXPORT') },
  { value: 1201, text: getLocale('Task_Type__TYPE_MERCHANDISE_SSU_EXPORT') },
  { value: 1202, text: getLocale('Task_Type__TYPE_MERCHANDISE_SKU_SSU_UPDATE') },
  { value: 1203, text: getLocale('Task_Type__TYPE_MERCHANDISE_CREATE_ALL') },
  { value: 1204, text: getLocale('Task_Type__TYPE_MERCHANDISE_CREATE_SSU') },
  { value: 1300, text: getLocale('Task_Type__TYPE_ENTERPRISE_CUSTOMER_EXPORT') },
  { value: 1301, text: getLocale('Task_Type__TYPE_ENTERPRISE_CUSTOMER_IMPORT') },
  { value: 1302, text: getLocale('Task_Type__TYPE_ENTERPRISE_SUPPLIER_EXPORT') },
  { value: 1400, text: getLocale('Task_Type__TYPE_INVENTORY_STOCK_SHEET_LIST_EXPORT') },
  { value: 1402, text: getLocale('Task_Type__TYPE_INVENTORY_BATCH_LOG_EXPORT') },
  { value: 1403, text: getLocale('Task_Type__TYPE_INVENTORY_COMMIT_ADJUST_SHEET') },
  { value: 1404, text: getLocale('Task_Type__TYPE_INVENTORY_BATCH_SYNC_STOCK_SHEET') },
  { value: 1410, text: getLocale('Task_Type__TYPE_INVENTORY_CHANGE_LOG_EXPORT') },
  { value: 1411, text: getLocale('Task_Type__TYPE_INVENTORY_PURCHASE_IN_LOG_EXPORT') },
  { value: 1412, text: getLocale('Task_Type__TYPE_INVENTORY_SALE_OUT_LOG_EXPORT') },
  { value: 1413, text: getLocale('Task_Type__TYPE_INVENTORY_MATERIAL_OUT_LOG_EXPORT') },
  { value: 1414, text: getLocale('Task_Type__TYPE_INVENTORY_MATERIAL_IN_LOG_EXPORT') },
  { value: 1415, text: getLocale('Task_Type__TYPE_INVENTORY_PRODUCT_IN_LOG_EXPORT') },
  { value: 1416, text: getLocale('Task_Type__TYPE_INVENTORY_REFOUND_OUT_LOG_EXPORT') },
  { value: 1417, text: getLocale('Task_Type__TYPE_INVENTORY_OTHER_OUT_LOG_EXPORT') },
  { value: 1418, text: getLocale('Task_Type__TYPE_INVENTORY_OTHER_IN_LOG_EXPORT') },
  { value: 1419, text: getLocale('Task_Type__TYPE_INVENTORY_INCREASE_IN_LOG_EXPORT') },
  { value: 1420, text: getLocale('Task_Type__TYPE_INVENTORY_LOSS_OUT_LOG_EXPORT') },
  { value: 1421, text: getLocale('Task_Type__TYPE_INVENTORY_CUSTOMER_TURNOVER_EXPORT') },
  { value: 1422, text: getLocale('Task_Type__TYPE_INVENTORY_LOAN_EXPORT') },
  { value: 1423, text: getLocale('Task_Type__TYPE_INVENTORY_REVERT_EXPORT') },
  { value: 1424, text: getLocale('Task_Type__TYPE_INVENTORY_CUSTOMER_TURNOVER_LOG_EXPORT') },
  { value: 1450, text: getLocale('Task_Type__TYPE_INVENTORY_SHEET_EXPORT') },
  { value: 1498, text: getLocale('Task_Type__TYPE_INVENTORY_BATCH_UPDATE_STOCK_SHEET') },
  { value: 1499, text: getLocale('Task_Type__TYPE_INVENTORY_SKU_STOCK_EXPORT') },
  { value: 1500, text: getLocale('Task_Type__TYPE_SORTING_SORTING_PERFORMANCE_EXPORT') },
  { value: 1601, text: getLocale('Task_Type__TYPE_PURCHASE_PURCHASE_SHEET_EXPORT') },
  { value: 1602, text: getLocale('Task_Type__TYPE_PURCHASE_INQUIRY_PRICE_IMPORT') },
  { value: 1603, text: getLocale('Task_Type__TYPE_PURCHASE_PURCHASE_TASK_RELEASE') },
  { value: 1604, text: getLocale('Task_Type__TYPE_PURCHASE_PURCHASE_TASK_SWITCH_SUPPLIER') },
  { value: 1605, text: getLocale('Task_Type__TYPE_PURCHASE_PURCHASE_TASK_SWITCH_PURCHASER') },
  { value: 1606, text: getLocale('Task_Type__TYPE_PURCHASE_PURCHASE_TASK_CREATE_PURCHASE_SHEET') },
  { value: 1607, text: getLocale('Task_Type__TYPE_PURCHASE_PURCHASE_TASK_EXPORT') },
  { value: 1608, text: getLocale('Task_Type__TYPE_PURCHASE_INQUIRY_PRICE_IMPORT_TMPL') },
  { value: 1706, text: getLocale('Task_Type__TYPE_ESHOP_ESHOP_ORDER_EXPORT') },
  { value: 1707, text: getLocale('Task_Type__TYPE_ESHOP_ESHOP_ORDER_UPDATE') },
  { value: 1801, text: getLocale('Task_Type__TYPE_ESHOP_LEAVEAPPLICATION_EXPORT') },
  { value: 1900, text: getLocale('Task_Type__TYPE_AFTER_SALE_EXPORT_ORDER') },
  { value: 1901, text: getLocale('Task_Type__TYPE_AFTER_SALE_EXPORT_TASK') },
  { value: 1902, text: getLocale('Task_Type__TYPE_AFTER_SALE_EXPORT_ORDER_SHEET') },
  { value: 1903, text: getLocale('Task_Type__TYPE_AFTER_SALE_EXPORT_ORDER_DETAIL') },
  { value: 2000, text: getLocale('Task_Type__TYPE_INITIALIZATION_CLEAN_GROUP_TICKET') },
  { value: 2001, text: getLocale('Task_Type__TYPE_INITIALIZATION_CLEAN_GROUP_DATA') },
  { value: 2100, text: getLocale('Task_Type__TYPE_FINANCE_SETTLE_EXPORT') },
  { value: 2101, text: getLocale('Task_Type__TYPE_FINANCE_BATCH_PAY_SETTLE') },
  { value: 2102, text: getLocale('Task_Type__TYPE_FINANCE_FLOW_EXPORT') },
  { value: 2103, text: getLocale('Task_Type__TYPE_FINANCE_ACCOUNT_BALANCE_EXPORT') },
  { value: 2104, text: getLocale('Task_Type__TYPE_FINANCE_BATCH_SUBMIT_SETTLE_SHEET') },
]
export const map_Task_Type: Enum_Map = {
  1: getLocale('Task_Type__TYPE_ASYNCTASK_FOO_TEST_NORMAL'),
  2: getLocale('Task_Type__TYPE_ASYNCTASK_FOO_TEST_MAX_FAILURE'),
  1000: getLocale('Task_Type__TYPE_ORDER_ORDER_IMPORT'),
  1001: getLocale('Task_Type__TYPE_ORDER_ORDERSTATE_UPDATE'),
  1002: getLocale('Task_Type__TYPE_ORDER_ORDEROUTSTOCK_UPDATE'),
  1003: getLocale('Task_Type__TYPE_ORDER_ORDERSSUPRICE_UPDATE'),
  1004: getLocale('Task_Type__TYPE_ORDER_SSUPRICE_SYNC'),
  1005: getLocale('Task_Type__TYPE_ORDER_ORDER_CREATE'),
  1006: getLocale('Task_Type__TYPE_ORDER_ORDER_EXPORT'),
  1007: getLocale('Task_Type__TYPE_ORDER_ORDERDRIVER_UPDATE'),
  1008: getLocale('Task_Type__TYPE_ORDER_ORDERDRIVER_AUTOUPDATE'),
  1009: getLocale('Task_Type__TYPE_ORDER_ORDER_SUBMIT_PURCHASE_TASK'),
  1010: getLocale('Task_Type__TYPE_ORDER_SCAN_RECEIPT'),
  1100: getLocale('Task_Type__TYPE_PRODUCTION_TASK_CREATE'),
  1101: getLocale('Task_Type__TYPE_PRODUCTION_PURCHASE_SKU_EXPORT'),
  1102: getLocale('Task_Type__TYPE_PRODUCTION_TASK_EXPORT'),
  1103: getLocale('Task_Type__TYPE_PRODUCTION_BOM_EXPORT'),
  1104: getLocale('Task_Type__TYPE_PRODUCTION_TASK_DATA_EXPORT'),
  1105: getLocale('Task_Type__TYPE_PRODUCTION_BOM_IMPORT'),
  1106: getLocale('Task_Type__TYPE_PRODUCTION_PROCESS_TEMPLATE_EXPORT'),
  1200: getLocale('Task_Type__TYPE_MERCHANDISE_SKU_EXPORT'),
  1201: getLocale('Task_Type__TYPE_MERCHANDISE_SSU_EXPORT'),
  1202: getLocale('Task_Type__TYPE_MERCHANDISE_SKU_SSU_UPDATE'),
  1203: getLocale('Task_Type__TYPE_MERCHANDISE_CREATE_ALL'),
  1204: getLocale('Task_Type__TYPE_MERCHANDISE_CREATE_SSU'),
  1300: getLocale('Task_Type__TYPE_ENTERPRISE_CUSTOMER_EXPORT'),
  1301: getLocale('Task_Type__TYPE_ENTERPRISE_CUSTOMER_IMPORT'),
  1302: getLocale('Task_Type__TYPE_ENTERPRISE_SUPPLIER_EXPORT'),
  1400: getLocale('Task_Type__TYPE_INVENTORY_STOCK_SHEET_LIST_EXPORT'),
  1402: getLocale('Task_Type__TYPE_INVENTORY_BATCH_LOG_EXPORT'),
  1403: getLocale('Task_Type__TYPE_INVENTORY_COMMIT_ADJUST_SHEET'),
  1404: getLocale('Task_Type__TYPE_INVENTORY_BATCH_SYNC_STOCK_SHEET'),
  1410: getLocale('Task_Type__TYPE_INVENTORY_CHANGE_LOG_EXPORT'),
  1411: getLocale('Task_Type__TYPE_INVENTORY_PURCHASE_IN_LOG_EXPORT'),
  1412: getLocale('Task_Type__TYPE_INVENTORY_SALE_OUT_LOG_EXPORT'),
  1413: getLocale('Task_Type__TYPE_INVENTORY_MATERIAL_OUT_LOG_EXPORT'),
  1414: getLocale('Task_Type__TYPE_INVENTORY_MATERIAL_IN_LOG_EXPORT'),
  1415: getLocale('Task_Type__TYPE_INVENTORY_PRODUCT_IN_LOG_EXPORT'),
  1416: getLocale('Task_Type__TYPE_INVENTORY_REFOUND_OUT_LOG_EXPORT'),
  1417: getLocale('Task_Type__TYPE_INVENTORY_OTHER_OUT_LOG_EXPORT'),
  1418: getLocale('Task_Type__TYPE_INVENTORY_OTHER_IN_LOG_EXPORT'),
  1419: getLocale('Task_Type__TYPE_INVENTORY_INCREASE_IN_LOG_EXPORT'),
  1420: getLocale('Task_Type__TYPE_INVENTORY_LOSS_OUT_LOG_EXPORT'),
  1421: getLocale('Task_Type__TYPE_INVENTORY_CUSTOMER_TURNOVER_EXPORT'),
  1422: getLocale('Task_Type__TYPE_INVENTORY_LOAN_EXPORT'),
  1423: getLocale('Task_Type__TYPE_INVENTORY_REVERT_EXPORT'),
  1424: getLocale('Task_Type__TYPE_INVENTORY_CUSTOMER_TURNOVER_LOG_EXPORT'),
  1450: getLocale('Task_Type__TYPE_INVENTORY_SHEET_EXPORT'),
  1498: getLocale('Task_Type__TYPE_INVENTORY_BATCH_UPDATE_STOCK_SHEET'),
  1499: getLocale('Task_Type__TYPE_INVENTORY_SKU_STOCK_EXPORT'),
  1500: getLocale('Task_Type__TYPE_SORTING_SORTING_PERFORMANCE_EXPORT'),
  1601: getLocale('Task_Type__TYPE_PURCHASE_PURCHASE_SHEET_EXPORT'),
  1602: getLocale('Task_Type__TYPE_PURCHASE_INQUIRY_PRICE_IMPORT'),
  1603: getLocale('Task_Type__TYPE_PURCHASE_PURCHASE_TASK_RELEASE'),
  1604: getLocale('Task_Type__TYPE_PURCHASE_PURCHASE_TASK_SWITCH_SUPPLIER'),
  1605: getLocale('Task_Type__TYPE_PURCHASE_PURCHASE_TASK_SWITCH_PURCHASER'),
  1606: getLocale('Task_Type__TYPE_PURCHASE_PURCHASE_TASK_CREATE_PURCHASE_SHEET'),
  1607: getLocale('Task_Type__TYPE_PURCHASE_PURCHASE_TASK_EXPORT'),
  1608: getLocale('Task_Type__TYPE_PURCHASE_INQUIRY_PRICE_IMPORT_TMPL'),
  1706: getLocale('Task_Type__TYPE_ESHOP_ESHOP_ORDER_EXPORT'),
  1707: getLocale('Task_Type__TYPE_ESHOP_ESHOP_ORDER_UPDATE'),
  1801: getLocale('Task_Type__TYPE_ESHOP_LEAVEAPPLICATION_EXPORT'),
  1900: getLocale('Task_Type__TYPE_AFTER_SALE_EXPORT_ORDER'),
  1901: getLocale('Task_Type__TYPE_AFTER_SALE_EXPORT_TASK'),
  1902: getLocale('Task_Type__TYPE_AFTER_SALE_EXPORT_ORDER_SHEET'),
  1903: getLocale('Task_Type__TYPE_AFTER_SALE_EXPORT_ORDER_DETAIL'),
  2000: getLocale('Task_Type__TYPE_INITIALIZATION_CLEAN_GROUP_TICKET'),
  2001: getLocale('Task_Type__TYPE_INITIALIZATION_CLEAN_GROUP_DATA'),
  2100: getLocale('Task_Type__TYPE_FINANCE_SETTLE_EXPORT'),
  2101: getLocale('Task_Type__TYPE_FINANCE_BATCH_PAY_SETTLE'),
  2102: getLocale('Task_Type__TYPE_FINANCE_FLOW_EXPORT'),
  2103: getLocale('Task_Type__TYPE_FINANCE_ACCOUNT_BALANCE_EXPORT'),
  2104: getLocale('Task_Type__TYPE_FINANCE_BATCH_SUBMIT_SETTLE_SHEET'),
}
export const list_Task_State: Enum_List_Item[] = [
  { value: 1, text: getLocale('Task_State__STATE_CREATED') },
  { value: 2, text: getLocale('Task_State__STATE_READY') },
  { value: 3, text: getLocale('Task_State__STATE_RUNNING') },
  { value: 4, text: getLocale('Task_State__STATE_CANCELED') },
  { value: 5, text: getLocale('Task_State__STATE_FAULTED') },
  { value: 6, text: getLocale('Task_State__STATE_COMPLETED') },
]
export const map_Task_State: Enum_Map = {
  1: getLocale('Task_State__STATE_CREATED'),
  2: getLocale('Task_State__STATE_READY'),
  3: getLocale('Task_State__STATE_RUNNING'),
  4: getLocale('Task_State__STATE_CANCELED'),
  5: getLocale('Task_State__STATE_FAULTED'),
  6: getLocale('Task_State__STATE_COMPLETED'),
}
