// gm_api/src/asynctask/pc.tsx
import React, { FC } from 'react'
import { Select, SelectProps } from '@gm-pc/react'
import {
  SelectList,
  SelectEnum,
  MoreSelectList,
  MapId,
} from '../util/pc_util'
import type {
  MapProps,
  SelectListProps,
  SelectEnumProps,
  MoreSelectListProps,
  MapIdProps,
} from '../util/pc_util'
import {
  list_Task_Category,
  map_Task_Category,
  list_Task_Type,
  map_Task_Type,
  list_Task_State,
  map_Task_State,
  Task,
  ListTask,
  ListTaskResponse,
} from './index'

export const Select_Task_Category: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Task_Category} />
}

export const Map_Task_Category: FC<MapProps> = ({ value }) => {
  return <>{map_Task_Category[value]}</>
}

export const Select_Task_Type: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Task_Type} />
}

export const Map_Task_Type: FC<MapProps> = ({ value }) => {
  return <>{map_Task_Type[value]}</>
}

export const Select_Task_State: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Task_State} />
}

export const Map_Task_State: FC<MapProps> = ({ value }) => {
  return <>{map_Task_State[value]}</>
}

export const Select_Task: FC<SelectListProps<
  Task,
  ListTaskResponse
>> = (props) => {
  return (
    <SelectList<Task, ListTaskResponse>
      {...props}
      messageName='Task'
      fetchList={ListTask}
    />
  )
}

export const MoreSelect_Task: FC<MoreSelectListProps<
  Task,
  ListTaskResponse
>> = (props) => {
  return (
    <MoreSelectList<Task, ListTaskResponse>
      {...props}
      messageName='Task'
      fetchList={ListTask}
    />
  )
}

export type {
  MapProps,
  SelectListProps,
  MoreSelectListProps,
}
