// gm_api/src/message_types.ts
export enum PackageName {
  account = 'account',
  analytics = 'analytics',
  asynctask = 'asynctask',
  cloudapi = 'cloudapi',
  common = 'common',
  databi = 'databi',
  delivery = 'delivery',
  driverapp = 'driverapp',
  enterprise = 'enterprise',
  eshop = 'eshop',
  idgenerator = 'idgenerator',
  initialization = 'initialization',
  inventory = 'inventory',
  merchandise = 'merchandise',
  oauth = 'oauth',
  operation = 'operation',
  order = 'order',
  orderlogic = 'orderlogic',
  preference = 'preference',
  production = 'production',
  purchase = 'purchase',
  sorting = 'sorting',
  toolbox = 'toolbox',
  verification = 'verification',
  wechat = 'wechat',
  aftersale = 'aftersale',
  finance = 'finance',
}
export enum MessageName_account {
  Account = 'Account',
  Profile = 'Profile',
}
export enum MessageName_analytics {
  Example = 'Example',
  Order = 'Order',
  OrderDetail = 'OrderDetail',
  Task = 'Task',
  TaskInput = 'TaskInput',
  TaskOutput = 'TaskOutput',
}
export enum MessageName_asynctask {
  Task = 'Task',
}
export enum MessageName_cloudapi {
}
export enum MessageName_common {
}
export enum MessageName_databi {
}
export enum MessageName_delivery {
  Route = 'Route',
  RouteCustomer = 'RouteCustomer',
}
export enum MessageName_driverapp {
}
export enum MessageName_enterprise {
  Group = 'Group',
  GroupExtraInfo = 'GroupExtraInfo',
  Station = 'Station',
  Supplier = 'Supplier',
  DistributionContractor = 'DistributionContractor',
  GroupUser = 'GroupUser',
  Role = 'Role',
  GroupPermissionRelation = 'GroupPermissionRelation',
  RolePermissionRelation = 'RolePermissionRelation',
  GroupUserRoleRelation = 'GroupUserRoleRelation',
  ServicePeriod = 'ServicePeriod',
  Country = 'Country',
  Province = 'Province',
  City = 'City',
  District = 'District',
  Street = 'Street',
  CustomerLabelRelation = 'CustomerLabelRelation',
  CustomerLabel = 'CustomerLabel',
  Customer = 'Customer',
  CustomerUser = 'CustomerUser',
  CustomerServicePeriodRelation = 'CustomerServicePeriodRelation',
  InvitationCode = 'InvitationCode',
}
export enum MessageName_eshop {
  LeaveApplication = 'LeaveApplication',
  Menu = 'Menu',
  CustomerMenuRelation = 'CustomerMenuRelation',
  MenuDetail = 'MenuDetail',
  MenuPeriodGroup = 'MenuPeriodGroup',
}
export enum MessageName_idgenerator {
}
export enum MessageName_initialization {
}
export enum MessageName_inventory {
  Shelf = 'Shelf',
  Batch = 'Batch',
  BatchLog = 'BatchLog',
  SkuUnitStock = 'SkuUnitStock',
  CostAllocation = 'CostAllocation',
  StockSheet = 'StockSheet',
  StockLog = 'StockLog',
  AdjustSheet = 'AdjustSheet',
  ReplaceBatch = 'ReplaceBatch',
}
export enum MessageName_merchandise {
  Unit = 'Unit',
  Category = 'Category',
  Spu = 'Spu',
  Sku = 'Sku',
  Ssu = 'Ssu',
  Quotation = 'Quotation',
  CustomerQuotationRelation = 'CustomerQuotationRelation',
  BasicPrice = 'BasicPrice',
  Favorite = 'Favorite',
  FavoriteSsu = 'FavoriteSsu',
  RuleSet = 'RuleSet',
}
export enum MessageName_oauth {
  Code = 'Code',
  Token = 'Token',
  Session = 'Session',
}
export enum MessageName_operation {
  Operation = 'Operation',
}
export enum MessageName_order {
  Order = 'Order',
  OrderDetail = 'OrderDetail',
  SortingNumCounter = 'SortingNumCounter',
  OrderCustomerLastUsedDriver = 'OrderCustomerLastUsedDriver',
}
export enum MessageName_orderlogic {
  OrderImportTemplete = 'OrderImportTemplete',
}
export enum MessageName_preference {
  PrintingTemplate = 'PrintingTemplate',
  PrintingTemplateRelation = 'PrintingTemplateRelation',
  ShopBanner = 'ShopBanner',
  Shop = 'Shop',
  Application = 'Application',
  PurchaseSettings = 'PurchaseSettings',
  AppointTimeSettings = 'AppointTimeSettings',
  PrecisionSettings = 'PrecisionSettings',
  SortingSettings = 'SortingSettings',
  InventorySettings = 'InventorySettings',
  ProductionSettings = 'ProductionSettings',
}
export enum MessageName_production {
  ProcessType = 'ProcessType',
  ProcessTemplate = 'ProcessTemplate',
  Processor = 'Processor',
  Bom = 'Bom',
  BomMaterial = 'BomMaterial',
  Task = 'Task',
  TaskInput = 'TaskInput',
  TaskOutput = 'TaskOutput',
  TaskProcess = 'TaskProcess',
  TaskOrder = 'TaskOrder',
  TaskWeight = 'TaskWeight',
  TaskSheet = 'TaskSheet',
}
export enum MessageName_purchase {
  PurchaseTaskSrcRelation = 'PurchaseTaskSrcRelation',
  PurchaseTask = 'PurchaseTask',
  Batch = 'Batch',
  InquiryPrice = 'InquiryPrice',
  PurchaseSheet = 'PurchaseSheet',
  PurchaseCreateRule = 'PurchaseCreateRule',
}
export enum MessageName_sorting {
  SortingTag = 'SortingTag',
  SortingLog = 'SortingLog',
  SortingSkuGroup = 'SortingSkuGroup',
  SortingSkuGroupSku = 'SortingSkuGroupSku',
  SortingBasket = 'SortingBasket',
}
export enum MessageName_toolbox {
}
export enum MessageName_verification {
}
export enum MessageName_wechat {
  WxaCustomMessageRecvType = 'WxaCustomMessageRecvType',
}
export enum MessageName_aftersale {
  AfterSaleOrderDetail = 'AfterSaleOrderDetail',
  AfterSaleOrder = 'AfterSaleOrder',
}
export enum MessageName_finance {
  SettleSheet = 'SettleSheet',
  AccountBalance = 'AccountBalance',
  TransactionFlow = 'TransactionFlow',
}
