// gm_api/src/finance/data.ts
import { getLocale } from './lng'

interface Enum_List_Item {
  text: string
  value: any
}

interface Enum_Map {
  [key: string]: string
}
export const list_Status_Code: Enum_List_Item[] = [
  { value: 2025001, text: getLocale('Status_Code__CODE_DIFFERENT_SUBJECT') },
  { value: 2025002, text: getLocale('Status_Code__CODE_NOW_ALLOW_OPERATE') },
  { value: 2025003, text: getLocale('Status_Code__CODE_NOT_ENOUGH_BALANCE') },
  { value: 2025004, text: getLocale('Status_Code__CODE_FAILED_CHECK_STOCK_SHEET') },
  { value: 2025005, text: getLocale('Status_Code__CODE_ONLY_ALLOW_APPROVED') },
  { value: 2025006, text: getLocale('Status_Code__CODE_DATA_EXPIRE') },
  { value: 2025007, text: getLocale('Status_Code__CODE_OPERATE_BY_OTHER') },
  { value: 2025008, text: getLocale('Status_Code__CODE_ITEM_JOIN_SHEET') },
  { value: 2025009, text: getLocale('Status_Code__CODE_NOT_PRODUCT_IN_REFUND_OUT') },
  { value: 2025010, text: getLocale('Status_Code__CODE_MANY_SETTLE_AMOUNT') },
  { value: 2025011, text: getLocale('Status_Code__CODE_ONLY_NOT_PAID') },
  { value: 2025012, text: getLocale('Status_Code__CODE_SHEET_NOT_NOT_ALLOW_PAY') },
  { value: 2025013, text: getLocale('Status_Code__CODE_SHEET_SETTLE_AMOUNT_LT_ZERO') },
  { value: 2025014, text: getLocale('Status_Code__CODE_ARRIVAL_SERIAL_NO_USED') },
]
export const map_Status_Code: Enum_Map = {
  2025001: getLocale('Status_Code__CODE_DIFFERENT_SUBJECT'),
  2025002: getLocale('Status_Code__CODE_NOW_ALLOW_OPERATE'),
  2025003: getLocale('Status_Code__CODE_NOT_ENOUGH_BALANCE'),
  2025004: getLocale('Status_Code__CODE_FAILED_CHECK_STOCK_SHEET'),
  2025005: getLocale('Status_Code__CODE_ONLY_ALLOW_APPROVED'),
  2025006: getLocale('Status_Code__CODE_DATA_EXPIRE'),
  2025007: getLocale('Status_Code__CODE_OPERATE_BY_OTHER'),
  2025008: getLocale('Status_Code__CODE_ITEM_JOIN_SHEET'),
  2025009: getLocale('Status_Code__CODE_NOT_PRODUCT_IN_REFUND_OUT'),
  2025010: getLocale('Status_Code__CODE_MANY_SETTLE_AMOUNT'),
  2025011: getLocale('Status_Code__CODE_ONLY_NOT_PAID'),
  2025012: getLocale('Status_Code__CODE_SHEET_NOT_NOT_ALLOW_PAY'),
  2025013: getLocale('Status_Code__CODE_SHEET_SETTLE_AMOUNT_LT_ZERO'),
  2025014: getLocale('Status_Code__CODE_ARRIVAL_SERIAL_NO_USED'),
}
export const list_AccountType: Enum_List_Item[] = [
  { value: 1, text: getLocale('AccountType__ACCOUNT_TYPE_SUPPLIER') },
  { value: 2, text: getLocale('AccountType__ACCOUNT_TYPE_CUSTOMER') },
]
export const map_AccountType: Enum_Map = {
  1: getLocale('AccountType__ACCOUNT_TYPE_SUPPLIER'),
  2: getLocale('AccountType__ACCOUNT_TYPE_CUSTOMER'),
}
export const list_TimeType: Enum_List_Item[] = [
  { value: 1, text: getLocale('TimeType__TIME_TYPE_CREATE_TIME') },
]
export const map_TimeType: Enum_Map = {
  1: getLocale('TimeType__TIME_TYPE_CREATE_TIME'),
}
export const list_SettleSheet_SheetStatus: Enum_List_Item[] = [
  { value: 1, text: getLocale('SettleSheet_SheetStatus__SHEET_STATUS_NOT_SUBMITTED') },
  { value: 2, text: getLocale('SettleSheet_SheetStatus__SHEET_STATUS_SUBMITTED_NOT_PAID') },
  { value: 3, text: getLocale('SettleSheet_SheetStatus__SHEET_STATUS_PAID') },
  { value: 4, text: getLocale('SettleSheet_SheetStatus__SHEET_STATUS_PART_PAID') },
  { value: 5, text: getLocale('SettleSheet_SheetStatus__SHEET_STATUS_NOT_APPROVED') },
  { value: 6, text: getLocale('SettleSheet_SheetStatus__SHEET_STATUS_DELETED') },
]
export const map_SettleSheet_SheetStatus: Enum_Map = {
  1: getLocale('SettleSheet_SheetStatus__SHEET_STATUS_NOT_SUBMITTED'),
  2: getLocale('SettleSheet_SheetStatus__SHEET_STATUS_SUBMITTED_NOT_PAID'),
  3: getLocale('SettleSheet_SheetStatus__SHEET_STATUS_PAID'),
  4: getLocale('SettleSheet_SheetStatus__SHEET_STATUS_PART_PAID'),
  5: getLocale('SettleSheet_SheetStatus__SHEET_STATUS_NOT_APPROVED'),
  6: getLocale('SettleSheet_SheetStatus__SHEET_STATUS_DELETED'),
}
export const list_TransactionFlow_ChangeType: Enum_List_Item[] = [
  { value: 1, text: getLocale('TransactionFlow_ChangeType__CHANGE_TYPE_RECHARGE') },
  { value: 2, text: getLocale('TransactionFlow_ChangeType__CHANGE_TYPE_REFUND') },
  { value: 3, text: getLocale('TransactionFlow_ChangeType__CHANGE_TYPE_SETTLE') },
  { value: 4, text: getLocale('TransactionFlow_ChangeType__CHANGE_TYPE_DEDUCTION') },
]
export const map_TransactionFlow_ChangeType: Enum_Map = {
  1: getLocale('TransactionFlow_ChangeType__CHANGE_TYPE_RECHARGE'),
  2: getLocale('TransactionFlow_ChangeType__CHANGE_TYPE_REFUND'),
  3: getLocale('TransactionFlow_ChangeType__CHANGE_TYPE_SETTLE'),
  4: getLocale('TransactionFlow_ChangeType__CHANGE_TYPE_DEDUCTION'),
}
export const list_CreateSettleSheetRequest_OperateType: Enum_List_Item[] = [
  { value: 1, text: getLocale('CreateSettleSheetRequest_OperateType__OPERATE_TYPE_JOIN_SHEET') },
  { value: 2, text: getLocale('CreateSettleSheetRequest_OperateType__OPERATE_TYPE_CREATE_SHEET') },
]
export const map_CreateSettleSheetRequest_OperateType: Enum_Map = {
  1: getLocale('CreateSettleSheetRequest_OperateType__OPERATE_TYPE_JOIN_SHEET'),
  2: getLocale('CreateSettleSheetRequest_OperateType__OPERATE_TYPE_CREATE_SHEET'),
}
