// gm_api/src/finance/methods.ts
import { Request } from '../util/request'
import {
  ListSettleSheetRequest, ListSettleSheetResponse,
  GetManySettleSheetRequest, GetManySettleSheetResponse,
  GetSettleSheetDetailRequest, GetSettleSheetDetailResponse,
  CreateSettleSheetRequest, CreateSettleSheetResponse,
  UpdateSettleSheetRequest, UpdateSettleSheetResponse,
  PaySettleRequest, PaySettleResponse,
  BatchPaySettleRequest, BatchPaySettleResponse,
  BatchSubmitSettleSheetRequest, BatchSubmitSettleSheetResponse,
  ExportSettleSheetRequest, ExportSettleSheetResponse,
  ExportAccountBalanceRequest, ExportAccountBalanceResponse,
  ExportFlowRequest, ExportFlowResponse,
  UpdateAccountBalanceRequest, UpdateAccountBalanceResponse,
  ListAccountBalanceRequest, ListAccountBalanceResponse,
  ListTransactionFlowRequest, ListTransactionFlowResponse,
} from './types'
export const ListSettleSheet = (r?: ListSettleSheetRequest, codes?: number[]) =>
  Request<ListSettleSheetResponse>('/ceres/finance/FinanceService/ListSettleSheet')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListSettleSheet.url = '/ceres/finance/FinanceService/ListSettleSheet'
export const GetManySettleSheet = (r?: GetManySettleSheetRequest, codes?: number[]) =>
  Request<GetManySettleSheetResponse>('/ceres/finance/FinanceService/GetManySettleSheet')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetManySettleSheet.url = '/ceres/finance/FinanceService/GetManySettleSheet'
export const GetSettleSheetDetail = (r?: GetSettleSheetDetailRequest, codes?: number[]) =>
  Request<GetSettleSheetDetailResponse>('/ceres/finance/FinanceService/GetSettleSheetDetail')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetSettleSheetDetail.url = '/ceres/finance/FinanceService/GetSettleSheetDetail'
export const CreateSettleSheet = (r?: CreateSettleSheetRequest, codes?: number[]) =>
  Request<CreateSettleSheetResponse>('/ceres/finance/FinanceService/CreateSettleSheet')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateSettleSheet.url = '/ceres/finance/FinanceService/CreateSettleSheet'
export const UpdateSettleSheet = (r?: UpdateSettleSheetRequest, codes?: number[]) =>
  Request<UpdateSettleSheetResponse>('/ceres/finance/FinanceService/UpdateSettleSheet')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateSettleSheet.url = '/ceres/finance/FinanceService/UpdateSettleSheet'
export const PaySettle = (r?: PaySettleRequest, codes?: number[]) =>
  Request<PaySettleResponse>('/ceres/finance/FinanceService/PaySettle')
    .code([...(codes || [])])
    .data(r || {})
    .run()
PaySettle.url = '/ceres/finance/FinanceService/PaySettle'
export const BatchPaySettle = (r?: BatchPaySettleRequest, codes?: number[]) =>
  Request<BatchPaySettleResponse>('/ceres/finance/FinanceService/BatchPaySettle')
    .code([...(codes || [])])
    .data(r || {})
    .run()
BatchPaySettle.url = '/ceres/finance/FinanceService/BatchPaySettle'
export const BatchSubmitSettleSheet = (r?: BatchSubmitSettleSheetRequest, codes?: number[]) =>
  Request<BatchSubmitSettleSheetResponse>('/ceres/finance/FinanceService/BatchSubmitSettleSheet')
    .code([...(codes || [])])
    .data(r || {})
    .run()
BatchSubmitSettleSheet.url = '/ceres/finance/FinanceService/BatchSubmitSettleSheet'
export const ExportSettleSheet = (r?: ExportSettleSheetRequest, codes?: number[]) =>
  Request<ExportSettleSheetResponse>('/ceres/finance/FinanceService/ExportSettleSheet')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ExportSettleSheet.url = '/ceres/finance/FinanceService/ExportSettleSheet'
export const ExportAccountBalance = (r?: ExportAccountBalanceRequest, codes?: number[]) =>
  Request<ExportAccountBalanceResponse>('/ceres/finance/FinanceService/ExportAccountBalance')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ExportAccountBalance.url = '/ceres/finance/FinanceService/ExportAccountBalance'
export const ExportFlow = (r?: ExportFlowRequest, codes?: number[]) =>
  Request<ExportFlowResponse>('/ceres/finance/FinanceService/ExportFlow')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ExportFlow.url = '/ceres/finance/FinanceService/ExportFlow'
export const UpdateAccountBalance = (r?: UpdateAccountBalanceRequest, codes?: number[]) =>
  Request<UpdateAccountBalanceResponse>('/ceres/finance/FinanceService/UpdateAccountBalance')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateAccountBalance.url = '/ceres/finance/FinanceService/UpdateAccountBalance'
export const ListAccountBalance = (r?: ListAccountBalanceRequest, codes?: number[]) =>
  Request<ListAccountBalanceResponse>('/ceres/finance/FinanceService/ListAccountBalance')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListAccountBalance.url = '/ceres/finance/FinanceService/ListAccountBalance'
export const ListTransactionFlow = (r?: ListTransactionFlowRequest, codes?: number[]) =>
  Request<ListTransactionFlowResponse>('/ceres/finance/FinanceService/ListTransactionFlow')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListTransactionFlow.url = '/ceres/finance/FinanceService/ListTransactionFlow'
