// gm_api/src/finance/pc.tsx
import React, { FC } from 'react'
import { Select, SelectProps } from '@gm-pc/react'
import {
  SelectList,
  SelectEnum,
  MoreSelectList,
  MapId,
} from '../util/pc_util'
import type {
  MapProps,
  SelectListProps,
  SelectEnumProps,
  MoreSelectListProps,
  MapIdProps,
} from '../util/pc_util'
import {
  list_Status_Code,
  map_Status_Code,
  list_AccountType,
  map_AccountType,
  list_TimeType,
  map_TimeType,
  list_SettleSheet_SheetStatus,
  map_SettleSheet_SheetStatus,
  list_TransactionFlow_ChangeType,
  map_TransactionFlow_ChangeType,
  list_CreateSettleSheetRequest_OperateType,
  map_CreateSettleSheetRequest_OperateType,
  SettleSheet,
  ListSettleSheet,
  ListSettleSheetResponse,
  AccountBalance,
  ListAccountBalance,
  ListAccountBalanceResponse,
  TransactionFlow,
  ListTransactionFlow,
  ListTransactionFlowResponse,
} from './index'

export const Select_Status_Code: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Status_Code} />
}

export const Map_Status_Code: FC<MapProps> = ({ value }) => {
  return <>{map_Status_Code[value]}</>
}

export const Select_AccountType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_AccountType} />
}

export const Map_AccountType: FC<MapProps> = ({ value }) => {
  return <>{map_AccountType[value]}</>
}

export const Select_TimeType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_TimeType} />
}

export const Map_TimeType: FC<MapProps> = ({ value }) => {
  return <>{map_TimeType[value]}</>
}

export const Select_SettleSheet_SheetStatus: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_SettleSheet_SheetStatus} />
}

export const Map_SettleSheet_SheetStatus: FC<MapProps> = ({ value }) => {
  return <>{map_SettleSheet_SheetStatus[value]}</>
}

export const Select_TransactionFlow_ChangeType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_TransactionFlow_ChangeType} />
}

export const Map_TransactionFlow_ChangeType: FC<MapProps> = ({ value }) => {
  return <>{map_TransactionFlow_ChangeType[value]}</>
}

export const Select_CreateSettleSheetRequest_OperateType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_CreateSettleSheetRequest_OperateType} />
}

export const Map_CreateSettleSheetRequest_OperateType: FC<MapProps> = ({ value }) => {
  return <>{map_CreateSettleSheetRequest_OperateType[value]}</>
}

export const Select_SettleSheet: FC<SelectListProps<
  SettleSheet,
  ListSettleSheetResponse
>> = (props) => {
  return (
    <SelectList<SettleSheet, ListSettleSheetResponse>
      {...props}
      messageName='SettleSheet'
      fetchList={ListSettleSheet}
    />
  )
}

export const MoreSelect_SettleSheet: FC<MoreSelectListProps<
  SettleSheet,
  ListSettleSheetResponse
>> = (props) => {
  return (
    <MoreSelectList<SettleSheet, ListSettleSheetResponse>
      {...props}
      messageName='SettleSheet'
      fetchList={ListSettleSheet}
    />
  )
}

export const Select_AccountBalance: FC<SelectListProps<
  AccountBalance,
  ListAccountBalanceResponse
>> = (props) => {
  return (
    <SelectList<AccountBalance, ListAccountBalanceResponse>
      {...props}
      messageName='AccountBalance'
      fetchList={ListAccountBalance}
    />
  )
}

export const MoreSelect_AccountBalance: FC<MoreSelectListProps<
  AccountBalance,
  ListAccountBalanceResponse
>> = (props) => {
  return (
    <MoreSelectList<AccountBalance, ListAccountBalanceResponse>
      {...props}
      messageName='AccountBalance'
      fetchList={ListAccountBalance}
    />
  )
}

export const Select_TransactionFlow: FC<SelectListProps<
  TransactionFlow,
  ListTransactionFlowResponse
>> = (props) => {
  return (
    <SelectList<TransactionFlow, ListTransactionFlowResponse>
      {...props}
      messageName='TransactionFlow'
      fetchList={ListTransactionFlow}
    />
  )
}

export const MoreSelect_TransactionFlow: FC<MoreSelectListProps<
  TransactionFlow,
  ListTransactionFlowResponse
>> = (props) => {
  return (
    <MoreSelectList<TransactionFlow, ListTransactionFlowResponse>
      {...props}
      messageName='TransactionFlow'
      fetchList={ListTransactionFlow}
    />
  )
}

export type {
  MapProps,
  SelectListProps,
  MoreSelectListProps,
}
