// gm_api/src/verification/pc.tsx
import React, { FC } from 'react'
import { Select, SelectProps } from '@gm-pc/react'
import {
  SelectList,
  SelectEnum,
  MoreSelectList,
  MapId,
} from '../util/pc_util'
import type {
  MapProps,
  SelectListProps,
  SelectEnumProps,
  MoreSelectListProps,
  MapIdProps,
} from '../util/pc_util'
import {
  list_Status_Code,
  map_Status_Code,
  list_SendType,
  map_SendType,
  list_CodeType,
  map_CodeType,
} from './index'

export const Select_Status_Code: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Status_Code} />
}

export const Map_Status_Code: FC<MapProps> = ({ value }) => {
  return <>{map_Status_Code[value]}</>
}

export const Select_SendType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_SendType} />
}

export const Map_SendType: FC<MapProps> = ({ value }) => {
  return <>{map_SendType[value]}</>
}

export const Select_CodeType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_CodeType} />
}

export const Map_CodeType: FC<MapProps> = ({ value }) => {
  return <>{map_CodeType[value]}</>
}

export type {
  MapProps,
  SelectListProps,
  MoreSelectListProps,
}
