// gm_api/src/aftersale/data.ts
import { getLocale } from './lng'

interface Enum_List_Item {
  text: string
  value: any
}

interface Enum_Map {
  [key: string]: string
}
export const list_AfterSaleOrderDetail_Type: Enum_List_Item[] = [
  { value: 1, text: getLocale('AfterSaleOrderDetail_Type__TYPE_ORDER') },
  { value: 2, text: getLocale('AfterSaleOrderDetail_Type__TYPE_REFUND') },
  { value: 3, text: getLocale('AfterSaleOrderDetail_Type__TYPE_REFUND_RETURN') },
]
export const map_AfterSaleOrderDetail_Type: Enum_Map = {
  1: getLocale('AfterSaleOrderDetail_Type__TYPE_ORDER'),
  2: getLocale('AfterSaleOrderDetail_Type__TYPE_REFUND'),
  3: getLocale('AfterSaleOrderDetail_Type__TYPE_REFUND_RETURN'),
}
export const list_AfterSaleOrderDetail_Reason: Enum_List_Item[] = [
  { value: 1, text: getLocale('AfterSaleOrderDetail_Reason__REASON_NOT_DELIVERED_ON_TIME') },
  { value: 2, text: getLocale('AfterSaleOrderDetail_Reason__REASON_MISSED_ORDERS') },
  { value: 3, text: getLocale('AfterSaleOrderDetail_Reason__REASON_SEND_THE_WRONG_ITEM') },
  { value: 4, text: getLocale('AfterSaleOrderDetail_Reason__REASON_WRONG_ORDER') },
  { value: 5, text: getLocale('AfterSaleOrderDetail_Reason__REASON_WEIGHT_WRONG') },
  { value: 6, text: getLocale('AfterSaleOrderDetail_Reason__REASON_QUALITY_ISSUES') },
  { value: 7, text: getLocale('AfterSaleOrderDetail_Reason__REASON_SPECIFICATION_ISSUE') },
  { value: 8, text: getLocale('AfterSaleOrderDetail_Reason__REASON_DRIVER_LOST_OR_BROKEN') },
  { value: 9, text: getLocale('AfterSaleOrderDetail_Reason__REASON_OUT_OF_STOCK_IN_THE_MARKET') },
  { value: 10, text: getLocale('AfterSaleOrderDetail_Reason__REASON_SYSTEM_PROBLEMS') },
  { value: 11, text: getLocale('AfterSaleOrderDetail_Reason__REASON_PROCUREMENT_ISSUES') },
  { value: 12, text: getLocale('AfterSaleOrderDetail_Reason__REASON_UNDELIVERABLE') },
  { value: 13, text: getLocale('AfterSaleOrderDetail_Reason__REASON_OTHER') },
]
export const map_AfterSaleOrderDetail_Reason: Enum_Map = {
  1: getLocale('AfterSaleOrderDetail_Reason__REASON_NOT_DELIVERED_ON_TIME'),
  2: getLocale('AfterSaleOrderDetail_Reason__REASON_MISSED_ORDERS'),
  3: getLocale('AfterSaleOrderDetail_Reason__REASON_SEND_THE_WRONG_ITEM'),
  4: getLocale('AfterSaleOrderDetail_Reason__REASON_WRONG_ORDER'),
  5: getLocale('AfterSaleOrderDetail_Reason__REASON_WEIGHT_WRONG'),
  6: getLocale('AfterSaleOrderDetail_Reason__REASON_QUALITY_ISSUES'),
  7: getLocale('AfterSaleOrderDetail_Reason__REASON_SPECIFICATION_ISSUE'),
  8: getLocale('AfterSaleOrderDetail_Reason__REASON_DRIVER_LOST_OR_BROKEN'),
  9: getLocale('AfterSaleOrderDetail_Reason__REASON_OUT_OF_STOCK_IN_THE_MARKET'),
  10: getLocale('AfterSaleOrderDetail_Reason__REASON_SYSTEM_PROBLEMS'),
  11: getLocale('AfterSaleOrderDetail_Reason__REASON_PROCUREMENT_ISSUES'),
  12: getLocale('AfterSaleOrderDetail_Reason__REASON_UNDELIVERABLE'),
  13: getLocale('AfterSaleOrderDetail_Reason__REASON_OTHER'),
}
export const list_AfterSaleOrderDetail_Method: Enum_List_Item[] = [
  { value: 1, text: getLocale('AfterSaleOrderDetail_Method__METHOD_GOODS_REDUCTION') },
  { value: 2, text: getLocale('AfterSaleOrderDetail_Method__METHOD_REPLENISHMENT') },
  { value: 3, text: getLocale('AfterSaleOrderDetail_Method__METHOD_EXCHANGE') },
  { value: 4, text: getLocale('AfterSaleOrderDetail_Method__METHOD_BILLING') },
  { value: 5, text: getLocale('AfterSaleOrderDetail_Method__METHOD_CUSTOMER_COMMUNICATE') },
  { value: 6, text: getLocale('AfterSaleOrderDetail_Method__METHOD_METHOD_OTHER') },
]
export const map_AfterSaleOrderDetail_Method: Enum_Map = {
  1: getLocale('AfterSaleOrderDetail_Method__METHOD_GOODS_REDUCTION'),
  2: getLocale('AfterSaleOrderDetail_Method__METHOD_REPLENISHMENT'),
  3: getLocale('AfterSaleOrderDetail_Method__METHOD_EXCHANGE'),
  4: getLocale('AfterSaleOrderDetail_Method__METHOD_BILLING'),
  5: getLocale('AfterSaleOrderDetail_Method__METHOD_CUSTOMER_COMMUNICATE'),
  6: getLocale('AfterSaleOrderDetail_Method__METHOD_METHOD_OTHER'),
}
export const list_AfterSaleOrderDetail_TaskStatus: Enum_List_Item[] = [
  { value: 1, text: getLocale('AfterSaleOrderDetail_TaskStatus__TASK_STATUS_UNDONE') },
  { value: 2, text: getLocale('AfterSaleOrderDetail_TaskStatus__TASK_STATUS_DONE') },
]
export const map_AfterSaleOrderDetail_TaskStatus: Enum_Map = {
  1: getLocale('AfterSaleOrderDetail_TaskStatus__TASK_STATUS_UNDONE'),
  2: getLocale('AfterSaleOrderDetail_TaskStatus__TASK_STATUS_DONE'),
}
export const list_AfterSaleOrderDetail_TaskMethod: Enum_List_Item[] = [
  { value: 1, text: getLocale('AfterSaleOrderDetail_TaskMethod__TASK_METHOD_PICK_UP') },
  { value: 2, text: getLocale('AfterSaleOrderDetail_TaskMethod__TASK_METHOD_GIVE_UP_PICKUP') },
]
export const map_AfterSaleOrderDetail_TaskMethod: Enum_Map = {
  1: getLocale('AfterSaleOrderDetail_TaskMethod__TASK_METHOD_PICK_UP'),
  2: getLocale('AfterSaleOrderDetail_TaskMethod__TASK_METHOD_GIVE_UP_PICKUP'),
}
export const list_AfterSaleOrder_Status: Enum_List_Item[] = [
  { value: 1, text: getLocale('AfterSaleOrder_Status__STATUS_TO_SUBMIT') },
  { value: 2, text: getLocale('AfterSaleOrder_Status__STATUS_SUBMITTED') },
  { value: 3, text: getLocale('AfterSaleOrder_Status__STATUS_COMPLETED') },
  { value: 4, text: getLocale('AfterSaleOrder_Status__STATUS_DELETED') },
]
export const map_AfterSaleOrder_Status: Enum_Map = {
  1: getLocale('AfterSaleOrder_Status__STATUS_TO_SUBMIT'),
  2: getLocale('AfterSaleOrder_Status__STATUS_SUBMITTED'),
  3: getLocale('AfterSaleOrder_Status__STATUS_COMPLETED'),
  4: getLocale('AfterSaleOrder_Status__STATUS_DELETED'),
}
export const list_TimeType: Enum_List_Item[] = [
  { value: 1, text: getLocale('TimeType__TIME_TYPE_CREATE_AFTER_SALE') },
  { value: 2, text: getLocale('TimeType__TIME_TYPE_PLACE_ORDER') },
  { value: 3, text: getLocale('TimeType__TIME_TYPE_RECEIVE_GOODS') },
]
export const map_TimeType: Enum_Map = {
  1: getLocale('TimeType__TIME_TYPE_CREATE_AFTER_SALE'),
  2: getLocale('TimeType__TIME_TYPE_PLACE_ORDER'),
  3: getLocale('TimeType__TIME_TYPE_RECEIVE_GOODS'),
}
