// gm_api/src/aftersale/methods.ts
import { Request } from '../util/request'
import {
  CreateAfterSaleOrderRequest, CreateAfterSaleOrderResponse,
  ListAfterSaleOrderRequest, ListAfterSaleOrderResponse,
  ExportAfterSaleOrderRequest, ExportAfterSaleOrderResponse,
  GetAfterSaleOrderRequest, GetAfterSaleOrderResponse,
  UpdateAfterSaleOrderRequest, UpdateAfterSaleOrderResponse,
  DeleteAfterSaleOrderRequest, DeleteAfterSaleOrderResponse,
  ListAfterSaleOrderTaskRequest, ListAfterSaleOrderTaskResponse,
  ExportAfterSaleOrderTaskRequest, ExportAfterSaleOrderTaskResponse,
  UpdateAfterSaleOrderTaskRequest, UpdateAfterSaleOrderTaskResponse,
  BatchUpdateAfterSaleOrderTaskMethodRequest, BatchUpdateAfterSaleOrderTaskMethodResponse,
  BatchCompleteAfterSaleOrderTaskRequest, BatchCompleteAfterSaleOrderTaskResponse,
  BatchUpdateAfterSaleOrderTaskDriverRequest, BatchUpdateAfterSaleOrderTaskDriverResponse,
  ListAfterSaleOrderSheetRequest, ListAfterSaleOrderSheetResponse,
  ExportAfterSaleOrderSheetRequest, ExportAfterSaleOrderSheetResponse,
  ListAfterSaleOrderDetailSheetRequest, ListAfterSaleOrderDetailSheetResponse,
  ExportAfterSaleOrderDetailSheetRequest, ExportAfterSaleOrderDetailSheetResponse,
  GetAfterSaleOrderDetailRequest, GetAfterSaleOrderDetailResponse,
} from './types'
export const CreateAfterSaleOrder = (r?: CreateAfterSaleOrderRequest, codes?: number[]) =>
  Request<CreateAfterSaleOrderResponse>('/ceres/aftersale/AfterSaleService/CreateAfterSaleOrder')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateAfterSaleOrder.url = '/ceres/aftersale/AfterSaleService/CreateAfterSaleOrder'
export const ListAfterSaleOrder = (r?: ListAfterSaleOrderRequest, codes?: number[]) =>
  Request<ListAfterSaleOrderResponse>('/ceres/aftersale/AfterSaleService/ListAfterSaleOrder')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListAfterSaleOrder.url = '/ceres/aftersale/AfterSaleService/ListAfterSaleOrder'
export const ExportAfterSaleOrder = (r?: ExportAfterSaleOrderRequest, codes?: number[]) =>
  Request<ExportAfterSaleOrderResponse>('/ceres/aftersale/AfterSaleService/ExportAfterSaleOrder')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ExportAfterSaleOrder.url = '/ceres/aftersale/AfterSaleService/ExportAfterSaleOrder'
export const GetAfterSaleOrder = (r?: GetAfterSaleOrderRequest, codes?: number[]) =>
  Request<GetAfterSaleOrderResponse>('/ceres/aftersale/AfterSaleService/GetAfterSaleOrder')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetAfterSaleOrder.url = '/ceres/aftersale/AfterSaleService/GetAfterSaleOrder'
export const UpdateAfterSaleOrder = (r?: UpdateAfterSaleOrderRequest, codes?: number[]) =>
  Request<UpdateAfterSaleOrderResponse>('/ceres/aftersale/AfterSaleService/UpdateAfterSaleOrder')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateAfterSaleOrder.url = '/ceres/aftersale/AfterSaleService/UpdateAfterSaleOrder'
export const DeleteAfterSaleOrder = (r?: DeleteAfterSaleOrderRequest, codes?: number[]) =>
  Request<DeleteAfterSaleOrderResponse>('/ceres/aftersale/AfterSaleService/DeleteAfterSaleOrder')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteAfterSaleOrder.url = '/ceres/aftersale/AfterSaleService/DeleteAfterSaleOrder'
export const ListAfterSaleOrderTask = (r?: ListAfterSaleOrderTaskRequest, codes?: number[]) =>
  Request<ListAfterSaleOrderTaskResponse>('/ceres/aftersale/AfterSaleService/ListAfterSaleOrderTask')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListAfterSaleOrderTask.url = '/ceres/aftersale/AfterSaleService/ListAfterSaleOrderTask'
export const ExportAfterSaleOrderTask = (r?: ExportAfterSaleOrderTaskRequest, codes?: number[]) =>
  Request<ExportAfterSaleOrderTaskResponse>('/ceres/aftersale/AfterSaleService/ExportAfterSaleOrderTask')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ExportAfterSaleOrderTask.url = '/ceres/aftersale/AfterSaleService/ExportAfterSaleOrderTask'
export const UpdateAfterSaleOrderTask = (r?: UpdateAfterSaleOrderTaskRequest, codes?: number[]) =>
  Request<UpdateAfterSaleOrderTaskResponse>('/ceres/aftersale/AfterSaleService/UpdateAfterSaleOrderTask')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateAfterSaleOrderTask.url = '/ceres/aftersale/AfterSaleService/UpdateAfterSaleOrderTask'
export const BatchUpdateAfterSaleOrderTaskMethod = (r?: BatchUpdateAfterSaleOrderTaskMethodRequest, codes?: number[]) =>
  Request<BatchUpdateAfterSaleOrderTaskMethodResponse>('/ceres/aftersale/AfterSaleService/BatchUpdateAfterSaleOrderTaskMethod')
    .code([...(codes || [])])
    .data(r || {})
    .run()
BatchUpdateAfterSaleOrderTaskMethod.url = '/ceres/aftersale/AfterSaleService/BatchUpdateAfterSaleOrderTaskMethod'
export const BatchCompleteAfterSaleOrderTask = (r?: BatchCompleteAfterSaleOrderTaskRequest, codes?: number[]) =>
  Request<BatchCompleteAfterSaleOrderTaskResponse>('/ceres/aftersale/AfterSaleService/BatchCompleteAfterSaleOrderTask')
    .code([...(codes || [])])
    .data(r || {})
    .run()
BatchCompleteAfterSaleOrderTask.url = '/ceres/aftersale/AfterSaleService/BatchCompleteAfterSaleOrderTask'
export const BatchUpdateAfterSaleOrderTaskDriver = (r?: BatchUpdateAfterSaleOrderTaskDriverRequest, codes?: number[]) =>
  Request<BatchUpdateAfterSaleOrderTaskDriverResponse>('/ceres/aftersale/AfterSaleService/BatchUpdateAfterSaleOrderTaskDriver')
    .code([...(codes || [])])
    .data(r || {})
    .run()
BatchUpdateAfterSaleOrderTaskDriver.url = '/ceres/aftersale/AfterSaleService/BatchUpdateAfterSaleOrderTaskDriver'
export const ListAfterSaleOrderSheet = (r?: ListAfterSaleOrderSheetRequest, codes?: number[]) =>
  Request<ListAfterSaleOrderSheetResponse>('/ceres/aftersale/AfterSaleService/ListAfterSaleOrderSheet')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListAfterSaleOrderSheet.url = '/ceres/aftersale/AfterSaleService/ListAfterSaleOrderSheet'
export const ExportAfterSaleOrderSheet = (r?: ExportAfterSaleOrderSheetRequest, codes?: number[]) =>
  Request<ExportAfterSaleOrderSheetResponse>('/ceres/aftersale/AfterSaleService/ExportAfterSaleOrderSheet')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ExportAfterSaleOrderSheet.url = '/ceres/aftersale/AfterSaleService/ExportAfterSaleOrderSheet'
export const ListAfterSaleOrderDetailSheet = (r?: ListAfterSaleOrderDetailSheetRequest, codes?: number[]) =>
  Request<ListAfterSaleOrderDetailSheetResponse>('/ceres/aftersale/AfterSaleService/ListAfterSaleOrderDetailSheet')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListAfterSaleOrderDetailSheet.url = '/ceres/aftersale/AfterSaleService/ListAfterSaleOrderDetailSheet'
export const ExportAfterSaleOrderDetailSheet = (r?: ExportAfterSaleOrderDetailSheetRequest, codes?: number[]) =>
  Request<ExportAfterSaleOrderDetailSheetResponse>('/ceres/aftersale/AfterSaleService/ExportAfterSaleOrderDetailSheet')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ExportAfterSaleOrderDetailSheet.url = '/ceres/aftersale/AfterSaleService/ExportAfterSaleOrderDetailSheet'
export const GetAfterSaleOrderDetail = (r?: GetAfterSaleOrderDetailRequest, codes?: number[]) =>
  Request<GetAfterSaleOrderDetailResponse>('/ceres/aftersale/AfterSaleService/GetAfterSaleOrderDetail')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetAfterSaleOrderDetail.url = '/ceres/aftersale/AfterSaleService/GetAfterSaleOrderDetail'
