// gm_api/src/aftersale/pc.tsx
import React, { FC } from 'react'
import { Select, SelectProps } from '@gm-pc/react'
import {
  SelectList,
  SelectEnum,
  MoreSelectList,
  MapId,
} from '../util/pc_util'
import type {
  MapProps,
  SelectListProps,
  SelectEnumProps,
  MoreSelectListProps,
  MapIdProps,
} from '../util/pc_util'
import {
  list_AfterSaleOrderDetail_Type,
  map_AfterSaleOrderDetail_Type,
  list_AfterSaleOrderDetail_Reason,
  map_AfterSaleOrderDetail_Reason,
  list_AfterSaleOrderDetail_Method,
  map_AfterSaleOrderDetail_Method,
  list_AfterSaleOrderDetail_TaskStatus,
  map_AfterSaleOrderDetail_TaskStatus,
  list_AfterSaleOrderDetail_TaskMethod,
  map_AfterSaleOrderDetail_TaskMethod,
  list_AfterSaleOrder_Status,
  map_AfterSaleOrder_Status,
  list_TimeType,
  map_TimeType,
  AfterSaleOrder,
  ListAfterSaleOrder,
  ListAfterSaleOrderResponse,
} from './index'

export const Select_AfterSaleOrderDetail_Type: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_AfterSaleOrderDetail_Type} />
}

export const Map_AfterSaleOrderDetail_Type: FC<MapProps> = ({ value }) => {
  return <>{map_AfterSaleOrderDetail_Type[value]}</>
}

export const Select_AfterSaleOrderDetail_Reason: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_AfterSaleOrderDetail_Reason} />
}

export const Map_AfterSaleOrderDetail_Reason: FC<MapProps> = ({ value }) => {
  return <>{map_AfterSaleOrderDetail_Reason[value]}</>
}

export const Select_AfterSaleOrderDetail_Method: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_AfterSaleOrderDetail_Method} />
}

export const Map_AfterSaleOrderDetail_Method: FC<MapProps> = ({ value }) => {
  return <>{map_AfterSaleOrderDetail_Method[value]}</>
}

export const Select_AfterSaleOrderDetail_TaskStatus: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_AfterSaleOrderDetail_TaskStatus} />
}

export const Map_AfterSaleOrderDetail_TaskStatus: FC<MapProps> = ({ value }) => {
  return <>{map_AfterSaleOrderDetail_TaskStatus[value]}</>
}

export const Select_AfterSaleOrderDetail_TaskMethod: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_AfterSaleOrderDetail_TaskMethod} />
}

export const Map_AfterSaleOrderDetail_TaskMethod: FC<MapProps> = ({ value }) => {
  return <>{map_AfterSaleOrderDetail_TaskMethod[value]}</>
}

export const Select_AfterSaleOrder_Status: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_AfterSaleOrder_Status} />
}

export const Map_AfterSaleOrder_Status: FC<MapProps> = ({ value }) => {
  return <>{map_AfterSaleOrder_Status[value]}</>
}

export const Select_TimeType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_TimeType} />
}

export const Map_TimeType: FC<MapProps> = ({ value }) => {
  return <>{map_TimeType[value]}</>
}

export const Select_AfterSaleOrder: FC<SelectListProps<
  AfterSaleOrder,
  ListAfterSaleOrderResponse
>> = (props) => {
  return (
    <SelectList<AfterSaleOrder, ListAfterSaleOrderResponse>
      {...props}
      messageName='AfterSaleOrder'
      fetchList={ListAfterSaleOrder}
    />
  )
}

export const MoreSelect_AfterSaleOrder: FC<MoreSelectListProps<
  AfterSaleOrder,
  ListAfterSaleOrderResponse
>> = (props) => {
  return (
    <MoreSelectList<AfterSaleOrder, ListAfterSaleOrderResponse>
      {...props}
      messageName='AfterSaleOrder'
      fetchList={ListAfterSaleOrder}
    />
  )
}

export type {
  MapProps,
  SelectListProps,
  MoreSelectListProps,
}
