// gm_api/src/purchase/data.ts
import { getLocale } from './lng'

interface Enum_List_Item {
  text: string
  value: any
}

interface Enum_Map {
  [key: string]: string
}
export const list_Status_Code: Enum_List_Item[] = [
  { value: 23450002, text: getLocale('Status_Code__SSU_SNAP_NOT_FOUND') },
  { value: 23450003, text: getLocale('Status_Code__PURCHASE_UNIT_NOT_FOUND') },
  { value: 23450004, text: getLocale('Status_Code__SKU_NOT_FOUND') },
  { value: 23450005, text: getLocale('Status_Code__UNIT_NOT_FOUND') },
  { value: 23450006, text: getLocale('Status_Code__PURCHASE_SHEET_SUBMIT_WITH_TASK_FINISHED') },
  { value: 23450007, text: getLocale('Status_Code__ASYNC_PARAMS_MISMATCH') },
  { value: 23450008, text: getLocale('Status_Code__SYNC_LACK_PURCHASE_TIME') },
  { value: 23450009, text: getLocale('Status_Code__SYNC_LACK_SHEET_ID') },
  { value: 23450010, text: getLocale('Status_Code__PURCHASE_TASK_DATA_ERROR') },
  { value: 23450011, text: getLocale('Status_Code__SYNC_LACK_SHEET_VALUE') },
  { value: 23450012, text: getLocale('Status_Code__PURCHASE_TASK_FAILURE_VERSION') },
]
export const map_Status_Code: Enum_Map = {
  23450002: getLocale('Status_Code__SSU_SNAP_NOT_FOUND'),
  23450003: getLocale('Status_Code__PURCHASE_UNIT_NOT_FOUND'),
  23450004: getLocale('Status_Code__SKU_NOT_FOUND'),
  23450005: getLocale('Status_Code__UNIT_NOT_FOUND'),
  23450006: getLocale('Status_Code__PURCHASE_SHEET_SUBMIT_WITH_TASK_FINISHED'),
  23450007: getLocale('Status_Code__ASYNC_PARAMS_MISMATCH'),
  23450008: getLocale('Status_Code__SYNC_LACK_PURCHASE_TIME'),
  23450009: getLocale('Status_Code__SYNC_LACK_SHEET_ID'),
  23450010: getLocale('Status_Code__PURCHASE_TASK_DATA_ERROR'),
  23450011: getLocale('Status_Code__SYNC_LACK_SHEET_VALUE'),
  23450012: getLocale('Status_Code__PURCHASE_TASK_FAILURE_VERSION'),
}
export const list_TimeType: Enum_List_Item[] = [
  { value: 256, text: getLocale('TimeType__CREATE_TIME') },
  { value: 512, text: getLocale('TimeType__PURCHASE_TIME') },
  { value: 1024, text: getLocale('TimeType__RELEASE_TIME') },
]
export const map_TimeType: Enum_Map = {
  256: getLocale('TimeType__CREATE_TIME'),
  512: getLocale('TimeType__PURCHASE_TIME'),
  1024: getLocale('TimeType__RELEASE_TIME'),
}
export const list_ListPurchaseTaskRequest_PagingField: Enum_List_Item[] = [
  { value: 1, text: getLocale('ListPurchaseTaskRequest_PagingField__CATEGORY') },
  { value: 2, text: getLocale('ListPurchaseTaskRequest_PagingField__SUPPLIER') },
  { value: 3, text: getLocale('ListPurchaseTaskRequest_PagingField__PURCHASER') },
  { value: 4, text: getLocale('ListPurchaseTaskRequest_PagingField__SKU') },
]
export const map_ListPurchaseTaskRequest_PagingField: Enum_Map = {
  1: getLocale('ListPurchaseTaskRequest_PagingField__CATEGORY'),
  2: getLocale('ListPurchaseTaskRequest_PagingField__SUPPLIER'),
  3: getLocale('ListPurchaseTaskRequest_PagingField__PURCHASER'),
  4: getLocale('ListPurchaseTaskRequest_PagingField__SKU'),
}
export const list_PurchaseTask_Status: Enum_List_Item[] = [
  { value: 256, text: getLocale('PurchaseTask_Status__PREPARE') },
  { value: 512, text: getLocale('PurchaseTask_Status__RELEASED') },
  { value: 1024, text: getLocale('PurchaseTask_Status__FINISHED') },
]
export const map_PurchaseTask_Status: Enum_Map = {
  256: getLocale('PurchaseTask_Status__PREPARE'),
  512: getLocale('PurchaseTask_Status__RELEASED'),
  1024: getLocale('PurchaseTask_Status__FINISHED'),
}
export const list_PurchaseTask_RequestSource: Enum_List_Item[] = [
  { value: 256, text: getLocale('PurchaseTask_RequestSource__ORDER') },
  { value: 512, text: getLocale('PurchaseTask_RequestSource__PROCESS') },
  { value: 1024, text: getLocale('PurchaseTask_RequestSource__STOCK') },
  { value: 2048, text: getLocale('PurchaseTask_RequestSource__MANUAL') },
  { value: 4096, text: getLocale('PurchaseTask_RequestSource__DIFF') },
  { value: 8192, text: getLocale('PurchaseTask_RequestSource__PACK') },
]
export const map_PurchaseTask_RequestSource: Enum_Map = {
  256: getLocale('PurchaseTask_RequestSource__ORDER'),
  512: getLocale('PurchaseTask_RequestSource__PROCESS'),
  1024: getLocale('PurchaseTask_RequestSource__STOCK'),
  2048: getLocale('PurchaseTask_RequestSource__MANUAL'),
  4096: getLocale('PurchaseTask_RequestSource__DIFF'),
  8192: getLocale('PurchaseTask_RequestSource__PACK'),
}
export const list_PurchaseTask_Type: Enum_List_Item[] = [
  { value: 256, text: getLocale('PurchaseTask_Type__TMP') },
  { value: 512, text: getLocale('PurchaseTask_Type__COMMON') },
]
export const map_PurchaseTask_Type: Enum_Map = {
  256: getLocale('PurchaseTask_Type__TMP'),
  512: getLocale('PurchaseTask_Type__COMMON'),
}
export const list_InquiryPrice_SourceType: Enum_List_Item[] = [
  { value: 256, text: getLocale('InquiryPrice_SourceType__STATION') },
  { value: 512, text: getLocale('InquiryPrice_SourceType__PURCHASE_APP') },
  { value: 1024, text: getLocale('InquiryPrice_SourceType__SUPPLIER_APP') },
]
export const map_InquiryPrice_SourceType: Enum_Map = {
  256: getLocale('InquiryPrice_SourceType__STATION'),
  512: getLocale('InquiryPrice_SourceType__PURCHASE_APP'),
  1024: getLocale('InquiryPrice_SourceType__SUPPLIER_APP'),
}
export const list_PurchaseSheet_Status: Enum_List_Item[] = [
  { value: 256, text: getLocale('PurchaseSheet_Status__DRAFT') },
  { value: 512, text: getLocale('PurchaseSheet_Status__COMMIT') },
]
export const map_PurchaseSheet_Status: Enum_Map = {
  256: getLocale('PurchaseSheet_Status__DRAFT'),
  512: getLocale('PurchaseSheet_Status__COMMIT'),
}
