// gm_api/src/purchase/methods.ts
import { Request } from '../util/request'
import {
  CreatePurchaseTaskRequest, CreatePurchaseTaskResponse,
  SyncPurchaseTaskRequest, SyncPurchaseTaskResponse,
  UpdatePurchaseTaskRequest, UpdatePurchaseTaskResponse,
  AsyncReleasePurchaseTaskRequest, AsyncReleasePurchaseTaskResponse,
  AsyncFinishPurchaseTaskRequest, AsyncFinishPurchaseTaskResponse,
  AsyncSwitchPurchaseTaskPurchaserRequest, AsyncSwitchPurchaseTaskPurchaserResponse,
  AsyncSwitchPurchaseTaskSupplierRequest, AsyncSwitchPurchaseTaskSupplierResponse,
  DeletePurchaseTaskRequest, DeletePurchaseTaskResponse,
  GetPurchaseTaskRequest, GetPurchaseTaskResponse,
  ListPurchaseTaskRequest, ListPurchaseTaskResponse,
  ExportPurchaseTaskRequest, ExportPurchaseTaskResponse,
  CreateSheetFromTaskRequest, CreateSheetFromTaskResponse,
  AsyncCreateSheetFromTaskRequest, AsyncCreateSheetFromTaskResponse,
  GetPurchaseTaskSummaryRequest, GetPurchaseTaskSummaryResponse,
  ListBatchRequest, ListBatchResponse,
  CreatePurchaseSheetRequest, CreatePurchaseSheetResponse,
  UpdatePurchaseSheetRequest, UpdatePurchaseSheetResponse,
  DeletePurchaseSheetRequest, DeletePurchaseSheetResponse,
  GetPurchaseSheetRequest, GetPurchaseSheetResponse,
  ListPurchaseSheetRequest, ListPurchaseSheetResponse,
  CreateInquiryPriceRequest, CreateInquiryPriceResponse,
  ListInquiryPriceRequest, ListInquiryPriceResponse,
  ImportInquiryPriceRequest, ImportInquiryPriceResponse,
  ImportInquiryPriceTmplRequest, ImportInquiryPriceTmplResponse,
} from './types'
export const CreatePurchaseTask = (r?: CreatePurchaseTaskRequest, codes?: number[]) =>
  Request<CreatePurchaseTaskResponse>('/ceres/purchase/PurchaseService/CreatePurchaseTask')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreatePurchaseTask.url = '/ceres/purchase/PurchaseService/CreatePurchaseTask'
export const SyncPurchaseTask = (r?: SyncPurchaseTaskRequest, codes?: number[]) =>
  Request<SyncPurchaseTaskResponse>('/ceres/purchase/PurchaseService/SyncPurchaseTask')
    .code([...(codes || [])])
    .data(r || {})
    .run()
SyncPurchaseTask.url = '/ceres/purchase/PurchaseService/SyncPurchaseTask'
export const UpdatePurchaseTask = (r?: UpdatePurchaseTaskRequest, codes?: number[]) =>
  Request<UpdatePurchaseTaskResponse>('/ceres/purchase/PurchaseService/UpdatePurchaseTask')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdatePurchaseTask.url = '/ceres/purchase/PurchaseService/UpdatePurchaseTask'
export const AsyncReleasePurchaseTask = (r?: AsyncReleasePurchaseTaskRequest, codes?: number[]) =>
  Request<AsyncReleasePurchaseTaskResponse>('/ceres/purchase/PurchaseService/AsyncReleasePurchaseTask')
    .code([...(codes || [])])
    .data(r || {})
    .run()
AsyncReleasePurchaseTask.url = '/ceres/purchase/PurchaseService/AsyncReleasePurchaseTask'
export const AsyncFinishPurchaseTask = (r?: AsyncFinishPurchaseTaskRequest, codes?: number[]) =>
  Request<AsyncFinishPurchaseTaskResponse>('/ceres/purchase/PurchaseService/AsyncFinishPurchaseTask')
    .code([...(codes || [])])
    .data(r || {})
    .run()
AsyncFinishPurchaseTask.url = '/ceres/purchase/PurchaseService/AsyncFinishPurchaseTask'
export const AsyncSwitchPurchaseTaskPurchaser = (r?: AsyncSwitchPurchaseTaskPurchaserRequest, codes?: number[]) =>
  Request<AsyncSwitchPurchaseTaskPurchaserResponse>('/ceres/purchase/PurchaseService/AsyncSwitchPurchaseTaskPurchaser')
    .code([...(codes || [])])
    .data(r || {})
    .run()
AsyncSwitchPurchaseTaskPurchaser.url = '/ceres/purchase/PurchaseService/AsyncSwitchPurchaseTaskPurchaser'
export const AsyncSwitchPurchaseTaskSupplier = (r?: AsyncSwitchPurchaseTaskSupplierRequest, codes?: number[]) =>
  Request<AsyncSwitchPurchaseTaskSupplierResponse>('/ceres/purchase/PurchaseService/AsyncSwitchPurchaseTaskSupplier')
    .code([...(codes || [])])
    .data(r || {})
    .run()
AsyncSwitchPurchaseTaskSupplier.url = '/ceres/purchase/PurchaseService/AsyncSwitchPurchaseTaskSupplier'
export const DeletePurchaseTask = (r?: DeletePurchaseTaskRequest, codes?: number[]) =>
  Request<DeletePurchaseTaskResponse>('/ceres/purchase/PurchaseService/DeletePurchaseTask')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeletePurchaseTask.url = '/ceres/purchase/PurchaseService/DeletePurchaseTask'
export const GetPurchaseTask = (r?: GetPurchaseTaskRequest, codes?: number[]) =>
  Request<GetPurchaseTaskResponse>('/ceres/purchase/PurchaseService/GetPurchaseTask')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetPurchaseTask.url = '/ceres/purchase/PurchaseService/GetPurchaseTask'
export const ListPurchaseTask = (r?: ListPurchaseTaskRequest, codes?: number[]) =>
  Request<ListPurchaseTaskResponse>('/ceres/purchase/PurchaseService/ListPurchaseTask')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListPurchaseTask.url = '/ceres/purchase/PurchaseService/ListPurchaseTask'
export const ExportPurchaseTask = (r?: ExportPurchaseTaskRequest, codes?: number[]) =>
  Request<ExportPurchaseTaskResponse>('/ceres/purchase/PurchaseService/ExportPurchaseTask')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ExportPurchaseTask.url = '/ceres/purchase/PurchaseService/ExportPurchaseTask'
export const CreateSheetFromTask = (r?: CreateSheetFromTaskRequest, codes?: number[]) =>
  Request<CreateSheetFromTaskResponse>('/ceres/purchase/PurchaseService/CreateSheetFromTask')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateSheetFromTask.url = '/ceres/purchase/PurchaseService/CreateSheetFromTask'
export const AsyncCreateSheetFromTask = (r?: AsyncCreateSheetFromTaskRequest, codes?: number[]) =>
  Request<AsyncCreateSheetFromTaskResponse>('/ceres/purchase/PurchaseService/AsyncCreateSheetFromTask')
    .code([...(codes || [])])
    .data(r || {})
    .run()
AsyncCreateSheetFromTask.url = '/ceres/purchase/PurchaseService/AsyncCreateSheetFromTask'
export const GetPurchaseTaskSummary = (r?: GetPurchaseTaskSummaryRequest, codes?: number[]) =>
  Request<GetPurchaseTaskSummaryResponse>('/ceres/purchase/PurchaseService/GetPurchaseTaskSummary')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetPurchaseTaskSummary.url = '/ceres/purchase/PurchaseService/GetPurchaseTaskSummary'
export const ListBatch = (r?: ListBatchRequest, codes?: number[]) =>
  Request<ListBatchResponse>('/ceres/purchase/PurchaseService/ListBatch')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListBatch.url = '/ceres/purchase/PurchaseService/ListBatch'
export const CreatePurchaseSheet = (r?: CreatePurchaseSheetRequest, codes?: number[]) =>
  Request<CreatePurchaseSheetResponse>('/ceres/purchase/PurchaseService/CreatePurchaseSheet')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreatePurchaseSheet.url = '/ceres/purchase/PurchaseService/CreatePurchaseSheet'
export const UpdatePurchaseSheet = (r?: UpdatePurchaseSheetRequest, codes?: number[]) =>
  Request<UpdatePurchaseSheetResponse>('/ceres/purchase/PurchaseService/UpdatePurchaseSheet')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdatePurchaseSheet.url = '/ceres/purchase/PurchaseService/UpdatePurchaseSheet'
export const DeletePurchaseSheet = (r?: DeletePurchaseSheetRequest, codes?: number[]) =>
  Request<DeletePurchaseSheetResponse>('/ceres/purchase/PurchaseService/DeletePurchaseSheet')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeletePurchaseSheet.url = '/ceres/purchase/PurchaseService/DeletePurchaseSheet'
export const GetPurchaseSheet = (r?: GetPurchaseSheetRequest, codes?: number[]) =>
  Request<GetPurchaseSheetResponse>('/ceres/purchase/PurchaseService/GetPurchaseSheet')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetPurchaseSheet.url = '/ceres/purchase/PurchaseService/GetPurchaseSheet'
export const ListPurchaseSheet = (r?: ListPurchaseSheetRequest, codes?: number[]) =>
  Request<ListPurchaseSheetResponse>('/ceres/purchase/PurchaseService/ListPurchaseSheet')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListPurchaseSheet.url = '/ceres/purchase/PurchaseService/ListPurchaseSheet'
export const CreateInquiryPrice = (r?: CreateInquiryPriceRequest, codes?: number[]) =>
  Request<CreateInquiryPriceResponse>('/ceres/purchase/PurchaseService/CreateInquiryPrice')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateInquiryPrice.url = '/ceres/purchase/PurchaseService/CreateInquiryPrice'
export const ListInquiryPrice = (r?: ListInquiryPriceRequest, codes?: number[]) =>
  Request<ListInquiryPriceResponse>('/ceres/purchase/PurchaseService/ListInquiryPrice')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListInquiryPrice.url = '/ceres/purchase/PurchaseService/ListInquiryPrice'
export const ImportInquiryPrice = (r?: ImportInquiryPriceRequest, codes?: number[]) =>
  Request<ImportInquiryPriceResponse>('/ceres/purchase/PurchaseService/ImportInquiryPrice')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ImportInquiryPrice.url = '/ceres/purchase/PurchaseService/ImportInquiryPrice'
export const ImportInquiryPriceTmpl = (r?: ImportInquiryPriceTmplRequest, codes?: number[]) =>
  Request<ImportInquiryPriceTmplResponse>('/ceres/purchase/PurchaseService/ImportInquiryPriceTmpl')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ImportInquiryPriceTmpl.url = '/ceres/purchase/PurchaseService/ImportInquiryPriceTmpl'
