// gm_api/src/purchase/pc.tsx
import React, { FC } from 'react'
import { Select, SelectProps } from '@gm-pc/react'
import {
  SelectList,
  SelectEnum,
  MoreSelectList,
  MapId,
} from '../util/pc_util'
import type {
  MapProps,
  SelectListProps,
  SelectEnumProps,
  MoreSelectListProps,
  MapIdProps,
} from '../util/pc_util'
import {
  list_Status_Code,
  map_Status_Code,
  list_TimeType,
  map_TimeType,
  list_ListPurchaseTaskRequest_PagingField,
  map_ListPurchaseTaskRequest_PagingField,
  list_PurchaseTask_Status,
  map_PurchaseTask_Status,
  list_PurchaseTask_RequestSource,
  map_PurchaseTask_RequestSource,
  list_PurchaseTask_Type,
  map_PurchaseTask_Type,
  list_InquiryPrice_SourceType,
  map_InquiryPrice_SourceType,
  list_PurchaseSheet_Status,
  map_PurchaseSheet_Status,
  PurchaseTask,
  ListPurchaseTask,
  ListPurchaseTaskResponse,
  Batch,
  ListBatch,
  ListBatchResponse,
  InquiryPrice,
  ListInquiryPrice,
  ListInquiryPriceResponse,
  PurchaseSheet,
  ListPurchaseSheet,
  ListPurchaseSheetResponse,
} from './index'

export const Select_Status_Code: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Status_Code} />
}

export const Map_Status_Code: FC<MapProps> = ({ value }) => {
  return <>{map_Status_Code[value]}</>
}

export const Select_TimeType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_TimeType} />
}

export const Map_TimeType: FC<MapProps> = ({ value }) => {
  return <>{map_TimeType[value]}</>
}

export const Select_ListPurchaseTaskRequest_PagingField: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_ListPurchaseTaskRequest_PagingField} />
}

export const Map_ListPurchaseTaskRequest_PagingField: FC<MapProps> = ({ value }) => {
  return <>{map_ListPurchaseTaskRequest_PagingField[value]}</>
}

export const Select_PurchaseTask_Status: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_PurchaseTask_Status} />
}

export const Map_PurchaseTask_Status: FC<MapProps> = ({ value }) => {
  return <>{map_PurchaseTask_Status[value]}</>
}

export const Select_PurchaseTask_RequestSource: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_PurchaseTask_RequestSource} />
}

export const Map_PurchaseTask_RequestSource: FC<MapProps> = ({ value }) => {
  return <>{map_PurchaseTask_RequestSource[value]}</>
}

export const Select_PurchaseTask_Type: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_PurchaseTask_Type} />
}

export const Map_PurchaseTask_Type: FC<MapProps> = ({ value }) => {
  return <>{map_PurchaseTask_Type[value]}</>
}

export const Select_InquiryPrice_SourceType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_InquiryPrice_SourceType} />
}

export const Map_InquiryPrice_SourceType: FC<MapProps> = ({ value }) => {
  return <>{map_InquiryPrice_SourceType[value]}</>
}

export const Select_PurchaseSheet_Status: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_PurchaseSheet_Status} />
}

export const Map_PurchaseSheet_Status: FC<MapProps> = ({ value }) => {
  return <>{map_PurchaseSheet_Status[value]}</>
}

export const Select_PurchaseTask: FC<SelectListProps<
  PurchaseTask,
  ListPurchaseTaskResponse
>> = (props) => {
  return (
    <SelectList<PurchaseTask, ListPurchaseTaskResponse>
      {...props}
      messageName='PurchaseTask'
      fetchList={ListPurchaseTask}
    />
  )
}

export const MoreSelect_PurchaseTask: FC<MoreSelectListProps<
  PurchaseTask,
  ListPurchaseTaskResponse
>> = (props) => {
  return (
    <MoreSelectList<PurchaseTask, ListPurchaseTaskResponse>
      {...props}
      messageName='PurchaseTask'
      fetchList={ListPurchaseTask}
    />
  )
}

export const Select_Batch: FC<SelectListProps<
  Batch,
  ListBatchResponse
>> = (props) => {
  return (
    <SelectList<Batch, ListBatchResponse>
      {...props}
      messageName='Batch'
      fetchList={ListBatch}
    />
  )
}

export const MoreSelect_Batch: FC<MoreSelectListProps<
  Batch,
  ListBatchResponse
>> = (props) => {
  return (
    <MoreSelectList<Batch, ListBatchResponse>
      {...props}
      messageName='Batch'
      fetchList={ListBatch}
    />
  )
}

export const Select_InquiryPrice: FC<SelectListProps<
  InquiryPrice,
  ListInquiryPriceResponse
>> = (props) => {
  return (
    <SelectList<InquiryPrice, ListInquiryPriceResponse>
      {...props}
      messageName='InquiryPrice'
      fetchList={ListInquiryPrice}
    />
  )
}

export const MoreSelect_InquiryPrice: FC<MoreSelectListProps<
  InquiryPrice,
  ListInquiryPriceResponse
>> = (props) => {
  return (
    <MoreSelectList<InquiryPrice, ListInquiryPriceResponse>
      {...props}
      messageName='InquiryPrice'
      fetchList={ListInquiryPrice}
    />
  )
}

export const Select_PurchaseSheet: FC<SelectListProps<
  PurchaseSheet,
  ListPurchaseSheetResponse
>> = (props) => {
  return (
    <SelectList<PurchaseSheet, ListPurchaseSheetResponse>
      {...props}
      messageName='PurchaseSheet'
      fetchList={ListPurchaseSheet}
    />
  )
}

export const MoreSelect_PurchaseSheet: FC<MoreSelectListProps<
  PurchaseSheet,
  ListPurchaseSheetResponse
>> = (props) => {
  return (
    <MoreSelectList<PurchaseSheet, ListPurchaseSheetResponse>
      {...props}
      messageName='PurchaseSheet'
      fetchList={ListPurchaseSheet}
    />
  )
}

export const MapId_PurchaseTask: FC<MapIdProps<PurchaseTask, ListPurchaseTaskResponse>> = (
  props,
) => {
  return (
    <MapId<PurchaseTask, ListPurchaseTaskResponse>
      {...props}
      packageName='purchase'
      messageName='PurchaseTask'
      fetchList={ListPurchaseTask}
    />
  )
}

export type {
  MapProps,
  SelectListProps,
  MoreSelectListProps,
}
