// gm_api/src/databi/types.ts
import * as analytics from '../analytics'
import * as common from '../common'
import * as enterprise from '../enterprise'
import * as account from '../account'
import * as dao from '../dao'
import * as idgenerator from '../idgenerator'
import * as order from '../order'
import * as merchandise from '../merchandise'
import * as asynctask from '../asynctask'
import * as oauth from '../oauth'
import * as inventory from '../inventory'
import * as preference from '../preference'
import * as production from '../production'
export interface TimeRange {
  begin_time?: string
  end_time?: string
  time_field?: string
}
export enum PresetType {
  PRESET_TYPE_UNSPECIFIED = 0,
  PRESET_TYPE_ANALYTICS_EXAMPLE_SUMMARY = 20030001,
  PRESET_TYPE_ANALYTICS_EXAMPLE_TREND = 20031001,
  PRESET_TYPE_ANALYTICS_EXAMPLE_RANK = 20032001,
  PRESET_TYPE_ORDER_ORDER_SUMMARY = 20480001,
  PRESET_TYPE_ORDER_ORDER_TREND = 20481001,
  PRESET_TYPE_ORDER_ORDER_RANK_CUSTOMER_ORDER_PRICE_SUM = 20482001,
  PRESET_TYPE_ORDER_ORDER_RANK_CUSTOMER_ORDER_ID_COUNT = 20482002,
  PRESET_TYPE_ORDER_ORDER_ESHOP_RANK = 20483001,
  PRESET_TYPE_ORDER_ORDER_ESHOP_CUSTOMER_TREND = 20483002,
  PRESET_TYPE_PRODUCTION_TASK_PRODUCE_SUMMARY = 85700001,
  PRESET_TYPE_PRODUCTION_TASK_PRODUCE_RANK = 85700002,
  PRESET_TYPE_PRODUCTION_TASK_PACK_RANK = 85700003,
  PRESET_TYPE_PRODUCTION_TASKINPUT_SUMMARY = 85701001,
  PRESET_TYPE_PRODUCTION_TASKOUTPUT_SUMMARY = 85702001,
  PRESET_TYPE_AFTER_SALE_CREATE_AMOUNT_TREND = 55505001,
  PRESET_TYPE_AFTER_SALE_PLACE_AMOUNT_TREND = 55505002,
  PRESET_TYPE_AFTER_SALE_RECEIVE_AMOUNT_TREND = 55505003,
  PRESET_TYPE_AFTER_SALE_CREATE_ORDER_NUM_TREND = 55505004,
  PRESET_TYPE_AFTER_SALE_PLACE_ORDER_NUM_TREND = 55505005,
  PRESET_TYPE_AFTER_SALE_RECIEVE_ORDER_NUM_TREND_TREND = 55505006,
  PRESET_TYPE_AFTER_SALE__CREATE_AMOUNT_RANK = 55505007,
  PRESET_TYPE_AFTER_SALE_PLACE_AMOUNT_RANK = 55505008,
  PRESET_TYPE_AFTER_SALE_RECIEVE_AMOUNT_RANK = 55505009,
}
export interface QueryRequest {
  time_ranges?: TimeRange[]
  preset_type?: PresetType
  expr?: analytics.QueryExpr
}
export interface QueryResponse {
  data?: analytics.QueryData[]
}
export interface GetProductionProduceTaskDataRequest {
  time_range?: TimeRange
  get_many_sku_request?: merchandise.GetManySkuRequest
  task_expr?: analytics.QueryExpr
  input_expr?: analytics.QueryExpr
  output_expr?: analytics.QueryExpr
}
export interface GetProductionProduceTaskDataResponse {
  task_data?: analytics.QueryData
  input_data?: analytics.QueryData
  output_data?: analytics.QueryData
}
export interface GetProductionPackTaskDataRequest {
  time_range?: TimeRange
  get_many_sku_request?: merchandise.GetManySkuRequest
  task_expr?: analytics.QueryExpr
  input_expr?: analytics.QueryExpr
  output_expr?: analytics.QueryExpr
}
export interface GetProductionPackTaskDataResponse {
  task_data?: analytics.QueryData
  input_data?: analytics.QueryData
  output_data?: analytics.QueryData
}
