// gm_api/src/sorting/data.ts
import { getLocale } from './lng'

interface Enum_List_Item {
  text: string
  value: any
}

interface Enum_Map {
  [key: string]: string
}
export const list_Status_Code: Enum_List_Item[] = [
  { value: 88480001, text: getLocale('Status_Code__SORTCONFLICT') },
  { value: 88480002, text: getLocale('Status_Code__CANOTREVERT') },
  { value: 88480003, text: getLocale('Status_Code__INVALIDMARKID') },
  { value: 88480004, text: getLocale('Status_Code__DUPLICATE_SORTNG_SKU_GROUP_NAME') },
  { value: 88480005, text: getLocale('Status_Code__DUPLICATE_SORTNG_BASKET_NAME') },
  { value: 8848006, text: getLocale('Status_Code__SORTLOCK') },
]
export const map_Status_Code: Enum_Map = {
  88480001: getLocale('Status_Code__SORTCONFLICT'),
  88480002: getLocale('Status_Code__CANOTREVERT'),
  88480003: getLocale('Status_Code__INVALIDMARKID'),
  88480004: getLocale('Status_Code__DUPLICATE_SORTNG_SKU_GROUP_NAME'),
  88480005: getLocale('Status_Code__DUPLICATE_SORTNG_BASKET_NAME'),
  8848006: getLocale('Status_Code__SORTLOCK'),
}
export const list_OpType: Enum_List_Item[] = [
  { value: 1, text: getLocale('OpType__SETWEIGHT') },
  { value: 2, text: getLocale('OpType__CANCELWEIGHT') },
  { value: 3, text: getLocale('OpType__SETOUTOFSTOCK') },
  { value: 4, text: getLocale('OpType__CANCELOUTOFSTOCK') },
  { value: 5, text: getLocale('OpType__SETSORTMARK') },
  { value: 6, text: getLocale('OpType__CANCELSORTMARK') },
  { value: 7, text: getLocale('OpType__REVERTLASTWEIGHT') },
]
export const map_OpType: Enum_Map = {
  1: getLocale('OpType__SETWEIGHT'),
  2: getLocale('OpType__CANCELWEIGHT'),
  3: getLocale('OpType__SETOUTOFSTOCK'),
  4: getLocale('OpType__CANCELOUTOFSTOCK'),
  5: getLocale('OpType__SETSORTMARK'),
  6: getLocale('OpType__CANCELSORTMARK'),
  7: getLocale('OpType__REVERTLASTWEIGHT'),
}
export const list_SortingTag_Type: Enum_List_Item[] = [
  { value: 1, text: getLocale('SortingTag_Type__NORMAL') },
  { value: 2, text: getLocale('SortingTag_Type__PRE') },
]
export const map_SortingTag_Type: Enum_Map = {
  1: getLocale('SortingTag_Type__NORMAL'),
  2: getLocale('SortingTag_Type__PRE'),
}
export const list_Weight: Enum_List_Item[] = [
  { value: 1, text: getLocale('Weight__WEIGHT') },
  { value: 2, text: getLocale('Weight__UNWEIGHT') },
]
export const map_Weight: Enum_Map = {
  1: getLocale('Weight__WEIGHT'),
  2: getLocale('Weight__UNWEIGHT'),
}
