// gm_api/src/sorting/methods.ts
import { Request } from '../util/request'
import {
  ListCategoryTreeRequest, ListCategoryTreeResponse,
  ListSortingSsuRequest, ListSortingSsuResponse,
  ListSortingTaskRequest, ListSortingTaskResponse,
  ListSsuSortingTaskRequest, ListSsuSortingTaskResponse,
  GetLastSortingLogRequest, GetLastSortingLogResponse,
  ListLastSortingLogRequest, ListLastSortingLogResponse,
  UpdateSortingTaskRequest, UpdateSortingTaskResponse,
  CreatePreSortingTagRequest, CreatePreSortingTagResponse,
  DeletePreSortingTagRequest, DeletePreSortingTagResponse,
  GetPreSortingTagRequest, GetPreSortingTagResponse,
  ListPreSortingTagRequest, ListPreSortingTagResponse,
  UpdatePreSortingTagRequest, UpdatePreSortingTagResponse,
  CreateSortingBasketRequest, CreateSortingBasketResponse,
  UpdateSortingBasketRequest, UpdateSortingBasketResponse,
  ListSortingBasketRequest, ListSortingBasketResponse,
  DeleteSortingBasketRequest, DeleteSortingBasketResponse,
  CreateSortingSkuGroupRequest, CreateSortingSkuGroupResponse,
  UpdateSortingSkuGroupRequest, UpdateSortingSkuGroupResponse,
  ListSortingSkuGroupRequest, ListSortingSkuGroupResponse,
  DeleteSortingSkuGroupRequest, DeleteSortingSkuGroupResponse,
  GetSortingInfoRequest, GetSortingInfoResponse,
} from './types'
export const ListCategoryTree = (r?: ListCategoryTreeRequest, codes?: number[]) =>
  Request<ListCategoryTreeResponse>('/ceres/sorting/SortingService/ListCategoryTree')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListCategoryTree.url = '/ceres/sorting/SortingService/ListCategoryTree'
export const ListSortingSsu = (r?: ListSortingSsuRequest, codes?: number[]) =>
  Request<ListSortingSsuResponse>('/ceres/sorting/SortingService/ListSortingSsu')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListSortingSsu.url = '/ceres/sorting/SortingService/ListSortingSsu'
export const ListSortingTask = (r?: ListSortingTaskRequest, codes?: number[]) =>
  Request<ListSortingTaskResponse>('/ceres/sorting/SortingService/ListSortingTask')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListSortingTask.url = '/ceres/sorting/SortingService/ListSortingTask'
export const ListSsuSortingTask = (r?: ListSsuSortingTaskRequest, codes?: number[]) =>
  Request<ListSsuSortingTaskResponse>('/ceres/sorting/SortingService/ListSsuSortingTask')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListSsuSortingTask.url = '/ceres/sorting/SortingService/ListSsuSortingTask'
export const GetLastSortingLog = (r?: GetLastSortingLogRequest, codes?: number[]) =>
  Request<GetLastSortingLogResponse>('/ceres/sorting/SortingService/GetLastSortingLog')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetLastSortingLog.url = '/ceres/sorting/SortingService/GetLastSortingLog'
export const ListLastSortingLog = (r?: ListLastSortingLogRequest, codes?: number[]) =>
  Request<ListLastSortingLogResponse>('/ceres/sorting/SortingService/ListLastSortingLog')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListLastSortingLog.url = '/ceres/sorting/SortingService/ListLastSortingLog'
export const UpdateSortingTask = (r?: UpdateSortingTaskRequest, codes?: number[]) =>
  Request<UpdateSortingTaskResponse>('/ceres/sorting/SortingService/UpdateSortingTask')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateSortingTask.url = '/ceres/sorting/SortingService/UpdateSortingTask'
export const CreatePreSortingTag = (r?: CreatePreSortingTagRequest, codes?: number[]) =>
  Request<CreatePreSortingTagResponse>('/ceres/sorting/SortingService/CreatePreSortingTag')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreatePreSortingTag.url = '/ceres/sorting/SortingService/CreatePreSortingTag'
export const DeletePreSortingTag = (r?: DeletePreSortingTagRequest, codes?: number[]) =>
  Request<DeletePreSortingTagResponse>('/ceres/sorting/SortingService/DeletePreSortingTag')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeletePreSortingTag.url = '/ceres/sorting/SortingService/DeletePreSortingTag'
export const GetPreSortingTag = (r?: GetPreSortingTagRequest, codes?: number[]) =>
  Request<GetPreSortingTagResponse>('/ceres/sorting/SortingService/GetPreSortingTag')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetPreSortingTag.url = '/ceres/sorting/SortingService/GetPreSortingTag'
export const ListPreSortingTag = (r?: ListPreSortingTagRequest, codes?: number[]) =>
  Request<ListPreSortingTagResponse>('/ceres/sorting/SortingService/ListPreSortingTag')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListPreSortingTag.url = '/ceres/sorting/SortingService/ListPreSortingTag'
export const UpdatePreSortingTag = (r?: UpdatePreSortingTagRequest, codes?: number[]) =>
  Request<UpdatePreSortingTagResponse>('/ceres/sorting/SortingService/UpdatePreSortingTag')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdatePreSortingTag.url = '/ceres/sorting/SortingService/UpdatePreSortingTag'
export const CreateSortingBasket = (r?: CreateSortingBasketRequest, codes?: number[]) =>
  Request<CreateSortingBasketResponse>('/ceres/sorting/SortingService/CreateSortingBasket')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateSortingBasket.url = '/ceres/sorting/SortingService/CreateSortingBasket'
export const UpdateSortingBasket = (r?: UpdateSortingBasketRequest, codes?: number[]) =>
  Request<UpdateSortingBasketResponse>('/ceres/sorting/SortingService/UpdateSortingBasket')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateSortingBasket.url = '/ceres/sorting/SortingService/UpdateSortingBasket'
export const ListSortingBasket = (r?: ListSortingBasketRequest, codes?: number[]) =>
  Request<ListSortingBasketResponse>('/ceres/sorting/SortingService/ListSortingBasket')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListSortingBasket.url = '/ceres/sorting/SortingService/ListSortingBasket'
export const DeleteSortingBasket = (r?: DeleteSortingBasketRequest, codes?: number[]) =>
  Request<DeleteSortingBasketResponse>('/ceres/sorting/SortingService/DeleteSortingBasket')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteSortingBasket.url = '/ceres/sorting/SortingService/DeleteSortingBasket'
export const CreateSortingSkuGroup = (r?: CreateSortingSkuGroupRequest, codes?: number[]) =>
  Request<CreateSortingSkuGroupResponse>('/ceres/sorting/SortingService/CreateSortingSkuGroup')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateSortingSkuGroup.url = '/ceres/sorting/SortingService/CreateSortingSkuGroup'
export const UpdateSortingSkuGroup = (r?: UpdateSortingSkuGroupRequest, codes?: number[]) =>
  Request<UpdateSortingSkuGroupResponse>('/ceres/sorting/SortingService/UpdateSortingSkuGroup')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateSortingSkuGroup.url = '/ceres/sorting/SortingService/UpdateSortingSkuGroup'
export const ListSortingSkuGroup = (r?: ListSortingSkuGroupRequest, codes?: number[]) =>
  Request<ListSortingSkuGroupResponse>('/ceres/sorting/SortingService/ListSortingSkuGroup')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListSortingSkuGroup.url = '/ceres/sorting/SortingService/ListSortingSkuGroup'
export const DeleteSortingSkuGroup = (r?: DeleteSortingSkuGroupRequest, codes?: number[]) =>
  Request<DeleteSortingSkuGroupResponse>('/ceres/sorting/SortingService/DeleteSortingSkuGroup')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteSortingSkuGroup.url = '/ceres/sorting/SortingService/DeleteSortingSkuGroup'
export const GetSortingInfo = (r?: GetSortingInfoRequest, codes?: number[]) =>
  Request<GetSortingInfoResponse>('/ceres/sorting/SortingService/GetSortingInfo')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetSortingInfo.url = '/ceres/sorting/SortingService/GetSortingInfo'
