// gm_api/src/sorting/pc.tsx
import React, { FC } from 'react'
import { Select, SelectProps } from '@gm-pc/react'
import {
  SelectList,
  SelectEnum,
  MoreSelectList,
  MapId,
} from '../util/pc_util'
import type {
  MapProps,
  SelectListProps,
  SelectEnumProps,
  MoreSelectListProps,
  MapIdProps,
} from '../util/pc_util'
import {
  list_Status_Code,
  map_Status_Code,
  list_OpType,
  map_OpType,
  list_SortingTag_Type,
  map_SortingTag_Type,
  list_Weight,
  map_Weight,
  SortingSkuGroup,
  ListSortingSkuGroup,
  ListSortingSkuGroupResponse,
  SortingBasket,
  ListSortingBasket,
  ListSortingBasketResponse,
} from './index'

export const Select_Status_Code: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Status_Code} />
}

export const Map_Status_Code: FC<MapProps> = ({ value }) => {
  return <>{map_Status_Code[value]}</>
}

export const Select_OpType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_OpType} />
}

export const Map_OpType: FC<MapProps> = ({ value }) => {
  return <>{map_OpType[value]}</>
}

export const Select_SortingTag_Type: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_SortingTag_Type} />
}

export const Map_SortingTag_Type: FC<MapProps> = ({ value }) => {
  return <>{map_SortingTag_Type[value]}</>
}

export const Select_Weight: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Weight} />
}

export const Map_Weight: FC<MapProps> = ({ value }) => {
  return <>{map_Weight[value]}</>
}

export const Select_SortingSkuGroup: FC<SelectListProps<
  SortingSkuGroup,
  ListSortingSkuGroupResponse
>> = (props) => {
  return (
    <SelectList<SortingSkuGroup, ListSortingSkuGroupResponse>
      {...props}
      messageName='SortingSkuGroup'
      fetchList={ListSortingSkuGroup}
    />
  )
}

export const MoreSelect_SortingSkuGroup: FC<MoreSelectListProps<
  SortingSkuGroup,
  ListSortingSkuGroupResponse
>> = (props) => {
  return (
    <MoreSelectList<SortingSkuGroup, ListSortingSkuGroupResponse>
      {...props}
      messageName='SortingSkuGroup'
      fetchList={ListSortingSkuGroup}
    />
  )
}

export const Select_SortingBasket: FC<SelectListProps<
  SortingBasket,
  ListSortingBasketResponse
>> = (props) => {
  return (
    <SelectList<SortingBasket, ListSortingBasketResponse>
      {...props}
      messageName='SortingBasket'
      fetchList={ListSortingBasket}
    />
  )
}

export const MoreSelect_SortingBasket: FC<MoreSelectListProps<
  SortingBasket,
  ListSortingBasketResponse
>> = (props) => {
  return (
    <MoreSelectList<SortingBasket, ListSortingBasketResponse>
      {...props}
      messageName='SortingBasket'
      fetchList={ListSortingBasket}
    />
  )
}

export type {
  MapProps,
  SelectListProps,
  MoreSelectListProps,
}
