// gm_api/src/merchandise/data.ts
import { getLocale } from './lng'

interface Enum_List_Item {
  text: string
  value: any
}

interface Enum_Map {
  [key: string]: string
}
export const list_Status_Code: Enum_List_Item[] = [
  { value: 22220001, text: getLocale('Status_Code__CATEGORY_EXISTS') },
  { value: 22220002, text: getLocale('Status_Code__PARENT_CATEGORY_NOT_EXISTS') },
  { value: 22220003, text: getLocale('Status_Code__PARENT_CATEGORY_CAN_NOT_CHANGE') },
  { value: 22220004, text: getLocale('Status_Code__CHILD_CATEGORY_EXISTS') },
  { value: 22220005, text: getLocale('Status_Code__CHILD_SPU_EXISTS') },
  { value: 22220006, text: getLocale('Status_Code__SPU_EXISTS') },
  { value: 22220007, text: getLocale('Status_Code__UNIT_NOT_EXISTS') },
  { value: 22220008, text: getLocale('Status_Code__BASE_UNIT_CAN_NOT_CHANGE') },
  { value: 22220009, text: getLocale('Status_Code__SKU_NOT_EXISTS') },
  { value: 22220010, text: getLocale('Status_Code__SKU_EXISTS') },
  { value: 22220011, text: getLocale('Status_Code__SSU_EXISTS') },
  { value: 22220012, text: getLocale('Status_Code__CUSTOMIZE_CODE_EXISTS') },
  { value: 22220030, text: getLocale('Status_Code__CUSTOMER_NOT_BAND_QUOTATION') },
  { value: 22220031, text: getLocale('Status_Code__DUPLICATE_QUOTATION_NAME') },
  { value: 22220032, text: getLocale('Status_Code__CANNOT_DELETE_DEFAULT_QUOTATION') },
  { value: 22220040, text: getLocale('Status_Code__DUPLICATE_FAVORITE_SSU') },
  { value: 22220041, text: getLocale('Status_Code__CANNOT_UPDATE_DEFAULT_FAVORITE') },
  { value: 22220050, text: getLocale('Status_Code__ASYNC_TASK_PARAM_ERROR') },
]
export const map_Status_Code: Enum_Map = {
  22220001: getLocale('Status_Code__CATEGORY_EXISTS'),
  22220002: getLocale('Status_Code__PARENT_CATEGORY_NOT_EXISTS'),
  22220003: getLocale('Status_Code__PARENT_CATEGORY_CAN_NOT_CHANGE'),
  22220004: getLocale('Status_Code__CHILD_CATEGORY_EXISTS'),
  22220005: getLocale('Status_Code__CHILD_SPU_EXISTS'),
  22220006: getLocale('Status_Code__SPU_EXISTS'),
  22220007: getLocale('Status_Code__UNIT_NOT_EXISTS'),
  22220008: getLocale('Status_Code__BASE_UNIT_CAN_NOT_CHANGE'),
  22220009: getLocale('Status_Code__SKU_NOT_EXISTS'),
  22220010: getLocale('Status_Code__SKU_EXISTS'),
  22220011: getLocale('Status_Code__SSU_EXISTS'),
  22220012: getLocale('Status_Code__CUSTOMIZE_CODE_EXISTS'),
  22220030: getLocale('Status_Code__CUSTOMER_NOT_BAND_QUOTATION'),
  22220031: getLocale('Status_Code__DUPLICATE_QUOTATION_NAME'),
  22220032: getLocale('Status_Code__CANNOT_DELETE_DEFAULT_QUOTATION'),
  22220040: getLocale('Status_Code__DUPLICATE_FAVORITE_SSU'),
  22220041: getLocale('Status_Code__CANNOT_UPDATE_DEFAULT_FAVORITE'),
  22220050: getLocale('Status_Code__ASYNC_TASK_PARAM_ERROR'),
}
export const list_SystemUnitId: Enum_List_Item[] = [
  { value: 200000, text: getLocale('SystemUnitId__SYSTEM_UNIT_ID_BEGIN') },
]
export const map_SystemUnitId: Enum_Map = {
  200000: getLocale('SystemUnitId__SYSTEM_UNIT_ID_BEGIN'),
}
export const list_UnitType: Enum_List_Item[] = [
  { value: 1, text: getLocale('UnitType__SYSTEM_UNIT') },
  { value: 2, text: getLocale('UnitType__SYSTEM_UNIT_ANCHOR') },
  { value: 3, text: getLocale('UnitType__SKU_BASE_UNIT') },
  { value: 4, text: getLocale('UnitType__SSU_UNIT') },
  { value: 5, text: getLocale('UnitType__SSU_UNIT_ANCHOR') },
]
export const map_UnitType: Enum_Map = {
  1: getLocale('UnitType__SYSTEM_UNIT'),
  2: getLocale('UnitType__SYSTEM_UNIT_ANCHOR'),
  3: getLocale('UnitType__SKU_BASE_UNIT'),
  4: getLocale('UnitType__SSU_UNIT'),
  5: getLocale('UnitType__SSU_UNIT_ANCHOR'),
}
export const list_Unit_Type: Enum_List_Item[] = [
  { value: 1, text: getLocale('Unit_Type__LENGTH') },
  { value: 2, text: getLocale('Unit_Type__AREA') },
  { value: 3, text: getLocale('Unit_Type__VOLUME') },
  { value: 4, text: getLocale('Unit_Type__MASS') },
  { value: 5, text: getLocale('Unit_Type__COUNT') },
]
export const map_Unit_Type: Enum_Map = {
  1: getLocale('Unit_Type__LENGTH'),
  2: getLocale('Unit_Type__AREA'),
  3: getLocale('Unit_Type__VOLUME'),
  4: getLocale('Unit_Type__MASS'),
  5: getLocale('Unit_Type__COUNT'),
}
export const list_Optype: Enum_List_Item[] = [
  { value: 1, text: getLocale('Optype__CREATE') },
  { value: 2, text: getLocale('Optype__UPDATE') },
  { value: 3, text: getLocale('Optype__DELETE') },
]
export const map_Optype: Enum_Map = {
  1: getLocale('Optype__CREATE'),
  2: getLocale('Optype__UPDATE'),
  3: getLocale('Optype__DELETE'),
}
export const list_Sku_PurchaseType: Enum_List_Item[] = [
  { value: 1, text: getLocale('Sku_PurchaseType__NO_TMP') },
  { value: 2, text: getLocale('Sku_PurchaseType__TMP') },
]
export const map_Sku_PurchaseType: Enum_Map = {
  1: getLocale('Sku_PurchaseType__NO_TMP'),
  2: getLocale('Sku_PurchaseType__TMP'),
}
export const list_Sku_DispatchType: Enum_List_Item[] = [
  { value: 1, text: getLocale('Sku_DispatchType__ORDER') },
  { value: 2, text: getLocale('Sku_DispatchType__DRIVER') },
]
export const map_Sku_DispatchType: Enum_Map = {
  1: getLocale('Sku_DispatchType__ORDER'),
  2: getLocale('Sku_DispatchType__DRIVER'),
}
export const list_Sku_Status: Enum_List_Item[] = [
  { value: 128, text: getLocale('Sku_Status__DETECT') },
  { value: 512, text: getLocale('Sku_Status__NUTRITION') },
]
export const map_Sku_Status: Enum_Map = {
  128: getLocale('Sku_Status__DETECT'),
  512: getLocale('Sku_Status__NUTRITION'),
}
export const list_Sku_SkuType: Enum_List_Item[] = [
  { value: 1, text: getLocale('Sku_SkuType__PACKAGE') },
  { value: 2, text: getLocale('Sku_SkuType__NOT_PACKAGE') },
]
export const map_Sku_SkuType: Enum_Map = {
  1: getLocale('Sku_SkuType__PACKAGE'),
  2: getLocale('Sku_SkuType__NOT_PACKAGE'),
}
export const list_Sku_NotPackageSubSkuType: Enum_List_Item[] = [
  { value: 1, text: getLocale('Sku_NotPackageSubSkuType__ORIGIN') },
  { value: 2, text: getLocale('Sku_NotPackageSubSkuType__CLEAN') },
  { value: 3, text: getLocale('Sku_NotPackageSubSkuType__DELICATESSEN') },
  { value: 4, text: getLocale('Sku_NotPackageSubSkuType__DISHES') },
  { value: 5, text: getLocale('Sku_NotPackageSubSkuType__COMBO') },
  { value: 6, text: getLocale('Sku_NotPackageSubSkuType__BOXED_LUNCH') },
  { value: 7, text: getLocale('Sku_NotPackageSubSkuType__CAKE') },
  { value: 8, text: getLocale('Sku_NotPackageSubSkuType__STAPLE') },
  { value: 9, text: getLocale('Sku_NotPackageSubSkuType__DRINK') },
  { value: 10, text: getLocale('Sku_NotPackageSubSkuType__WATER') },
  { value: 11, text: getLocale('Sku_NotPackageSubSkuType__INGREDIENTS') },
  { value: 12, text: getLocale('Sku_NotPackageSubSkuType__NON_STAPLE_FOOD') },
  { value: 13, text: getLocale('Sku_NotPackageSubSkuType__OUTSOURCED_FINISHED_PRODUCTS') },
  { value: 14, text: getLocale('Sku_NotPackageSubSkuType__HOMEMADE_PRODUCTS') },
  { value: 15, text: getLocale('Sku_NotPackageSubSkuType__MATERIALS') },
  { value: 16, text: getLocale('Sku_NotPackageSubSkuType__GRAIN_AND_OIL') },
]
export const map_Sku_NotPackageSubSkuType: Enum_Map = {
  1: getLocale('Sku_NotPackageSubSkuType__ORIGIN'),
  2: getLocale('Sku_NotPackageSubSkuType__CLEAN'),
  3: getLocale('Sku_NotPackageSubSkuType__DELICATESSEN'),
  4: getLocale('Sku_NotPackageSubSkuType__DISHES'),
  5: getLocale('Sku_NotPackageSubSkuType__COMBO'),
  6: getLocale('Sku_NotPackageSubSkuType__BOXED_LUNCH'),
  7: getLocale('Sku_NotPackageSubSkuType__CAKE'),
  8: getLocale('Sku_NotPackageSubSkuType__STAPLE'),
  9: getLocale('Sku_NotPackageSubSkuType__DRINK'),
  10: getLocale('Sku_NotPackageSubSkuType__WATER'),
  11: getLocale('Sku_NotPackageSubSkuType__INGREDIENTS'),
  12: getLocale('Sku_NotPackageSubSkuType__NON_STAPLE_FOOD'),
  13: getLocale('Sku_NotPackageSubSkuType__OUTSOURCED_FINISHED_PRODUCTS'),
  14: getLocale('Sku_NotPackageSubSkuType__HOMEMADE_PRODUCTS'),
  15: getLocale('Sku_NotPackageSubSkuType__MATERIALS'),
  16: getLocale('Sku_NotPackageSubSkuType__GRAIN_AND_OIL'),
}
export const list_Sku_PackageSubSkuType: Enum_List_Item[] = [
  { value: 1, text: getLocale('Sku_PackageSubSkuType__TURNOVER') },
  { value: 2, text: getLocale('Sku_PackageSubSkuType__ONE_TIME') },
]
export const map_Sku_PackageSubSkuType: Enum_Map = {
  1: getLocale('Sku_PackageSubSkuType__TURNOVER'),
  2: getLocale('Sku_PackageSubSkuType__ONE_TIME'),
}
export const list_ModuleType: Enum_List_Item[] = [
  { value: 1, text: getLocale('ModuleType__SORTING') },
  { value: 2, text: getLocale('ModuleType__ORDER') },
  { value: 3, text: getLocale('ModuleType__DISTRIBUTE') },
  { value: 4, text: getLocale('ModuleType__PURCHASE') },
  { value: 5, text: getLocale('ModuleType__STOCK') },
]
export const map_ModuleType: Enum_Map = {
  1: getLocale('ModuleType__SORTING'),
  2: getLocale('ModuleType__ORDER'),
  3: getLocale('ModuleType__DISTRIBUTE'),
  4: getLocale('ModuleType__PURCHASE'),
  5: getLocale('ModuleType__STOCK'),
}
export const list_Ssu_PackageCalculateType: Enum_List_Item[] = [
  { value: 1, text: getLocale('Ssu_PackageCalculateType__FIXED') },
  { value: 2, text: getLocale('Ssu_PackageCalculateType__BY_ORDER_NUM') },
]
export const map_Ssu_PackageCalculateType: Enum_Map = {
  1: getLocale('Ssu_PackageCalculateType__FIXED'),
  2: getLocale('Ssu_PackageCalculateType__BY_ORDER_NUM'),
}
export const list_Ssu_NutritionType: Enum_List_Item[] = [
  { value: 1, text: getLocale('Ssu_NutritionType__PROTEIN') },
  { value: 2, text: getLocale('Ssu_NutritionType__FAT') },
  { value: 3, text: getLocale('Ssu_NutritionType__CARBOHYDRATE') },
]
export const map_Ssu_NutritionType: Enum_Map = {
  1: getLocale('Ssu_NutritionType__PROTEIN'),
  2: getLocale('Ssu_NutritionType__FAT'),
  3: getLocale('Ssu_NutritionType__CARBOHYDRATE'),
}
export const list_Ssu_SplitFlag: Enum_List_Item[] = [
  { value: 1, text: getLocale('Ssu_SplitFlag__THROW_BASK_BET') },
  { value: 2, text: getLocale('Ssu_SplitFlag__DIVIDE') },
  { value: 3, text: getLocale('Ssu_SplitFlag__INVESTMENT') },
]
export const map_Ssu_SplitFlag: Enum_Map = {
  1: getLocale('Ssu_SplitFlag__THROW_BASK_BET'),
  2: getLocale('Ssu_SplitFlag__DIVIDE'),
  3: getLocale('Ssu_SplitFlag__INVESTMENT'),
}
export const list_Ssu_ShippingFeeUnit: Enum_List_Item[] = [
  { value: 1, text: getLocale('Ssu_ShippingFeeUnit__BASE') },
  { value: 2, text: getLocale('Ssu_ShippingFeeUnit__PACKAGE') },
]
export const map_Ssu_ShippingFeeUnit: Enum_Map = {
  1: getLocale('Ssu_ShippingFeeUnit__BASE'),
  2: getLocale('Ssu_ShippingFeeUnit__PACKAGE'),
}
export const list_Favorite_FavoriteType: Enum_List_Item[] = [
  { value: 1, text: getLocale('Favorite_FavoriteType__BSHOP') },
]
export const map_Favorite_FavoriteType: Enum_Map = {
  1: getLocale('Favorite_FavoriteType__BSHOP'),
}
export const list_Favorite_FavoriteContentType: Enum_List_Item[] = [
  { value: 1, text: getLocale('Favorite_FavoriteContentType__SSU') },
]
export const map_Favorite_FavoriteContentType: Enum_Map = {
  1: getLocale('Favorite_FavoriteContentType__SSU'),
}
export const list_RuleSet_State: Enum_List_Item[] = [
  { value: 1, text: getLocale('RuleSet_State__ON') },
  { value: 2, text: getLocale('RuleSet_State__OFF') },
]
export const map_RuleSet_State: Enum_Map = {
  1: getLocale('RuleSet_State__ON'),
  2: getLocale('RuleSet_State__OFF'),
}
export const list_RuleSet_Type: Enum_List_Item[] = [
  { value: 1, text: getLocale('RuleSet_Type__RANGE') },
  { value: 2, text: getLocale('RuleSet_Type__PRICE') },
]
export const map_RuleSet_Type: Enum_Map = {
  1: getLocale('RuleSet_Type__RANGE'),
  2: getLocale('RuleSet_Type__PRICE'),
}
export const list_ChangeMerchandiseRankRequest_NodeType: Enum_List_Item[] = [
  { value: 1, text: getLocale('ChangeMerchandiseRankRequest_NodeType__CATEGORY') },
  { value: 2, text: getLocale('ChangeMerchandiseRankRequest_NodeType__SPU') },
  { value: 3, text: getLocale('ChangeMerchandiseRankRequest_NodeType__SKU') },
]
export const map_ChangeMerchandiseRankRequest_NodeType: Enum_Map = {
  1: getLocale('ChangeMerchandiseRankRequest_NodeType__CATEGORY'),
  2: getLocale('ChangeMerchandiseRankRequest_NodeType__SPU'),
  3: getLocale('ChangeMerchandiseRankRequest_NodeType__SKU'),
}
export const list_ListSkuRequest_RequestData: Enum_List_Item[] = [
  { value: 256, text: getLocale('ListSkuRequest_RequestData__SSU') },
  { value: 512, text: getLocale('ListSkuRequest_RequestData__PRICE') },
  { value: 1024, text: getLocale('ListSkuRequest_RequestData__CATEGORY') },
  { value: 2048, text: getLocale('ListSkuRequest_RequestData__QUOTATION') },
]
export const map_ListSkuRequest_RequestData: Enum_Map = {
  256: getLocale('ListSkuRequest_RequestData__SSU'),
  512: getLocale('ListSkuRequest_RequestData__PRICE'),
  1024: getLocale('ListSkuRequest_RequestData__CATEGORY'),
  2048: getLocale('ListSkuRequest_RequestData__QUOTATION'),
}
export const list_GetManySkuRequest_RequestData: Enum_List_Item[] = [
  { value: 256, text: getLocale('GetManySkuRequest_RequestData__SSU') },
  { value: 512, text: getLocale('GetManySkuRequest_RequestData__PRICE') },
  { value: 1024, text: getLocale('GetManySkuRequest_RequestData__CATEGORY') },
]
export const map_GetManySkuRequest_RequestData: Enum_Map = {
  256: getLocale('GetManySkuRequest_RequestData__SSU'),
  512: getLocale('GetManySkuRequest_RequestData__PRICE'),
  1024: getLocale('GetManySkuRequest_RequestData__CATEGORY'),
}
