// gm_api/src/merchandise/methods.ts
import { Request } from '../util/request'
import {
  SetCustomerQuotationRelationRequest, SetCustomerQuotationRelationResponse,
  GetCustomerIdsByQuotationIdRequest, GetCustomerIdsByQuotationIdResponse,
  GetQuotationByCustomerIdRequest, GetQuotationByCustomerIdResponse,
  GetQuotationIdsByCustomerIdsRequest, GetQuotationIdsByCustomerIdsResponse,
  GetCustomerQuotationRelationRequest, GetCustomerQuotationRelationResponse,
  DeleteCustomerQuotationRelationRequest, DeleteCustomerQuotationRelationResponse,
  GetCategoryRequest, GetCategoryResponse,
  CreateCategoryRequest, CreateCategoryResponse,
  UpdateCategoryRequest, UpdateCategoryResponse,
  DeleteCategoryRequest, DeleteCategoryResponse,
  ListCategoryRequest, ListCategoryResponse,
  GetSpuRequest, GetSpuResponse,
  GetManySpuRequest, GetManySpuResponse,
  CreateSpuRequest, CreateSpuResponse,
  UpdateSpuRequest, UpdateSpuResponse,
  DeleteSpuRequest, DeleteSpuResponse,
  ListSpuRequest, ListSpuResponse,
  GetCategoryTreeRequest, GetCategoryTreeResponse,
  ChangeMerchandiseRankRequest, ChangeMerchandiseRankResponse,
  GetManyCategoryRequest, GetManyCategoryResponse,
  GetSkuRequest, GetSkuResponse,
  CreateSkuRequest, CreateSkuResponse,
  UpdateSkuRequest, UpdateSkuResponse,
  DeleteSkuRequest, DeleteSkuResponse,
  ListSkuRequest, ListSkuResponse,
  ExportSkuRequest, ExportSkuResponse,
  ExportSsuRequest, ExportSsuResponse,
  GetManySkuRequest, GetManySkuResponse,
  GetSkuSnapRequest, GetSkuSnapResponse,
  GetManySkuSnapRequest, GetManySkuSnapResponse,
  GetSsuSnapRequest, GetSsuSnapResponse,
  GetManySsuSnapRequest, GetManySsuSnapResponse,
  GetSsuRequest, GetSsuResponse,
  UpdateSsuRequest, UpdateSsuResponse,
  CreateSsuRequest, CreateSsuResponse,
  DeleteSsuRequest, DeleteSsuResponse,
  BatchUpdateSkuSsuRequest, BatchUpdateSkuSsuResponse,
  BatchCreateCategorySkuRequest, BatchCreateCategorySkuResponse,
  BatchCreateSsuRequest, BatchCreateSsuResponse,
  ListSsuRequest, ListSsuResponse,
  ListSsuByCustomerOrQuotationRequest, ListSsuByCustomerOrQuotationResponse,
  GetRuleSetRequest, GetRuleSetResponse,
  CreateRuleSetRequest, CreateRuleSetResponse,
  UpdateRuleSetRequest, UpdateRuleSetResponse,
  DeleteRuleSetRequest, DeleteRuleSetResponse,
  ListRuleSetRequest, ListRuleSetResponse,
  CreateUnitRequest, CreateUnitResponse,
  ListUnitRequest, ListUnitResponse,
  GetUnitRateGroupRequest, GetUnitRateGroupResponse,
  InitUnitRequest, InitUnitResponse,
  InitCategoryImageRequest, InitCategoryImageResponse,
  ListCategoryImageRequest, ListCategoryImageResponse,
  GetQuotationRequest, GetQuotationResponse,
  ListQuotationRequest, ListQuotationResponse,
  GetManyQuotationRequest, GetManyQuotationResponse,
  CreateQuotationRequest, CreateQuotationResponse,
  UpdateQuotationRequest, UpdateQuotationResponse,
  DeleteQuotationRequest, DeleteQuotationResponse,
  DeleteBasicPriceRequest, DeleteBasicPriceResponse,
  GetManySsuPriceRequest, GetManySsuPriceResponse,
  SyncSsuImageRequest, SyncSsuImageResponse,
  CreateFavoriteRequest, CreateFavoriteResponse,
  UpdateFavoriteRequest, UpdateFavoriteResponse,
  DeleteFavoriteRequest, DeleteFavoriteResponse,
  ListFavoriteRequest, ListFavoriteResponse,
  CreateFavoriteSsuRequest, CreateFavoriteSsuResponse,
  DeleteFavoriteSsuRequest, DeleteFavoriteSsuResponse,
  ListFavoriteSsuRequest, ListFavoriteSsuResponse,
} from './types'
export const SetCustomerQuotationRelation = (r?: SetCustomerQuotationRelationRequest, codes?: number[]) =>
  Request<SetCustomerQuotationRelationResponse>('/ceres/merchandise/MerchandiseService/SetCustomerQuotationRelation')
    .code([...(codes || [])])
    .data(r || {})
    .run()
SetCustomerQuotationRelation.url = '/ceres/merchandise/MerchandiseService/SetCustomerQuotationRelation'
export const GetCustomerIdsByQuotationId = (r?: GetCustomerIdsByQuotationIdRequest, codes?: number[]) =>
  Request<GetCustomerIdsByQuotationIdResponse>('/ceres/merchandise/MerchandiseService/GetCustomerIdsByQuotationId')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetCustomerIdsByQuotationId.url = '/ceres/merchandise/MerchandiseService/GetCustomerIdsByQuotationId'
export const GetQuotationByCustomerId = (r?: GetQuotationByCustomerIdRequest, codes?: number[]) =>
  Request<GetQuotationByCustomerIdResponse>('/ceres/merchandise/MerchandiseService/GetQuotationByCustomerId')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetQuotationByCustomerId.url = '/ceres/merchandise/MerchandiseService/GetQuotationByCustomerId'
export const GetQuotationIdsByCustomerIds = (r?: GetQuotationIdsByCustomerIdsRequest, codes?: number[]) =>
  Request<GetQuotationIdsByCustomerIdsResponse>('/ceres/merchandise/MerchandiseService/GetQuotationIdsByCustomerIds')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetQuotationIdsByCustomerIds.url = '/ceres/merchandise/MerchandiseService/GetQuotationIdsByCustomerIds'
export const GetCustomerQuotationRelation = (r?: GetCustomerQuotationRelationRequest, codes?: number[]) =>
  Request<GetCustomerQuotationRelationResponse>('/ceres/merchandise/MerchandiseService/GetCustomerQuotationRelation')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetCustomerQuotationRelation.url = '/ceres/merchandise/MerchandiseService/GetCustomerQuotationRelation'
export const DeleteCustomerQuotationRelation = (r?: DeleteCustomerQuotationRelationRequest, codes?: number[]) =>
  Request<DeleteCustomerQuotationRelationResponse>('/ceres/merchandise/MerchandiseService/DeleteCustomerQuotationRelation')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteCustomerQuotationRelation.url = '/ceres/merchandise/MerchandiseService/DeleteCustomerQuotationRelation'
export const GetCategory = (r?: GetCategoryRequest, codes?: number[]) =>
  Request<GetCategoryResponse>('/ceres/merchandise/MerchandiseService/GetCategory')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetCategory.url = '/ceres/merchandise/MerchandiseService/GetCategory'
export const CreateCategory = (r?: CreateCategoryRequest, codes?: number[]) =>
  Request<CreateCategoryResponse>('/ceres/merchandise/MerchandiseService/CreateCategory')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateCategory.url = '/ceres/merchandise/MerchandiseService/CreateCategory'
export const UpdateCategory = (r?: UpdateCategoryRequest, codes?: number[]) =>
  Request<UpdateCategoryResponse>('/ceres/merchandise/MerchandiseService/UpdateCategory')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateCategory.url = '/ceres/merchandise/MerchandiseService/UpdateCategory'
export const DeleteCategory = (r?: DeleteCategoryRequest, codes?: number[]) =>
  Request<DeleteCategoryResponse>('/ceres/merchandise/MerchandiseService/DeleteCategory')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteCategory.url = '/ceres/merchandise/MerchandiseService/DeleteCategory'
export const ListCategory = (r?: ListCategoryRequest, codes?: number[]) =>
  Request<ListCategoryResponse>('/ceres/merchandise/MerchandiseService/ListCategory')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListCategory.url = '/ceres/merchandise/MerchandiseService/ListCategory'
export const GetSpu = (r?: GetSpuRequest, codes?: number[]) =>
  Request<GetSpuResponse>('/ceres/merchandise/MerchandiseService/GetSpu')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetSpu.url = '/ceres/merchandise/MerchandiseService/GetSpu'
export const GetManySpu = (r?: GetManySpuRequest, codes?: number[]) =>
  Request<GetManySpuResponse>('/ceres/merchandise/MerchandiseService/GetManySpu')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetManySpu.url = '/ceres/merchandise/MerchandiseService/GetManySpu'
export const CreateSpu = (r?: CreateSpuRequest, codes?: number[]) =>
  Request<CreateSpuResponse>('/ceres/merchandise/MerchandiseService/CreateSpu')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateSpu.url = '/ceres/merchandise/MerchandiseService/CreateSpu'
export const UpdateSpu = (r?: UpdateSpuRequest, codes?: number[]) =>
  Request<UpdateSpuResponse>('/ceres/merchandise/MerchandiseService/UpdateSpu')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateSpu.url = '/ceres/merchandise/MerchandiseService/UpdateSpu'
export const DeleteSpu = (r?: DeleteSpuRequest, codes?: number[]) =>
  Request<DeleteSpuResponse>('/ceres/merchandise/MerchandiseService/DeleteSpu')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteSpu.url = '/ceres/merchandise/MerchandiseService/DeleteSpu'
export const ListSpu = (r?: ListSpuRequest, codes?: number[]) =>
  Request<ListSpuResponse>('/ceres/merchandise/MerchandiseService/ListSpu')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListSpu.url = '/ceres/merchandise/MerchandiseService/ListSpu'
export const GetCategoryTree = (r?: GetCategoryTreeRequest, codes?: number[]) =>
  Request<GetCategoryTreeResponse>('/ceres/merchandise/MerchandiseService/GetCategoryTree')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetCategoryTree.url = '/ceres/merchandise/MerchandiseService/GetCategoryTree'
export const ChangeMerchandiseRank = (r?: ChangeMerchandiseRankRequest, codes?: number[]) =>
  Request<ChangeMerchandiseRankResponse>('/ceres/merchandise/MerchandiseService/ChangeMerchandiseRank')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ChangeMerchandiseRank.url = '/ceres/merchandise/MerchandiseService/ChangeMerchandiseRank'
export const GetManyCategory = (r?: GetManyCategoryRequest, codes?: number[]) =>
  Request<GetManyCategoryResponse>('/ceres/merchandise/MerchandiseService/GetManyCategory')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetManyCategory.url = '/ceres/merchandise/MerchandiseService/GetManyCategory'
export const GetSku = (r?: GetSkuRequest, codes?: number[]) =>
  Request<GetSkuResponse>('/ceres/merchandise/MerchandiseService/GetSku')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetSku.url = '/ceres/merchandise/MerchandiseService/GetSku'
export const CreateSku = (r?: CreateSkuRequest, codes?: number[]) =>
  Request<CreateSkuResponse>('/ceres/merchandise/MerchandiseService/CreateSku')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateSku.url = '/ceres/merchandise/MerchandiseService/CreateSku'
export const UpdateSku = (r?: UpdateSkuRequest, codes?: number[]) =>
  Request<UpdateSkuResponse>('/ceres/merchandise/MerchandiseService/UpdateSku')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateSku.url = '/ceres/merchandise/MerchandiseService/UpdateSku'
export const DeleteSku = (r?: DeleteSkuRequest, codes?: number[]) =>
  Request<DeleteSkuResponse>('/ceres/merchandise/MerchandiseService/DeleteSku')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteSku.url = '/ceres/merchandise/MerchandiseService/DeleteSku'
export const ListSku = (r?: ListSkuRequest, codes?: number[]) =>
  Request<ListSkuResponse>('/ceres/merchandise/MerchandiseService/ListSku')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListSku.url = '/ceres/merchandise/MerchandiseService/ListSku'
export const ExportSku = (r?: ExportSkuRequest, codes?: number[]) =>
  Request<ExportSkuResponse>('/ceres/merchandise/MerchandiseService/ExportSku')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ExportSku.url = '/ceres/merchandise/MerchandiseService/ExportSku'
export const ExportSsu = (r?: ExportSsuRequest, codes?: number[]) =>
  Request<ExportSsuResponse>('/ceres/merchandise/MerchandiseService/ExportSsu')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ExportSsu.url = '/ceres/merchandise/MerchandiseService/ExportSsu'
export const GetManySku = (r?: GetManySkuRequest, codes?: number[]) =>
  Request<GetManySkuResponse>('/ceres/merchandise/MerchandiseService/GetManySku')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetManySku.url = '/ceres/merchandise/MerchandiseService/GetManySku'
export const GetSkuSnap = (r?: GetSkuSnapRequest, codes?: number[]) =>
  Request<GetSkuSnapResponse>('/ceres/merchandise/MerchandiseService/GetSkuSnap')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetSkuSnap.url = '/ceres/merchandise/MerchandiseService/GetSkuSnap'
export const GetManySkuSnap = (r?: GetManySkuSnapRequest, codes?: number[]) =>
  Request<GetManySkuSnapResponse>('/ceres/merchandise/MerchandiseService/GetManySkuSnap')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetManySkuSnap.url = '/ceres/merchandise/MerchandiseService/GetManySkuSnap'
export const GetSsuSnap = (r?: GetSsuSnapRequest, codes?: number[]) =>
  Request<GetSsuSnapResponse>('/ceres/merchandise/MerchandiseService/GetSsuSnap')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetSsuSnap.url = '/ceres/merchandise/MerchandiseService/GetSsuSnap'
export const GetManySsuSnap = (r?: GetManySsuSnapRequest, codes?: number[]) =>
  Request<GetManySsuSnapResponse>('/ceres/merchandise/MerchandiseService/GetManySsuSnap')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetManySsuSnap.url = '/ceres/merchandise/MerchandiseService/GetManySsuSnap'
export const GetSsu = (r?: GetSsuRequest, codes?: number[]) =>
  Request<GetSsuResponse>('/ceres/merchandise/MerchandiseService/GetSsu')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetSsu.url = '/ceres/merchandise/MerchandiseService/GetSsu'
export const UpdateSsu = (r?: UpdateSsuRequest, codes?: number[]) =>
  Request<UpdateSsuResponse>('/ceres/merchandise/MerchandiseService/UpdateSsu')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateSsu.url = '/ceres/merchandise/MerchandiseService/UpdateSsu'
export const CreateSsu = (r?: CreateSsuRequest, codes?: number[]) =>
  Request<CreateSsuResponse>('/ceres/merchandise/MerchandiseService/CreateSsu')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateSsu.url = '/ceres/merchandise/MerchandiseService/CreateSsu'
export const DeleteSsu = (r?: DeleteSsuRequest, codes?: number[]) =>
  Request<DeleteSsuResponse>('/ceres/merchandise/MerchandiseService/DeleteSsu')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteSsu.url = '/ceres/merchandise/MerchandiseService/DeleteSsu'
export const BatchUpdateSkuSsu = (r?: BatchUpdateSkuSsuRequest, codes?: number[]) =>
  Request<BatchUpdateSkuSsuResponse>('/ceres/merchandise/MerchandiseService/BatchUpdateSkuSsu')
    .code([...(codes || [])])
    .data(r || {})
    .run()
BatchUpdateSkuSsu.url = '/ceres/merchandise/MerchandiseService/BatchUpdateSkuSsu'
export const BatchCreateCategorySku = (r?: BatchCreateCategorySkuRequest, codes?: number[]) =>
  Request<BatchCreateCategorySkuResponse>('/ceres/merchandise/MerchandiseService/BatchCreateCategorySku')
    .code([...(codes || [])])
    .data(r || {})
    .run()
BatchCreateCategorySku.url = '/ceres/merchandise/MerchandiseService/BatchCreateCategorySku'
export const BatchCreateSsu = (r?: BatchCreateSsuRequest, codes?: number[]) =>
  Request<BatchCreateSsuResponse>('/ceres/merchandise/MerchandiseService/BatchCreateSsu')
    .code([...(codes || [])])
    .data(r || {})
    .run()
BatchCreateSsu.url = '/ceres/merchandise/MerchandiseService/BatchCreateSsu'
export const ListSsu = (r?: ListSsuRequest, codes?: number[]) =>
  Request<ListSsuResponse>('/ceres/merchandise/MerchandiseService/ListSsu')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListSsu.url = '/ceres/merchandise/MerchandiseService/ListSsu'
export const ListSsuByCustomerOrQuotation = (r?: ListSsuByCustomerOrQuotationRequest, codes?: number[]) =>
  Request<ListSsuByCustomerOrQuotationResponse>('/ceres/merchandise/MerchandiseService/ListSsuByCustomerOrQuotation')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListSsuByCustomerOrQuotation.url = '/ceres/merchandise/MerchandiseService/ListSsuByCustomerOrQuotation'
export const GetRuleSet = (r?: GetRuleSetRequest, codes?: number[]) =>
  Request<GetRuleSetResponse>('/ceres/merchandise/MerchandiseService/GetRuleSet')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetRuleSet.url = '/ceres/merchandise/MerchandiseService/GetRuleSet'
export const CreateRuleSet = (r?: CreateRuleSetRequest, codes?: number[]) =>
  Request<CreateRuleSetResponse>('/ceres/merchandise/MerchandiseService/CreateRuleSet')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateRuleSet.url = '/ceres/merchandise/MerchandiseService/CreateRuleSet'
export const UpdateRuleSet = (r?: UpdateRuleSetRequest, codes?: number[]) =>
  Request<UpdateRuleSetResponse>('/ceres/merchandise/MerchandiseService/UpdateRuleSet')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateRuleSet.url = '/ceres/merchandise/MerchandiseService/UpdateRuleSet'
export const DeleteRuleSet = (r?: DeleteRuleSetRequest, codes?: number[]) =>
  Request<DeleteRuleSetResponse>('/ceres/merchandise/MerchandiseService/DeleteRuleSet')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteRuleSet.url = '/ceres/merchandise/MerchandiseService/DeleteRuleSet'
export const ListRuleSet = (r?: ListRuleSetRequest, codes?: number[]) =>
  Request<ListRuleSetResponse>('/ceres/merchandise/MerchandiseService/ListRuleSet')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListRuleSet.url = '/ceres/merchandise/MerchandiseService/ListRuleSet'
export const CreateUnit = (r?: CreateUnitRequest, codes?: number[]) =>
  Request<CreateUnitResponse>('/ceres/merchandise/MerchandiseService/CreateUnit')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateUnit.url = '/ceres/merchandise/MerchandiseService/CreateUnit'
export const ListUnit = (r?: ListUnitRequest, codes?: number[]) =>
  Request<ListUnitResponse>('/ceres/merchandise/MerchandiseService/ListUnit')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListUnit.url = '/ceres/merchandise/MerchandiseService/ListUnit'
export const GetUnitRateGroup = (r?: GetUnitRateGroupRequest, codes?: number[]) =>
  Request<GetUnitRateGroupResponse>('/ceres/merchandise/MerchandiseService/GetUnitRateGroup')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetUnitRateGroup.url = '/ceres/merchandise/MerchandiseService/GetUnitRateGroup'
export const InitUnit = (r?: InitUnitRequest, codes?: number[]) =>
  Request<InitUnitResponse>('/ceres/merchandise/MerchandiseService/InitUnit')
    .code([...(codes || [])])
    .data(r || {})
    .run()
InitUnit.url = '/ceres/merchandise/MerchandiseService/InitUnit'
export const InitCategoryImage = (r?: InitCategoryImageRequest, codes?: number[]) =>
  Request<InitCategoryImageResponse>('/ceres/merchandise/MerchandiseService/InitCategoryImage')
    .code([...(codes || [])])
    .data(r || {})
    .run()
InitCategoryImage.url = '/ceres/merchandise/MerchandiseService/InitCategoryImage'
export const ListCategoryImage = (r?: ListCategoryImageRequest, codes?: number[]) =>
  Request<ListCategoryImageResponse>('/ceres/merchandise/MerchandiseService/ListCategoryImage')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListCategoryImage.url = '/ceres/merchandise/MerchandiseService/ListCategoryImage'
export const GetQuotation = (r?: GetQuotationRequest, codes?: number[]) =>
  Request<GetQuotationResponse>('/ceres/merchandise/MerchandiseService/GetQuotation')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetQuotation.url = '/ceres/merchandise/MerchandiseService/GetQuotation'
export const ListQuotation = (r?: ListQuotationRequest, codes?: number[]) =>
  Request<ListQuotationResponse>('/ceres/merchandise/MerchandiseService/ListQuotation')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListQuotation.url = '/ceres/merchandise/MerchandiseService/ListQuotation'
export const GetManyQuotation = (r?: GetManyQuotationRequest, codes?: number[]) =>
  Request<GetManyQuotationResponse>('/ceres/merchandise/MerchandiseService/GetManyQuotation')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetManyQuotation.url = '/ceres/merchandise/MerchandiseService/GetManyQuotation'
export const CreateQuotation = (r?: CreateQuotationRequest, codes?: number[]) =>
  Request<CreateQuotationResponse>('/ceres/merchandise/MerchandiseService/CreateQuotation')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateQuotation.url = '/ceres/merchandise/MerchandiseService/CreateQuotation'
export const UpdateQuotation = (r?: UpdateQuotationRequest, codes?: number[]) =>
  Request<UpdateQuotationResponse>('/ceres/merchandise/MerchandiseService/UpdateQuotation')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateQuotation.url = '/ceres/merchandise/MerchandiseService/UpdateQuotation'
export const DeleteQuotation = (r?: DeleteQuotationRequest, codes?: number[]) =>
  Request<DeleteQuotationResponse>('/ceres/merchandise/MerchandiseService/DeleteQuotation')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteQuotation.url = '/ceres/merchandise/MerchandiseService/DeleteQuotation'
export const DeleteBasicPrice = (r?: DeleteBasicPriceRequest, codes?: number[]) =>
  Request<DeleteBasicPriceResponse>('/ceres/merchandise/MerchandiseService/DeleteBasicPrice')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteBasicPrice.url = '/ceres/merchandise/MerchandiseService/DeleteBasicPrice'
export const GetManySsuPrice = (r?: GetManySsuPriceRequest, codes?: number[]) =>
  Request<GetManySsuPriceResponse>('/ceres/merchandise/MerchandiseService/GetManySsuPrice')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetManySsuPrice.url = '/ceres/merchandise/MerchandiseService/GetManySsuPrice'
export const SyncSsuImage = (r?: SyncSsuImageRequest, codes?: number[]) =>
  Request<SyncSsuImageResponse>('/ceres/merchandise/MerchandiseService/SyncSsuImage')
    .code([...(codes || [])])
    .data(r || {})
    .run()
SyncSsuImage.url = '/ceres/merchandise/MerchandiseService/SyncSsuImage'
export const CreateFavorite = (r?: CreateFavoriteRequest, codes?: number[]) =>
  Request<CreateFavoriteResponse>('/ceres/merchandise/MerchandiseService/CreateFavorite')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateFavorite.url = '/ceres/merchandise/MerchandiseService/CreateFavorite'
export const UpdateFavorite = (r?: UpdateFavoriteRequest, codes?: number[]) =>
  Request<UpdateFavoriteResponse>('/ceres/merchandise/MerchandiseService/UpdateFavorite')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateFavorite.url = '/ceres/merchandise/MerchandiseService/UpdateFavorite'
export const DeleteFavorite = (r?: DeleteFavoriteRequest, codes?: number[]) =>
  Request<DeleteFavoriteResponse>('/ceres/merchandise/MerchandiseService/DeleteFavorite')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteFavorite.url = '/ceres/merchandise/MerchandiseService/DeleteFavorite'
export const ListFavorite = (r?: ListFavoriteRequest, codes?: number[]) =>
  Request<ListFavoriteResponse>('/ceres/merchandise/MerchandiseService/ListFavorite')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListFavorite.url = '/ceres/merchandise/MerchandiseService/ListFavorite'
export const CreateFavoriteSsu = (r?: CreateFavoriteSsuRequest, codes?: number[]) =>
  Request<CreateFavoriteSsuResponse>('/ceres/merchandise/MerchandiseService/CreateFavoriteSsu')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateFavoriteSsu.url = '/ceres/merchandise/MerchandiseService/CreateFavoriteSsu'
export const DeleteFavoriteSsu = (r?: DeleteFavoriteSsuRequest, codes?: number[]) =>
  Request<DeleteFavoriteSsuResponse>('/ceres/merchandise/MerchandiseService/DeleteFavoriteSsu')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteFavoriteSsu.url = '/ceres/merchandise/MerchandiseService/DeleteFavoriteSsu'
export const ListFavoriteSsu = (r?: ListFavoriteSsuRequest, codes?: number[]) =>
  Request<ListFavoriteSsuResponse>('/ceres/merchandise/MerchandiseService/ListFavoriteSsu')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListFavoriteSsu.url = '/ceres/merchandise/MerchandiseService/ListFavoriteSsu'
