// gm_api/src/merchandise/pc.tsx
import React, { FC } from 'react'
import { Select, SelectProps } from '@gm-pc/react'
import {
  SelectList,
  SelectEnum,
  MoreSelectList,
  MapId,
} from '../util/pc_util'
import type {
  MapProps,
  SelectListProps,
  SelectEnumProps,
  MoreSelectListProps,
  MapIdProps,
} from '../util/pc_util'
import {
  list_Status_Code,
  map_Status_Code,
  list_SystemUnitId,
  map_SystemUnitId,
  list_UnitType,
  map_UnitType,
  list_Unit_Type,
  map_Unit_Type,
  list_Optype,
  map_Optype,
  list_Sku_PurchaseType,
  map_Sku_PurchaseType,
  list_Sku_DispatchType,
  map_Sku_DispatchType,
  list_Sku_Status,
  map_Sku_Status,
  list_Sku_SkuType,
  map_Sku_SkuType,
  list_Sku_NotPackageSubSkuType,
  map_Sku_NotPackageSubSkuType,
  list_Sku_PackageSubSkuType,
  map_Sku_PackageSubSkuType,
  list_ModuleType,
  map_ModuleType,
  list_Ssu_PackageCalculateType,
  map_Ssu_PackageCalculateType,
  list_Ssu_NutritionType,
  map_Ssu_NutritionType,
  list_Ssu_SplitFlag,
  map_Ssu_SplitFlag,
  list_Ssu_ShippingFeeUnit,
  map_Ssu_ShippingFeeUnit,
  list_Favorite_FavoriteType,
  map_Favorite_FavoriteType,
  list_Favorite_FavoriteContentType,
  map_Favorite_FavoriteContentType,
  list_RuleSet_State,
  map_RuleSet_State,
  list_RuleSet_Type,
  map_RuleSet_Type,
  list_ChangeMerchandiseRankRequest_NodeType,
  map_ChangeMerchandiseRankRequest_NodeType,
  list_ListSkuRequest_RequestData,
  map_ListSkuRequest_RequestData,
  list_GetManySkuRequest_RequestData,
  map_GetManySkuRequest_RequestData,
  Unit,
  ListUnit,
  ListUnitResponse,
  Category,
  ListCategory,
  ListCategoryResponse,
  Spu,
  ListSpu,
  ListSpuResponse,
  Sku,
  ListSku,
  ListSkuResponse,
  Ssu,
  ListSsu,
  ListSsuResponse,
  Quotation,
  ListQuotation,
  ListQuotationResponse,
  Favorite,
  ListFavorite,
  ListFavoriteResponse,
  FavoriteSsu,
  ListFavoriteSsu,
  ListFavoriteSsuResponse,
  RuleSet,
  ListRuleSet,
  ListRuleSetResponse,
} from './index'

export const Select_Status_Code: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Status_Code} />
}

export const Map_Status_Code: FC<MapProps> = ({ value }) => {
  return <>{map_Status_Code[value]}</>
}

export const Select_SystemUnitId: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_SystemUnitId} />
}

export const Map_SystemUnitId: FC<MapProps> = ({ value }) => {
  return <>{map_SystemUnitId[value]}</>
}

export const Select_UnitType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_UnitType} />
}

export const Map_UnitType: FC<MapProps> = ({ value }) => {
  return <>{map_UnitType[value]}</>
}

export const Select_Unit_Type: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Unit_Type} />
}

export const Map_Unit_Type: FC<MapProps> = ({ value }) => {
  return <>{map_Unit_Type[value]}</>
}

export const Select_Optype: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Optype} />
}

export const Map_Optype: FC<MapProps> = ({ value }) => {
  return <>{map_Optype[value]}</>
}

export const Select_Sku_PurchaseType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Sku_PurchaseType} />
}

export const Map_Sku_PurchaseType: FC<MapProps> = ({ value }) => {
  return <>{map_Sku_PurchaseType[value]}</>
}

export const Select_Sku_DispatchType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Sku_DispatchType} />
}

export const Map_Sku_DispatchType: FC<MapProps> = ({ value }) => {
  return <>{map_Sku_DispatchType[value]}</>
}

export const Select_Sku_Status: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Sku_Status} />
}

export const Map_Sku_Status: FC<MapProps> = ({ value }) => {
  return <>{map_Sku_Status[value]}</>
}

export const Select_Sku_SkuType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Sku_SkuType} />
}

export const Map_Sku_SkuType: FC<MapProps> = ({ value }) => {
  return <>{map_Sku_SkuType[value]}</>
}

export const Select_Sku_NotPackageSubSkuType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Sku_NotPackageSubSkuType} />
}

export const Map_Sku_NotPackageSubSkuType: FC<MapProps> = ({ value }) => {
  return <>{map_Sku_NotPackageSubSkuType[value]}</>
}

export const Select_Sku_PackageSubSkuType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Sku_PackageSubSkuType} />
}

export const Map_Sku_PackageSubSkuType: FC<MapProps> = ({ value }) => {
  return <>{map_Sku_PackageSubSkuType[value]}</>
}

export const Select_ModuleType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_ModuleType} />
}

export const Map_ModuleType: FC<MapProps> = ({ value }) => {
  return <>{map_ModuleType[value]}</>
}

export const Select_Ssu_PackageCalculateType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Ssu_PackageCalculateType} />
}

export const Map_Ssu_PackageCalculateType: FC<MapProps> = ({ value }) => {
  return <>{map_Ssu_PackageCalculateType[value]}</>
}

export const Select_Ssu_NutritionType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Ssu_NutritionType} />
}

export const Map_Ssu_NutritionType: FC<MapProps> = ({ value }) => {
  return <>{map_Ssu_NutritionType[value]}</>
}

export const Select_Ssu_SplitFlag: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Ssu_SplitFlag} />
}

export const Map_Ssu_SplitFlag: FC<MapProps> = ({ value }) => {
  return <>{map_Ssu_SplitFlag[value]}</>
}

export const Select_Ssu_ShippingFeeUnit: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Ssu_ShippingFeeUnit} />
}

export const Map_Ssu_ShippingFeeUnit: FC<MapProps> = ({ value }) => {
  return <>{map_Ssu_ShippingFeeUnit[value]}</>
}

export const Select_Favorite_FavoriteType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Favorite_FavoriteType} />
}

export const Map_Favorite_FavoriteType: FC<MapProps> = ({ value }) => {
  return <>{map_Favorite_FavoriteType[value]}</>
}

export const Select_Favorite_FavoriteContentType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Favorite_FavoriteContentType} />
}

export const Map_Favorite_FavoriteContentType: FC<MapProps> = ({ value }) => {
  return <>{map_Favorite_FavoriteContentType[value]}</>
}

export const Select_RuleSet_State: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_RuleSet_State} />
}

export const Map_RuleSet_State: FC<MapProps> = ({ value }) => {
  return <>{map_RuleSet_State[value]}</>
}

export const Select_RuleSet_Type: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_RuleSet_Type} />
}

export const Map_RuleSet_Type: FC<MapProps> = ({ value }) => {
  return <>{map_RuleSet_Type[value]}</>
}

export const Select_ChangeMerchandiseRankRequest_NodeType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_ChangeMerchandiseRankRequest_NodeType} />
}

export const Map_ChangeMerchandiseRankRequest_NodeType: FC<MapProps> = ({ value }) => {
  return <>{map_ChangeMerchandiseRankRequest_NodeType[value]}</>
}

export const Select_ListSkuRequest_RequestData: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_ListSkuRequest_RequestData} />
}

export const Map_ListSkuRequest_RequestData: FC<MapProps> = ({ value }) => {
  return <>{map_ListSkuRequest_RequestData[value]}</>
}

export const Select_GetManySkuRequest_RequestData: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_GetManySkuRequest_RequestData} />
}

export const Map_GetManySkuRequest_RequestData: FC<MapProps> = ({ value }) => {
  return <>{map_GetManySkuRequest_RequestData[value]}</>
}

export const Select_Unit: FC<SelectListProps<
  Unit,
  ListUnitResponse
>> = (props) => {
  return (
    <SelectList<Unit, ListUnitResponse>
      {...props}
      messageName='Unit'
      fetchList={ListUnit}
    />
  )
}

export const MoreSelect_Unit: FC<MoreSelectListProps<
  Unit,
  ListUnitResponse
>> = (props) => {
  return (
    <MoreSelectList<Unit, ListUnitResponse>
      {...props}
      messageName='Unit'
      fetchList={ListUnit}
    />
  )
}

export const Select_Category: FC<SelectListProps<
  Category,
  ListCategoryResponse
>> = (props) => {
  return (
    <SelectList<Category, ListCategoryResponse>
      {...props}
      messageName='Category'
      fetchList={ListCategory}
    />
  )
}

export const MoreSelect_Category: FC<MoreSelectListProps<
  Category,
  ListCategoryResponse
>> = (props) => {
  return (
    <MoreSelectList<Category, ListCategoryResponse>
      {...props}
      messageName='Category'
      fetchList={ListCategory}
    />
  )
}

export const Select_Spu: FC<SelectListProps<
  Spu,
  ListSpuResponse
>> = (props) => {
  return (
    <SelectList<Spu, ListSpuResponse>
      {...props}
      messageName='Spu'
      fetchList={ListSpu}
    />
  )
}

export const MoreSelect_Spu: FC<MoreSelectListProps<
  Spu,
  ListSpuResponse
>> = (props) => {
  return (
    <MoreSelectList<Spu, ListSpuResponse>
      {...props}
      messageName='Spu'
      fetchList={ListSpu}
    />
  )
}

export const Select_Sku: FC<SelectListProps<
  Sku,
  ListSkuResponse
>> = (props) => {
  return (
    <SelectList<Sku, ListSkuResponse>
      {...props}
      messageName='Sku'
      fetchList={ListSku}
    />
  )
}

export const MoreSelect_Sku: FC<MoreSelectListProps<
  Sku,
  ListSkuResponse
>> = (props) => {
  return (
    <MoreSelectList<Sku, ListSkuResponse>
      {...props}
      messageName='Sku'
      fetchList={ListSku}
    />
  )
}

export const Select_Ssu: FC<SelectListProps<
  Ssu,
  ListSsuResponse
>> = (props) => {
  return (
    <SelectList<Ssu, ListSsuResponse>
      {...props}
      messageName='Ssu'
      fetchList={ListSsu}
    />
  )
}

export const MoreSelect_Ssu: FC<MoreSelectListProps<
  Ssu,
  ListSsuResponse
>> = (props) => {
  return (
    <MoreSelectList<Ssu, ListSsuResponse>
      {...props}
      messageName='Ssu'
      fetchList={ListSsu}
    />
  )
}

export const Select_Quotation: FC<SelectListProps<
  Quotation,
  ListQuotationResponse
>> = (props) => {
  return (
    <SelectList<Quotation, ListQuotationResponse>
      {...props}
      messageName='Quotation'
      fetchList={ListQuotation}
    />
  )
}

export const MoreSelect_Quotation: FC<MoreSelectListProps<
  Quotation,
  ListQuotationResponse
>> = (props) => {
  return (
    <MoreSelectList<Quotation, ListQuotationResponse>
      {...props}
      messageName='Quotation'
      fetchList={ListQuotation}
    />
  )
}

export const Select_Favorite: FC<SelectListProps<
  Favorite,
  ListFavoriteResponse
>> = (props) => {
  return (
    <SelectList<Favorite, ListFavoriteResponse>
      {...props}
      messageName='Favorite'
      fetchList={ListFavorite}
    />
  )
}

export const MoreSelect_Favorite: FC<MoreSelectListProps<
  Favorite,
  ListFavoriteResponse
>> = (props) => {
  return (
    <MoreSelectList<Favorite, ListFavoriteResponse>
      {...props}
      messageName='Favorite'
      fetchList={ListFavorite}
    />
  )
}

export const Select_FavoriteSsu: FC<SelectListProps<
  FavoriteSsu,
  ListFavoriteSsuResponse
>> = (props) => {
  return (
    <SelectList<FavoriteSsu, ListFavoriteSsuResponse>
      {...props}
      messageName='FavoriteSsu'
      fetchList={ListFavoriteSsu}
    />
  )
}

export const MoreSelect_FavoriteSsu: FC<MoreSelectListProps<
  FavoriteSsu,
  ListFavoriteSsuResponse
>> = (props) => {
  return (
    <MoreSelectList<FavoriteSsu, ListFavoriteSsuResponse>
      {...props}
      messageName='FavoriteSsu'
      fetchList={ListFavoriteSsu}
    />
  )
}

export const Select_RuleSet: FC<SelectListProps<
  RuleSet,
  ListRuleSetResponse
>> = (props) => {
  return (
    <SelectList<RuleSet, ListRuleSetResponse>
      {...props}
      messageName='RuleSet'
      fetchList={ListRuleSet}
    />
  )
}

export const MoreSelect_RuleSet: FC<MoreSelectListProps<
  RuleSet,
  ListRuleSetResponse
>> = (props) => {
  return (
    <MoreSelectList<RuleSet, ListRuleSetResponse>
      {...props}
      messageName='RuleSet'
      fetchList={ListRuleSet}
    />
  )
}

export const MapId_Unit: FC<MapIdProps<Unit, ListUnitResponse>> = (
  props,
) => {
  return (
    <MapId<Unit, ListUnitResponse>
      {...props}
      packageName='merchandise'
      messageName='Unit'
      fetchList={ListUnit}
    />
  )
}

export const MapId_Category: FC<MapIdProps<Category, ListCategoryResponse>> = (
  props,
) => {
  return (
    <MapId<Category, ListCategoryResponse>
      {...props}
      packageName='merchandise'
      messageName='Category'
      fetchList={ListCategory}
    />
  )
}

export const MapId_Spu: FC<MapIdProps<Spu, ListSpuResponse>> = (
  props,
) => {
  return (
    <MapId<Spu, ListSpuResponse>
      {...props}
      packageName='merchandise'
      messageName='Spu'
      fetchList={ListSpu}
    />
  )
}

export const MapId_Sku: FC<MapIdProps<Sku, ListSkuResponse>> = (
  props,
) => {
  return (
    <MapId<Sku, ListSkuResponse>
      {...props}
      packageName='merchandise'
      messageName='Sku'
      fetchList={ListSku}
    />
  )
}

export const MapId_Ssu: FC<MapIdProps<Ssu, ListSsuResponse>> = (
  props,
) => {
  return (
    <MapId<Ssu, ListSsuResponse>
      {...props}
      packageName='merchandise'
      messageName='Ssu'
      fetchList={ListSsu}
    />
  )
}

export type {
  MapProps,
  SelectListProps,
  MoreSelectListProps,
}
