// gm_api/src/account/methods.ts
import { Request } from '../util/request'
import {
  CreateAccountRequest, CreateAccountResponse,
  UpdateAccountRequest, UpdateAccountResponse,
  DeleteAccountRequest, DeleteAccountResponse,
  GetAccountRequest, GetAccountResponse,
  CreateProfileRequest, CreateProfileResponse,
  UpdateProfileRequest, UpdateProfileResponse,
  GetProfileRequest, GetProfileResponse,
  ListProfileRequest, ListProfileResponse,
} from './types'
export const CreateAccount = (r?: CreateAccountRequest, codes?: number[]) =>
  Request<CreateAccountResponse>('/ceres/account/AccountService/CreateAccount')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateAccount.url = '/ceres/account/AccountService/CreateAccount'
export const UpdateAccount = (r?: UpdateAccountRequest, codes?: number[]) =>
  Request<UpdateAccountResponse>('/ceres/account/AccountService/UpdateAccount')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateAccount.url = '/ceres/account/AccountService/UpdateAccount'
export const DeleteAccount = (r?: DeleteAccountRequest, codes?: number[]) =>
  Request<DeleteAccountResponse>('/ceres/account/AccountService/DeleteAccount')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteAccount.url = '/ceres/account/AccountService/DeleteAccount'
export const GetAccount = (r?: GetAccountRequest, codes?: number[]) =>
  Request<GetAccountResponse>('/ceres/account/AccountService/GetAccount')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetAccount.url = '/ceres/account/AccountService/GetAccount'
export const CreateProfile = (r?: CreateProfileRequest, codes?: number[]) =>
  Request<CreateProfileResponse>('/ceres/account/AccountService/CreateProfile')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateProfile.url = '/ceres/account/AccountService/CreateProfile'
export const UpdateProfile = (r?: UpdateProfileRequest, codes?: number[]) =>
  Request<UpdateProfileResponse>('/ceres/account/AccountService/UpdateProfile')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateProfile.url = '/ceres/account/AccountService/UpdateProfile'
export const GetProfile = (r?: GetProfileRequest, codes?: number[]) =>
  Request<GetProfileResponse>('/ceres/account/AccountService/GetProfile')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetProfile.url = '/ceres/account/AccountService/GetProfile'
export const ListProfile = (r?: ListProfileRequest, codes?: number[]) =>
  Request<ListProfileResponse>('/ceres/account/AccountService/ListProfile')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListProfile.url = '/ceres/account/AccountService/ListProfile'
