// gm_api/src/account/pc.tsx
import React, { FC } from 'react'
import { Select, SelectProps } from '@gm-pc/react'
import {
  SelectList,
  SelectEnum,
  MoreSelectList,
  MapId,
} from '../util/pc_util'
import type {
  MapProps,
  SelectListProps,
  SelectEnumProps,
  MoreSelectListProps,
  MapIdProps,
} from '../util/pc_util'
import {
  list_Account_Type,
  map_Account_Type,
  list_Account_Status,
  map_Account_Status,
  Profile,
  ListProfile,
  ListProfileResponse,
} from './index'

export const Select_Account_Type: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Account_Type} />
}

export const Map_Account_Type: FC<MapProps> = ({ value }) => {
  return <>{map_Account_Type[value]}</>
}

export const Select_Account_Status: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Account_Status} />
}

export const Map_Account_Status: FC<MapProps> = ({ value }) => {
  return <>{map_Account_Status[value]}</>
}

export const Select_Profile: FC<SelectListProps<
  Profile,
  ListProfileResponse
>> = (props) => {
  return (
    <SelectList<Profile, ListProfileResponse>
      {...props}
      messageName='Profile'
      fetchList={ListProfile}
    />
  )
}

export const MoreSelect_Profile: FC<MoreSelectListProps<
  Profile,
  ListProfileResponse
>> = (props) => {
  return (
    <MoreSelectList<Profile, ListProfileResponse>
      {...props}
      messageName='Profile'
      fetchList={ListProfile}
    />
  )
}

export const MapId_Profile: FC<MapIdProps<Profile, ListProfileResponse>> = (
  props,
) => {
  return (
    <MapId<Profile, ListProfileResponse>
      {...props}
      packageName='account'
      messageName='Profile'
      fetchList={ListProfile}
    />
  )
}

export type {
  MapProps,
  SelectListProps,
  MoreSelectListProps,
}
