// gm_api/src/account/types.ts
export enum Account_Type {
  TYPE_UNSPECIFIED = 0,
  TYPE_PUBLIC = 1,
  TYPE_PRIVATE = 2,
}
export enum Account_Status {
  STATUS_UNSPECIFIED = 0,
  STATUS_DISABLED = 256,
}
export interface Account {
  account_id: string
  status?: string
  revision?: string
  create_time?: string
  update_time?: string
  delete_time?: string
  type: Account_Type
  scope?: string
  subtype?: string
  username?: string
  password?: string
  phone?: string
  email?: string
  wechat_openid?: string
  wechat_unionid?: string
  wechat_nickname?: string
  wechat_avatar?: string
}
export interface ReqCreateAccount {
  // account_id: string
  status?: string
  revision?: string
  // create_time?: string
  // update_time?: string
  // delete_time?: string
  type: Account_Type
  scope?: string
  subtype?: string
  username?: string
  password?: string
  phone?: string
  email?: string
  wechat_openid?: string
  wechat_unionid?: string
  wechat_nickname?: string
  wechat_avatar?: string
}
export interface Profile {
  profile_id: string
  status?: string
  revision?: string
  create_time?: string
  update_time?: string
  delete_time?: string
  nickname: string
  avatar?: string
}
export interface ReqCreateProfile {
  // profile_id: string
  status?: string
  revision?: string
  // create_time?: string
  // update_time?: string
  // delete_time?: string
  nickname: string
  avatar?: string
}
export interface CreateAccountRequest {
  account?: ReqCreateAccount
}
export interface CreateAccountResponse {
  account: Account
}
export interface UpdateAccountRequest {
  account?: Account
}
export interface UpdateAccountResponse {
  account: Account
}
export interface DeleteAccountRequest {
  account_id?: string
}
export interface DeleteAccountResponse {
  account: Account
}
export interface GetAccountRequest {
  account_id?: string
  type?: Account_Type
  scope?: string
  subtype?: string
  username?: string
  phone?: string
  email?: string
  wechat_unionid?: string
}
export interface GetAccountResponse {
  account: Account
}
export interface CreateProfileRequest {
  profile?: ReqCreateProfile
}
export interface CreateProfileResponse {
  profile: Profile
}
export interface UpdateProfileRequest {
  profile?: Profile
}
export interface UpdateProfileResponse {
  profile: Profile
}
export interface GetProfileRequest {
  profile_id?: string
}
export interface GetProfileResponse {
  profile: Profile
}
export interface ListProfileRequest {
  profile_ids?: string[]
}
export interface ListProfileResponse {
  profiles: Profile[]
}
