// gm_api/src/eshop/data.ts
import { getLocale } from './lng'

interface Enum_List_Item {
  text: string
  value: any
}

interface Enum_Map {
  [key: string]: string
}
export const list_Status_Code: Enum_List_Item[] = [
  { value: 20200001, text: getLocale('Status_Code__CODE_ERROR_LOAD_LOCATION') },
  { value: 20200002, text: getLocale('Status_Code__CODE_ERROR_RECEIVE_TIME') },
  { value: 20200003, text: getLocale('Status_Code__CODE_ERROR_NO_DETAIL') },
  { value: 20200004, text: getLocale('Status_Code__CODE_ERROR_NO_ADDRESS') },
  { value: 20200005, text: getLocale('Status_Code__CODE_ERROR_EXISTED_UNRELEASE_ORDER') },
]
export const map_Status_Code: Enum_Map = {
  20200001: getLocale('Status_Code__CODE_ERROR_LOAD_LOCATION'),
  20200002: getLocale('Status_Code__CODE_ERROR_RECEIVE_TIME'),
  20200003: getLocale('Status_Code__CODE_ERROR_NO_DETAIL'),
  20200004: getLocale('Status_Code__CODE_ERROR_NO_ADDRESS'),
  20200005: getLocale('Status_Code__CODE_ERROR_EXISTED_UNRELEASE_ORDER'),
}
export const list_Menu_CycleType: Enum_List_Item[] = [
  { value: 1, text: getLocale('Menu_CycleType__CYCLETYPE_DAILY') },
  { value: 2, text: getLocale('Menu_CycleType__CYCLETYPE_WEEKLY') },
  { value: 3, text: getLocale('Menu_CycleType__CYCLETYPE_MONTHLY') },
]
export const map_Menu_CycleType: Enum_Map = {
  1: getLocale('Menu_CycleType__CYCLETYPE_DAILY'),
  2: getLocale('Menu_CycleType__CYCLETYPE_WEEKLY'),
  3: getLocale('Menu_CycleType__CYCLETYPE_MONTHLY'),
}
export const list_MenuDetail_State: Enum_List_Item[] = [
  { value: 1, text: getLocale('MenuDetail_State__STATE_PREPARE') },
  { value: 2, text: getLocale('MenuDetail_State__STATE_FINISHED') },
]
export const map_MenuDetail_State: Enum_Map = {
  1: getLocale('MenuDetail_State__STATE_PREPARE'),
  2: getLocale('MenuDetail_State__STATE_FINISHED'),
}
