// gm_api/src/eshop/methods.ts
import { Request } from '../util/request'
import {
  GetEshopCartOrderRequest, GetEshopCartOrderResponse,
  UpdateEshopCartOrderRequest, UpdateEshopCartOrderResponse,
  CreateEshopOrderRequest, CreateEshopOrderResponse,
  CreateProductOrderRequest, CreateProductOrderResponse,
  GetLeaveApplicationRequest, GetLeaveApplicationResponse,
  CreateLeaveApplicationRequest, CreateLeaveApplicationResponse,
  UpdateLeaveApplicationRequest, UpdateLeaveApplicationResponse,
  DeleteLeaveApplicationRequest, DeleteLeaveApplicationResponse,
  ListLeaveApplicationRequest, ListLeaveApplicationResponse,
  GetMenuRequest, GetMenuResponse,
  ListMenuRequest, ListMenuResponse,
  ListMenuDetailRequest, ListMenuDetailResponse,
  CreateMenuRequest, CreateMenuResponse,
  UpdateMenuRequest, UpdateMenuResponse,
  CreateMenuDetailRequest, CreateMenuDetailResponse,
  UpdateMenuDetailRequest, UpdateMenuDetailResponse,
  DeleteMenuRequest, DeleteMenuResponse,
  GetCustomerMenuRelationRequest, GetCustomerMenuRelationResponse,
  ListMenuPeriodGroupRequest, ListMenuPeriodGroupResponse,
  CreateMenuPeriodGroupRequest, CreateMenuPeriodGroupResponse,
  UpdateMenuPeriodGroupRequest, UpdateMenuPeriodGroupResponse,
  DeleteMenuPeriodGroupRequest, DeleteMenuPeriodGroupResponse,
  GetManyMenuPeriodGroupRequest, GetManyMenuPeriodGroupResponse,
  ExportEshopOrderRequest, ExportEshopOrderResponse,
  BatchUpdateEshopOrderStateRequest, BatchUpdateEshopOrderStateResponse,
  ExportLeaveApplicationRequest, ExportLeaveApplicationResponse,
} from './types'
export const GetEshopCartOrder = (r?: GetEshopCartOrderRequest, codes?: number[]) =>
  Request<GetEshopCartOrderResponse>('/ceres/eshop/EshopService/GetEshopCartOrder')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetEshopCartOrder.url = '/ceres/eshop/EshopService/GetEshopCartOrder'
export const UpdateEshopCartOrder = (r?: UpdateEshopCartOrderRequest, codes?: number[]) =>
  Request<UpdateEshopCartOrderResponse>('/ceres/eshop/EshopService/UpdateEshopCartOrder')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateEshopCartOrder.url = '/ceres/eshop/EshopService/UpdateEshopCartOrder'
export const CreateEshopOrder = (r?: CreateEshopOrderRequest, codes?: number[]) =>
  Request<CreateEshopOrderResponse>('/ceres/eshop/EshopService/CreateEshopOrder')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateEshopOrder.url = '/ceres/eshop/EshopService/CreateEshopOrder'
export const CreateProductOrder = (r?: CreateProductOrderRequest, codes?: number[]) =>
  Request<CreateProductOrderResponse>('/ceres/eshop/EshopService/CreateProductOrder')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateProductOrder.url = '/ceres/eshop/EshopService/CreateProductOrder'
export const GetLeaveApplication = (r?: GetLeaveApplicationRequest, codes?: number[]) =>
  Request<GetLeaveApplicationResponse>('/ceres/eshop/EshopService/GetLeaveApplication')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetLeaveApplication.url = '/ceres/eshop/EshopService/GetLeaveApplication'
export const CreateLeaveApplication = (r?: CreateLeaveApplicationRequest, codes?: number[]) =>
  Request<CreateLeaveApplicationResponse>('/ceres/eshop/EshopService/CreateLeaveApplication')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateLeaveApplication.url = '/ceres/eshop/EshopService/CreateLeaveApplication'
export const UpdateLeaveApplication = (r?: UpdateLeaveApplicationRequest, codes?: number[]) =>
  Request<UpdateLeaveApplicationResponse>('/ceres/eshop/EshopService/UpdateLeaveApplication')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateLeaveApplication.url = '/ceres/eshop/EshopService/UpdateLeaveApplication'
export const DeleteLeaveApplication = (r?: DeleteLeaveApplicationRequest, codes?: number[]) =>
  Request<DeleteLeaveApplicationResponse>('/ceres/eshop/EshopService/DeleteLeaveApplication')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteLeaveApplication.url = '/ceres/eshop/EshopService/DeleteLeaveApplication'
export const ListLeaveApplication = (r?: ListLeaveApplicationRequest, codes?: number[]) =>
  Request<ListLeaveApplicationResponse>('/ceres/eshop/EshopService/ListLeaveApplication')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListLeaveApplication.url = '/ceres/eshop/EshopService/ListLeaveApplication'
export const GetMenu = (r?: GetMenuRequest, codes?: number[]) =>
  Request<GetMenuResponse>('/ceres/eshop/EshopService/GetMenu')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetMenu.url = '/ceres/eshop/EshopService/GetMenu'
export const ListMenu = (r?: ListMenuRequest, codes?: number[]) =>
  Request<ListMenuResponse>('/ceres/eshop/EshopService/ListMenu')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListMenu.url = '/ceres/eshop/EshopService/ListMenu'
export const ListMenuDetail = (r?: ListMenuDetailRequest, codes?: number[]) =>
  Request<ListMenuDetailResponse>('/ceres/eshop/EshopService/ListMenuDetail')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListMenuDetail.url = '/ceres/eshop/EshopService/ListMenuDetail'
export const CreateMenu = (r?: CreateMenuRequest, codes?: number[]) =>
  Request<CreateMenuResponse>('/ceres/eshop/EshopService/CreateMenu')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateMenu.url = '/ceres/eshop/EshopService/CreateMenu'
export const UpdateMenu = (r?: UpdateMenuRequest, codes?: number[]) =>
  Request<UpdateMenuResponse>('/ceres/eshop/EshopService/UpdateMenu')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateMenu.url = '/ceres/eshop/EshopService/UpdateMenu'
export const CreateMenuDetail = (r?: CreateMenuDetailRequest, codes?: number[]) =>
  Request<CreateMenuDetailResponse>('/ceres/eshop/EshopService/CreateMenuDetail')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateMenuDetail.url = '/ceres/eshop/EshopService/CreateMenuDetail'
export const UpdateMenuDetail = (r?: UpdateMenuDetailRequest, codes?: number[]) =>
  Request<UpdateMenuDetailResponse>('/ceres/eshop/EshopService/UpdateMenuDetail')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateMenuDetail.url = '/ceres/eshop/EshopService/UpdateMenuDetail'
export const DeleteMenu = (r?: DeleteMenuRequest, codes?: number[]) =>
  Request<DeleteMenuResponse>('/ceres/eshop/EshopService/DeleteMenu')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteMenu.url = '/ceres/eshop/EshopService/DeleteMenu'
export const GetCustomerMenuRelation = (r?: GetCustomerMenuRelationRequest, codes?: number[]) =>
  Request<GetCustomerMenuRelationResponse>('/ceres/merchandise/MerchandiseService/GetCustomerMenuRelation')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetCustomerMenuRelation.url = '/ceres/merchandise/MerchandiseService/GetCustomerMenuRelation'
export const ListMenuPeriodGroup = (r?: ListMenuPeriodGroupRequest, codes?: number[]) =>
  Request<ListMenuPeriodGroupResponse>('/ceres/eshop/EshopService/ListMenuPeriodGroup')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ListMenuPeriodGroup.url = '/ceres/eshop/EshopService/ListMenuPeriodGroup'
export const CreateMenuPeriodGroup = (r?: CreateMenuPeriodGroupRequest, codes?: number[]) =>
  Request<CreateMenuPeriodGroupResponse>('/ceres/eshop/EshopService/CreateMenuPeriodGroup')
    .code([...(codes || [])])
    .data(r || {})
    .run()
CreateMenuPeriodGroup.url = '/ceres/eshop/EshopService/CreateMenuPeriodGroup'
export const UpdateMenuPeriodGroup = (r?: UpdateMenuPeriodGroupRequest, codes?: number[]) =>
  Request<UpdateMenuPeriodGroupResponse>('/ceres/eshop/EshopService/UpdateMenuPeriodGroup')
    .code([...(codes || [])])
    .data(r || {})
    .run()
UpdateMenuPeriodGroup.url = '/ceres/eshop/EshopService/UpdateMenuPeriodGroup'
export const DeleteMenuPeriodGroup = (r?: DeleteMenuPeriodGroupRequest, codes?: number[]) =>
  Request<DeleteMenuPeriodGroupResponse>('/ceres/eshop/EshopService/DeleteMenuPeriodGroup')
    .code([...(codes || [])])
    .data(r || {})
    .run()
DeleteMenuPeriodGroup.url = '/ceres/eshop/EshopService/DeleteMenuPeriodGroup'
export const GetManyMenuPeriodGroup = (r?: GetManyMenuPeriodGroupRequest, codes?: number[]) =>
  Request<GetManyMenuPeriodGroupResponse>('/ceres/eshop/EshopService/GetManyMenuPeriodGroup')
    .code([...(codes || [])])
    .data(r || {})
    .run()
GetManyMenuPeriodGroup.url = '/ceres/eshop/EshopService/GetManyMenuPeriodGroup'
export const ExportEshopOrder = (r?: ExportEshopOrderRequest, codes?: number[]) =>
  Request<ExportEshopOrderResponse>('/ceres/eshop/EshopService/ExportEshopOrder')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ExportEshopOrder.url = '/ceres/eshop/EshopService/ExportEshopOrder'
export const BatchUpdateEshopOrderState = (r?: BatchUpdateEshopOrderStateRequest, codes?: number[]) =>
  Request<BatchUpdateEshopOrderStateResponse>('/ceres/eshop/EshopService/BatchUpdateEshopOrderState')
    .code([...(codes || [])])
    .data(r || {})
    .run()
BatchUpdateEshopOrderState.url = '/ceres/eshop/EshopService/BatchUpdateEshopOrderState'
export const ExportLeaveApplication = (r?: ExportLeaveApplicationRequest, codes?: number[]) =>
  Request<ExportLeaveApplicationResponse>('/ceres/eshop/EshopService/ExportLeaveApplication')
    .code([...(codes || [])])
    .data(r || {})
    .run()
ExportLeaveApplication.url = '/ceres/eshop/EshopService/ExportLeaveApplication'
