// gm_api/src/eshop/pc.tsx
import React, { FC } from 'react'
import { Select, SelectProps } from '@gm-pc/react'
import {
  SelectList,
  SelectEnum,
  MoreSelectList,
  MapId,
} from '../util/pc_util'
import type {
  MapProps,
  SelectListProps,
  SelectEnumProps,
  MoreSelectListProps,
  MapIdProps,
} from '../util/pc_util'
import {
  list_Status_Code,
  map_Status_Code,
  list_Menu_CycleType,
  map_Menu_CycleType,
  list_MenuDetail_State,
  map_MenuDetail_State,
  LeaveApplication,
  ListLeaveApplication,
  ListLeaveApplicationResponse,
  Menu,
  ListMenu,
  ListMenuResponse,
  MenuDetail,
  ListMenuDetail,
  ListMenuDetailResponse,
  MenuPeriodGroup,
  ListMenuPeriodGroup,
  ListMenuPeriodGroupResponse,
} from './index'

export const Select_Status_Code: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Status_Code} />
}

export const Map_Status_Code: FC<MapProps> = ({ value }) => {
  return <>{map_Status_Code[value]}</>
}

export const Select_Menu_CycleType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_Menu_CycleType} />
}

export const Map_Menu_CycleType: FC<MapProps> = ({ value }) => {
  return <>{map_Menu_CycleType[value]}</>
}

export const Select_MenuDetail_State: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_MenuDetail_State} />
}

export const Map_MenuDetail_State: FC<MapProps> = ({ value }) => {
  return <>{map_MenuDetail_State[value]}</>
}

export const Select_LeaveApplication: FC<SelectListProps<
  LeaveApplication,
  ListLeaveApplicationResponse
>> = (props) => {
  return (
    <SelectList<LeaveApplication, ListLeaveApplicationResponse>
      {...props}
      messageName='LeaveApplication'
      fetchList={ListLeaveApplication}
    />
  )
}

export const MoreSelect_LeaveApplication: FC<MoreSelectListProps<
  LeaveApplication,
  ListLeaveApplicationResponse
>> = (props) => {
  return (
    <MoreSelectList<LeaveApplication, ListLeaveApplicationResponse>
      {...props}
      messageName='LeaveApplication'
      fetchList={ListLeaveApplication}
    />
  )
}

export const Select_Menu: FC<SelectListProps<
  Menu,
  ListMenuResponse
>> = (props) => {
  return (
    <SelectList<Menu, ListMenuResponse>
      {...props}
      messageName='Menu'
      fetchList={ListMenu}
    />
  )
}

export const MoreSelect_Menu: FC<MoreSelectListProps<
  Menu,
  ListMenuResponse
>> = (props) => {
  return (
    <MoreSelectList<Menu, ListMenuResponse>
      {...props}
      messageName='Menu'
      fetchList={ListMenu}
    />
  )
}

export const Select_MenuDetail: FC<SelectListProps<
  MenuDetail,
  ListMenuDetailResponse
>> = (props) => {
  return (
    <SelectList<MenuDetail, ListMenuDetailResponse>
      {...props}
      messageName='MenuDetail'
      fetchList={ListMenuDetail}
    />
  )
}

export const MoreSelect_MenuDetail: FC<MoreSelectListProps<
  MenuDetail,
  ListMenuDetailResponse
>> = (props) => {
  return (
    <MoreSelectList<MenuDetail, ListMenuDetailResponse>
      {...props}
      messageName='MenuDetail'
      fetchList={ListMenuDetail}
    />
  )
}

export const Select_MenuPeriodGroup: FC<SelectListProps<
  MenuPeriodGroup,
  ListMenuPeriodGroupResponse
>> = (props) => {
  return (
    <SelectList<MenuPeriodGroup, ListMenuPeriodGroupResponse>
      {...props}
      messageName='MenuPeriodGroup'
      fetchList={ListMenuPeriodGroup}
    />
  )
}

export const MoreSelect_MenuPeriodGroup: FC<MoreSelectListProps<
  MenuPeriodGroup,
  ListMenuPeriodGroupResponse
>> = (props) => {
  return (
    <MoreSelectList<MenuPeriodGroup, ListMenuPeriodGroupResponse>
      {...props}
      messageName='MenuPeriodGroup'
      fetchList={ListMenuPeriodGroup}
    />
  )
}

export const MapId_LeaveApplication: FC<MapIdProps<LeaveApplication, ListLeaveApplicationResponse>> = (
  props,
) => {
  return (
    <MapId<LeaveApplication, ListLeaveApplicationResponse>
      {...props}
      packageName='eshop'
      messageName='LeaveApplication'
      fetchList={ListLeaveApplication}
    />
  )
}

export const MapId_MenuDetail: FC<MapIdProps<MenuDetail, ListMenuDetailResponse>> = (
  props,
) => {
  return (
    <MapId<MenuDetail, ListMenuDetailResponse>
      {...props}
      packageName='eshop'
      messageName='MenuDetail'
      fetchList={ListMenuDetail}
    />
  )
}

export type {
  MapProps,
  SelectListProps,
  MoreSelectListProps,
}
