// gm_api/src/cloudapi/data.ts
import { getLocale } from './lng'

interface Enum_List_Item {
  text: string
  value: any
}

interface Enum_Map {
  [key: string]: string
}
export const list_CloudType: Enum_List_Item[] = [
  { value: 1, text: getLocale('CloudType__CLOUD_TYPE_TENCENT') },
  { value: 2, text: getLocale('CloudType__CLOUD_TYPE_ALIBABA') },
  { value: 3, text: getLocale('CloudType__CLOUD_TYPE_QINIU') },
]
export const map_CloudType: Enum_Map = {
  1: getLocale('CloudType__CLOUD_TYPE_TENCENT'),
  2: getLocale('CloudType__CLOUD_TYPE_ALIBABA'),
  3: getLocale('CloudType__CLOUD_TYPE_QINIU'),
}
export const list_BucketType: Enum_List_Item[] = [
  { value: 1, text: getLocale('BucketType__BUCKET_TYPE_GMFILE') },
]
export const map_BucketType: Enum_Map = {
  1: getLocale('BucketType__BUCKET_TYPE_GMFILE'),
}
export const list_FileType: Enum_List_Item[] = [
  { value: 10, text: getLocale('FileType__FILE_TYPE_MERCHANDISE_CATEGORY_DEFAULT_IMAGE') },
  { value: 11, text: getLocale('FileType__FILE_TYPE_MERCHANDISE_CATEGORY_IMAGE') },
  { value: 12, text: getLocale('FileType__FILE_TYPE_MERCHANDISE_SKU_IMAGE') },
  { value: 13, text: getLocale('FileType__FILE_TYPE_MERCHANDISE_SSU_IMAGE') },
  { value: 14, text: getLocale('FileType__FILE_TYPE_MERCHANDISE_SKU_IMPORT') },
  { value: 20, text: getLocale('FileType__FILE_TYPE_ENTERPRISE_LOGO') },
  { value: 21, text: getLocale('FileType__FILE_TYPE_ENTERPRISE_CUSTOMER_IMPORT') },
  { value: 22, text: getLocale('FileType__FILE_TYPE_ENTERPRISE_CUSTOMER_EXPORT') },
  { value: 30, text: getLocale('FileType__FILE_TYPE_DELIVERY_RECEIVE_IMAGE') },
  { value: 40, text: getLocale('FileType__FILE_TYPE_PREFERENCE_BANNER_IMAGE') },
  { value: 41, text: getLocale('FileType__FILE_TYPE_PREFERENCE_SHOP_LOGO_IMAGE') },
  { value: 42, text: getLocale('FileType__FILE_TYPE_PREFERENCE_SHOP_REFERENCE') },
  { value: 100, text: getLocale('FileType__FILE_TYPE_ASYNCTASK_EXPORT') },
  { value: 150, text: getLocale('FileType__FILE_TYPE_PRODUCTION_BOM_IMPORT') },
  { value: 200, text: getLocale('FileType__FILE_TYPE_PURCHASE_PURCHASE_TASK_EXPORT') },
  { value: 201, text: getLocale('FileType__FILE_TYPE_PURCHASE_PURCHASE_SHEET_EXPORT') },
  { value: 202, text: getLocale('FileType__FILE_TYPE_PURCHASE_INQUIRY_PRICE_IMPORT_TMPL') },
  { value: 203, text: getLocale('FileType__FILE_TYPE_PURCHASE_INQUIRY_PRICE_IMPORT') },
  { value: 204, text: getLocale('FileType__FILE_TYPE_PURCHASE_INQUIRY_PRICE_IMPORT_ERR') },
  { value: 300, text: getLocale('FileType__FILE_TYPE_AFTER_SALE_ORDER_EXPORT') },
  { value: 301, text: getLocale('FileType__FILE_TYPE_AFTER_SALE_SHEET_EXPORT') },
  { value: 302, text: getLocale('FileType__FILE_TYPE_AFTER_SALE_TASK_EXPORT') },
  { value: 303, text: getLocale('FileType__FILE_TYPE_AFTER_SALE_DETAIL_EXPORT') },
  { value: 400, text: getLocale('FileType__FILE_TYPE_FINANCE_SETTLE_SHEET_EXPORT') },
  { value: 401, text: getLocale('FileType__FILE_TYPE_FINANCE_ACCOUNT_BALANCE_EXPORT') },
  { value: 402, text: getLocale('FileType__FILE_TYPE_FINANCE_FLOW_EXPORT') },
]
export const map_FileType: Enum_Map = {
  10: getLocale('FileType__FILE_TYPE_MERCHANDISE_CATEGORY_DEFAULT_IMAGE'),
  11: getLocale('FileType__FILE_TYPE_MERCHANDISE_CATEGORY_IMAGE'),
  12: getLocale('FileType__FILE_TYPE_MERCHANDISE_SKU_IMAGE'),
  13: getLocale('FileType__FILE_TYPE_MERCHANDISE_SSU_IMAGE'),
  14: getLocale('FileType__FILE_TYPE_MERCHANDISE_SKU_IMPORT'),
  20: getLocale('FileType__FILE_TYPE_ENTERPRISE_LOGO'),
  21: getLocale('FileType__FILE_TYPE_ENTERPRISE_CUSTOMER_IMPORT'),
  22: getLocale('FileType__FILE_TYPE_ENTERPRISE_CUSTOMER_EXPORT'),
  30: getLocale('FileType__FILE_TYPE_DELIVERY_RECEIVE_IMAGE'),
  40: getLocale('FileType__FILE_TYPE_PREFERENCE_BANNER_IMAGE'),
  41: getLocale('FileType__FILE_TYPE_PREFERENCE_SHOP_LOGO_IMAGE'),
  42: getLocale('FileType__FILE_TYPE_PREFERENCE_SHOP_REFERENCE'),
  100: getLocale('FileType__FILE_TYPE_ASYNCTASK_EXPORT'),
  150: getLocale('FileType__FILE_TYPE_PRODUCTION_BOM_IMPORT'),
  200: getLocale('FileType__FILE_TYPE_PURCHASE_PURCHASE_TASK_EXPORT'),
  201: getLocale('FileType__FILE_TYPE_PURCHASE_PURCHASE_SHEET_EXPORT'),
  202: getLocale('FileType__FILE_TYPE_PURCHASE_INQUIRY_PRICE_IMPORT_TMPL'),
  203: getLocale('FileType__FILE_TYPE_PURCHASE_INQUIRY_PRICE_IMPORT'),
  204: getLocale('FileType__FILE_TYPE_PURCHASE_INQUIRY_PRICE_IMPORT_ERR'),
  300: getLocale('FileType__FILE_TYPE_AFTER_SALE_ORDER_EXPORT'),
  301: getLocale('FileType__FILE_TYPE_AFTER_SALE_SHEET_EXPORT'),
  302: getLocale('FileType__FILE_TYPE_AFTER_SALE_TASK_EXPORT'),
  303: getLocale('FileType__FILE_TYPE_AFTER_SALE_DETAIL_EXPORT'),
  400: getLocale('FileType__FILE_TYPE_FINANCE_SETTLE_SHEET_EXPORT'),
  401: getLocale('FileType__FILE_TYPE_FINANCE_ACCOUNT_BALANCE_EXPORT'),
  402: getLocale('FileType__FILE_TYPE_FINANCE_FLOW_EXPORT'),
}
export const list_SmsType: Enum_List_Item[] = [
  { value: 49688, text: getLocale('SmsType__SMS_TYPE_VERIFICATION_CODE') },
]
export const map_SmsType: Enum_Map = {
  49688: getLocale('SmsType__SMS_TYPE_VERIFICATION_CODE'),
}
