// gm_api/src/cloudapi/pc.tsx
import React, { FC } from 'react'
import { Select, SelectProps } from '@gm-pc/react'
import {
  SelectList,
  SelectEnum,
  MoreSelectList,
  MapId,
} from '../util/pc_util'
import type {
  MapProps,
  SelectListProps,
  SelectEnumProps,
  MoreSelectListProps,
  MapIdProps,
} from '../util/pc_util'
import {
  list_CloudType,
  map_CloudType,
  list_BucketType,
  map_BucketType,
  list_FileType,
  map_FileType,
  list_SmsType,
  map_SmsType,
} from './index'

export const Select_CloudType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_CloudType} />
}

export const Map_CloudType: FC<MapProps> = ({ value }) => {
  return <>{map_CloudType[value]}</>
}

export const Select_BucketType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_BucketType} />
}

export const Map_BucketType: FC<MapProps> = ({ value }) => {
  return <>{map_BucketType[value]}</>
}

export const Select_FileType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_FileType} />
}

export const Map_FileType: FC<MapProps> = ({ value }) => {
  return <>{map_FileType[value]}</>
}

export const Select_SmsType: FC<SelectEnumProps<number>> = ({ ...rest }) => {
  return <SelectEnum {...rest} data={list_SmsType} />
}

export const Map_SmsType: FC<MapProps> = ({ value }) => {
  return <>{map_SmsType[value]}</>
}

export type {
  MapProps,
  SelectListProps,
  MoreSelectListProps,
}
