// gm_api/src/cloudapi/types.ts
export enum CloudType {
  CLOUD_TYPE_UNSPECIFIED = 0,
  CLOUD_TYPE_TENCENT = 1,
  CLOUD_TYPE_ALIBABA = 2,
  CLOUD_TYPE_QINIU = 3,
}
export enum BucketType {
  BUCKET_TYPE_UNSPECIFIED = 0,
  BUCKET_TYPE_GMFILE = 1,
}
export enum FileType {
  FILE_TYPE_UNSPECIFIED = 0,
  FILE_TYPE_MERCHANDISE_CATEGORY_DEFAULT_IMAGE = 10,
  FILE_TYPE_MERCHANDISE_CATEGORY_IMAGE = 11,
  FILE_TYPE_MERCHANDISE_SKU_IMAGE = 12,
  FILE_TYPE_MERCHANDISE_SSU_IMAGE = 13,
  FILE_TYPE_MERCHANDISE_SKU_IMPORT = 14,
  FILE_TYPE_ENTERPRISE_LOGO = 20,
  FILE_TYPE_ENTERPRISE_CUSTOMER_IMPORT = 21,
  FILE_TYPE_ENTERPRISE_CUSTOMER_EXPORT = 22,
  FILE_TYPE_DELIVERY_RECEIVE_IMAGE = 30,
  FILE_TYPE_PREFERENCE_BANNER_IMAGE = 40,
  FILE_TYPE_PREFERENCE_SHOP_LOGO_IMAGE = 41,
  FILE_TYPE_PREFERENCE_SHOP_REFERENCE = 42,
  FILE_TYPE_ASYNCTASK_EXPORT = 100,
  FILE_TYPE_PRODUCTION_BOM_IMPORT = 150,
  FILE_TYPE_PURCHASE_PURCHASE_TASK_EXPORT = 200,
  FILE_TYPE_PURCHASE_PURCHASE_SHEET_EXPORT = 201,
  FILE_TYPE_PURCHASE_INQUIRY_PRICE_IMPORT_TMPL = 202,
  FILE_TYPE_PURCHASE_INQUIRY_PRICE_IMPORT = 203,
  FILE_TYPE_PURCHASE_INQUIRY_PRICE_IMPORT_ERR = 204,
  FILE_TYPE_AFTER_SALE_ORDER_EXPORT = 300,
  FILE_TYPE_AFTER_SALE_SHEET_EXPORT = 301,
  FILE_TYPE_AFTER_SALE_TASK_EXPORT = 302,
  FILE_TYPE_AFTER_SALE_DETAIL_EXPORT = 303,
  FILE_TYPE_FINANCE_SETTLE_SHEET_EXPORT = 400,
  FILE_TYPE_FINANCE_ACCOUNT_BALANCE_EXPORT = 401,
  FILE_TYPE_FINANCE_FLOW_EXPORT = 402,
}
export enum SmsType {
  SMS_TYPE_UNSPECIFIED = 0,
  SMS_TYPE_VERIFICATION_CODE = 49688,
}
export interface SmsConfig {
}
export interface TencentConfig_FileConfig {
  file_type?: FileType
  prefix?: string
}
export interface TencentConfig_BucketConfig {
  secret_id?: string
  secret_key?: string
  app_id?: string
  endpoint?: string
  base_url?: string
  file_configs?: { [key: string]: TencentConfig_FileConfig }
}
export interface TencentConfig_SmsConfig {
  sms_type?: SmsType
  secret_id?: string
  secret_key?: string
  app_id?: string
  sign?: string
  endpoint?: string
  template_id?: string
}
export interface TencentConfig {
  bucket_configs?: { [key: string]: TencentConfig_BucketConfig }
  sms_configs?: { [key: string]: TencentConfig_SmsConfig }
}
export interface UploadFileRequest {
  cloud_type: CloudType
  file_type: FileType
  filename: string
  data?: bytes
}
export interface UploadFileResponse {
  url?: string
}
export interface GetPresignedURLRequest {
  cloud_type: CloudType
  file_type: FileType
  filename: string
}
export interface GetPresignedURLResponse {
  presigned_url?: string
  download_url?: string
}
export interface SendSmsRequest {
  cloud_type?: CloudType
  sms_type?: SmsType
  phone_numbers?: string[]
  parameters?: string[]
}
export interface SendSmsResponse {
}
export interface GetQiniuUploadTokenRequest {
  file_type: FileType
}
export interface GetQiniuUploadTokenResponse {
  upload_token: string
  dir_path: string
  expire_time: string
}
